<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: MSSim Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a05822.html">OpenMS</a></li><li class="navelem"><a class="el" href="a10845.html">MSSim</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="a10842.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">MSSim Class Reference<div class="ingroups"><a class="el" href="a05811.html">Simulation</a></div></div>  </div>
</div><!--header-->
<div class="contents">

<p>Central class for simulation of mass spectrometry experiments.  
 <a href="a10845.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="a04610_source.html">OpenMS/SIMULATION/MSSim.h</a>&gt;</code></p>
<div class="dynheader">
Inheritance diagram for MSSim:</div>
<div class="dyncontent">
 <div class="center">
  <img src="a10845.png" usemap="#MSSim_map" alt=""/>
  <map id="MSSim_map" name="MSSim_map">
<area href="a08589.html" title="A base class for all classes handling default parameters." alt="DefaultParamHandler" shape="rect" coords="0,0,133,24"/>
<area href="a08481.html" title="Base class for all classes that want to report their progress." alt="ProgressLogger" shape="rect" coords="143,0,276,24"/>
  </map>
</div></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="member-group"></a>
Constructors and Destructors</h2></td></tr>
<tr class="memitem:a6bf6c31430e50c6d99744da96cdca044"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a6bf6c31430e50c6d99744da96cdca044">experiment_</a></td></tr>
<tr class="memdesc:a6bf6c31430e50c6d99744da96cdca044"><td class="mdescLeft">&#160;</td><td class="mdescRight">Holds the simulated data.  <a href="a10845.html#a6bf6c31430e50c6d99744da96cdca044">More...</a><br /></td></tr>
<tr class="separator:a6bf6c31430e50c6d99744da96cdca044"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adc073663b8fca88af1da2a8bd16c8aad"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#adc073663b8fca88af1da2a8bd16c8aad">peak_map_</a></td></tr>
<tr class="memdesc:adc073663b8fca88af1da2a8bd16c8aad"><td class="mdescLeft">&#160;</td><td class="mdescRight">Holds the ground-truth on generated peaks positions and intensities.  <a href="a10845.html#adc073663b8fca88af1da2a8bd16c8aad">More...</a><br /></td></tr>
<tr class="separator:adc073663b8fca88af1da2a8bd16c8aad"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af0d97a3e4120e8f57d2a0f2c5e10b4df"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a05853.html#a5377a8d620ef376a0b1eb2503b847e79">SimTypes::FeatureMapSimVector</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#af0d97a3e4120e8f57d2a0f2c5e10b4df">feature_maps_</a></td></tr>
<tr class="memdesc:af0d97a3e4120e8f57d2a0f2c5e10b4df"><td class="mdescLeft">&#160;</td><td class="mdescRight">Holds the ground-truth on generated features.  <a href="a10845.html#af0d97a3e4120e8f57d2a0f2c5e10b4df">More...</a><br /></td></tr>
<tr class="separator:af0d97a3e4120e8f57d2a0f2c5e10b4df"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f04b32b80833aabd4ab8bc8c2c9ecb4"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a10021.html">ConsensusMap</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a9f04b32b80833aabd4ab8bc8c2c9ecb4">consensus_map_</a></td></tr>
<tr class="memdesc:a9f04b32b80833aabd4ab8bc8c2c9ecb4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Holds consensus ground-truth about the charge associations.  <a href="a10845.html#a9f04b32b80833aabd4ab8bc8c2c9ecb4">More...</a><br /></td></tr>
<tr class="separator:a9f04b32b80833aabd4ab8bc8c2c9ecb4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4c6c540ae92e189dbbf8dbb5bb0fa2a3"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a4c6c540ae92e189dbbf8dbb5bb0fa2a3">contaminants_map_</a></td></tr>
<tr class="memdesc:a4c6c540ae92e189dbbf8dbb5bb0fa2a3"><td class="mdescLeft">&#160;</td><td class="mdescRight">Holds the ground-truth on generated contaminants.  <a href="a10845.html#a4c6c540ae92e189dbbf8dbb5bb0fa2a3">More...</a><br /></td></tr>
<tr class="separator:a4c6c540ae92e189dbbf8dbb5bb0fa2a3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a813cce92a0e34c5f83c0798cae8a61ae"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a10821.html">BaseLabeler</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a813cce92a0e34c5f83c0798cae8a61ae">labeler_</a></td></tr>
<tr class="memdesc:a813cce92a0e34c5f83c0798cae8a61ae"><td class="mdescLeft">&#160;</td><td class="mdescRight">Labeling functionality.  <a href="a10845.html#a813cce92a0e34c5f83c0798cae8a61ae">More...</a><br /></td></tr>
<tr class="separator:a813cce92a0e34c5f83c0798cae8a61ae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a063de8c0272b877c59781b899546e057"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a063de8c0272b877c59781b899546e057">MSSim</a> ()</td></tr>
<tr class="memdesc:a063de8c0272b877c59781b899546e057"><td class="mdescLeft">&#160;</td><td class="mdescRight">Default constructor.  <a href="a10845.html#a063de8c0272b877c59781b899546e057">More...</a><br /></td></tr>
<tr class="separator:a063de8c0272b877c59781b899546e057"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a91363e6770cbe9bade26f523bf9662c7"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a91363e6770cbe9bade26f523bf9662c7">~MSSim</a> () override</td></tr>
<tr class="memdesc:a91363e6770cbe9bade26f523bf9662c7"><td class="mdescLeft">&#160;</td><td class="mdescRight">Destructor.  <a href="a10845.html#a91363e6770cbe9bade26f523bf9662c7">More...</a><br /></td></tr>
<tr class="separator:a91363e6770cbe9bade26f523bf9662c7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1ff9361d142cb1722cf7758d1b9c5ea2"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a1ff9361d142cb1722cf7758d1b9c5ea2">simulate</a> (<a class="el" href="a05853.html#a00c57778f72fd6a465d82be2bb231dfe">SimTypes::MutableSimRandomNumberGeneratorPtr</a> rnd_gen, <a class="el" href="a05853.html#abd6a489f13876548dd540ccefd087b0b">SimTypes::SampleChannels</a> &amp;peptides)</td></tr>
<tr class="memdesc:a1ff9361d142cb1722cf7758d1b9c5ea2"><td class="mdescLeft">&#160;</td><td class="mdescRight">General purpose function to simulate a mass spectrometry run.  <a href="a10845.html#a1ff9361d142cb1722cf7758d1b9c5ea2">More...</a><br /></td></tr>
<tr class="separator:a1ff9361d142cb1722cf7758d1b9c5ea2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aba841147967347f8b403f39b0ab7019b"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#aba841147967347f8b403f39b0ab7019b">getExperiment</a> () const</td></tr>
<tr class="memdesc:aba841147967347f8b403f39b0ab7019b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the simulated experiment.  <a href="a10845.html#aba841147967347f8b403f39b0ab7019b">More...</a><br /></td></tr>
<tr class="separator:aba841147967347f8b403f39b0ab7019b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8c12281c11eedb6817ab040852b01c72"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a8c12281c11eedb6817ab040852b01c72">getSimulatedFeatures</a> () const</td></tr>
<tr class="memdesc:a8c12281c11eedb6817ab040852b01c72"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the simulated features.  <a href="a10845.html#a8c12281c11eedb6817ab040852b01c72">More...</a><br /></td></tr>
<tr class="separator:a8c12281c11eedb6817ab040852b01c72"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a22d35cb0ae2a4d19e44f6ac8cd4f57c9"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a10021.html">ConsensusMap</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a22d35cb0ae2a4d19e44f6ac8cd4f57c9">getChargeConsensus</a> ()</td></tr>
<tr class="memdesc:a22d35cb0ae2a4d19e44f6ac8cd4f57c9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the charge consensus map of simulated features.  <a href="a10845.html#a22d35cb0ae2a4d19e44f6ac8cd4f57c9">More...</a><br /></td></tr>
<tr class="separator:a22d35cb0ae2a4d19e44f6ac8cd4f57c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a19f1fb58afc66fd144252470ddb3d176"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a19f1fb58afc66fd144252470ddb3d176">getContaminants</a> () const</td></tr>
<tr class="memdesc:a19f1fb58afc66fd144252470ddb3d176"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the contaminants feature map of simulated features.  <a href="a10845.html#a19f1fb58afc66fd144252470ddb3d176">More...</a><br /></td></tr>
<tr class="separator:a19f1fb58afc66fd144252470ddb3d176"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aac38d440d2e98f886eb076696c8c507b"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a10021.html">ConsensusMap</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#aac38d440d2e98f886eb076696c8c507b">getLabelingConsensus</a> ()</td></tr>
<tr class="memdesc:aac38d440d2e98f886eb076696c8c507b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the labeling consensus map of simulated features.  <a href="a10845.html#aac38d440d2e98f886eb076696c8c507b">More...</a><br /></td></tr>
<tr class="separator:aac38d440d2e98f886eb076696c8c507b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a960ecae2325e1418a4371a0b1a9ac083"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a960ecae2325e1418a4371a0b1a9ac083">getPeakMap</a> () const</td></tr>
<tr class="memdesc:a960ecae2325e1418a4371a0b1a9ac083"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the picked (centroided) experiment.  <a href="a10845.html#a960ecae2325e1418a4371a0b1a9ac083">More...</a><br /></td></tr>
<tr class="separator:a960ecae2325e1418a4371a0b1a9ac083"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaa266aa79d1729ff264838a2f37b1bd4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#aaa266aa79d1729ff264838a2f37b1bd4">getIdentifications</a> (std::vector&lt; <a class="el" href="a10677.html">ProteinIdentification</a> &gt; &amp;proteins, std::vector&lt; <a class="el" href="a10645.html">PeptideIdentification</a> &gt; &amp;peptides) const</td></tr>
<tr class="memdesc:aaa266aa79d1729ff264838a2f37b1bd4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the simulated identifications (proteins and peptides)  <a href="a10845.html#aaa266aa79d1729ff264838a2f37b1bd4">More...</a><br /></td></tr>
<tr class="separator:aaa266aa79d1729ff264838a2f37b1bd4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3a22e0505a27c8a2a39970888c7b99fc"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a3a22e0505a27c8a2a39970888c7b99fc">getMS2Identifications</a> (std::vector&lt; <a class="el" href="a10677.html">ProteinIdentification</a> &gt; &amp;proteins, std::vector&lt; <a class="el" href="a10645.html">PeptideIdentification</a> &gt; &amp;peptides) const</td></tr>
<tr class="memdesc:a3a22e0505a27c8a2a39970888c7b99fc"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the simulated MS2 identifications (proteins and peptides)  <a href="a10845.html#a3a22e0505a27c8a2a39970888c7b99fc">More...</a><br /></td></tr>
<tr class="separator:a3a22e0505a27c8a2a39970888c7b99fc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae587207f751bcfe0ba94031d5b75654b"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#ae587207f751bcfe0ba94031d5b75654b">getFeatureIdentifications</a> (std::vector&lt; <a class="el" href="a10677.html">ProteinIdentification</a> &gt; &amp;proteins, std::vector&lt; <a class="el" href="a10645.html">PeptideIdentification</a> &gt; &amp;peptides) const</td></tr>
<tr class="memdesc:ae587207f751bcfe0ba94031d5b75654b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Access the simulated identifications (proteins and peptides) from feature annotations.  <a href="a10845.html#ae587207f751bcfe0ba94031d5b75654b">More...</a><br /></td></tr>
<tr class="separator:ae587207f751bcfe0ba94031d5b75654b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a301e5b8ce00c61b1b4efaf09f213b619"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08745.html">Param</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a301e5b8ce00c61b1b4efaf09f213b619">getParameters</a> () const</td></tr>
<tr class="memdesc:a301e5b8ce00c61b1b4efaf09f213b619"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns the default parameters for simulation including the labeling technique with name <code>labeling_name</code>.  <a href="a10845.html#a301e5b8ce00c61b1b4efaf09f213b619">More...</a><br /></td></tr>
<tr class="separator:a301e5b8ce00c61b1b4efaf09f213b619"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a63efff66245e023f272fa69fee0631fe"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#a63efff66245e023f272fa69fee0631fe">syncParams_</a> (<a class="el" href="a08745.html">Param</a> &amp;p, bool to_outer)</td></tr>
<tr class="memdesc:a63efff66245e023f272fa69fee0631fe"><td class="mdescLeft">&#160;</td><td class="mdescRight">handle global params  <a href="a10845.html#a63efff66245e023f272fa69fee0631fe">More...</a><br /></td></tr>
<tr class="separator:a63efff66245e023f272fa69fee0631fe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af4298dd2e9804f7799b6f265b15c161a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#af4298dd2e9804f7799b6f265b15c161a">createFeatureMap_</a> (const <a class="el" href="a05853.html#af8f836c78cff89a3bc6f1b4920100ace">SimTypes::SampleProteins</a> &amp;peptides, <a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a> &amp;features, <a class="el" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> map_index)</td></tr>
<tr class="memdesc:af4298dd2e9804f7799b6f265b15c161a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert a list of peptides with given abundance values into a <a class="el" href="a10057.html" title="A container for features.">FeatureMap</a>.  <a href="a10845.html#af4298dd2e9804f7799b6f265b15c161a">More...</a><br /></td></tr>
<tr class="separator:af4298dd2e9804f7799b6f265b15c161a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abff6e392ce6da7af8f083397494a7971"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a10845.html#abff6e392ce6da7af8f083397494a7971">updateMembers_</a> () override</td></tr>
<tr class="memdesc:abff6e392ce6da7af8f083397494a7971"><td class="mdescLeft">&#160;</td><td class="mdescRight">Synchronize members with param class.  <a href="a10845.html#abff6e392ce6da7af8f083397494a7971">More...</a><br /></td></tr>
<tr class="separator:abff6e392ce6da7af8f083397494a7971"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="inherited"></a>
Additional Inherited Members</h2></td></tr>
<tr class="inherit_header pub_types_a08481"><td colspan="2" onclick="javascript:toggleInherit('pub_types_a08481')"><img src="closed.png" alt="-"/>&#160;Public Types inherited from <a class="el" href="a08481.html">ProgressLogger</a></td></tr>
<tr class="memitem:af67907baa897e9fb84df0cb89795b87c inherit pub_types_a08481"><td class="memItemLeft" align="right" valign="top">enum &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87c">LogType</a> { <a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87ca3ab34dd50405af2d5a62627d572f7860">CMD</a>, 
<a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87ca73c6901c7c648a6a735770bc038bb26a">GUI</a>, 
<a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87cac157bdf0b85a40d2619cbc8bc1ae5fe2">NONE</a>
 }</td></tr>
<tr class="memdesc:af67907baa897e9fb84df0cb89795b87c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Possible log types.  <a href="a08481.html#af67907baa897e9fb84df0cb89795b87c">More...</a><br /></td></tr>
<tr class="separator:af67907baa897e9fb84df0cb89795b87c inherit pub_types_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_a08589"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_a08589')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="a08589.html">DefaultParamHandler</a></td></tr>
<tr class="memitem:afb6802fda161cb0c1a4a05597f94aecc inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#afb6802fda161cb0c1a4a05597f94aecc">DefaultParamHandler</a> (const <a class="el" href="a08769.html">String</a> &amp;name)</td></tr>
<tr class="memdesc:afb6802fda161cb0c1a4a05597f94aecc inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Constructor with name that is displayed in error messages.  <a href="a08589.html#afb6802fda161cb0c1a4a05597f94aecc">More...</a><br /></td></tr>
<tr class="separator:afb6802fda161cb0c1a4a05597f94aecc inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a954da7eef3cfddda446574eaadd20abb inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a954da7eef3cfddda446574eaadd20abb">DefaultParamHandler</a> (const <a class="el" href="a08589.html">DefaultParamHandler</a> &amp;rhs)</td></tr>
<tr class="memdesc:a954da7eef3cfddda446574eaadd20abb inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Copy constructor.  <a href="a08589.html#a954da7eef3cfddda446574eaadd20abb">More...</a><br /></td></tr>
<tr class="separator:a954da7eef3cfddda446574eaadd20abb inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a11e9d6dd34fe8860874aa8abe0ec121b inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">virtual&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a11e9d6dd34fe8860874aa8abe0ec121b">~DefaultParamHandler</a> ()</td></tr>
<tr class="memdesc:a11e9d6dd34fe8860874aa8abe0ec121b inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Destructor.  <a href="a08589.html#a11e9d6dd34fe8860874aa8abe0ec121b">More...</a><br /></td></tr>
<tr class="separator:a11e9d6dd34fe8860874aa8abe0ec121b inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae6b6d68036a573a44f6609460ba2dfb6 inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">virtual <a class="el" href="a08589.html">DefaultParamHandler</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#ae6b6d68036a573a44f6609460ba2dfb6">operator=</a> (const <a class="el" href="a08589.html">DefaultParamHandler</a> &amp;rhs)</td></tr>
<tr class="memdesc:ae6b6d68036a573a44f6609460ba2dfb6 inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Assignment operator.  <a href="a08589.html#ae6b6d68036a573a44f6609460ba2dfb6">More...</a><br /></td></tr>
<tr class="separator:ae6b6d68036a573a44f6609460ba2dfb6 inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1a859bdd46d8068663f8acefed5230eb inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">virtual bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a1a859bdd46d8068663f8acefed5230eb">operator==</a> (const <a class="el" href="a08589.html">DefaultParamHandler</a> &amp;rhs) const</td></tr>
<tr class="memdesc:a1a859bdd46d8068663f8acefed5230eb inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Equality operator.  <a href="a08589.html#a1a859bdd46d8068663f8acefed5230eb">More...</a><br /></td></tr>
<tr class="separator:a1a859bdd46d8068663f8acefed5230eb inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abc8067511cfc949ae942ea2b08588f81 inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#abc8067511cfc949ae942ea2b08588f81">setParameters</a> (const <a class="el" href="a08745.html">Param</a> &amp;param)</td></tr>
<tr class="memdesc:abc8067511cfc949ae942ea2b08588f81 inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Sets the parameters.  <a href="a08589.html#abc8067511cfc949ae942ea2b08588f81">More...</a><br /></td></tr>
<tr class="separator:abc8067511cfc949ae942ea2b08588f81 inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1decb59b26f2579931e3b4850df53d98 inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a08745.html">Param</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a1decb59b26f2579931e3b4850df53d98">getParameters</a> () const</td></tr>
<tr class="memdesc:a1decb59b26f2579931e3b4850df53d98 inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Non-mutable access to the parameters.  <a href="a08589.html#a1decb59b26f2579931e3b4850df53d98">More...</a><br /></td></tr>
<tr class="separator:a1decb59b26f2579931e3b4850df53d98 inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac5e5ebf235cb30ceccad12cb15d46886 inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a08745.html">Param</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#ac5e5ebf235cb30ceccad12cb15d46886">getDefaults</a> () const</td></tr>
<tr class="memdesc:ac5e5ebf235cb30ceccad12cb15d46886 inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Non-mutable access to the default parameters.  <a href="a08589.html#ac5e5ebf235cb30ceccad12cb15d46886">More...</a><br /></td></tr>
<tr class="separator:ac5e5ebf235cb30ceccad12cb15d46886 inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aba14af7abcdff896b4673851c4b6248f inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">const <a class="el" href="a08769.html">String</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#aba14af7abcdff896b4673851c4b6248f">getName</a> () const</td></tr>
<tr class="memdesc:aba14af7abcdff896b4673851c4b6248f inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Non-mutable access to the name.  <a href="a08589.html#aba14af7abcdff896b4673851c4b6248f">More...</a><br /></td></tr>
<tr class="separator:aba14af7abcdff896b4673851c4b6248f inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a77d37f2e1c12e8bd33ea0e660a89e927 inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a77d37f2e1c12e8bd33ea0e660a89e927">setName</a> (const <a class="el" href="a08769.html">String</a> &amp;name)</td></tr>
<tr class="memdesc:a77d37f2e1c12e8bd33ea0e660a89e927 inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Mutable access to the name.  <a href="a08589.html#a77d37f2e1c12e8bd33ea0e660a89e927">More...</a><br /></td></tr>
<tr class="separator:a77d37f2e1c12e8bd33ea0e660a89e927 inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4c4e51ec6dfdafaafa06854f9b13f62c inherit pub_methods_a08589"><td class="memItemLeft" align="right" valign="top">const std::vector&lt; <a class="el" href="a08769.html">String</a> &gt; &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a4c4e51ec6dfdafaafa06854f9b13f62c">getSubsections</a> () const</td></tr>
<tr class="memdesc:a4c4e51ec6dfdafaafa06854f9b13f62c inherit pub_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Non-mutable access to the registered subsections.  <a href="a08589.html#a4c4e51ec6dfdafaafa06854f9b13f62c">More...</a><br /></td></tr>
<tr class="separator:a4c4e51ec6dfdafaafa06854f9b13f62c inherit pub_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pub_methods_a08481"><td colspan="2" onclick="javascript:toggleInherit('pub_methods_a08481')"><img src="closed.png" alt="-"/>&#160;Public Member Functions inherited from <a class="el" href="a08481.html">ProgressLogger</a></td></tr>
<tr class="memitem:a2a106011aec95b2013d7a545f2063d01 inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a2a106011aec95b2013d7a545f2063d01">ProgressLogger</a> ()</td></tr>
<tr class="memdesc:a2a106011aec95b2013d7a545f2063d01 inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Constructor.  <a href="a08481.html#a2a106011aec95b2013d7a545f2063d01">More...</a><br /></td></tr>
<tr class="separator:a2a106011aec95b2013d7a545f2063d01 inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a19c882d907f90f7f88c7436d82cd2a13 inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a19c882d907f90f7f88c7436d82cd2a13">~ProgressLogger</a> ()</td></tr>
<tr class="memdesc:a19c882d907f90f7f88c7436d82cd2a13 inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Destructor.  <a href="a08481.html#a19c882d907f90f7f88c7436d82cd2a13">More...</a><br /></td></tr>
<tr class="separator:a19c882d907f90f7f88c7436d82cd2a13 inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a50978f110c7c899ac59df876726417d0 inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a50978f110c7c899ac59df876726417d0">ProgressLogger</a> (const <a class="el" href="a08481.html">ProgressLogger</a> &amp;other)</td></tr>
<tr class="memdesc:a50978f110c7c899ac59df876726417d0 inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Copy constructor.  <a href="a08481.html#a50978f110c7c899ac59df876726417d0">More...</a><br /></td></tr>
<tr class="separator:a50978f110c7c899ac59df876726417d0 inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a09816be277dd46e84d5a637e749cdfce inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08481.html">ProgressLogger</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a09816be277dd46e84d5a637e749cdfce">operator=</a> (const <a class="el" href="a08481.html">ProgressLogger</a> &amp;other)</td></tr>
<tr class="memdesc:a09816be277dd46e84d5a637e749cdfce inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Assignment Operator.  <a href="a08481.html#a09816be277dd46e84d5a637e749cdfce">More...</a><br /></td></tr>
<tr class="separator:a09816be277dd46e84d5a637e749cdfce inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2ce21e810918d3b0be795bda3fa7136a inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a2ce21e810918d3b0be795bda3fa7136a">setLogType</a> (<a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87c">LogType</a> type) const</td></tr>
<tr class="memdesc:a2ce21e810918d3b0be795bda3fa7136a inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Sets the progress log that should be used. The default type is NONE!  <a href="a08481.html#a2ce21e810918d3b0be795bda3fa7136a">More...</a><br /></td></tr>
<tr class="separator:a2ce21e810918d3b0be795bda3fa7136a inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad79c8c7d9f3e1cacb20e84c0cf408c7b inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87c">LogType</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#ad79c8c7d9f3e1cacb20e84c0cf408c7b">getLogType</a> () const</td></tr>
<tr class="memdesc:ad79c8c7d9f3e1cacb20e84c0cf408c7b inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns the type of progress log being used.  <a href="a08481.html#ad79c8c7d9f3e1cacb20e84c0cf408c7b">More...</a><br /></td></tr>
<tr class="separator:ad79c8c7d9f3e1cacb20e84c0cf408c7b inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6ca049c9417a2a648b4d41aec766d56b inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a6ca049c9417a2a648b4d41aec766d56b">startProgress</a> (<a class="el" href="a05782.html#gae113cb41304c6692c7bf5d0624dc6f0f">SignedSize</a> begin, <a class="el" href="a05782.html#gae113cb41304c6692c7bf5d0624dc6f0f">SignedSize</a> end, const <a class="el" href="a08769.html">String</a> &amp;label) const</td></tr>
<tr class="memdesc:a6ca049c9417a2a648b4d41aec766d56b inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Initializes the progress display.  <a href="a08481.html#a6ca049c9417a2a648b4d41aec766d56b">More...</a><br /></td></tr>
<tr class="separator:a6ca049c9417a2a648b4d41aec766d56b inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05d564f4a203341c9f1b1e6c8ec1068d inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a05d564f4a203341c9f1b1e6c8ec1068d">setProgress</a> (<a class="el" href="a05782.html#gae113cb41304c6692c7bf5d0624dc6f0f">SignedSize</a> value) const</td></tr>
<tr class="memdesc:a05d564f4a203341c9f1b1e6c8ec1068d inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Sets the current progress.  <a href="a08481.html#a05d564f4a203341c9f1b1e6c8ec1068d">More...</a><br /></td></tr>
<tr class="separator:a05d564f4a203341c9f1b1e6c8ec1068d inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab11bd2bba91e2c1055b2eeb376a25d0e inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#ab11bd2bba91e2c1055b2eeb376a25d0e">endProgress</a> () const</td></tr>
<tr class="memdesc:ab11bd2bba91e2c1055b2eeb376a25d0e inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Ends the progress display.  <a href="a08481.html#ab11bd2bba91e2c1055b2eeb376a25d0e">More...</a><br /></td></tr>
<tr class="separator:ab11bd2bba91e2c1055b2eeb376a25d0e inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adf9a755cba9ecf32628c8dfc327afb27 inherit pub_methods_a08481"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#adf9a755cba9ecf32628c8dfc327afb27">nextProgress</a> () const</td></tr>
<tr class="memdesc:adf9a755cba9ecf32628c8dfc327afb27 inherit pub_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">increment progress by 1 (according to range begin-end)  <a href="a08481.html#adf9a755cba9ecf32628c8dfc327afb27">More...</a><br /></td></tr>
<tr class="separator:adf9a755cba9ecf32628c8dfc327afb27 inherit pub_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_methods_a08589"><td colspan="2" onclick="javascript:toggleInherit('pro_methods_a08589')"><img src="closed.png" alt="-"/>&#160;Protected Member Functions inherited from <a class="el" href="a08589.html">DefaultParamHandler</a></td></tr>
<tr class="memitem:a2802c591e6d7fd3fd9cc35f5e875d0ba inherit pro_methods_a08589"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a2802c591e6d7fd3fd9cc35f5e875d0ba">defaultsToParam_</a> ()</td></tr>
<tr class="memdesc:a2802c591e6d7fd3fd9cc35f5e875d0ba inherit pro_methods_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Updates the parameters after the defaults have been set in the constructor.  <a href="a08589.html#a2802c591e6d7fd3fd9cc35f5e875d0ba">More...</a><br /></td></tr>
<tr class="separator:a2802c591e6d7fd3fd9cc35f5e875d0ba inherit pro_methods_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_static_methods_a08481"><td colspan="2" onclick="javascript:toggleInherit('pro_static_methods_a08481')"><img src="closed.png" alt="-"/>&#160;Static Protected Member Functions inherited from <a class="el" href="a08481.html">ProgressLogger</a></td></tr>
<tr class="memitem:a4e8b785e7dc68933455069c7730e344e inherit pro_static_methods_a08481"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="a08769.html">String</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a4e8b785e7dc68933455069c7730e344e">logTypeToFactoryName_</a> (<a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87c">LogType</a> type)</td></tr>
<tr class="memdesc:a4e8b785e7dc68933455069c7730e344e inherit pro_static_methods_a08481"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the name of the factory product used for this log type.  <a href="a08481.html#a4e8b785e7dc68933455069c7730e344e">More...</a><br /></td></tr>
<tr class="separator:a4e8b785e7dc68933455069c7730e344e inherit pro_static_methods_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_attribs_a08589"><td colspan="2" onclick="javascript:toggleInherit('pro_attribs_a08589')"><img src="closed.png" alt="-"/>&#160;Protected Attributes inherited from <a class="el" href="a08589.html">DefaultParamHandler</a></td></tr>
<tr class="memitem:a28c73e623c63a4fe3bfceb1ae8274f39 inherit pro_attribs_a08589"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08745.html">Param</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a28c73e623c63a4fe3bfceb1ae8274f39">param_</a></td></tr>
<tr class="memdesc:a28c73e623c63a4fe3bfceb1ae8274f39 inherit pro_attribs_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Container for current parameters.  <a href="a08589.html#a28c73e623c63a4fe3bfceb1ae8274f39">More...</a><br /></td></tr>
<tr class="separator:a28c73e623c63a4fe3bfceb1ae8274f39 inherit pro_attribs_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a761b1937850c141d0c6fff0e4d633a04 inherit pro_attribs_a08589"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08745.html">Param</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a761b1937850c141d0c6fff0e4d633a04">defaults_</a></td></tr>
<tr class="memdesc:a761b1937850c141d0c6fff0e4d633a04 inherit pro_attribs_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Container for default parameters. This member should be filled in the constructor of derived classes!  <a href="a08589.html#a761b1937850c141d0c6fff0e4d633a04">More...</a><br /></td></tr>
<tr class="separator:a761b1937850c141d0c6fff0e4d633a04 inherit pro_attribs_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4b0e20bc59c7a5afe016c69067e428c inherit pro_attribs_a08589"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="a08769.html">String</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#ac4b0e20bc59c7a5afe016c69067e428c">subsections_</a></td></tr>
<tr class="memdesc:ac4b0e20bc59c7a5afe016c69067e428c inherit pro_attribs_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Container for registered subsections. This member should be filled in the constructor of derived classes!  <a href="a08589.html#ac4b0e20bc59c7a5afe016c69067e428c">More...</a><br /></td></tr>
<tr class="separator:ac4b0e20bc59c7a5afe016c69067e428c inherit pro_attribs_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5a51f3f150b09068256195e934339a19 inherit pro_attribs_a08589"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08769.html">String</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a5a51f3f150b09068256195e934339a19">error_name_</a></td></tr>
<tr class="memdesc:a5a51f3f150b09068256195e934339a19 inherit pro_attribs_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">Name that is displayed in error messages during the parameter checking.  <a href="a08589.html#a5a51f3f150b09068256195e934339a19">More...</a><br /></td></tr>
<tr class="separator:a5a51f3f150b09068256195e934339a19 inherit pro_attribs_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7c3b265b697a57615da5cd3274780efd inherit pro_attribs_a08589"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#a7c3b265b697a57615da5cd3274780efd">check_defaults_</a></td></tr>
<tr class="memdesc:a7c3b265b697a57615da5cd3274780efd inherit pro_attribs_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">If this member is set to false no checking if parameters in done;.  <a href="a08589.html#a7c3b265b697a57615da5cd3274780efd">More...</a><br /></td></tr>
<tr class="separator:a7c3b265b697a57615da5cd3274780efd inherit pro_attribs_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aba444910e4fccb93f91bf7ab135f508a inherit pro_attribs_a08589"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08589.html#aba444910e4fccb93f91bf7ab135f508a">warn_empty_defaults_</a></td></tr>
<tr class="memdesc:aba444910e4fccb93f91bf7ab135f508a inherit pro_attribs_a08589"><td class="mdescLeft">&#160;</td><td class="mdescRight">If this member is set to false no warning is emitted when defaults are empty;.  <a href="a08589.html#aba444910e4fccb93f91bf7ab135f508a">More...</a><br /></td></tr>
<tr class="separator:aba444910e4fccb93f91bf7ab135f508a inherit pro_attribs_a08589"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_attribs_a08481"><td colspan="2" onclick="javascript:toggleInherit('pro_attribs_a08481')"><img src="closed.png" alt="-"/>&#160;Protected Attributes inherited from <a class="el" href="a08481.html">ProgressLogger</a></td></tr>
<tr class="memitem:a7c691d7de2a43bc5606e96cb15da3641 inherit pro_attribs_a08481"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08481.html#af67907baa897e9fb84df0cb89795b87c">LogType</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a7c691d7de2a43bc5606e96cb15da3641">type_</a></td></tr>
<tr class="separator:a7c691d7de2a43bc5606e96cb15da3641 inherit pro_attribs_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a79bacf60c4109d7517590441c7f85f9c inherit pro_attribs_a08481"><td class="memItemLeft" align="right" valign="top">time_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a79bacf60c4109d7517590441c7f85f9c">last_invoke_</a></td></tr>
<tr class="separator:a79bacf60c4109d7517590441c7f85f9c inherit pro_attribs_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac9469092d200ce7a427bea73b1086192 inherit pro_attribs_a08481"><td class="memItemLeft" align="right" valign="top"><a class="el" href="a08485.html">ProgressLoggerImpl</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#ac9469092d200ce7a427bea73b1086192">current_logger_</a></td></tr>
<tr class="separator:ac9469092d200ce7a427bea73b1086192 inherit pro_attribs_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="inherit_header pro_static_attribs_a08481"><td colspan="2" onclick="javascript:toggleInherit('pro_static_attribs_a08481')"><img src="closed.png" alt="-"/>&#160;Static Protected Attributes inherited from <a class="el" href="a08481.html">ProgressLogger</a></td></tr>
<tr class="memitem:a94eb3ef5ef94b83eb348dd00a2b53119 inherit pro_static_attribs_a08481"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="a00001.html#a12329">int</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="a08481.html#a94eb3ef5ef94b83eb348dd00a2b53119">recursion_depth_</a></td></tr>
<tr class="separator:a94eb3ef5ef94b83eb348dd00a2b53119 inherit pro_static_attribs_a08481"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Central class for simulation of mass spectrometry experiments. </p>
<p>This implementation is an extended and rewritten version of the concepts and ideas presented in:<br  />
 </p>
<p>Ole Schulz-Trieglaff, Nico Pfeifer, Clemens Gropl, Oliver Kohlbacher, and Knut Reinert.<br  />
 LC-MSsim - A simulation software for liquid chromatography mass spectrometry data.<br  />
 <em>BMC Bioinformatics</em> 9:423, 2008. </p>
<B>Parameters of this class are:</B><BR><BR>
<table border="1" style="border-style:solid; border-collapse:collapse; border-color:#c0c0c0;" width="100%" cellpadding="4">
<tr style="border-bottom:1px solid black; background:#fffff0"><th>Name</th><th>Type</th><th>Default</th><th>Restrictions</th><th>Description</th></tr>
<tr>
  <td style="vertical-align:top"><b>Digestion:enzyme</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">Trypsin</td>
  <td style="vertical-align:top">V8-E, leukocyte elastase, proline endopeptidase, glutamyl endopeptidase, Arg-C, Chymotrypsin, Chymotrypsin/P, Trypsin/P, Trypsin, Asp-N, Asp-N/B, Formic_acid, Lys-C, CNBr, Lys-C/P, PepsinA, TrypChymo, Asp-N_ambic, Lys-N, Arg-C/P, V8-DE, Glu-C+P, unspecific cleavage, proline-endopeptidase/HKR, Alpha-lytic protease, 2-iodobenzoate, Clostripain/P, elastase-trypsin-chymotrypsin, cyanogen-bromide, no cleavage, staphylococcal protease/D, iodosobenzoate, PepsinA + P</td><td style="vertical-align:top">Enzyme to use for digestion (select 'no cleavage' to skip digestion)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Digestion:model</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">naive</td>
  <td style="vertical-align:top">trained, naive</td><td style="vertical-align:top">The cleavage model to use for digestion. 'Trained' is based on a log likelihood model (see DOI:10.1021/pr060507u).</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Digestion:min_peptide_length</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">3</td>
  <td style="vertical-align:top">min: 1</td><td style="vertical-align:top">Minimum peptide length after digestion (shorter ones will be discarded)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Digestion:model_trained:threshold</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.5</td>
  <td style="vertical-align:top">min: -2.0 max: 4.0</td><td style="vertical-align:top">Model threshold for calling a cleavage. Higher values increase the number of cleavages. -2 will give no cleavages, +4 almost full cleavage.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Digestion:model_naive:missed_cleavages</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">1</td>
  <td style="vertical-align:top">min: 0</td><td style="vertical-align:top">Maximum number of missed cleavages considered. All possible resulting peptides will be created.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:rt_column</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">HPLC</td>
  <td style="vertical-align:top">none, HPLC, CE</td><td style="vertical-align:top">Modelling of an RT or CE column</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:auto_scale</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">true</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Scale predicted RT's/MT's to given 'total_gradient_time'? If 'true', for CE this means that 'CE:lenght_d', 'CE:length_total', 'CE:voltage' have no influence.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:total_gradient_time</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">2500.0</td>
  <td style="vertical-align:top">min: 1.0e-05</td><td style="vertical-align:top">The duration [s] of the gradient.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:sampling_rate</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">2.0</td>
  <td style="vertical-align:top">min: 0.01 max: 60.0</td><td style="vertical-align:top">Time interval [s] between consecutive scans</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:scan_window:min</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">500.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Start of RT Scan Window [s]</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:scan_window:max</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1500.0</td>
  <td style="vertical-align:top">min: 1.0</td><td style="vertical-align:top">End of RT Scan Window [s]</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:<span title="Random component that simulates technical/biological variation">variation</span>:feature_stddev</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">3</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Standard deviation of shift in retention time [s] from predicted model (applied to every single feature independently)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:<span title="Random component that simulates technical/biological variation">variation</span>:affine_offset</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Global offset in retention time [s] from predicted model</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:<span title="Random component that simulates technical/biological variation">variation</span>:affine_scale</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">1</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Global scaling in retention time from predicted model</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:column_condition:distortion</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">0</td>
  <td style="vertical-align:top">min: 0 max: 10</td><td style="vertical-align:top">Distortion of the elution profiles. Good presets are 0 for a perfect elution profile, 1 for a slightly distorted elution profile etc... For trapping instruments (e.g. Orbitrap) distortion should be >4.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:profile_shape:<span title="Width of the EGH elution shape, i.e. the sigma^2 parameter, which is computed using 'value' + rnd_cauchy('variance')">width</span>:value</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">9.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Width of the Exponential Gaussian Hybrid distribution shape of the elution profile. This does not correspond directly to the width in [s].</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:profile_shape:<span title="Width of the EGH elution shape, i.e. the sigma^2 parameter, which is computed using 'value' + rnd_cauchy('variance')">width</span>:variance</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.6</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Random component of the width (set to 0 to disable randomness), i.e. scale parameter for the lorentzian variation of the variance (Note: The scale parameter has to be >= 0).</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:profile_shape:<span title="Skewness of the EGH elution shape, i.e. the tau parameter, which is computed using 'value' + rnd_cauchy('variance')">skewness</span>:value</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.1</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Asymmetric component of the EGH. Higher absolute(!) values lead to more skewness (negative values cause fronting, positive values cause tailing). Tau parameter of the EGH, i.e. time constant of the exponential decay of the Exponential Gaussian Hybrid distribution shape of the elution profile.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:profile_shape:<span title="Skewness of the EGH elution shape, i.e. the tau parameter, which is computed using 'value' + rnd_cauchy('variance')">skewness</span>:variance</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.3</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Random component of skewness (set to 0 to disable randomness), i.e. scale parameter for the lorentzian variation of the time constant (Note: The scale parameter has to be > 0).</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:HPLC:model_file</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">examples/simulation/RTPredict.model</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">SVM model for retention time prediction</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:CE:pH</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">3.0</td>
  <td style="vertical-align:top">min: 0.0 max: 14.0</td><td style="vertical-align:top">pH of buffer</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:CE:alpha</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.5</td>
  <td style="vertical-align:top">min: 0.0 max: 1.0</td><td style="vertical-align:top">Exponent Alpha used to calculate mobility</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:CE:mu_eo</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">min: 0.0 max: 5.0</td><td style="vertical-align:top">Electroosmotic flow</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:CE:lenght_d</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">70.0</td>
  <td style="vertical-align:top">min: 0.0 max: 1000.0</td><td style="vertical-align:top">Length of capillary [cm] from injection site to MS</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:CE:length_total</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">75.0</td>
  <td style="vertical-align:top">min: 0.0 max: 1000.0</td><td style="vertical-align:top">Total length of capillary [cm]</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RT:CE:voltage</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1000.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Voltage applied to capillary</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Detectability:dt_simulation_on</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Modelling detectibility enabled? This can serve as a filter to remove peptides which ionize badly, thus reducing peptide count</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Detectability:min_detect</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.5</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Minimum peptide detectability accepted. Peptides with a lower score will be removed</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Detectability:dt_model_file</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">examples/simulation/DTPredict.model</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">SVM model for peptide detectability prediction</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Ionization:esi:ionized_residues</b></td>
  <td style="vertical-align:top">string list</td><td style="vertical-align:top">[Arg, Lys, His]</td>
  <td style="vertical-align:top">Ala, Cys, Asp, Glu, Phe, Gly, His, Ile, Lys, Leu, Met, Asn, Pro, Gln, Arg, Sec, Ser, Thr, Val, Trp, Tyr</td><td style="vertical-align:top">List of residues (as three letter code) that will be considered during ES ionization. The N-term is always assumed to carry a charge. This parameter will be ignored during MALDI ionization</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Ionization:esi:charge_impurity</b></td>
  <td style="vertical-align:top">string list</td><td style="vertical-align:top">[H+:1]</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">List of charged ions that contribute to charge with weight of occurrence (their sum is scaled to 1 internally), e.g. ['H:1'] or ['H:0.7' 'Na:0.3'], ['H:4' 'Na:1'] (which internally translates to ['H:0.8' 'Na:0.2'])</td>
</tr>
<tr>
  <td style="vertical-align:top"><i>Ionization:esi:max_impurity_set_size</i></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">3</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Maximal @#combinations of charge impurities allowed (each generating one feature) per charge state. E.g. assuming charge=3 and this parameter is 2, then we could choose to allow '3H+, 2H+Na+' features (given a certain 'charge_impurity' constraints), but no '3H+, 2H+Na+, 3Na+'</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Ionization:esi:ionization_probability</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.8</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Probability for the binomial distribution of the ESI charge states</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Ionization:maldi:ionization_probabilities</b></td>
  <td style="vertical-align:top">float list</td><td style="vertical-align:top">[0.9, 0.1, 0.0]</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">List of probabilities for different charge states (starting at charge=1, 2, ...) during MALDI ionization (the list must sum up to 1.0)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Ionization:mz:lower_measurement_limit</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">200.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Lower m/z detector limit</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Ionization:mz:upper_measurement_limit</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1200.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Upper m/z detector limit</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:enabled</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">true</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Enable RAW signal simulation? (select 'false' if you only need feature-maps)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:peak_shape</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">Gaussian</td>
  <td style="vertical-align:top">Gaussian, Lorentzian</td><td style="vertical-align:top">Peak Shape used around each isotope peak (be aware that the area under the curve is constant for both types, but the maximal height will differ (~ 2:3 = Lorentz:Gaussian) due to the wider base of the Lorentzian</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:resolution:value</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">50000</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Instrument resolution at 400 Th</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:resolution:type</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">linear</td>
  <td style="vertical-align:top">constant, linear, sqrt</td><td style="vertical-align:top">How does resolution change with increasing m/z?! QTOFs usually show 'constant' behavior, FTs have linear degradation, and on Orbitraps the resolution decreases with square root of mass</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Baseline modeling for MALDI ionization">baseline</span>:scaling</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Scale of baseline. Set to 0 to disable simulation of baseline</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Baseline modeling for MALDI ionization">baseline</span>:shape</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.5</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">The baseline is modeled by an exponential probability density function (pdf) with f(x) = shape*e^(- shape*x)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:mz:sampling_points</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">3</td>
  <td style="vertical-align:top">min: 2</td><td style="vertical-align:top">Number of raw data points per FWHM of the peak</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:contaminants:file</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">examples/simulation/contaminants.csv</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Contaminants file with sum formula and absolute RT interval. See 'OpenMS/examples/simulation/contaminants.txt' for details</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Random components that simulate biological and technical variations of the simulated data">variation</span>:<span title="Shifts in mass to charge dimension of the simulated signals">mz</span>:error_mean</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Average systematic m/z error (in Da)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Random components that simulate biological and technical variations of the simulated data">variation</span>:<span title="Shifts in mass to charge dimension of the simulated signals">mz</span>:error_stddev</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Standard deviation for m/z errors. Set to 0 to disable simulation of m/z errors</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Random components that simulate biological and technical variations of the simulated data">variation</span>:<span title="Variations in intensity to model randomness in feature intensity">intensity</span>:scale</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">100.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Constant scale factor of the feature intensity. Set to 1.0 to get the real intensity values provided in the FASTA file</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Random components that simulate biological and technical variations of the simulated data">variation</span>:<span title="Variations in intensity to model randomness in feature intensity">intensity</span>:scale_stddev</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Standard deviation of peak intensity (relative to the scaled peak height). Set to 0 to get simple rescaled intensities</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Parameters modeling noise in mass spectrometry measurements">noise</span>:<span title="Parameters of Poisson and Exponential for shot noise modeling (set :rate OR :mean = 0 to disable)">shot</span>:rate</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Poisson rate of shot noise per unit m/z (random peaks in m/z, where the number of peaks per unit m/z follows a Poisson distribution). Set this to 0 to disable simulation of shot noise</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Parameters modeling noise in mass spectrometry measurements">noise</span>:<span title="Parameters of Poisson and Exponential for shot noise modeling (set :rate OR :mean = 0 to disable)">shot</span>:intensity-mean</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Shot noise intensity mean (exponentially distributed with given mean)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Parameters modeling noise in mass spectrometry measurements">noise</span>:<span title="Parameters of Gaussian distribution for white noise modeling (set :mean AND :stddev = 0 to disable). No new peaks are generated; only intensity of existing ones is changed">white</span>:mean</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Mean value of white noise (Gaussian) being added to each *measured* signal intensity</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Parameters modeling noise in mass spectrometry measurements">noise</span>:<span title="Parameters of Gaussian distribution for white noise modeling (set :mean AND :stddev = 0 to disable). No new peaks are generated; only intensity of existing ones is changed">white</span>:stddev</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Standard deviation of white noise being added to each *measured* signal intensity</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Parameters modeling noise in mass spectrometry measurements">noise</span>:<span title="Parameters of Gaussian distribution for detector noise modeling (set :mean AND :stddev = 0 to disable). If enabled, ALL possible m/z positions (up to sampling frequency of detector) will receive an intensity increase/decrease according to the specified Gaussian intensity distribution (similar to a noisy baseline)">detector</span>:mean</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Mean intensity value of the detector noise (Gaussian distribution)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawSignal:<span title="Parameters modeling noise in mass spectrometry measurements">noise</span>:<span title="Parameters of Gaussian distribution for detector noise modeling (set :mean AND :stddev = 0 to disable). If enabled, ALL possible m/z positions (up to sampling frequency of detector) will receive an intensity increase/decrease according to the specified Gaussian intensity distribution (similar to a noisy baseline)">detector</span>:stddev</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Standard deviation of the detector noise (Gaussian distribution)</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:status</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">disabled</td>
  <td style="vertical-align:top">disabled, precursor, MS^E</td><td style="vertical-align:top">Create Tandem-MS scans?</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:tandem_mode</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">0</td>
  <td style="vertical-align:top">min: 0 max: 2</td><td style="vertical-align:top">Algorithm to generate the tandem-MS spectra. 0 - fixed intensities, 1 - SVC prediction (abundant/missing), 2 - SVR prediction of peak intensity <BR></td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:svm_model_set_file</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">examples/simulation/SvmModelSet.model</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">File containing the filenames of SVM Models for different charge variants</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ms2_spectra_per_rt_bin</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">5</td>
  <td style="vertical-align:top">min: 1</td><td style="vertical-align:top">Number of allowed MS/MS spectra in a retention time bin.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:min_mz_peak_distance</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">2.0</td>
  <td style="vertical-align:top">min: 1.0e-04</td><td style="vertical-align:top">The minimal distance (in Th) between two peaks for concurrent selection for fragmentation. Also used to define the m/z width of an exclusion window (distance +/- from m/z of precursor). If you set this lower than the isotopic envelope of a peptide, you might get multiple fragment spectra pointing to the same precursor.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:mz_isolation_window</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">2.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">All peaks within a mass window (in Th) of a selected peak are also selected for fragmentation.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:exclude_overlapping_peaks</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If true, overlapping or nearby peaks (within 'min_mz_peak_distance') are excluded for selection.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:charge_filter</b></td>
  <td style="vertical-align:top">int list</td><td style="vertical-align:top">[2, 3]</td>
  <td style="vertical-align:top">min: 1 max: 5</td><td style="vertical-align:top">Charges considered for MS2 fragmentation.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:Exclusion:use_dynamic_exclusion</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If true dynamic exclusion is applied.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:Exclusion:exclusion_time</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">100.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">The time (in seconds) a feature is excluded.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:max_list_size</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">1000</td>
  <td style="vertical-align:top">min: 1</td><td style="vertical-align:top">The maximal number of precursors in the inclusion list.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:rt:min_rt</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">960.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Minimal rt in seconds.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:rt:max_rt</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">3840.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Maximal rt in seconds.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:rt:rt_step_size</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">30.0</td>
  <td style="vertical-align:top">min: 1.0</td><td style="vertical-align:top">rt step size in seconds.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:rt:rt_window_size</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">100</td>
  <td style="vertical-align:top">min: 1</td><td style="vertical-align:top">rt window size in seconds.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:min_protein_id_probability</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.95</td>
  <td style="vertical-align:top">min: 0.0 max: 1.0</td><td style="vertical-align:top">Minimal protein probability for a protein to be considered identified.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:min_pt_weight</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.5</td>
  <td style="vertical-align:top">min: 0.0 max: 1.0</td><td style="vertical-align:top">Minimal pt weight of a precursor</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:min_mz</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">500.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Minimal mz to be considered in protein based LP formulation.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:max_mz</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">5000.0</td>
  <td style="vertical-align:top">min: 0.0</td><td style="vertical-align:top">Minimal mz to be considered in protein based LP formulation.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:use_peptide_rule</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Use peptide rule instead of minimal protein id probability</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:min_peptide_ids</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">2</td>
  <td style="vertical-align:top">min: 1</td><td style="vertical-align:top">If use_peptide_rule is true, this parameter sets the minimal number of peptide ids for a protein id</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:Precursor:ProteinBasedInclusion:thresholds:min_peptide_probability</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.95</td>
  <td style="vertical-align:top">min: 0.0 max: 1.0</td><td style="vertical-align:top">If use_peptide_rule is true, this parameter sets the minimal probability for a peptide to be safely identified</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:MS_E:add_single_spectra</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If true, the MS2 spectra for each peptide signal are included in the output (might be a lot). They will have a meta value 'MSE_DebugSpectrum' attached, so they can be filtered out. Native MS_E spectra will have 'MSE_Spectrum' instead.</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_isotopes</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If set to 'true' isotope peaks of the product ion peaks are added</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:max_isotope</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">2</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Defines the maximal isotopic peak which is added if 'add_isotopes' is 'true'</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_metainfo</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Adds the type of peaks as metainfo to the peaks, like y8+, [M-H2O+2H]++</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_losses</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Adds common losses to those ion expect to have them, only water and ammonia loss is considered</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_precursor_peaks</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Adds peaks of the unfragmented precursor ion to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_all_precursor_charges</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Adds precursor peaks with all charges in the given range</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_abundant_immonium_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add most abundant immonium ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_first_prefix_ion</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If set to true e.g. b1 ions are added</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_y_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">true</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of y-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_b_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">true</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of b-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_a_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of a-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_c_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of c-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_x_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of  x-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:add_z_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of z-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:y_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the y-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:b_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the b-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:a_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the a-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:c_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the c-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:x_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the x-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:z_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the z-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:relative_loss_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.1</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of loss ions, in relation to the intact ion intensity</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:precursor_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the precursor peak</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:precursor_H2O_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the H2O loss peak of the precursor</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:Simple:precursor_NH3_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the NH3 loss peak of the precursor</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:add_isotopes</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If set to 1 isotope peaks of the product ion peaks are added</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:max_isotope</b></td>
  <td style="vertical-align:top">int</td><td style="vertical-align:top">2</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Defines the maximal isotopic peak which is added, add_isotopes must be set to 1</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:add_metainfo</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Adds the type of peaks as metainfo to the peaks, like y8+, [M-H2O+2H]++</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:add_first_prefix_ion</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">If set to true e.g. b1 ions are added</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_y_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of y-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_y2_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of y-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_b_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of b-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_b2_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of b-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_a_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of a-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_c_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of c-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_x_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of  x-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_z_ions</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Add peaks of z-ions to the spectrum</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:hide_losses</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">false</td>
  <td style="vertical-align:top">true, false</td><td style="vertical-align:top">Adds common losses to those ion expect to have them, only water and ammonia loss is considered</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:y_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the y-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:b_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the b-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:a_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the a-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:c_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the c-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:x_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the x-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:z_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">1.0</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of the z-ions</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>RawTandemSignal:TandemSim:SVM:relative_loss_intensity</b></td>
  <td style="vertical-align:top">float</td><td style="vertical-align:top">0.1</td>
  <td style="vertical-align:top">&nbsp;</td><td style="vertical-align:top">Intensity of loss ions, in relation to the intact ion intensity</td>
</tr>
<tr>
  <td style="vertical-align:top"><b>Global:ionization_type</b></td>
  <td style="vertical-align:top">string</td><td style="vertical-align:top">ESI</td>
  <td style="vertical-align:top">MALDI, ESI</td><td style="vertical-align:top">Type of Ionization (MALDI or ESI)</td>
</tr>
</table>
<br>
<b>Note:</b>
<UL style="margin-top:0px;">
  <LI> If a section name is documented, the documentation is displayed as tooltip.
  <LI> Advanced parameter names are italic.
</UL>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="a10793.html" title="Simulates peptide detectability.">DetectabilitySimulation</a> </dd>
<dd>
<a class="el" href="a10797.html" title="Simulates protein digestion.">DigestSimulation</a> </dd>
<dd>
<a class="el" href="a10817.html" title="Simulates Protein ionization.">IonizationSimulation</a> </dd>
<dd>
<a class="el" href="a10849.html" title="Simulates MS signals for a given set of peptides.">RawMSSignalSimulation</a> </dd>
<dd>
<a class="el" href="a10861.html" title="Simulates/Predicts retention times for peptides or peptide separation.">RTSimulation</a> </dd></dl>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a id="a063de8c0272b877c59781b899546e057"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a063de8c0272b877c59781b899546e057">&#9670;&nbsp;</a></span>MSSim()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a10845.html">MSSim</a> </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Default constructor. </p>

</div>
</div>
<a id="a91363e6770cbe9bade26f523bf9662c7"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a91363e6770cbe9bade26f523bf9662c7">&#9670;&nbsp;</a></span>~MSSim()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">~<a class="el" href="a10845.html">MSSim</a> </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">override</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Destructor. </p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a id="af4298dd2e9804f7799b6f265b15c161a"></a>
<h2 class="memtitle"><span class="permalink"><a href="#af4298dd2e9804f7799b6f265b15c161a">&#9670;&nbsp;</a></span>createFeatureMap_()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void createFeatureMap_ </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="a05853.html#af8f836c78cff89a3bc6f1b4920100ace">SimTypes::SampleProteins</a> &amp;&#160;</td>
          <td class="paramname"><em>peptides</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a> &amp;&#160;</td>
          <td class="paramname"><em>features</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a>&#160;</td>
          <td class="paramname"><em>map_index</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Convert a list of peptides with given abundance values into a <a class="el" href="a10057.html" title="A container for features.">FeatureMap</a>. </p>

</div>
</div>
<a id="a22d35cb0ae2a4d19e44f6ac8cd4f57c9"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a22d35cb0ae2a4d19e44f6ac8cd4f57c9">&#9670;&nbsp;</a></span>getChargeConsensus()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a10021.html">ConsensusMap</a>&amp; getChargeConsensus </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the charge consensus map of simulated features. </p>

</div>
</div>
<a id="a19f1fb58afc66fd144252470ddb3d176"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a19f1fb58afc66fd144252470ddb3d176">&#9670;&nbsp;</a></span>getContaminants()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const <a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a>&amp; getContaminants </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the contaminants feature map of simulated features. </p>

</div>
</div>
<a id="aba841147967347f8b403f39b0ab7019b"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aba841147967347f8b403f39b0ab7019b">&#9670;&nbsp;</a></span>getExperiment()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const <a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a>&amp; getExperiment </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the simulated experiment. </p>

</div>
</div>
<a id="ae587207f751bcfe0ba94031d5b75654b"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae587207f751bcfe0ba94031d5b75654b">&#9670;&nbsp;</a></span>getFeatureIdentifications()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void getFeatureIdentifications </td>
          <td>(</td>
          <td class="paramtype">std::vector&lt; <a class="el" href="a10677.html">ProteinIdentification</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>proteins</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::vector&lt; <a class="el" href="a10645.html">PeptideIdentification</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>peptides</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the simulated identifications (proteins and peptides) from feature annotations. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">proteins</td><td>Will be filled with a single <a class="el" href="a10677.html" title="Representation of a protein identification run.">ProteinIdentification</a> holding all ProteinHits. </td></tr>
    <tr><td class="paramname">peptides</td><td>Will be filled with PeptideIdentifications for all simulated features. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a id="aaa266aa79d1729ff264838a2f37b1bd4"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aaa266aa79d1729ff264838a2f37b1bd4">&#9670;&nbsp;</a></span>getIdentifications()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void getIdentifications </td>
          <td>(</td>
          <td class="paramtype">std::vector&lt; <a class="el" href="a10677.html">ProteinIdentification</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>proteins</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::vector&lt; <a class="el" href="a10645.html">PeptideIdentification</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>peptides</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the simulated identifications (proteins and peptides) </p>
<p>If an MS2 signal has been simulated, identifications are generated based on their corresponding MS2 spectra. If MS2 simulation has been disabled, peptide IDs are generated from feature annotations. Otherwise, the -out_id idXML file would be empty if MS2 simulation is disabled (which is the default).</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">proteins</td><td>Will be filled with a single <a class="el" href="a10677.html" title="Representation of a protein identification run.">ProteinIdentification</a> holding all ProteinHits. </td></tr>
    <tr><td class="paramname">peptides</td><td>Will be filled with PeptideIdentifications. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a id="aac38d440d2e98f886eb076696c8c507b"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aac38d440d2e98f886eb076696c8c507b">&#9670;&nbsp;</a></span>getLabelingConsensus()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a10021.html">ConsensusMap</a>&amp; getLabelingConsensus </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the labeling consensus map of simulated features. </p>

</div>
</div>
<a id="a3a22e0505a27c8a2a39970888c7b99fc"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a3a22e0505a27c8a2a39970888c7b99fc">&#9670;&nbsp;</a></span>getMS2Identifications()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void getMS2Identifications </td>
          <td>(</td>
          <td class="paramtype">std::vector&lt; <a class="el" href="a10677.html">ProteinIdentification</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>proteins</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::vector&lt; <a class="el" href="a10645.html">PeptideIdentification</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>peptides</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the simulated MS2 identifications (proteins and peptides) </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">proteins</td><td>Will be filled with a single <a class="el" href="a10677.html" title="Representation of a protein identification run.">ProteinIdentification</a> holding all ProteinHits used in the simulated MS2 spectra. </td></tr>
    <tr><td class="paramname">peptides</td><td>Will be filled with PeptideIdentifications for each simulated MS2 spectra holding all contributing peptides scored by their intensity contribution. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a id="a301e5b8ce00c61b1b4efaf09f213b619"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a301e5b8ce00c61b1b4efaf09f213b619">&#9670;&nbsp;</a></span>getParameters()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a08745.html">Param</a> getParameters </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Returns the default parameters for simulation including the labeling technique with name <code>labeling_name</code>. </p>

</div>
</div>
<a id="a960ecae2325e1418a4371a0b1a9ac083"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a960ecae2325e1418a4371a0b1a9ac083">&#9670;&nbsp;</a></span>getPeakMap()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const <a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a>&amp; getPeakMap </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the picked (centroided) experiment. </p>

</div>
</div>
<a id="a8c12281c11eedb6817ab040852b01c72"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a8c12281c11eedb6817ab040852b01c72">&#9670;&nbsp;</a></span>getSimulatedFeatures()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const <a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a>&amp; getSimulatedFeatures </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Access the simulated features. </p>

</div>
</div>
<a id="a1ff9361d142cb1722cf7758d1b9c5ea2"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a1ff9361d142cb1722cf7758d1b9c5ea2">&#9670;&nbsp;</a></span>simulate()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void simulate </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="a05853.html#a00c57778f72fd6a465d82be2bb231dfe">SimTypes::MutableSimRandomNumberGeneratorPtr</a>&#160;</td>
          <td class="paramname"><em>rnd_gen</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="a05853.html#abd6a489f13876548dd540ccefd087b0b">SimTypes::SampleChannels</a> &amp;&#160;</td>
          <td class="paramname"><em>peptides</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>General purpose function to simulate a mass spectrometry run. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">rnd_gen</td><td>random number generator which will be passed to the different classes </td></tr>
    <tr><td class="paramname">peptides</td><td>List of peptides and abundances that will be simulated </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a id="a63efff66245e023f272fa69fee0631fe"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a63efff66245e023f272fa69fee0631fe">&#9670;&nbsp;</a></span>syncParams_()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void syncParams_ </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="a08745.html">Param</a> &amp;&#160;</td>
          <td class="paramname"><em>p</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>to_outer</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>handle global params </p>

</div>
</div>
<a id="abff6e392ce6da7af8f083397494a7971"></a>
<h2 class="memtitle"><span class="permalink"><a href="#abff6e392ce6da7af8f083397494a7971">&#9670;&nbsp;</a></span>updateMembers_()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void updateMembers_ </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">override</span><span class="mlabel">protected</span><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Synchronize members with param class. </p>

<p>Reimplemented from <a class="el" href="a08589.html#a776f7572fc4f466e9527ca233471920d">DefaultParamHandler</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a id="a9f04b32b80833aabd4ab8bc8c2c9ecb4"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a9f04b32b80833aabd4ab8bc8c2c9ecb4">&#9670;&nbsp;</a></span>consensus_map_</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a10021.html">ConsensusMap</a> consensus_map_</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Holds consensus ground-truth about the charge associations. </p>

</div>
</div>
<a id="a4c6c540ae92e189dbbf8dbb5bb0fa2a3"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a4c6c540ae92e189dbbf8dbb5bb0fa2a3">&#9670;&nbsp;</a></span>contaminants_map_</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a05853.html#a9396dc872898d760f035465bc56585d8">SimTypes::FeatureMapSim</a> contaminants_map_</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Holds the ground-truth on generated contaminants. </p>

</div>
</div>
<a id="a6bf6c31430e50c6d99744da96cdca044"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a6bf6c31430e50c6d99744da96cdca044">&#9670;&nbsp;</a></span>experiment_</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a> experiment_</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Holds the simulated data. </p>

</div>
</div>
<a id="af0d97a3e4120e8f57d2a0f2c5e10b4df"></a>
<h2 class="memtitle"><span class="permalink"><a href="#af0d97a3e4120e8f57d2a0f2c5e10b4df">&#9670;&nbsp;</a></span>feature_maps_</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a05853.html#a5377a8d620ef376a0b1eb2503b847e79">SimTypes::FeatureMapSimVector</a> feature_maps_</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Holds the ground-truth on generated features. </p>

</div>
</div>
<a id="a813cce92a0e34c5f83c0798cae8a61ae"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a813cce92a0e34c5f83c0798cae8a61ae">&#9670;&nbsp;</a></span>labeler_</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a10821.html">BaseLabeler</a>* labeler_</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Labeling functionality. </p>

</div>
</div>
<a id="adc073663b8fca88af1da2a8bd16c8aad"></a>
<h2 class="memtitle"><span class="permalink"><a href="#adc073663b8fca88af1da2a8bd16c8aad">&#9670;&nbsp;</a></span>peak_map_</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="a05853.html#a40bccd8402f1abc9eaa68422b0085ce0">SimTypes::MSSimExperiment</a> peak_map_</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Holds the ground-truth on generated peaks positions and intensities. </p>

</div>
</div>
</div><!-- contents -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:49 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
