<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: TOPP documentation</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">TOPP documentation </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>'TOPP - The OpenMS Proteomics Pipeline' is a pipeline for the analysis of HPLC-MS data. It consists of several small applications that can be chained to create analysis pipelines tailored for a specific problem.</p>
<p>The TOPP tools are divided into several subgroups:</p>
<p><b>Graphical Tools</b></p><ul>
<li><a class="el" href="a16257.html">TOPPView</a> - A viewer for mass spectrometry data.</li>
<li><a class="el" href="a16256.html">TOPPAS</a> - An assistant for GUI-driven TOPP workflow design.</li>
<li><a class="el" href="a16255.html">INIFileEditor</a> - An editor for OpenMS configuration files.</li>
</ul>
<p><b>File Handling</b></p><ul>
<li><a class="el" href="a16266.html">DTAExtractor</a> - Extracts spectra of an MS run file to several files in DTA format.</li>
<li><a class="el" href="a16283.html">FileConverter</a> - Converts between different MS file formats.</li>
<li><a class="el" href="a16284.html">FileFilter</a> - Extracts or manipulates portions of data from peak, feature or consensus feature files.</li>
<li><a class="el" href="a16285.html">FileInfo</a> - Shows basic information about the file, such as data ranges and file type.</li>
<li><a class="el" href="a16286.html">FileMerger</a> - Merges several MS files into one file.</li>
<li><a class="el" href="a16294.html">IDMerger</a> - Merges several protein/peptide identification files into one file.</li>
<li><a class="el" href="a16296.html">IDRipper</a> - Splits protein/peptide identifications according their file-origin.</li>
<li><a class="el" href="a16291.html">IDFileConverter</a> - Converts identification engine file formats.</li>
<li><a class="el" href="a16308.html">MapStatistics</a> - Extract extended statistics on the features of a map for quality control.</li>
<li><a class="el" href="a16425.html">TargetedFileConverter</a> - Converts targeted files (such as tsv or TraML files).</li>
<li><a class="el" href="a16356.html">TextExporter</a> - Exports various XML formats to a text file.</li>
<li><a class="el" href="a16316.html">MzTabExporter</a> - Exports various XML formats to an mzTab file.</li>
</ul>
<p><b>Signal Processing and Preprocessing</b></p><ul>
<li><a class="el" href="a16258.html">BaselineFilter</a> - Removes the baseline from profile spectra using a top-hat filter.</li>
<li><a class="el" href="a16300.html">InternalCalibration</a> - Applies an internal mass calibration.</li>
<li><a class="el" href="a16269.html">ExternalCalibration</a> - Applies an external mass calibration.</li>
<li><a class="el" href="a16306.html">MapNormalizer</a> - Normalizes peak intensities in an MS run.</li>
<li><a class="el" href="a16312.html">MassTraceExtractor</a> - Annotates mass traces in centroided LC-MS maps.</li>
<li><a class="el" href="a16317.html">NoiseFilterGaussian</a> - Removes noise from profile spectra by using different smoothing techniques.</li>
<li><a class="el" href="a16318.html">NoiseFilterSGolay</a> - Removes noise from profile spectra by using different smoothing techniques.</li>
<li><a class="el" href="a16327.html">PeakPickerHiRes</a> - Finds mass spectrometric peaks in profile mass spectra.</li>
<li><a class="el" href="a16328.html">PeakPickerWavelet</a> - Finds mass spectrometric peaks in profile mass spectra.</li>
<li><a class="el" href="a16334.html">PrecursorMassCorrector</a> - Correct the precursor entries of tandem MS scans.</li>
<li><a class="el" href="a16290.html">HighResPrecursorMassCorrector</a> - Correct the precursor entries of tandem MS scans for high resolution data.</li>
<li><a class="el" href="a16341.html">Resampler</a> - Transforms an LC-MS map into an equally-spaced (in RT and m/z) map.</li>
<li><a class="el" href="a16346.html">SpectraFilterBernNorm</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16347.html">SpectraFilterMarkerMower</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16348.html">SpectraFilterNLargest</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16349.html">SpectraFilterNormalizer</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16350.html">SpectraFilterParentPeakMower</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16351.html">SpectraFilterScaler</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16352.html">SpectraFilterSqrtMower</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16353.html">SpectraFilterThresholdMower</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16354.html">SpectraFilterWindowMower</a> - Applies a filter to peak spectra.</li>
<li><a class="el" href="a16355.html">SpectraMerger</a> - Merges spectra from an LC-MS map, either by precursor or by RT blocks.</li>
<li><a class="el" href="a16357.html">TOFCalibration</a> - Applies time of flight mass calibration.</li>
</ul>
<p><b>Quantitation</b></p><ul>
<li><a class="el" href="a16265.html">Decharger</a> - Decharges and merges different feature charge variants of the same chemical entity.</li>
<li><a class="el" href="a16267.html">EICExtractor</a> - Quantifies signals at given positions in (raw or picked) LC-MS maps.</li>
<li><a class="el" href="a16271.html">FeatureFinderCentroided</a> - Detects two-dimensional features in centroided LC-MS data.</li>
<li><a class="el" href="a16272.html">FeatureFinderIdentification</a> - Detects two-dimensional features in MS1 data based on peptide identifications.</li>
<li><a class="el" href="a16273.html">FeatureFinderIsotopeWavelet</a> - Detects two-dimensional features in uncentroided (=raw) LC-MS data.</li>
<li><a class="el" href="a16274.html">FeatureFinderMetabo</a> - Detects two-dimensional features in centroided LC-MS data of metabolites.</li>
<li><a class="el" href="a16275.html">FeatureFinderMRM</a> - Quantifies features LC-MS/MS MRM data.</li>
<li><a class="el" href="a16276.html">FeatureFinderMultiplex</a> - Identifies peptide multiplets (pairs, triplets etc., e.g. for SILAC or Dimethyl labeling) and determines their relative abundance.</li>
<li><a class="el" href="a16301.html">IsobaricAnalyzer</a> - Extracts and normalizes TMT and iTRAQ information from an MS experiment.</li>
<li><a class="el" href="a16336.html">ProteinQuantifier</a> - Computes protein abundances from annotated feature/consensus maps.</li>
<li><a class="el" href="a16337.html">ProteinResolver</a> - A peptide-centric algorithm for protein inference.</li>
<li><a class="el" href="a16344.html">SeedListGenerator</a> - Generates seed lists for feature detection.</li>
</ul>
<p><b>Map Alignment</b></p><ul>
<li><a class="el" href="a16263.html">ConsensusMapNormalizer</a> - Normalizes maps of one consensusXML file (after linking).</li>
<li><a class="el" href="a16303.html">MapAlignerIdentification</a> - Corrects retention time distortions between maps based on common peptide identifications.</li>
<li><a class="el" href="a16304.html">MapAlignerPoseClustering</a> - Corrects retention time distortions between maps using a pose clustering approach.</li>
<li><a class="el" href="a16305.html">MapAlignerSpectrum</a> - Corrects retention time distortions between maps by spectrum alignment.</li>
<li><a class="el" href="a16307.html">MapRTTransformer</a> - Applies retention time transformations to maps.</li>
<li><a class="el" href="a16278.html">FeatureLinkerLabeled</a> - Groups corresponding isotope-labeled features in a feature map.</li>
<li><a class="el" href="a16279.html">FeatureLinkerUnlabeled</a> - Groups corresponding features from multiple maps.</li>
<li><a class="el" href="a16281.html">FeatureLinkerUnlabeledQT</a> - Groups corresponding features from multiple maps using a QT clustering approach.</li>
<li><a class="el" href="a16280.html">FeatureLinkerUnlabeledKD</a> - Groups corresponding features from multiple maps using a KD tree.</li>
</ul>
<p><b>Protein/Peptide Identification</b></p><ul>
<li><a class="el" href="a16259.html">CometAdapter</a> - Identifies MS/MS spectra using Comet (external).</li>
<li><a class="el" href="a16260.html">CompNovo</a> - Performs a peptide/protein identification with the CompNovo engine.</li>
<li><a class="el" href="a16261.html">CompNovoCID</a> - Performs a peptide/protein identification with the CompNovo engine in CID mode.</li>
<li><a class="el" href="a16264.html">CruxAdapter</a> - Identifies peptides in MS/MS spectra via Crux and tide-search (external).</li>
<li><a class="el" href="a16299.html">InspectAdapter</a> - Identifies MS/MS spectra using Inspect (external).</li>
<li><a class="el" href="a16310.html">MascotAdapter</a> - Identifies MS/MS spectra using Mascot (external).</li>
<li><a class="el" href="a16311.html">MascotAdapterOnline</a> - Identifies MS/MS spectra using Mascot (external).</li>
<li><a class="el" href="a16314.html">MSGFPlusAdapter</a> - Identifies MS/MS spectra using MSGFPlus (external).</li>
<li><a class="el" href="a16315.html">MyriMatchAdapter</a> - Identifies MS/MS spectra using MyriMatch (external).</li>
<li><a class="el" href="a16319.html">OMSSAAdapter</a> - Identifies MS/MS spectra using OMSSA (external).</li>
<li><a class="el" href="a16329.html">PepNovoAdapter</a> - Identifies MS/MS spectra using PepNovo (external).</li>
<li><a class="el" href="a16345.html">SpecLibSearcher</a> - Identifies peptide MS/MS spectra by spectral matching with a searchable spectral library.</li>
<li><a class="el" href="a16420.html">SimpleSearchEngine</a> - A simple database search engine for annotating MS/MS spectra.</li>
<li><a class="el" href="a16358.html">XTandemAdapter</a> - Identifies MS/MS spectra using XTandem (external).</li>
</ul>
<p><b>Protein/Peptide Processing</b></p><ul>
<li><a class="el" href="a16262.html">ConsensusID</a> - Computes a consensus identification from peptide identifications of several identification engines.</li>
<li><a class="el" href="a16270.html">FalseDiscoveryRate</a> - Estimates the false discovery rate on peptide and protein level using decoy searches.</li>
<li><a class="el" href="a16282.html">FidoAdapter</a> - Runs the protein inference engine Fido.</li>
<li><a class="el" href="a16250.html">IDConflictResolver</a> - Resolves ambiguous annotations of features with peptide identifications.</li>
<li><a class="el" href="a16292.html">IDFilter</a> - Filters results from protein or peptide identification engines based on different criteria.</li>
<li><a class="el" href="a16293.html">IDMapper</a> - Assigns protein/peptide identifications to feature or consensus features.</li>
<li><a class="el" href="a16295.html">IDPosteriorErrorProbability</a> - Estimates posterior error probabilities using a mixture model.</li>
<li><a class="el" href="a16297.html">IDRTCalibration</a> - Can be used to calibrate RTs of peptide hits linearly to standards.</li>
<li><a class="el" href="a16302.html">LuciphorAdapter</a> - Scores potential phosphorylation sites in order to localize the most probable sites.</li>
<li><a class="el" href="a16330.html">PeptideIndexer</a> - Refreshes the protein references for all peptide hits.</li>
<li><a class="el" href="a16332.html">PhosphoScoring</a> - Scores potential phosphorylation sites in order to localize the most probable sites.</li>
<li><a class="el" href="a16335.html">ProteinInference</a> - Infer proteins from a list of (high-confidence) peptides.</li>
<li><a class="el" href="a16331.html">PercolatorAdapter</a> - Applies the percolator algorithm to protein/peptide identifications.</li>
</ul>
<p><b>Targeted Experiments and OpenSWATH</b></p><ul>
<li><a class="el" href="a16402.html">OpenSwathWorkflow</a> - Complete workflow to run OpenSWATH.</li>
<li><a class="el" href="a16360.html">AssayGeneratorMetabo</a> - Generates an assay library using DDA data (Metabolomics).</li>
<li><a class="el" href="a16298.html">InclusionExclusionListCreator</a> - Creates inclusion and/or exclusion lists for LC-MS/MS experiments.</li>
<li><a class="el" href="a16313.html">MRMMapper</a> - MRMMapper maps measured chromatograms (mzML) and the transitions used (TraML).</li>
<li><a class="el" href="a16390.html">MRMTransitionGroupPicker</a> - Picks peaks in MRM chromatograms.</li>
<li><a class="el" href="a16333.html">PrecursorIonSelector</a> - A tool for precursor ion selection based on identification results.</li>
<li><a class="el" href="a16320.html">OpenSwathAnalyzer</a> - Picks peaks and finds features in an SRM experiment.</li>
<li><a class="el" href="a16321.html">OpenSwathAssayGenerator</a> - Generates filtered and optimized assays using TraML files.</li>
<li><a class="el" href="a16322.html">OpenSwathChromatogramExtractor</a> - Extract chromatograms (XIC) from a MS2 map file.</li>
<li><a class="el" href="a16323.html">OpenSwathConfidenceScoring</a> - Computes confidence scores for <a class="el" href="a05820.html">OpenSwath</a> results.</li>
<li><a class="el" href="a16324.html">OpenSwathDecoyGenerator</a> - Generates decoys according to different models for a specific TraML.</li>
<li><a class="el" href="a16325.html">OpenSwathFeatureXMLToTSV</a> - Converts a featureXML to a tsv.</li>
<li><a class="el" href="a16326.html">OpenSwathRTNormalizer</a> - This tool will align an SRM / SWATH run to a normalized retention time space.</li>
</ul>
<p><b>Peptide Property Prediction</b></p><ul>
<li><a class="el" href="a16338.html">PTModel</a> - Trains a model for the prediction of proteotypic peptides from a training set.</li>
<li><a class="el" href="a16339.html">PTPredict</a> - Predicts the likelihood of peptides to be proteotypic using a model trained by PTModel.</li>
<li><a class="el" href="a16342.html">RTModel</a> - Trains a model for the retention time prediction of peptides from a training set.</li>
<li><a class="el" href="a16343.html">RTPredict</a> - Predicts retention times for peptides using a model trained by RTModel.</li>
</ul>
<p><b>Misc</b></p><ul>
<li><a class="el" href="a16340.html">QualityControl</a> - A one-in-all QC tool to generate an augmented mzTab</li>
<li><a class="el" href="a16288.html">GenericWrapper</a> - Allows the generic wrapping of external tools.</li>
<li><a class="el" href="a16268.html">ExecutePipeline</a> - Executes workflows created by TOPPAS.</li>
<li><a class="el" href="a16289.html">GNPSExport</a> - Export MS/MS data in .MGF format for GNPS.</li>
<li><a class="el" href="a16309.html">MaRaClusterAdapter</a> - Run the spectral clustering implemented in MaRaCluster.</li>
</ul>
<p><b>Top down</b></p><ul>
<li><a class="el" href="a16287.html">TOPP_FLASHDeconv</a> - Computes a feature deconvolution from Top down MS data. </li>
</ul>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:27 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
