<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: CruxAdapter</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16240.html">TOPP documentation</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">CruxAdapter </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Identifies peptides in MS/MS spectra via Crux and tide-search.</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="2"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> CruxAdapter <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1">any signal-/preprocessing tool <br  />
 (in mzML format) </td><td valign="middle" align="center" rowspan="1"><a class="el" href="a16292.html">IDFilter</a> or <br  />
 any protein/peptide processing tool  </td></tr>
</table>
</center><p><em>Crux</em> must be installed before this wrapper can be used. This should be the case for the installers.</p>
<p>The default parameters are set for a high resolution instrument. See the following publication: <br  />
 Christopher Y. Park, Aaron A. Klammer, Lukas Käll, Michael J. MacCoss and William Stafford Noble. "Rapid and accurate peptide identification from tandem mass spectra." Journal of Proteome Research. 7(7):3022-3027, 2008. doi: 10.1021/pr800127y</p>
<p><b>The command line parameters of this tool are:</b> </p><pre class="fragment">

CruxAdapter -- Identifies MS/MS spectra using Crux.

Full documentation: http://www.openms.de/documentation/TOPP_CruxAdapter.html

Version: 2.4.0-HEAD-2019-12-09 Dec  9 2019, 13:47:09, Revision: af8a11e

To cite OpenMS:

  Rost HL, Sachsenberg T, Aiche S, Bielow C et al.. OpenMS: a flexible open-source software platform for mass spectrometry data analysis. Nat Meth. 2016; 13, 9: 741-748. doi:10.1038/nmeth.3959.

To cite CruxAdapter:

  Park CI, Klammer AA, Käll L, MacCoss MJ, Noble WS. Rapid and accurate peptide identification from tandem mass spectra. J Proteome Res 7(7):3022-3027, 2008.. doi:10.1021/pr800127y.



Usage:

  CruxAdapter &lt;options&gt;



Options (mandatory options marked with '*'):

  -in &lt;file&gt;*                            Input file (valid formats: 'mzML')

  -out &lt;file&gt;*                           Output file (valid formats: 'idXML')

  -database &lt;file&gt;*                      FASTA file (valid formats: 'FASTA')

  -crux_executable &lt;executable&gt;*         Crux executable of the installation e.g. 'crux.exe'

  -extra_index_args &lt;choice&gt;             Extra arguments to be passed to tide-index

  -extra_search_args &lt;choice&gt;            Extra arguments to be passed to tide-search

  -extra_percolator_args &lt;choice&gt;        Extra arguments to be passed to percolator

  -precursor_mass_tolerance &lt;tolerance&gt;  Precursor monoisotopic mass tolerance (Crux parameter: peptide_mass_

                                         tolerance) (default: '10.0')

  -precursor_mass_units &lt;choice&gt;         Unit of precursor mass tolerance (amu, m/z or ppm) (default: 'ppm' 

                                         valid: 'mass', 'mz', 'ppm')

  -fragment_bin_offset &lt;offset&gt;          In the discretization of the m/z axes of the observed and theoretica

                                         l spectra, this parameter specifies the location of the left edge

                                         of the first bin, relative to mass = 0 (i.e., mz-bin-offset = 0.xx

                                         means the left edge of the first bin will be located at +0.xx Da).

                                         (default: '0.0')

  -fragment_bin_width &lt;width&gt;            Before calculation of the XCorr score, the m/z axes of the observed 

                                         and theoretical spectra are discretized. This parameter specifies

                                         the size of each bin. The exact formula for computing the discretize

                                         d m/z value is floor((x/mz-bin-width) + 1.0 - mz-bin-offset), where

                                         x is the observed m/z value. For low resolution ion trap ms/ms data

                                         1.0005079 and for high resolution ms/ms 0.02 is recommended. (defaul

                                         t: '0.02')

  -isotope_error &lt;choice&gt;                List of positive, non-zero integers.

  -run_percolator &lt;true/false&gt;           Whether to run percolator after tide-search (default: 'true' valid: 

                                         'true', 'false')

  -enzyme &lt;cleavage site&gt;                The enzyme used for peptide digestion. (default: 'trypsin' valid: 

                                         'custom-enzyme', 'clostripain', 'staph-protease', 'pepsin-a', 'glu-c

                                         ', 'elastase-trypsin-chymotrypsin', 'no-enzyme', 'proline-endopeptid

                                         ase', 'cyanogen-bromide', 'iodosobenzoate', 'elastase', 'arg-c',

                                         'chymotrypsin', 'lys-c', 'trypsin/p', 'trypsin', 'asp-n', 'lys-n')

  -digestion &lt;choice&gt;                    Full, partial or non specific digestion (default: 'full-digest' vali

                                         d: 'full-digest', 'partial-digest', 'non-specific-digest')

  -allowed_missed_cleavages &lt;num&gt;        Number of possible cleavage sites missed by the enzyme, maximum valu

                                         e is 5; for enzyme search (default: '0')

  -decoy-format &lt;choice&gt;                 Decoy generation method either by reversing the sequence or shufflin

                                         g it. (default: 'shuffle' valid: 'none', 'shuffle', 'peptide-reverse

                                         ', 'protein-reverse')

  -keep-terminal-aminos &lt;choice&gt;         Whether to keep N and C terminal in place or also shuffled / reverse

                                         d. (default: 'NC' valid: 'N', 'C', 'NC', 'none')

  -cterm_modifications &lt;mods&gt;            Specifies C-terminal static and variable mass modifications on pepti

                                         des.  Specify a comma-separated list of C-terminal modification sequ

                                         ences of the form: X+21.9819 Default = &lt;empty&gt;.

  -nterm_modifications &lt;mods&gt;            Specifies N-terminal static and variable mass modifications on pepti

                                         des.  Specify a comma-separated list of N-terminal modification sequ

                                         ences of the form: 1E-18.0106,C-17.0265 Default = &lt;empty&gt;.

  -modifications &lt;mods&gt;                  Expression for static and variable mass modifications to include. 

                                         Specify a comma-separated list of modification sequences of the form

                                         : C+57.02146,2M+15.9949,1STY+79.966331,... Default = C+57.02146.

  -test_fdr &lt;fdr&gt;                        False discovery rate threshold used in selecting hyperparameters 

                                         during internal cross-validation and for reporting the final results

                                         . (default: '0.01')

  -train_fdr &lt;fdr&gt;                       False discovery rate threshold to define positive examples in traini

                                         ng. (default: '0.01')

                                         

Common TOPP options:

  -ini &lt;file&gt;                            Use the given TOPP INI file

  -threads &lt;n&gt;                           Sets the number of threads allowed to be used by the TOPP tool (defa

                                         ult: '1')

  -write_ini &lt;file&gt;                      Writes the default configuration file

  --help                                 Shows options

  --helphelp                             Shows all options (including advanced)



</pre><p> <b>INI file documentation of this tool:</b> <div class="ini_global">
<div class="legend">
<b>Legend:</b><br>
 <div class="item item_required">required parameter</div>
 <div class="item item_advanced">advanced parameter</div>
</div>
  <div class="node"><span class="node_name">+CruxAdapter</span><span class="node_description">Identifies MS/MS spectra using Crux.</span></div>
    <div class="item item_advanced"><span class="item_name" style="padding-left:16px;">version</span><span class="item_value">2.4.0-HEAD-2019-12-09</span>
<span class="item_description">Version of the tool that generated this parameters file.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>    <div class="node"><span class="node_name">++1</span><span class="node_description">Instance '1' section for 'CruxAdapter'</span></div>
      <div class="item"><span class="item_name item_required" style="padding-left:24px;">in</span><span class="item_value"></span>
<span class="item_description">Input file</span><span class="item_tags">input file</span><span class="item_restrictions">*.mzML</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">out</span><span class="item_value"></span>
<span class="item_description">Output file</span><span class="item_tags">output file</span><span class="item_restrictions">*.idXML</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">database</span><span class="item_value"></span>
<span class="item_description">FASTA file</span><span class="item_tags">input file</span><span class="item_restrictions">*.FASTA</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">crux_executable</span><span class="item_value">crux.exe</span>
<span class="item_description">Crux executable of the installation e.g. 'crux.exe'</span><span class="item_tags">input file</span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">extra_index_args</span><span class="item_value"></span>
<span class="item_description">Extra arguments to be passed to tide-index</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">extra_search_args</span><span class="item_value"></span>
<span class="item_description">Extra arguments to be passed to tide-search</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">extra_percolator_args</span><span class="item_value"></span>
<span class="item_description">Extra arguments to be passed to percolator</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">precursor_mass_tolerance</span><span class="item_value">10.0</span>
<span class="item_description">Precursor monoisotopic mass tolerance (Crux parameter: peptide_mass_tolerance)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">precursor_mass_units</span><span class="item_value">ppm</span>
<span class="item_description">Unit of precursor mass tolerance (amu, m/z or ppm)</span><span class="item_tags"></span><span class="item_restrictions">mass,mz,ppm</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">fragment_bin_offset</span><span class="item_value">0.0</span>
<span class="item_description">In the discretization of the m/z axes of the observed and theoretical spectra, this parameter specifies the location of the left edge of the first bin, relative to mass = 0 (i.e., mz-bin-offset = 0.xx means the left edge of the first bin will be located at +0.xx Da).</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">fragment_bin_width</span><span class="item_value">0.02</span>
<span class="item_description">Before calculation of the XCorr score, the m/z axes of the observed and theoretical spectra are discretized. This parameter specifies the size of each bin. The exact formula for computing the discretized m/z value is floor((x/mz-bin-width) + 1.0 - mz-bin-offset), where x is the observed m/z value. For low resolution ion trap ms/ms data 1.0005079 and for high resolution ms/ms 0.02 is recommended.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">isotope_error</span><span class="item_value"></span>
<span class="item_description">List of positive, non-zero integers.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">run_percolator</span><span class="item_value">true</span>
<span class="item_description">Whether to run percolator after tide-search</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">enzyme</span><span class="item_value">trypsin</span>
<span class="item_description">The enzyme used for peptide digestion.</span><span class="item_tags"></span><span class="item_restrictions">custom-enzyme,chymotrypsin,trypsin,arg-c,asp-n,lys-n,trypsin/p,proline-endopeptidase,glu-c,clostripain,no-enzyme,elastase,lys-c,iodosobenzoate,staph-protease,pepsin-a,cyanogen-bromide,elastase-trypsin-chymotrypsin</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">digestion</span><span class="item_value">full-digest</span>
<span class="item_description">Full, partial or non specific digestion</span><span class="item_tags"></span><span class="item_restrictions">full-digest,partial-digest,non-specific-digest</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">allowed_missed_cleavages</span><span class="item_value">0</span>
<span class="item_description">Number of possible cleavage sites missed by the enzyme, maximum value is 5; for enzyme search</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">custom_enzyme</span><span class="item_value"></span>
<span class="item_description">Specify rules for in silico digestion of protein sequences. Overrides the enzyme option. Two lists of residues are given enclosed in square brackets or curly braces and separated by a |. The first list contains residues required/prohibited before the cleavage site and the second list is residues after the cleavage site.  </span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">decoy_prefix</span><span class="item_value">decoy_</span>
<span class="item_description">Specifies the prefix of the protein names that indicate a decoy</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">decoy-format</span><span class="item_value">shuffle</span>
<span class="item_description">Decoy generation method either by reversing the sequence or shuffling it.</span><span class="item_tags"></span><span class="item_restrictions">none,shuffle,peptide-reverse,protein-reverse</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">keep-terminal-aminos</span><span class="item_value">NC</span>
<span class="item_description">Whether to keep N and C terminal in place or also shuffled / reversed.</span><span class="item_tags"></span><span class="item_restrictions">N,C,NC,none</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">cterm_modifications</span><span class="item_value"></span>
<span class="item_description">Specifies C-terminal static and variable mass modifications on peptides.  Specify a comma-separated list of C-terminal modification sequences of the form: X+21.9819 Default = <empty>.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">nterm_modifications</span><span class="item_value"></span>
<span class="item_description">Specifies N-terminal static and variable mass modifications on peptides.  Specify a comma-separated list of N-terminal modification sequences of the form: 1E-18.0106,C-17.0265 Default = <empty>.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">modifications</span><span class="item_value"></span>
<span class="item_description">Expression for static and variable mass modifications to include. Specify a comma-separated list of modification sequences of the form: C+57.02146,2M+15.9949,1STY+79.966331,... Default = C+57.02146.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">test_fdr</span><span class="item_value">0.01</span>
<span class="item_description">False discovery rate threshold used in selecting hyperparameters during internal cross-validation and for reporting the final results.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">train_fdr</span><span class="item_value">0.01</span>
<span class="item_description">False discovery rate threshold to define positive examples in training.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">deisotope</span><span class="item_value">false</span>
<span class="item_description">Deisotope spectra before searching</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">report_decoys</span><span class="item_value">false</span>
<span class="item_description">Include decoys in the final reported dataset</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">log</span><span class="item_value"></span>
<span class="item_description">Name of log file (created only when specified)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">debug</span><span class="item_value">0</span>
<span class="item_description">Sets the debug level</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">threads</span><span class="item_value">1</span>
<span class="item_description">Sets the number of threads allowed to be used by the TOPP tool</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">no_progress</span><span class="item_value">false</span>
<span class="item_description">Disables progress logging to command line</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">force</span><span class="item_value">false</span>
<span class="item_description">Overwrite tool specific checks.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">test</span><span class="item_value">false</span>
<span class="item_description">Enables the test mode (needed for internal use only)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div></div>
 </div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:28 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
