<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: OpenMS Developer Quickstart Guide</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title"><a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> Developer Quickstart Guide </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h1><a class="anchor" id="tutorial_introduction"></a>
Introduction</h1>
<h2><a class="anchor" id="tutorial_gf"></a>
General Information</h2>
<p>Mass spectrometry (MS) is an essential analytical technique for high-throughput analysis in proteomics and metabolomics. The development of new separation techniques, precise mass analyzers and experimental protocols is a very active field of research. This leads to more complex experimental setups yielding ever increasing amounts of data. Consequently, analysis of the data is currently often the bottleneck for experimental studies. Although software tools for many data analysis tasks are available today, they are often hard to combine with each other or not flexible enough to allow for rapid prototyping of a new analysis workflow. <br  />
</p>
<p><b>OpenMS</b>, a software framework for rapid <b>application and method development</b> in mass spectrometry has been designed to be portable, easy-to-use, and robust while offering a rich functionality ranging from basic data structures to sophisticated algorithms for data analysis (<a href="https://www.nature.com/nmeth/journal/v13/n9/abs/nmeth.3959.html">https://www.nature.com/nmeth/journal/v13/n9/abs/nmeth.3959.html</a>). <br  />
</p>
<p><b>Ease of use:</b> OpenMS follows the <b>object-oriented</b> programming paradigm, which aims at mapping real-world entities to comprehensible data structures and interfaces. OpenMS enforces a <b>coding style</b> that ensures consistent names of classes, methods and member variables which increases the usability as a software library. Another important feature of a software framework is documentation. We decided to use <b>doxygen</b> (www.doxygen.org/) to generate the class documentation from the <b>source code</b>, which ensures consistency of code and documentation. The documentation is generated in HTML format making it easy to read with a web browser. <br  />
</p>
<p><b>Robustness:</b> Robustness of algorithms is essential if a new method will be applied routinely to large scale datasets. Typically, there is a trade-off between performance and robustness. OpenMS tries to address both issues equally. In general, we try to tolerate recoverable errors, e.g. files that do not entirely fulfill the format specifications. On the other hand, <b>exceptions</b> are used to handle fatal errors. To check for correctness, more than 1000 <b>unit tests</b> are implemented in total, covering public methods of classes. These tests check the behavior for both valid and invalid use. Additionally, <b>preprocessor macros</b> are used to enable additional consistency checks in debug mode, enforce <b>pre- and post-conditions</b>, and are then disabled in productive mode for performance reasons. <br  />
</p>
<p><b>Extensibility:</b> Since OpenMS is based on several <b>external libraries</b> it is designed for the integration of external code. All classes are encapsulated in the OpenMS namespace to avoid symbol clashes with other libraries. Through the use of <b>C++ templates</b>, many data structures are adaptable to specific use cases. Also, OpenMS supports <b>standard formats</b> and is itself open-source software. The use of standard formats ensures that applications developed with OpenMS can be easily integrated into existing analysis pipelines. OpenMS source code is released under the permissive <b>BSD 3 license</b> and located on <b>GitHub</b>, a repository for open-source software. This allows users to participate in the project and to contribute to the code base. <br  />
</p>
<p><b>Scriptable:</b> OpenMS allows exposing its functionality through python bindings (<b>pyOpenMS</b>). This eases the rapid development of algorithms in Python that later can be translated to C++. Please see our <a href="https://pyopenms.readthedocs.io">pyOpenMS documentation</a> for a description and walk-through of the pyOpenMS capabilities. <br  />
</p>
<p><b>Portability:</b> OpenMS supports <b>Windows</b>, <b>Linux</b>, and <b>OS X</b> platforms. <br  />
</p>
<h2><a class="anchor" id="tutorial_structure"></a>
The structure of the OpenMS Framework</h2>
<p>The following image shows the overall structure of OpenMS:</p>
<p> <style>div.image img[src="OpenMS_overview.png"]{max-width:1200px;}</style>  </p><div class="image">
<img src="OpenMS_overview.png" alt="" width="1200px"/>
<div class="caption">
Overall design of OpenMS. Kindly provided by Timo Sachsenberg.</div></div>
 <p>The structure of the <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> framework. <br  />
</p>
<p>The OpenMS software framework consists of three main layers:</p><ul>
<li><b>OpenMS Library:</b> the object-oriented OpenMS core library contains over 1,300 classes and is built on modern C++ infrastructure with native compiler support on Windows, Linux and OS X. The classes are representing core concepts in mass spectrometry as well as the corresponding ontologies defined by the Human Proteome Organization Proteomics Standard Initiative (HUPO-PSI). <br  />
</li>
<li><b>Scripting:</b> a well-defined Python API offers scripting for rapid software prototyping and interactive data exploration by researchers with advanced scripting skills. The pyOpenMS interactive Python interface, providing easy integration of the OpenMS library with other scientific Python libraries. <br  />
</li>
<li><b>TOPP tools:</b> a set of pre-built tools covering most core tasks in computational mass spectrometry. These tools are created using the OpenMS library. These tools form the building blocks that can be chained together to form complex workflows. <br  />
</li>
<li><b>Workflow:</b> a set of over 185 different tools for common mass spectrometric tasks can be accessed by routine users through the KNIME, and Galaxy workflow systems. <br  />
</li>
</ul>
<p>Each level of increasing abstraction provides better usability, but limits the extensibility as the Python and workflow levels only have access to the exposed Python API or the available set of TOPP tools respectively. Increasing abstraction, however, makes it easier to design and execute complex analyses, even across multiple omics types. By following a layered design the different needs of bioinformaticians and life scientists are addressed. <br  />
</p>
<h2><a class="anchor" id="tutorial_developing"></a>
Developing with OpenMS</h2>
<p>Before we get started developing with OpenMS, we would like to point to some information on the development model and conventions we follow to maintain a coherent code base. <br  />
 <b>Development model</b><br  />
 OpenMS follows the Gitflow development workflow which is excellently describedI <a href="http://nvie.com/posts/a-successful-git-branching-model/">here</a>. Additionally we encourage every developer (even if he is eligible to push directly to OpenMS) to create his own fork (e.g. username). The GitHub people provide superb documentation on <a href="https://help.github.com/articles/fork-a-repo">forking</a> and how to keep your fork <a href="https://help.github.com/articles/syncing-a-fork">up-to-date</a>. With your own fork you can follow the Gitflow development model directly, but instead of merging into "develop" in your own fork you can open a <a href="https://help.github.com/articles/using-pull-requests">pull request</a>. Before opening the pull request, please check the <a href="https://github.com/OpenMS/OpenMS/wiki/Pull-Request-Checklist">checklist</a>. <br  />
 Some more details and tips are collected here. <br  />
</p>
<p><b>Conventions</b><br  />
 See the manual for proper coding style: <a href="https://github.com/OpenMS/OpenMS/wiki/Coding-conventions">Coding conventions</a> also see: <a href="https://github.com/OpenMS/OpenMS/wiki/C&amp;%2343;&amp;%2343;-Guide">C++ Guide</a>. We automatically test for common coding convention violations using a modified version of cpplint. Style testing can be enabled using CMake options. We also provide a configuration file for Uncrustify for automated style corrections (see "tools/uncrustify.cfg"). <br  />
</p>
<p><b>Commit Messages</b><br  />
 In order to ease the creation of a CHANGELOG we use a defined format for our commit messages. See the manual for proper commit messages: <a href="https://github.com/OpenMS/OpenMS/wiki/HowTo---Write-Commit-Messages">How to write commit messages</a>.<br  />
 <br  />
 <b>Automated Unit Tests</b><br  />
 Pull requests are automatically tested using our continuous integration platform. In addition we perform nightly test runs covering different platforms. Even if everything compiled well on your machine and all tests passed, please check if you broke another platform on the next day. Nightly tests: <a href="http://cdash.openms.de/index.php?project=OpenMS">CDASH</a> <br  />
 <br  />
 <b>Experimental Installers</b><br  />
 We automatically build installers for different platforms. These usually contain unstable or partially untested code - so use them at your own risk. The nightly (unstable) installers are available <a href="http://ftp.mi.fu-berlin.de/OpenMS/nightly_binaries/">here</a>. <br  />
 <br  />
 <b>Technical Documentation</b><br  />
 Documentation of classes and tools is automatically generated using doxygen: See the documentation for <a href="http://www.openms.de/current_doxygen/html/">HEAD</a> See the documentation for the latest <a href="https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/index.html">release branch</a> <br  />
 <br  />
 <b>Building OpenMS</b><br  />
 Before you get started coding with OpenMS you need to build it for your operating system. Please follow the <b>build instructions</b> from the documentation. <br  />
 <a href="https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/2.3.0/html/install_linux.html"><b>Building OpenMS on GNU/Linux</b></a> <br  />
 <a href="https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/2.3.0/html/install_mac.html"><b>Building OpenMS on Mac OS X</b></a> <br  />
 <a href="https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/2.3.0/html/install_win.html"><b>Building OpenMS on Windows</b></a> <br  />
 Note that for development purposes, you might want to set the variable <em>CMAKE_BUILD_TYPE</em> to <em>Debug</em>. Otherwise, the default <em>Release</em> will be applied and disables pre-condition and post-condition checks, and assertions. <br  />
 <br  />
 <b>Choice of an IDE</b> <br  />
 You are, of course, free to choose your favorite (or even no) IDE for OpenMS development but given the size of OpenMS, not all IDEs perform equally well. We have good experiences with Qt Creator on Linux and Mac, because it can directly import CMake Projects and is rather fast in indexing all files. On Windows, Visual Studio is currently the preferred solution. Additionally, you may want to try JetBrains CLion (it is free for students, teachers and open source projects). Another option is Eclipse with C++ support, which can also import CMake projects directly with the respective CMake generator.</p>
<h2><a class="anchor" id="tutorial_terms"></a>
Mass spectrometry terms</h2>
<p>The following terms for MS-related data are used in this tutorial and the OpenMS class documentation:</p><ul>
<li><b>Raw or profile peak</b>: a typically Gaussian shaped mass peak measured by the instrument. <br  />
</li>
<li><b>Centroid or picked peak</b>: a single m/z, intensity pair as obtained after using a peak picking (also: peak centroiding) algorithm. <br  />
</li>
<li><p class="startli"><b>Spectrum / Scan</b>: a mass spectrum containing profile or centroided peaks (profile spectrum) or centroided peaks (peak spectrum). E.g. a low resolution profile (blue) and a centroided peak spectrum (pink) are shown in the figure below. <br  />
</p>
<div class="image">
<img src="Terms_Spectrum.png" alt=""/>
<div class="caption">
Part of a raw spectrum (blue) with three peaks (red)</div></div>
 </li>
<li><b>(Peak or Raw) Map</b>: a collection of spectra of a single LC-MS run. If spectra are recorded in profile mode, we usually use the term raw map. If spectra are already centroided we usually refer to them as peak map. <br  />
</li>
<li><b>Feature</b>: a signal from a chemical entity detected in an HPLC-MS experiment, typically a peptide. <br  />
</li>
</ul>
<p>The image below shows a peak map and the red circle highlights a feature.</p>
<div class="image">
<img src="Terms_Map.png" alt=""/>
<div class="caption">
Peak map with a marked feature (red)</div></div>
 <h1><a class="anchor" id="tutorial_library"></a>
OpenMS Library</h1>
<p>The extensible OpenMS library implements common mass spectrometric data processing tasks through a well defined API in C++ and Python using standardized open data formats.</p>
<h2><a class="anchor" id="tutorial_library_overview"></a>
Overview on Central Algorithms and Methods</h2>
<p>OpenMS provides algorithms in many fields of computational metabolomics and proteomics. <br  />
 <b>The following list is intended to algorithm and tool developers a starting point to tools and classes relevant to their scientific question at hand</b>. It does not include third-party tools but only tools that were implemented in OpenMS. <br  />
</p>
<ul>
<li>Proteomics:<ul>
<li>Signal processing:<ul>
<li>Conversion from profile to centroided spectra (Tool PeakPickerHiRes)</li>
<li>Precursor mass correction (Tool HiResPrecursorMassCorrector)</li>
</ul>
</li>
<li>Filtering:<ul>
<li>Large number of basic filters applicable to different types of data (e.g., remove identified spectra, filter MS2, extract m/z ranges, … in Tool FileFilter and IDFilter)</li>
</ul>
</li>
<li>Identification:<ul>
<li>Database search:<ul>
<li>Peptides (Tool SimpleSearchEngine and its classes - started simple but is, by now, rather complete peptide identification engine)</li>
<li>Protein-RNA cross-links (Tool RNPxlSearch and its classes)</li>
<li>Protein-Protein cross-links (Tool OpenPepXL)</li>
</ul>
</li>
<li>Spectral library search:<ul>
<li>Tool SpecLibSearcher and its classes</li>
</ul>
</li>
<li>DeNovo:<ul>
<li>Tool CompNovoCID and its classes</li>
</ul>
</li>
</ul>
</li>
<li>Quantification:<ul>
<li>Peptide Feature Detection:<ul>
<li>Untargeted, label-free (Tools FeatureFinderCentroided, FeatureFinderMultiplex, and its classes)</li>
<li>ID-based label-free (Tool FeatureFinderIdentification “new”)</li>
<li>SILAC-labeling (Tool FeatureFinderMultiplex)</li>
<li>iTRAQ/TMT (Tool IsobaricAnalyzer)</li>
<li>Dynamically labeled (SIP) peptides (Tool MetaProSIP)</li>
</ul>
</li>
<li>Retention Time Alignment:<ul>
<li>Linear map alignment (Tool MapAlignerPoseClustering)</li>
<li>(Non-)linear map alignment (Tool MapAlignerIdentification “new”)</li>
</ul>
</li>
<li>Peptide Feature linking (matching of features between runs):<ul>
<li>fast, KD-tree based linking (Tool FeatureLinkerUnlabeledKD)</li>
<li>QT based clustering and linking (Tool FeatureLinkerUnlabeledQT)</li>
</ul>
</li>
<li>Protein inference:<ul>
<li>WIP (currently via third-party tool FIDO and Wrapper FidoAdapter)</li>
</ul>
</li>
<li>Protein Quantification:<ul>
<li>Tool ProteinQuantifier</li>
</ul>
</li>
<li>Targeted data extraction:<ul>
<li>Analysis of data-independent acquisition or SWATH-MS data (Tool OpenSWATH)</li>
</ul>
</li>
<li>Misc:<ul>
<li>Theoretical spectra generators <br  />
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>Metabolomics:<ul>
<li>Quantification:<ul>
<li>Small molecule feature detection:<ul>
<li>Untargeted, label-free (Tool FeatureFinderMetabo)</li>
</ul>
</li>
<li>Retention Time Alignment:<ul>
<li>Linear map alignment (Tool MagAlignerPoseClustering)</li>
</ul>
</li>
<li>Small molecule feature linking:<ul>
<li>QT based clustering and linking (Tool FeatureLinkerUnlabeledQT)</li>
<li>fast, KD-tree based linking (Tool FeatureLinkerUnlabeledKD)</li>
</ul>
</li>
<li>Adduct decharing:<ul>
<li>Linear programming based determination of small molecule ion adducts and charges (Tool MetboliteAdductDecharger)</li>
</ul>
</li>
<li>Targeted data extraction:<ul>
<li>Analysis of data-independent acquisition or SWATH-MS data (Tool OpenSWATH)</li>
</ul>
</li>
</ul>
</li>
<li>Identification:<ul>
<li>Spectral library search:<ul>
<li>Tool MetaboliteSpectralMatcher</li>
</ul>
</li>
<li>Accurate mass search:<ul>
<li>Tool AccurateMassSearch</li>
</ul>
</li>
<li>De novo identification:<ul>
<li>Tool SiriusAdapter</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><br  />
</p><ul>
<li>General:<ul>
<li>Mass decomposition algorithms</li>
<li>Isotope pattern generators</li>
<li>Quality control (Tools QCCalculator, QCExtractor) metrics and file format (QcML)</li>
</ul>
</li>
</ul>
<p><br  />
 </p><table class="doxtable">
<caption>Directory structure of src folder (/src)</caption>
<tr>
<th>Folder </th><th>Description </th></tr>
<tr>
<td>openms</td><td>Source code of core library </td></tr>
<tr>
<td>openms_gui</td><td>Source code of GUI applications (e.g.: TOPPView) </td></tr>
<tr>
<td>topp</td><td>Source code of (stable) OpenMS Tools </td></tr>
<tr>
<td>util</td><td>Source code of (experimental) OpenMS Tools </td></tr>
<tr>
<td>pyOpenMS</td><td>Source files providing the python bindings </td></tr>
<tr>
<td>tests</td><td>Source code of class and tool tests </td></tr>
</table>
<p><br  />
 </p><table class="doxtable">
<caption>Directory structure of core library (/src/openms/include/OpenMS)</caption>
<tr>
<th>Folder </th><th>Description </th></tr>
<tr>
<td>ANALYSIS</td><td>Source code of high-level analysis like PeakPicking, Quantitation, Identification, MapAlignment </td></tr>
<tr>
<td>APPLICATIONS</td><td>Source code for tool base and handling </td></tr>
<tr>
<td>CHEMISTRY</td><td>Source code dealing with Elements, Enzymes, Residues, Modifications, Isotope distributions and amino acid sequences </td></tr>
<tr>
<td>COMPARISON</td><td>Different scoring functions for clustering and spectra comparison </td></tr>
<tr>
<td>CONCEPT</td><td>OpenMS concepts (types, macros, ...) </td></tr>
<tr>
<td>DATASTRUCTURES</td><td>Auxiliary data structures </td></tr>
<tr>
<td>FILTERING</td><td>Filter </td></tr>
<tr>
<td>FORMAT</td><td>Source code for I/O classes and file formats </td></tr>
<tr>
<td>INTERFACES</td><td>Interfaces (WIP) </td></tr>
<tr>
<td>KERNEL</td><td>Core data structures </td></tr>
<tr>
<td>MATH</td><td>Source code for math functions and classes </td></tr>
<tr>
<td>METADATA</td><td>Source code for classes that capture metadata about a MS or HPLC-MS experiment </td></tr>
<tr>
<td>SIMULATION</td><td>Source code of MS simulator </td></tr>
<tr>
<td>SYSTEM</td><td>Source code for basic functionality (file system, stopwatch) </td></tr>
<tr>
<td>TRANSFORMATIONS</td><td>Feature detection (MS1 label-free and isotopic labelling) and PeakPickers (centroiding algorithms) </td></tr>
</table>
<p><br  />
</p>
<p>Within the ANALYSIS folder, you can find several important tools</p>
<p><br  />
 </p><table class="doxtable">
<caption>Directory structure of the algorithmic part of the library (/src/openms/include/OpenMS/ANALYSIS)</caption>
<tr>
<th>Folder </th><th>Description </th></tr>
<tr>
<td>DECHARGING</td><td>Algorithms for de-charging (charge analysis) for peptides and metabolites </td></tr>
<tr>
<td>DENOVO</td><td>Algorithms for "de-novo" identification tools including CompNovo </td></tr>
<tr>
<td>ID</td><td>Source code dealing with identifications including ID conflict resolvers, metabolite spectrum matching and target-decoy models </td></tr>
<tr>
<td>MAPMATCHING</td><td>Algorithms for retention time correction and feature matching (matching between runs) </td></tr>
<tr>
<td>MRM</td><td>Algorithms for MRM Fragment selection </td></tr>
<tr>
<td>OPENSWATH</td><td>OpenSWATH algorithms for targeted, chromatogram-based analysis of MRM, SRM, PRM, DIA and SWATH-MS data </td></tr>
<tr>
<td>PIP</td><td>Peak intensity predictor </td></tr>
<tr>
<td>QUANTITATION</td><td>Algorithms for quantitative analysis including isobaric labelling </td></tr>
<tr>
<td>RNPXL</td><td>Algorithms for RNA cross-linking </td></tr>
<tr>
<td>SVM</td><td>Algorithms for SVM </td></tr>
<tr>
<td>TARGETED</td><td>Algorithms for targeted proteomics (MRM, SRM) </td></tr>
<tr>
<td>XLMS</td><td>Algorithms for Cross-link mass spectrometry </td></tr>
</table>
<p><br  />
</p>
<p>For the sake of completeness you will find a short list of the THIRDPARTY tools, which are integrated via wrappers into the OpenMS framework (usually called -Adapter e.g. SiriusAdapter)</p>
<p>Wrapper to third-party tools:</p><ul>
<li>Search Engines (MSGFPLUS, XTandem, OMSSA, Comet, MyriMatch)</li>
<li>Protein Inference (Fido)</li>
<li>Spectral Library Search (SpectraST)</li>
<li>Metabolite Identification (Sirius)</li>
<li>Score calibration and FDR calculation (Percolator)</li>
</ul>
<h2><a class="anchor" id="tutorial_library_kernelclasses"></a>
Kernel Classes</h2>
<p>The OpenMS kernel contains the data structures that store the actual MS data.</p>
<p>For storing the basic MS data (spectra, chromatograms, and full runs) OpenMS uses</p><ul>
<li>Peaks (Peak1D and ChromatogramPeak) stored in</li>
<li>MSSpectrum and MSChromatogram, which in turn can both be stored in an</li>
<li>MSExperiment</li>
</ul>
<p>For storing quantified peptides or analytes in single MS runs, OpenMS uses so called feature maps.</p>
<p>The main data structures for quantitative information are</p><ul>
<li>Features (for quantitative information in MS1 maps)</li>
<li>MRMFeatures (for quantitative information in XIC traces on MS1 and MS2 level)<ul>
<li>which are both stored in a FeatureMap</li>
</ul>
</li>
</ul>
<p>To store quantified peptides or analytes over several MS runs, OpenMS uses so called consensus maps.</p><ul>
<li>ConsensusFeatures are stored in a</li>
<li>ConsensusMap</li>
</ul>
<p>To store identified peptides OpenMS has classes</p><ul>
<li>PeptideHit, which corresponds to a Peptide-Spectrum-Matching stored in a</li>
<li>PeptideIdentification object (which is associated with a single spectrum)</li>
</ul>
<p><br  />
 </p><table class="doxtable">
<caption>Directory structure of core library (/src/openms)</caption>
<tr>
<th>Stored Entity </th><th>Class Name </th></tr>
<tr>
<td>Mass Peak (m/z + intensity)</td><td>Peak1D </td></tr>
<tr>
<td>Elution Peak (rt + intensity)</td><td>ChromatogramPeak </td></tr>
<tr>
<td>Spectrum of Mass Peaks</td><td>MSSpectrum </td></tr>
<tr>
<td>Chromatogram of Elution Peaks</td><td>MSChromatogram </td></tr>
<tr>
<td>Mass trace for small molecule detection</td><td>MassTrace </td></tr>
<tr>
<td>Full MS run, containing both spectra and chromatograms</td><td>MSExperiment (alias PeakMap) </td></tr>
<tr>
<td>Feature (isotopic pattern of eluting analyte)</td><td>Feature </td></tr>
<tr>
<td>All features detected in an MS Run</td><td>FeatureMap </td></tr>
<tr>
<td>Linked / Grouped feature (e.g., same Peptide quantified in several MS runs) </td><td>ConsensusFeature </td></tr>
<tr>
<td>All grouped ConsensusFeatures of a multi-run experiment</td><td>ConsensusMap </td></tr>
<tr>
<td>Peptide Spectrum Match</td><td>PeptideHit </td></tr>
<tr>
<td>Identified Spectrum with one or several PSMs</td><td>PeptideIdentification </td></tr>
<tr>
<td>Identified Protein</td><td>ProteinHit </td></tr>
</table>
<p><br  />
</p>
<h2><a class="anchor" id="tutorial_library_kernelclasses_peaks"></a>
Peaks</h2>
<p>OpenMS provides one-, two- and d-dimensional data points, either with or without metadata attached to them.</p>
<div class="image">
<img src="Kernel_DataPoints.png" alt=""/>
<div class="caption">
Data structure for MS data points</div></div>
 <p>One-dimensional data points: One-dimensional data points (Peak1D) are the most important ones and used throughout OpenMS. The two-dimensional and d-dimensional data points are needed rarely and used for special purposes only. Peak1D provides getter and setter methods to store the mass-to-charge ratio and intensity. <br  />
 Two-dimensional data points: The two-dimensional data points store mass-to-charge, retention time and intensity. The most prominent example we will later take a closer look at is the Feature class, which stores a two-dimensional position (m/z and RT) and intensity of the eluting peptide or analyte. <br  />
 The base class of the two-dimensional data points is Peak2D. It provides the same interface as Peak1D and additional getter and setter methods for the retention time. RichPeak2D is derived from Peak2D and adds an interface for metadata. The Feature is derived from RichPeak2D and adds information about the convex hull of the feature, quality and so on. <br  />
 For information on d-dimensional data points see the appendix.</p>
<h2><a class="anchor" id="tutorial_library_kernelclasses_spectra"></a>
Spectra</h2>
<p>The most important container for raw/profile data and centroided peaks is MSSpectrum. The elements of a MSSpectrum are peaks (Peak1D). In fact it is so common that it has its own typedef PeakSpectrum. MSSpectrum is derived from SpectrumSettings, a container for the metadata of a spectrum (e.g. precursor information). Here, only MS data handling is explained, SpectrumSettings is described in subsection meta data of a spectrum. In the following example (Tutorial_MSSpectrum.cpp) program, a MSSpectrum is filled with peaks, sorted according to mass-to-charge ratio and a selection of peak positions is displaOne-dimensional data points: <br  />
<br  />
 <b>Example: Tutorial_MSSpectrum.cpp</b> <br  />
 In this example, we create MS1 spectrum at 1 minute and insert peaks with descending mass-to-charge ratios (for educational reasons). We sort the peaks according to ascending mass-to-charge ratio. Finally we print the peak positions of those peaks between 800 and 1000 Thomson. For printing all the peaks in the spectrum, we simply would have used the STL-conform methods begin() and end(). In addition to the iterator access, we can also directly access the peaks via vector indices (e.g. spectrum[0] is the first Peak1D object of the MSSpectrum).</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04229.html">OpenMS/KERNEL/MSSpectrum.h</a>&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// Create spectrum</span></div>
<div class="line">  <a class="code" href="a10097.html">MSSpectrum</a> spectrum;</div>
<div class="line">  <a class="code" href="a10109.html">Peak1D</a> peak;</div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keywordtype">float</span> mz = 1500.0; mz &gt;= 500; mz -= 100.0)</div>
<div class="line">  {</div>
<div class="line">    peak.<a class="code" href="a10109.html#ab9c4ce5d172a75d6de2c6f5f1847c9ab">setMZ</a>(mz);</div>
<div class="line">    spectrum.push_back(peak);</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Sort the peaks according to ascending mass-to-charge ratio</span></div>
<div class="line">  spectrum.<a class="code" href="a10097.html#a86f7b200260209865e730a2e4fcaa628">sortByPosition</a>();</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Iterate over spectrum of those peaks between 800 and 1000 Thomson</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">auto</span> it = spectrum.<a class="code" href="a10097.html#a808327302a2ba3fb630ea132d32db942">MZBegin</a>(800.0); it != spectrum.<a class="code" href="a10097.html#add06770712321b6e489bbdbcd123c888">MZEnd</a>(1000.0); ++it)</div>
<div class="line">  {</div>
<div class="line">    cout &lt;&lt; it-&gt;getMZ() &lt;&lt; endl;</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Access a peak by index</span></div>
<div class="line">  cout &lt;&lt; spectrum[1].getMZ() &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; spectrum[1].getIntensity() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// ... and many more</span></div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">}</div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_library_kernelclasses_chrom"></a>
Chromatograms</h2>
<p>The most important container for targeted analysis / XIC data is MSChromatogram. The elements of a MSChromatogram are chromatogram peaks (Peak1D). MSChromatogram is derived from ChromatogramSettings, a container for the metadata of a chromatogram (e.g. containing precursor and product information), similarly to SpectrumSettings. In the following example (Tutorial_MSChromatogram.cpp) program, a MSChromatogram is filled with chromatographic peaks, sorted according to retention time and a selection of peak positions is displayed. <br  />
<br  />
 <b>Example: Tutorial_MSChromatogram</b> <br  />
 Fill MSChromatogram with chromatographic peaks, sorted according to retention time </p><div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04223.html">OpenMS/KERNEL/MSChromatogram.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04361.html">OpenMS/METADATA/ChromatogramSettings.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04184.html">OpenMS/KERNEL/ChromatogramPeak.h</a>&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// create a chromatogram </span></div>
<div class="line">  <a class="code" href="a10073.html">MSChromatogram</a> chromatogram;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// fill it with metadata information</span></div>
<div class="line">  chromatogram.<a class="code" href="a10389.html#afa370200c7cf12d530364dec51830515">setNativeID</a>(<span class="stringliteral">&quot;transition_300.9_188.0&quot;</span>);</div>
<div class="line">  chromatogram.<a class="code" href="a10389.html#a397cbc084d939f590f1bfaa5eb7be4e2">getProduct</a>().<a class="code" href="a10653.html#a566e1d015a01fa37751e0a77c64d70a8">setMZ</a>(188.0);</div>
<div class="line">  chromatogram.<a class="code" href="a10389.html#aab74cfd047d44d3b939b327a16e4d6f1">getPrecursor</a>().<a class="code" href="a10109.html#ab9c4ce5d172a75d6de2c6f5f1847c9ab">setMZ</a>(300.9);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// fill chromatogram with peaks</span></div>
<div class="line">  <a class="code" href="a09945.html">ChromatogramPeak</a> peak;</div>
<div class="line">  peak.<a class="code" href="a09945.html#ab875c2479c6cd2a0189e4684bfa4f97d">setIntensity</a>(1.0);</div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keywordtype">float</span> rt = 200.0; rt &gt;= 100; rt -= 1.0)</div>
<div class="line">  {</div>
<div class="line">    peak.<a class="code" href="a09945.html#a98fe0c3fba1788864d35d8ef6159459f">setRT</a>(rt);</div>
<div class="line">    chromatogram.push_back(peak);</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">// end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --><p> Since much of the functionality is shared between MSChromatogram and MSSpectrum, further examples can be gathered from the MSSpectrum subsection.</p>
<h2><a class="anchor" id="tutorial_library_kernelclasses_precursor"></a>
Precursor</h2>
<p>The precursor data stored along with MS/MS spectra contains invaluable information for MS/MS analysis (e.g, m/z, charge, activation mode, collision energy). This information is stored in Precursor objects that can be retrieved from each spectrum. For a complete list of functions please see the Precursor class documentation. <br  />
<br  />
 <b>Example: Tutorial_Precursor</b> <br  />
 Retrieve precursor information</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04226.html">OpenMS/KERNEL/MSExperiment.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04505.html">OpenMS/METADATA/Precursor.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04040.html">OpenMS/FORMAT/MzMLFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03488.html">OpenMS/CONCEPT/Exception.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>(<span class="keywordtype">int</span> argc, <span class="keyword">const</span> <span class="keywordtype">char</span>** argv)</div>
<div class="line">{</div>
<div class="line">  </div>
<div class="line">  <span class="keywordflow">if</span> (argc &lt; 2) <span class="keywordflow">return</span> 1;</div>
<div class="line">  </div>
<div class="line">  <span class="comment">// the path to the data should be given on the command line</span></div>
<div class="line">  <a class="code" href="a08769.html">String</a> tutorial_data_path(argv[1]);</div>
<div class="line">  </div>
<div class="line">  <a class="code" href="a10081.html">MSExperiment</a> spectra;</div>
<div class="line">  <a class="code" href="a09537.html">MzMLFile</a> f;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// load mzML from code examples folder</span></div>
<div class="line">  f.<a class="code" href="a09537.html#af242617eaf1d0f50a8ab0648823a4f23">load</a>(tutorial_data_path + <span class="stringliteral">&quot;/data/Tutorial_GaussFilter.mzML&quot;</span>, spectra);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// iterate over map and output MS2 precursor information</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">auto</span> s_it = spectra.<a class="code" href="a10081.html#a2387033802383edbdc95f9bbb12a707e">begin</a>(); s_it != spectra.<a class="code" href="a10081.html#ab45dae688fc5d8983727abffa4389003">end</a>(); ++s_it)</div>
<div class="line">  {</div>
<div class="line">    <span class="comment">// we are only interested in MS2 spectra so we skip all other levels</span></div>
<div class="line">    <span class="keywordflow">if</span> (s_it-&gt;getMSLevel() != 2) <span class="keywordflow">continue</span>;</div>
<div class="line"> </div>
<div class="line">    <span class="comment">// get a reference to the precursor information</span></div>
<div class="line">    <span class="keyword">const</span> <a class="code" href="a10097.html">MSSpectrum</a>&amp; spectrum = *s_it;</div>
<div class="line">    <span class="keyword">const</span> vector&lt;Precursor&gt;&amp; precursors = spectrum.<a class="code" href="a10725.html#ac69ba691dca00671a379aa956816cac5">getPrecursors</a>();</div>
<div class="line"> </div>
<div class="line">    <span class="comment">// size check &amp; throw exception if needed </span></div>
<div class="line">    <span class="keywordflow">if</span> (precursors.empty()) <span class="keywordflow">throw</span> <a class="code" href="a08305.html">Exception::InvalidSize</a>(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, precursors.size());</div>
<div class="line"> </div>
<div class="line">    <span class="comment">// get m/z and intensity of precursor</span></div>
<div class="line">    <span class="keywordtype">double</span> precursor_mz = precursors[0].getMZ();</div>
<div class="line">    <span class="keywordtype">float</span> precursor_int = precursors[0].getIntensity();</div>
<div class="line">  </div>
<div class="line">    <span class="comment">// retrieve the precursor spectrum (the most recent MS1 spectrum)</span></div>
<div class="line">    <a class="code" href="a10081.html#aee0a1a68d0fab63eefa6e14010b4a7f0">PeakMap::ConstIterator</a> precursor_spectrum = spectra.<a class="code" href="a10081.html#ae3d9a0d5a931f4978e44abb667dec531">getPrecursorSpectrum</a>(s_it);</div>
<div class="line">    <span class="keywordtype">double</span> precursor_rt = precursor_spectrum-&gt;getRT();</div>
<div class="line">  </div>
<div class="line">    <span class="comment">// output precursor information</span></div>
<div class="line">    std::cout &lt;&lt; <span class="stringliteral">&quot; precusor m/z: &quot;</span> &lt;&lt; precursor_mz</div>
<div class="line">              &lt;&lt; <span class="stringliteral">&quot; intensity: &quot;</span> &lt;&lt; precursor_int</div>
<div class="line">              &lt;&lt; <span class="stringliteral">&quot; retention time (sec.): &quot;</span> &lt;&lt; precursor_rt </div>
<div class="line">              &lt;&lt; std::endl;</div>
<div class="line">   }</div>
<div class="line">                                                            </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">// end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_library_kernelclasses_mrm"></a>
MRMTransitionGroup</h2>
<p>The targeted analysis of SRM or DIA (SWATH-MS) type of data requires a set of targeted assays as well as raw data chromatograms. The MRMTransitionGroup class allows users to map these two types of information and store them together with identified features conveniently in a single object. <br  />
<br  />
 <b>Example: Tutorial_MRMTransitionGroup</b> <br  />
 Create an empty MRMTransitionGroup with two dummy transitions</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="keyword">typedef</span> MRMTransitionGroup&lt;MSChromatogram, ReactionMonitoringTransition&gt; TrGroup;</div>
<div class="line">  TrGroup createTransitionGroup()</div>
<div class="line">  {</div>
<div class="line">    TrGroup tr_group;</div>
<div class="line">    tr_group.addChromatogram(MSChromatogram(), “transition1”);</div>
<div class="line">    tr_group.addTransition(ReactionMonitoringTransition(), “transition1”);</div>
<div class="line">    tr_group.addChromatogram(MSChromatogram(), “transition2”);</div>
<div class="line">    tr_group.addTransition(ReactionMonitoringTransition(), “transition2”);</div>
<div class="line">    tr_group.setTransitionGroupID(“tr_peptideA”);</div>
<div class="line">    <span class="keywordflow">return</span> tr_group;</div>
<div class="line">  }</div>
<div class="line"> </div>
</div><!-- fragment --><p> Note how the identifiers of the chromatograms and the assay information (ReactionMonitoringTransition) are matched so that downstream algorithms can utilize the meta-information stored in the assays for data analysis.</p>
<h2><a class="anchor" id="tutorial_library_kernelclasses_map"></a>
Maps</h2>
<p>Although raw data maps, peak maps and feature maps are conceptually very similar they are stored in different data types. For raw data and peak maps, the default container is MSExperiment, which is an array of MSSpectrum instances. In contrast to raw data and peak maps, feature maps are not a collection of one-dimensional spectra, but an array of two-dimensional feature instances. The main data structure for feature maps is called FeatureMap.</p>
<p>Although MSExperiment and FeatureMap differ in the data they store, they also have things in common. Both store metadata that is valid for the whole map, i.e. sample description and instrument description. This data is stored in the common base class ExperimentalSettings.</p>
<h2><a class="anchor" id="tutorial_library_kernelclasses_mse"></a>
MSExperiment</h2>
<p>MSExperiment contains ExperimentalSettings (metadata of the MS run) and a vector&lt;MSSpectrum&gt;. The one-dimensional spectrum MSSpectrum is derived from SpectrumSettings (metadata of a spectrum). <br  />
<br  />
 <b>Example: Tutorial_MSExperiment.cpp</b> <br  />
 The following example creates a MSExperiment containing four MSSpectrum instances. We then iterate over RT range (2,3) and m/z range (603,802) and print the peak positions using an AreaIterator. Then we show how we iterate over all spectra and peaks. In the commented out part, we show how to load/store all spectra and associated metadata from/to an mzML file.</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03533.html">OpenMS/CONCEPT/Types.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04226.html">OpenMS/KERNEL/MSExperiment.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// create a peak map containing 4 dummy spectra and peaks</span></div>
<div class="line">  <a class="code" href="a10081.html">MSExperiment</a> exp;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// The following examples creates a MSExperiment containing four MSSpectrum instances.</span></div>
<div class="line">  <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> i = 0; i &lt; 4; ++i)</div>
<div class="line">  {</div>
<div class="line">    <a class="code" href="a10097.html">MSSpectrum</a> spectrum;</div>
<div class="line">    spectrum.<a class="code" href="a10097.html#af4bee28529a0258718cd7269369886f9">setRT</a>(i);</div>
<div class="line">    spectrum.<a class="code" href="a10097.html#a24f081fe74bbbf773533ad19e96d8358">setMSLevel</a>(1);</div>
<div class="line">    <span class="keywordflow">for</span> (<span class="keywordtype">float</span> mz = 500.0; mz &lt;= 900; mz += 100.0)</div>
<div class="line">    {</div>
<div class="line">      <a class="code" href="a10109.html">Peak1D</a> peak;</div>
<div class="line">      peak.<a class="code" href="a10109.html#ab9c4ce5d172a75d6de2c6f5f1847c9ab">setMZ</a>(mz + i);</div>
<div class="line">      spectrum.push_back(peak);</div>
<div class="line">    }</div>
<div class="line">    </div>
<div class="line">    exp.<a class="code" href="a10081.html#a1c97663e0b6a5cc44a3b189b89210bac">addSpectrum</a>(spectrum);</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Iteration over the RT range (2,3) and the m/z range (603,802) and print the peak positions.</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">auto</span> it = exp.<a class="code" href="a10081.html#a878abca09cb3f6810e634b72b99007eb">areaBegin</a>(2.0, 3.0, 603.0, 802.0); it != exp.<a class="code" href="a10081.html#afb936fe46d35a9d29e36552a68c1ef2a">areaEnd</a>(); ++it)</div>
<div class="line">  {</div>
<div class="line">    cout &lt;&lt; it.getRT() &lt;&lt; <span class="stringliteral">&quot; - &quot;</span> &lt;&lt; it-&gt;getMZ() &lt;&lt; endl;</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Iteration over all peaks in the experiment. </span></div>
<div class="line">  <span class="comment">// Output: RT, m/z, and intensity</span></div>
<div class="line">  <span class="comment">// Note that the retention time is stored in the spectrum (not in the peak object)</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">auto</span> s_it = exp.<a class="code" href="a10081.html#a2387033802383edbdc95f9bbb12a707e">begin</a>(); s_it != exp.<a class="code" href="a10081.html#ab45dae688fc5d8983727abffa4389003">end</a>(); ++s_it)</div>
<div class="line">  {</div>
<div class="line">    <span class="keywordflow">for</span> (<span class="keyword">auto</span> p_it = s_it-&gt;begin(); p_it != s_it-&gt;end(); ++p_it)</div>
<div class="line">    {</div>
<div class="line">      cout &lt;&lt; s_it-&gt;getRT() &lt;&lt; <span class="stringliteral">&quot; - &quot;</span> &lt;&lt; p_it-&gt;getMZ() &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; p_it-&gt;getIntensity() &lt;&lt; endl;</div>
<div class="line">    }</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// We could store the spectra to a mzML file with:</span></div>
<div class="line">  <span class="comment">// MzMLFile mzml;</span></div>
<div class="line">  <span class="comment">// mzml.store(filename, exp);</span></div>
<div class="line">  </div>
<div class="line">  <span class="comment">// And load it with</span></div>
<div class="line">  <span class="comment">// mzml.load(filename, exp);</span></div>
<div class="line">  <span class="comment">// If we wanted to load only the MS2 spectra we could speed up reading by setting:</span></div>
<div class="line">  <span class="comment">// mzml.getOptions().addMSLevel(2);</span></div>
<div class="line">  <span class="comment">// before executing: mzml.load(filename, exp);</span></div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_library_kernelclasses_fmap"></a>
FeatureMap</h2>
<p>FeatureMap, the container for features, is simply a vector&lt;Feature&gt;. Additionally, it is derived from ExperimentalSettings, to store the meta information. All peak and feature containers (MSSpectrum, MSExperiment, FeatureMap) are also derived from RangeManager. This class facilitates the handling of MS data ranges. It allows to calculate and store both the position range and the intensity range of the container. <br  />
<br  />
 <b>Example: Tutorial_FeatureMap.cpp</b> <br  />
 The following examples creates a FeatureMap containing two Feature instances. Then we iterate over all features and output the retention time and m/z. We then show, how to use the underlying range manager to retrieve FeatureMap boundaries in rt, m/z, and intensity.</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04211.html">OpenMS/KERNEL/FeatureMap.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// Insert of two features into a map and iterate over the features.</span></div>
<div class="line">  <a class="code" href="a10057.html">FeatureMap</a> map;</div>
<div class="line"> </div>
<div class="line">  <a class="code" href="a10037.html">Feature</a> feature;</div>
<div class="line">  feature.<a class="code" href="a10125.html#a396c5ebcac2e70f9b9a478bb2456985b">setRT</a>(15.0);</div>
<div class="line">  feature.<a class="code" href="a10125.html#a2694f0775c46d9076355cb85d9815425">setMZ</a>(571.3);</div>
<div class="line">  map.push_back(feature); <span class="comment">//append feature 1</span></div>
<div class="line">  feature.<a class="code" href="a10125.html#a396c5ebcac2e70f9b9a478bb2456985b">setRT</a>(23.3);</div>
<div class="line">  feature.<a class="code" href="a10125.html#a2694f0775c46d9076355cb85d9815425">setMZ</a>(1311.3);</div>
<div class="line">  map.push_back(feature); <span class="comment">//append feature 2</span></div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Iteration over FeatureMap</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">auto</span> it = map.begin(); it != map.end(); ++it)</div>
<div class="line">  {</div>
<div class="line">    cout &lt;&lt; it-&gt;getRT() &lt;&lt; <span class="stringliteral">&quot; - &quot;</span> &lt;&lt; it-&gt;getMZ() &lt;&lt; endl;</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Calculate and output the ranges</span></div>
<div class="line">  map.<a class="code" href="a10057.html#ae8d7d7b28cd7807d573d0f75da4201f0">updateRanges</a>();</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Int: &quot;</span> &lt;&lt; map.<a class="code" href="a10149.html#aeab815807dd490278441926c58ebf3b8">getMinInt</a>() &lt;&lt; <span class="stringliteral">&quot; - &quot;</span> &lt;&lt; map.<a class="code" href="a10149.html#a75a60db5a9ee9fc609b6d4caaf4d82b3">getMaxInt</a>() &lt;&lt; endl;</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;RT:  &quot;</span> &lt;&lt; map.<a class="code" href="a10149.html#a3cc15077963df5fb4e2b0e68fdac3fda">getMin</a>()[0] &lt;&lt; <span class="stringliteral">&quot; - &quot;</span> &lt;&lt; map.<a class="code" href="a10149.html#ab76aaa57fa843ab12a5239af35775eb9">getMax</a>()[0] &lt;&lt; endl;</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;m/z: &quot;</span> &lt;&lt; map.<a class="code" href="a10149.html#a3cc15077963df5fb4e2b0e68fdac3fda">getMin</a>()[1] &lt;&lt; <span class="stringliteral">&quot; - &quot;</span> &lt;&lt; map.<a class="code" href="a10149.html#ab76aaa57fa843ab12a5239af35775eb9">getMax</a>()[1] &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// ... and many more</span></div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_fileformat"></a>
File Formats</h2>
<table class="doxtable">
<tr>
<td>mzML</td><td>The HUPO-PSI standard format for mass spectrometry data </td></tr>
<tr>
<td>mzIdentML</td><td>The HUPO-PSI standard format for identification results data from any search engines </td></tr>
<tr>
<td>mzTAB</td><td>The HUPO-PSI standard format for reporting MS-based proteomics and metabolomics results </td></tr>
<tr>
<td>traML</td><td>The HUPO-PSI standard format for exchange and transmission lists for selected reaction monitoring (SRM) experiments </td></tr>
<tr>
<td>featureXML</td><td>The OpenMS format for quantitation results </td></tr>
<tr>
<td>consensusXML</td><td>The OpenMS format for grouping features in one map or across several maps </td></tr>
<tr>
<td>idXML</td><td>The OpenMS format for identification results </td></tr>
<tr>
<td>trafoXML</td><td>The OpenMS format for storing of transformations </td></tr>
<tr>
<td>OpenSWATH</td><td></td></tr>
</table>
<p>For further information of the HUPO Proteomics Standards Initiative please visit: <a href="http://www.psidev.info/">http://www.psidev.info/</a></p>
<h2><a class="anchor" id="tutorial_logging"></a>
Logging</h2>
<p>To make direct output to std::out and std::err more consistent, OpenMS provides several low-level macros: <br  />
 OPENMS_LOG_FATAL_ERROR, <br  />
 OPENMS_LOG_ERROR <br  />
 OPENMS_LOG_WARN, <br  />
 OPENMS_LOG_INFO and <br  />
 OPENMS_LOG_DEBUG <br  />
 which should be used instead of the less descriptive std::out and std::err streams.</p>
<p>If you are writing an OpenMS tool, you can also use the ProgressLogger to indicate how many percent of the processing has already been performed: <br  />
<br  />
 <b>Example: Tutorial_Logger.cpp</b> <br  />
 Logging the Tool Progress</p>
<div class="fragment"><div class="line"> </div>
<div class="line">  ProgressLogger progresslogger;</div>
<div class="line">    progresslogger.setLogType(log_type_);  <span class="comment">// set the log type (command line or a file)</span></div>
<div class="line"> </div>
<div class="line">    <span class="comment">// set start progress (0) and end (ms_run.size() = the number of spectra)</span></div>
<div class="line">    progresslogger.startProgress(0, ms_run.size(), <span class="stringliteral">&quot;Doing some calculation...&quot;</span>);</div>
<div class="line"> </div>
<div class="line">    <span class="keywordflow">for</span> (PeakMap::Iterator it = ms_run.begin(); it != ms_run.end(); ++it)</div>
<div class="line">    {</div>
<div class="line">    <span class="comment">// update progress</span></div>
<div class="line">         progresslogger.setProgress(ms_run.end() - it);</div>
<div class="line">      </div>
<div class="line">    <span class="comment">// do the actual calculations and processing ...</span></div>
<div class="line">    }</div>
<div class="line">    progresslogger.endProgress();</div>
<div class="line"> </div>
</div><!-- fragment --><p> Depending on how the user configures the tool, this output is written to the command line or a log file.</p>
<h2><a class="anchor" id="tutorial_identifications"></a>
Identifications</h2>
<p>Identifications of proteins, peptides, and the mapping between peptides and proteins (or groups of proteins) are stored in dedicated data structures. These data structures are typically stored to disc as idXML or mzIdentML file. The highest-level structure is ProteinIdentification. It stores all identified proteins of an identification run as ProteinHit objects + additional metadata (search parameters, etc.). Each ProteinHit contains the actual Proteinaccession, an associated score, and (optionally) the protein sequence. A ProteinIdentification object stores the data corresponding to a single identified spectrum or feature. It has members for the retention time, m/z, and a vector of PeptideHits. Each PeptideHit stores the information of a specific peptide-to-spectrum match (e.g., the score and the peptide sequence). Each PeptideHit also contains a vector of PeptideEvidence objects which store the reference to one (or in the case the peptide maps to multiple proteins multiple) Proteins and the position therein. <br  />
<br  />
 <b>Example: Tutorial_IdentificationClasses.cpp</b> <br  />
 Create all identification data needed to store an idXML file</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04502.html">OpenMS/METADATA/PeptideIdentification.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04514.html">OpenMS/METADATA/ProteinIdentification.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04499.html">OpenMS/METADATA/PeptideHit.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03650.html">OpenMS/DATASTRUCTURES/String.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03257.html">OpenMS/CHEMISTRY/AASequence.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03980.html">OpenMS/FORMAT/IdXMLFile.h</a>&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// Create new protein identification object corresponding to a single search</span></div>
<div class="line">  vector&lt;ProteinIdentification&gt; protein_ids;</div>
<div class="line">  <a class="code" href="a10677.html">ProteinIdentification</a> protein_id;</div>
<div class="line"> </div>
<div class="line">  protein_id.<a class="code" href="a10677.html#a00f1456779f764c2738c35be2c04f6e8">setIdentifier</a>(<span class="stringliteral">&quot;Identifier&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Each ProteinIdentification object stores a vector of protein hits</span></div>
<div class="line">  vector&lt;ProteinHit&gt; protein_hits;</div>
<div class="line">  <a class="code" href="a10657.html">ProteinHit</a> protein_hit = <a class="code" href="a10657.html">ProteinHit</a>();</div>
<div class="line">  protein_hit.<a class="code" href="a10657.html#aba1fff7a176ff80d53f932769e606980">setAccession</a>(<span class="stringliteral">&quot;MyAccession&quot;</span>);</div>
<div class="line">  protein_hit.<a class="code" href="a10657.html#a2b8bf5f3febdb76461bdea9ff39d6b6d">setSequence</a>(<span class="stringliteral">&quot;PEPTIDEPEPTIDEPEPTIDEPEPTIDER&quot;</span>);</div>
<div class="line">  protein_hit.<a class="code" href="a10657.html#a64df7d8252c536e76dd3ef6468fc61b8">setScore</a>(1.0);</div>
<div class="line">  protein_hits.push_back(protein_hit);</div>
<div class="line"> </div>
<div class="line">  protein_id.<a class="code" href="a10677.html#a86f5c6d7c17c3aa94ec734e2a4fb50fa">setHits</a>(protein_hits);</div>
<div class="line"> </div>
<div class="line">  <a class="code" href="a08581.html">DateTime</a> now = <a class="code" href="a08581.html#a27fa24aa34288a3934c3d5a936463973">DateTime::now</a>();</div>
<div class="line">  <a class="code" href="a08769.html">String</a> date_string = now.<a class="code" href="a08581.html#a56a6f29cf9fa7e13a8df5c871afe4b94">getDate</a>();</div>
<div class="line">  protein_id.<a class="code" href="a10677.html#ae0728c54c04bcc99a4c6ebfa961ed2c5">setDateTime</a>(now);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Example of possible search parameters</span></div>
<div class="line">  <a class="code" href="a10685.html">ProteinIdentification::SearchParameters</a> search_parameters;</div>
<div class="line">  search_parameters.<a class="code" href="a10685.html#ae525bb5a71aa3ae9d811588db4652c00">db</a> = <span class="stringliteral">&quot;database&quot;</span>;</div>
<div class="line">  search_parameters.<a class="code" href="a10685.html#a6b3f6ef19caf97c201fba99444bf0663">charges</a> = <span class="stringliteral">&quot;+2&quot;</span>;</div>
<div class="line">  protein_id.<a class="code" href="a10677.html#afb0f58ddeea0e6f58da132fae1421563">setSearchParameters</a>(search_parameters);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Some search engine meta data</span></div>
<div class="line">  protein_id.<a class="code" href="a10677.html#a82e595befae76c1f7c22918420d2d9e8">setSearchEngineVersion</a>(<span class="stringliteral">&quot;v1.0.0&quot;</span>);</div>
<div class="line">  protein_id.<a class="code" href="a10677.html#a15a14866884b8b4dea8950b2151a0963">setSearchEngine</a>(<span class="stringliteral">&quot;SearchEngine&quot;</span>);</div>
<div class="line">  protein_id.<a class="code" href="a10677.html#a915462ec4a7a731fdd71f28e8a3463b1">setScoreType</a>(<span class="stringliteral">&quot;HyperScore&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  protein_ids.push_back(protein_id);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Iterate over protein identifications and protein hits</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">auto</span> it = protein_ids.begin(); it != protein_ids.end(); ++it)</div>
<div class="line">  {</div>
<div class="line">    <span class="keywordflow">for</span> (<span class="keyword">auto</span> hit = it-&gt;getHits().begin(); hit &lt; it-&gt;getHits().end(); ++hit)</div>
<div class="line">    {</div>
<div class="line">      cout &lt;&lt; <span class="stringliteral">&quot;Protein hit accession: &quot;</span> &lt;&lt; hit-&gt;getAccession() &lt;&lt; endl;</div>
<div class="line">      cout &lt;&lt; <span class="stringliteral">&quot;Protein hit sequence: &quot;</span> &lt;&lt; hit-&gt;getSequence() &lt;&lt; endl;</div>
<div class="line">      cout &lt;&lt; <span class="stringliteral">&quot;Protein hit score: &quot;</span> &lt;&lt; hit-&gt;getScore() &lt;&lt; endl;</div>
<div class="line">    }</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Create new peptide identifications</span></div>
<div class="line">  vector&lt;PeptideIdentification&gt; peptide_ids;</div>
<div class="line">  <a class="code" href="a10645.html">PeptideIdentification</a> peptide_id;</div>
<div class="line"> </div>
<div class="line">  peptide_id.<a class="code" href="a10645.html#af4bee28529a0258718cd7269369886f9">setRT</a>(1243.56);</div>
<div class="line">  peptide_id.<a class="code" href="a10645.html#a566e1d015a01fa37751e0a77c64d70a8">setMZ</a>(440.0);</div>
<div class="line">  peptide_id.<a class="code" href="a10645.html#a915462ec4a7a731fdd71f28e8a3463b1">setScoreType</a>(<span class="stringliteral">&quot;ScoreType&quot;</span>);</div>
<div class="line">  peptide_id.<a class="code" href="a10645.html#a98447dc4641b69db36fcc5087cfe73e6">setHigherScoreBetter</a>(<span class="keyword">false</span>);</div>
<div class="line">  peptide_id.<a class="code" href="a10645.html#a00f1456779f764c2738c35be2c04f6e8">setIdentifier</a>(<span class="stringliteral">&quot;Identifier&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// define additional meta value for the peptide identification</span></div>
<div class="line">  peptide_id.<a class="code" href="a10585.html#a135a7e818125b31198a8a65d9ac7a3d1">setMetaValue</a>(<span class="stringliteral">&quot;AdditionalMetaValue&quot;</span>, <span class="stringliteral">&quot;Value&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// add PeptideHit to a PeptideIdentification</span></div>
<div class="line">  vector&lt;PeptideHit&gt; peptide_hits;</div>
<div class="line">  <a class="code" href="a10617.html">PeptideHit</a> peptide_hit;</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#af26eef0efdc5d34a5a280d2bb42f79e9">setScore</a>(1.0);</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#a7daed733ac80f8002f3a1fbdc97c21e1">setRank</a>(1);</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#aca40a4630f5f8a0568899b74a62dcd4b">setCharge</a>(2);</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#a209d1e10f38a766c3f73c9666e89e14d">setSequence</a>(<a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;DLQM(Oxidation)TQSPSSLSVSVGDR&quot;</span>));</div>
<div class="line">  peptide_hits.push_back(peptide_hit);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// add second best PeptideHit to the PeptideIdentification</span></div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#af26eef0efdc5d34a5a280d2bb42f79e9">setScore</a>(1.5);</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#a7daed733ac80f8002f3a1fbdc97c21e1">setRank</a>(2);</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#aca40a4630f5f8a0568899b74a62dcd4b">setCharge</a>(2);</div>
<div class="line">  peptide_hit.<a class="code" href="a10617.html#a209d1e10f38a766c3f73c9666e89e14d">setSequence</a>(<a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;QLDM(Oxidation)TQSPSSLSVSVGDR&quot;</span>));</div>
<div class="line">  peptide_hits.push_back(peptide_hit);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// add PeptideHit to PeptideIdentification</span></div>
<div class="line">  peptide_id.<a class="code" href="a10645.html#ab4d15b02dbc53727d0b6dae927ad6688">setHits</a>(peptide_hits);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// add PeptideIdentification</span></div>
<div class="line">  peptide_ids.push_back(peptide_id);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// We could now store the identification data in an idXML file</span></div>
<div class="line">  <span class="comment">// IdXMLFile().store(outfile, protein_ids, peptide_ids);</span></div>
<div class="line">  <span class="comment">// And load it back with</span></div>
<div class="line">  <span class="comment">// IdXMLFile().load(outfile, protein_ids, peptide_ids);</span></div>
<div class="line"> </div>
<div class="line">  <span class="comment">// Iterate over PeptideIdentification</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">const</span> <span class="keyword">auto</span>&amp; peptide_id : peptide_ids)</div>
<div class="line">  {</div>
<div class="line">    <span class="comment">// Peptide identification values</span></div>
<div class="line">    cout &lt;&lt; <span class="stringliteral">&quot;Peptide ID m/z: &quot;</span> &lt;&lt; peptide_id.<a class="code" href="a10645.html#afe08b9bf9f9599bffa043dad0feeea43">getMZ</a>() &lt;&lt; endl;</div>
<div class="line">    cout &lt;&lt; <span class="stringliteral">&quot;Peptide ID rt: &quot;</span> &lt;&lt; peptide_id.<a class="code" href="a10645.html#a1529ca20d43500a280e162a759dd993f">getRT</a>() &lt;&lt; endl;</div>
<div class="line">    cout &lt;&lt; <span class="stringliteral">&quot;Peptide ID score type: &quot;</span> &lt;&lt; peptide_id.<a class="code" href="a10645.html#aee2ef4fb4e6965786d394761543a27b3">getScoreType</a>() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">    <span class="comment">// PeptideHits</span></div>
<div class="line">    <span class="keywordflow">for</span> (<span class="keyword">const</span> <span class="keyword">auto</span>&amp; scored_hit : peptide_id.<a class="code" href="a10645.html#a24d0fe4bb2a1dd9a237cc580cb48d318">getHits</a>())</div>
<div class="line">    {</div>
<div class="line">      cout &lt;&lt; <span class="stringliteral">&quot; - Peptide hit rank: &quot;</span> &lt;&lt; scored_hit.getRank() &lt;&lt; endl;</div>
<div class="line">      cout &lt;&lt; <span class="stringliteral">&quot; - Peptide hit sequence: &quot;</span> &lt;&lt; scored_hit.getSequence().toString() &lt;&lt; endl;</div>
<div class="line">      cout &lt;&lt; <span class="stringliteral">&quot; - Peptide hit score: &quot;</span> &lt;&lt; scored_hit.getScore() &lt;&lt; endl;</div>
<div class="line">    }</div>
<div class="line">  }</div>
<div class="line">  <span class="comment">// ...</span></div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">}</div>
<div class="line"> </div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_chemistry"></a>
Chemistry</h2>
<h2><a class="anchor" id="tutorial_element"></a>
Element, ElementDB, EmpiricalFormula</h2>
<p>An Element object is the representation of an element. It can store the name, symbol and mass (average/mono) and natural abundances of isotopes. Elements are retrieved from the ElementDB singleton which is created from the file “/OpenMS/CHEMISTRY/Elements.xml”. The EmpiricalFormula object can be used to represent the empirical formula of a compound as well as to extract its natural isotope abundance and weight. <br  />
<br  />
 <b>Example: Tutorial_Element.cpp</b> <br  />
 Work with Element object </p><div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03278.html">OpenMS/CHEMISTRY/ElementDB.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03275.html">OpenMS/CHEMISTRY/Element.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><a class="code" href="a00001.html#a12329">Int</a> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="keyword">const</span> <a class="code" href="a07901.html">ElementDB</a> * db = <a class="code" href="a07901.html#ae141930c87ab8851ec484bc1fd0ccb90">ElementDB::getInstance</a>();</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// extract carbon element from ElementDB</span></div>
<div class="line">  <span class="comment">// .getResidue(&quot;C&quot;) would work as well</span></div>
<div class="line">  <a class="code" href="a07897.html">Element</a> carbon = *db-&gt;<a class="code" href="a07901.html#a2af9395d26b51383a18fa548f9789eb9">getElement</a>(<span class="stringliteral">&quot;Carbon&quot;</span>); </div>
<div class="line"> </div>
<div class="line">  <span class="comment">// output name, symbol, monoisotopic weight and average weight</span></div>
<div class="line">  cout &lt;&lt; carbon.<a class="code" href="a07897.html#aba14af7abcdff896b4673851c4b6248f">getName</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; carbon.<a class="code" href="a07897.html#a45fe2b2630d5ff57c151320f156a1b34">getSymbol</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; carbon.<a class="code" href="a07897.html#a3a98b87fe586720f67f6b18c41778772">getMonoWeight</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; carbon.<a class="code" href="a07897.html#a94ec5aa6fb849e4859971fc78afa68de">getAverageWeight</a>() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --><p><br  />
<br  />
 <b>Example: Tutorial_EmpiricalFormula.cpp</b> <br  />
 Extract isotope distribution and monoisotopic weight of an EmpiricalFormula object</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03281.html">OpenMS/CHEMISTRY/EmpiricalFormula.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03278.html">OpenMS/CHEMISTRY/ElementDB.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03290.html">OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><a class="code" href="a00001.html#a12329">Int</a> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <a class="code" href="a07905.html">EmpiricalFormula</a> methanol(<span class="stringliteral">&quot;CH3OH&quot;</span>), water(<span class="stringliteral">&quot;H2O&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// sum up empirical formula</span></div>
<div class="line">  <a class="code" href="a07905.html">EmpiricalFormula</a> <a class="code" href="a05786.html#ga4ed94a7147b86680101341d95e24253c">sum</a> = methanol + water;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// get element from ElementDB</span></div>
<div class="line">  <span class="keyword">const</span> <a class="code" href="a07897.html">Element</a> * carbon = <a class="code" href="a07901.html#ae141930c87ab8851ec484bc1fd0ccb90">ElementDB::getInstance</a>()-&gt;<a class="code" href="a07901.html#a2af9395d26b51383a18fa548f9789eb9">getElement</a>(<span class="stringliteral">&quot;Carbon&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// output number of carbon atoms and average weight </span></div>
<div class="line">  cout &lt;&lt; <a class="code" href="a05786.html#ga4ed94a7147b86680101341d95e24253c">sum</a> &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; <a class="code" href="a05786.html#ga4ed94a7147b86680101341d95e24253c">sum</a>.getNumberOf(carbon) &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; <a class="code" href="a05786.html#ga4ed94a7147b86680101341d95e24253c">sum</a>.getAverageWeight() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// extract the isotope distribution</span></div>
<div class="line">  <a class="code" href="a07961.html">IsotopeDistribution</a> iso_dist = <a class="code" href="a05786.html#ga4ed94a7147b86680101341d95e24253c">sum</a>.getIsotopeDistribution(<a class="code" href="a07925.html">CoarseIsotopePatternGenerator</a>(3));</div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">const</span> <span class="keyword">auto</span>&amp; it : iso_dist)</div>
<div class="line">  {</div>
<div class="line">    cout &lt;&lt; it.getMZ() &lt;&lt; <span class="stringliteral">&quot; &quot;</span> &lt;&lt; it.getIntensity() &lt;&lt; endl;</div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_aaseq"></a>
AASequence - Representing a Peptide</h2>
<p>An AASequence object stores a (potentially chemically modified) peptide. It can conveniently be constructed from the amino acid sequence (e.g., a string or a string literal “DEFIANGR”). Modifications may be encoded using the unimod name. Once constructed, many convenient functions are available to calculate peptide or ion properties. <br  />
<br  />
 <b>Example: Tutorial_AASequence.cpp</b> <br  />
 Compute and output basic AASequence properties</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03257.html">OpenMS/CHEMISTRY/AASequence.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// generate AASequence object from String</span></div>
<div class="line">  <span class="keyword">const</span> <a class="code" href="a08769.html">String</a> s = <span class="stringliteral">&quot;DEFIANGER&quot;</span>;</div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> peptide1 = <a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(s);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// generate AASequence object from string literal</span></div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> peptide2 = <a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;PEPTIDER&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// extract prefix and suffix</span></div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> prefix(peptide1.<a class="code" href="a07865.html#a63b98c65a63f87f41a8a74ac1e04138b">getPrefix</a>(2));</div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> suffix(peptide1.<a class="code" href="a07865.html#ab197aa53161e28b8f340ae72c9458f8a">getSuffix</a>(3));</div>
<div class="line">  cout &lt;&lt; peptide1.<a class="code" href="a07865.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; prefix &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; suffix &lt;&lt; endl;</div>
<div class="line">  </div>
<div class="line">  <span class="comment">// create chemically modified peptide</span></div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> peptide_meth_ox = <a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;PEPTIDESEKUEM(Oxidation)CER&quot;</span>);</div>
<div class="line">  cout &lt;&lt; peptide_meth_ox.<a class="code" href="a07865.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; peptide_meth_ox.<a class="code" href="a07865.html#a9c074ddb7e1c4f2539770fdaf31fe853">toUnmodifiedString</a>()</div>
<div class="line">       &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// mass of the full, uncharged peptide</span></div>
<div class="line">  <span class="keywordtype">double</span> peptide_mass_mono = peptide_meth_ox.<a class="code" href="a07865.html#a6cd2e12e525408ce2615d7877dfaa08a">getMonoWeight</a>();</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Monoisotopic mass of the uncharged, full peptide: &quot;</span> &lt;&lt; peptide_mass_mono &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="keywordtype">double</span> peptide_mass_avg = peptide_meth_ox.<a class="code" href="a07865.html#a448ebefe744de451f7cce4b02160fa87">getAverageWeight</a>();</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Average mass of the uncharged, full peptide: &quot;</span> &lt;&lt; peptide_mass_avg &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// mass of the 2+ charged b-ion with the given sequence</span></div>
<div class="line">  <span class="keywordtype">double</span> ion_mass_2plus = peptide_meth_ox.<a class="code" href="a07865.html#a6cd2e12e525408ce2615d7877dfaa08a">getMonoWeight</a>(<a class="code" href="a08061.html#a7651af21f9cf8ed6445415903fc6cb48a7894cbe7cb5a3c5508b8becc24ca040e">Residue::BIon</a>, 2);</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Mass of the doubly positively charged b-ion: &quot;</span> &lt;&lt; ion_mass_2plus &lt;&lt; endl;</div>
<div class="line">         </div>
<div class="line">  <span class="comment">// ... many more</span></div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">}</div>
<div class="line"> </div>
</div><!-- fragment --><p> Internally, an AASequence object is composed of Residues.</p>
<h2><a class="anchor" id="tutorial_residue"></a>
Residue, ResidueDB</h2>
<p>Residues are the building blocks of AASequence objects. They store physico-chemical properties of specific amino acids. ResidueDB stores that data and is initialized from the file “data/CHEMISTRY/residues.xml”. <br  />
<br  />
 <b>Example: Tutorial_Residue.cpp</b> <br  />
 Compute and output basic Residue properties</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03365.html">OpenMS/CHEMISTRY/ResidueDB.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03362.html">OpenMS/CHEMISTRY/Residue.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03257.html">OpenMS/CHEMISTRY/AASequence.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// get ResidueDB singleton</span></div>
<div class="line">  <a class="code" href="a08065.html">ResidueDB</a> <span class="keyword">const</span> * res_db = <a class="code" href="a08065.html#a416f4888cc1f2b8acb849cc5347b4342">ResidueDB::getInstance</a>();</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// query Lysine</span></div>
<div class="line">  <a class="code" href="a08061.html">Residue</a> <span class="keyword">const</span> * lys = res_db-&gt;<a class="code" href="a08065.html#aa01a846d5ba4f9ea4128685f55247967">getResidue</a>(<span class="stringliteral">&quot;Lysine&quot;</span>);</div>
<div class="line"> </div>
<div class="line">  cout &lt;&lt; lys-&gt;<a class="code" href="a08061.html#aba14af7abcdff896b4673851c4b6248f">getName</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; lys-&gt;<a class="code" href="a08061.html#a78b142edbd067ad26f26373afa189db0">getThreeLetterCode</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; lys-&gt;<a class="code" href="a08061.html#a61ccbfa578837af65174c4ee27101b5d">getOneLetterCode</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; lys-&gt;<a class="code" href="a08061.html#a47fd4b1ed0d95b8649da8dc5748fbc30">getFormula</a>().<a class="code" href="a07905.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; lys-&gt;<a class="code" href="a08061.html#ade734ac92287f1969e1f38d97fd34cbe">getAverageWeight</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; lys-&gt;<a class="code" href="a08061.html#af294687b8b378a19671bcd0cc5e3d01f">getMonoWeight</a>() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// one letter code query of Arginine</span></div>
<div class="line">  <a class="code" href="a08061.html">Residue</a> <span class="keyword">const</span> * arg = res_db-&gt;<a class="code" href="a08065.html#aa01a846d5ba4f9ea4128685f55247967">getResidue</a>(<span class="charliteral">&#39;R&#39;</span>);</div>
<div class="line"> </div>
<div class="line">  cout &lt;&lt; arg-&gt;<a class="code" href="a08061.html#aba14af7abcdff896b4673851c4b6248f">getName</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; arg-&gt;<a class="code" href="a08061.html#a47fd4b1ed0d95b8649da8dc5748fbc30">getFormula</a>().<a class="code" href="a07905.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; arg-&gt;<a class="code" href="a08061.html#af294687b8b378a19671bcd0cc5e3d01f">getMonoWeight</a>() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// construct a AASequence object, query a residue </span></div>
<div class="line">  <span class="comment">// and output some of its properties</span></div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> aas = <a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;DEFIANGER&quot;</span>);</div>
<div class="line">  cout &lt;&lt; aas[3].getName() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[3].<a class="code" href="a07865.html#a2aa42ac42db18219a6f89ecb7bde389a">getFormula</a>().<a class="code" href="a07905.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[3].<a class="code" href="a07865.html#a6cd2e12e525408ce2615d7877dfaa08a">getMonoWeight</a>() &lt;&lt; endl; </div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_residuemod"></a>
ResidueModification, ModificationsDB</h2>
<p>If a residue is modified (e.g. phosphorylation of an amino acid) it can be stored in the ResidueModification class. The ResidueModification class stores information about chemical modifications of residues. Each ResidueModification has an ID, the residue that can be modified with this modification and the difference in mass between the unmodified and the modified residue, among other information. The Residue class allows to set one modification per residue and the mass difference of the modification is accounted for in the mass of the residue. The class ModificationsDB is a database of ResidueModifications. These are mostly initialized from the file “/share/CHEMISTRY/unimod.xml” containing a slightly modified version of the UniMod database of modifications. ModificationsDB has functions to search for modifications by name or mass. <br  />
<br  />
 <b>Example: Tutorial_ResidueModification.cpp</b> <br  />
 Set a ResidueModification on a Residue</p>
<div class="fragment"><div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">//                   OpenMS -- Open-Source Mass Spectrometry</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,</span></div>
<div class="line"><span class="comment">// ETH Zurich, and Freie Universitaet Berlin 2002-2018.</span></div>
<div class="line"><span class="comment">//</span></div>
<div class="line"><span class="comment">// This software is released under a three-clause BSD license:</span></div>
<div class="line"><span class="comment">//  * Redistributions of source code must retain the above copyright</span></div>
<div class="line"><span class="comment">//    notice, this list of conditions and the following disclaimer.</span></div>
<div class="line"><span class="comment">//  * Redistributions in binary form must reproduce the above copyright</span></div>
<div class="line"><span class="comment">//    notice, this list of conditions and the following disclaimer in the</span></div>
<div class="line"><span class="comment">//    documentation and/or other materials provided with the distribution.</span></div>
<div class="line"><span class="comment">//  * Neither the name of any author or any participating institution</span></div>
<div class="line"><span class="comment">//    may be used to endorse or promote products derived from this software</span></div>
<div class="line"><span class="comment">//    without specific prior written permission.</span></div>
<div class="line"><span class="comment">// For a full list of authors, refer to the file AUTHORS.</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS &quot;AS IS&quot;</span></div>
<div class="line"><span class="comment">// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE</span></div>
<div class="line"><span class="comment">// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE</span></div>
<div class="line"><span class="comment">// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING</span></div>
<div class="line"><span class="comment">// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,</span></div>
<div class="line"><span class="comment">// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,</span></div>
<div class="line"><span class="comment">// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;</span></div>
<div class="line"><span class="comment">// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,</span></div>
<div class="line"><span class="comment">// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR</span></div>
<div class="line"><span class="comment">// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF</span></div>
<div class="line"><span class="comment">// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</span></div>
<div class="line"><span class="comment">//</span></div>
<div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03257.html">OpenMS/CHEMISTRY/AASequence.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03368.html">OpenMS/CHEMISTRY/ResidueModification.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03344.html">OpenMS/CHEMISTRY/ModificationsDB.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// construct a AASequence object, query a residue</span></div>
<div class="line">  <span class="comment">// and output some of its properties</span></div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> aas = <a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;DECIANGER&quot;</span>);</div>
<div class="line">  cout &lt;&lt; aas[2].getName() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[2].<a class="code" href="a07865.html#a2aa42ac42db18219a6f89ecb7bde389a">getFormula</a>().<a class="code" href="a07905.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[2].getModificationName() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[2].<a class="code" href="a07865.html#a6cd2e12e525408ce2615d7877dfaa08a">getMonoWeight</a>() &lt;&lt; endl;</div>
<div class="line">  </div>
<div class="line">  <span class="comment">// find a modification in ModificationsDB</span></div>
<div class="line">  <span class="comment">// and output some of its properties</span></div>
<div class="line">  <span class="comment">// getInstance() returns a pointer to a ModsDB instance</span></div>
<div class="line">  <span class="keyword">const</span> <a class="code" href="a08069.html">ResidueModification</a>* mod = <a class="code" href="a08029.html#aef99622c54e40b2bae02a5ceb5c81eef">ModificationsDB::getInstance</a>()-&gt;<a class="code" href="a08029.html#aaaa3cb3bf4886e9ab4d0ceb2905224a8">getModification</a>(<span class="stringliteral">&quot;Carbamidomethyl (C)&quot;</span>);</div>
<div class="line">  cout &lt;&lt; mod-&gt;<a class="code" href="a08069.html#abae0525ed84c3fc9a6a14abe8f2f8ec9">getOrigin</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; mod-&gt;<a class="code" href="a08069.html#afca32dec9307255ca6a01424b0f15452">getFullId</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; mod-&gt;<a class="code" href="a08069.html#a8c9a8ccd7f43d0592f5b996d232f1eb9">getDiffMonoMass</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; mod-&gt;<a class="code" href="a08069.html#aca9fce25db8a5971b38cbce06abf44a0">getMonoMass</a>() &lt;&lt; endl;</div>
<div class="line">  </div>
<div class="line">  <span class="comment">// set the modification on a residue of a peptide</span></div>
<div class="line">  <span class="comment">// and output some of its properties (the formula and mass have changed)</span></div>
<div class="line">  <span class="comment">// in this case ModificationsDB is used in the background</span></div>
<div class="line">  <span class="comment">// to relate the name of the mod to its attributes</span></div>
<div class="line">  aas.<a class="code" href="a07865.html#a3a77e574f42bcdb282c1df6a82a6cc5c">setModification</a>(2, <span class="stringliteral">&quot;Carbamidomethyl (C)&quot;</span>);</div>
<div class="line">  cout &lt;&lt; aas[2].getName() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[2].<a class="code" href="a07865.html#a2aa42ac42db18219a6f89ecb7bde389a">getFormula</a>().<a class="code" href="a07905.html#a26b2839c6b49fe42966074e2375553d7">toString</a>() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[2].getModificationName() &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; aas[2].<a class="code" href="a07865.html#a6cd2e12e525408ce2615d7877dfaa08a">getMonoWeight</a>() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_tsg"></a>
TheoreticalSpectrumGenerator</h2>
<p>The TheoreticalSpectrumGenerator generates ion ladders from AASequences. <br  />
<br  />
 <b>Example: Tutorial_TheoreticalSpectrumGenerator.cpp</b> <br  />
 Generate theoretical spectra</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03398.html">OpenMS/CHEMISTRY/TheoreticalSpectrumGenerator.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03257.html">OpenMS/CHEMISTRY/AASequence.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04229.html">OpenMS/KERNEL/MSSpectrum.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04226.html">OpenMS/KERNEL/MSExperiment.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <span class="comment">// initialize a TheoreticalSpectrumGenerator</span></div>
<div class="line">  <a class="code" href="a08133.html">TheoreticalSpectrumGenerator</a> tsg;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// get current parameters</span></div>
<div class="line">  <span class="comment">// in this case default parameters, since we have not changed any yet</span></div>
<div class="line">  <a class="code" href="a08745.html">Param</a> tsg_settings = tsg.<a class="code" href="a08589.html#a1decb59b26f2579931e3b4850df53d98">getParameters</a>();</div>
<div class="line">    </div>
<div class="line">  <span class="comment">// with default parameters, only b- and y-ions are generated,</span></div>
<div class="line">  <span class="comment">// so we will add a-ions</span></div>
<div class="line">  tsg_settings.<a class="code" href="a08745.html#a432099944e11263aa20aaf8ea1389790">setValue</a>(<span class="stringliteral">&quot;add_a_ions&quot;</span>, <span class="stringliteral">&quot;true&quot;</span>);</div>
<div class="line">    </div>
<div class="line">  <span class="comment">// store ion types for each peak</span></div>
<div class="line">  tsg_settings.<a class="code" href="a08745.html#a432099944e11263aa20aaf8ea1389790">setValue</a>(<span class="stringliteral">&quot;add_metainfo&quot;</span>, <span class="stringliteral">&quot;true&quot;</span>);</div>
<div class="line">    </div>
<div class="line">  <span class="comment">// set the changed parameters for the TSG</span></div>
<div class="line">  tsg.<a class="code" href="a08589.html#abc8067511cfc949ae942ea2b08588f81">setParameters</a>(tsg_settings);</div>
<div class="line">                     </div>
<div class="line"> </div>
<div class="line">  <a class="code" href="a10097.html">PeakSpectrum</a> theoretical_spectrum;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// initialize peptide to be fragmented</span></div>
<div class="line">  <a class="code" href="a07865.html">AASequence</a> peptide = <a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;DEFIANGER&quot;</span>);</div>
<div class="line">  </div>
<div class="line">  <span class="comment">// generate a-, b- and y- ion spectrum of the peptide</span></div>
<div class="line">  <span class="comment">// with all fragment charges from 1 to 2</span></div>
<div class="line">  tsg.<a class="code" href="a08133.html#a8bf29e7f36f5d0f3a533cb590554a894">getSpectrum</a>(theoretical_spectrum, peptide, 1, 2);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// output of masses and meta information (ion-types) of some peaks</span></div>
<div class="line">  <span class="keyword">const</span> <a class="code" href="a10421.html">PeakSpectrum::StringDataArray</a>&amp; ion_types = theoretical_spectrum.<a class="code" href="a10097.html#ade02ee1e066e024a94db1c1753f11991">getStringDataArrays</a>().at(0);</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Mass of second peak: &quot;</span> &lt;&lt; theoretical_spectrum[1].getMZ()</div>
<div class="line">       &lt;&lt; <span class="stringliteral">&quot; | Ion type of second peak: &quot;</span> &lt;&lt; ion_types[1] &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Mass of tenth peak: &quot;</span> &lt;&lt; theoretical_spectrum[9].getMZ()</div>
<div class="line">       &lt;&lt; <span class="stringliteral">&quot; | Ion type of tenth peak: &quot;</span> &lt;&lt; ion_types[9] &lt;&lt; endl;</div>
<div class="line">  </div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">} <span class="comment">//end of main</span></div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_dep"></a>
DigestionEnzymeProtein, ProteaseDB and ProteaseDigestion</h2>
<p>OpenMS provides the most common digestion enzymes (DigestionEnzymeProtein) used in MS. They are stored in the ProteaseDB singleton and loaded from “/share/CHEMISTRY/Enzymes.xml”. <br  />
<br  />
 <b>Example: Tutorial_Enzyme.cpp</b> <br  />
 Digest amino acid sequence</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03257.html">OpenMS/CHEMISTRY/AASequence.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03359.html">OpenMS/CHEMISTRY/ProteaseDigestion.h</a>&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;vector&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>()</div>
<div class="line">{</div>
<div class="line">  <a class="code" href="a08057.html">ProteaseDigestion</a> protease;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// in this example, we don&#39;t produce peptides with missed cleavages</span></div>
<div class="line">  protease.<a class="code" href="a07909.html#a6ea268a321b164aaa6246c0909bed97f">setMissedCleavages</a>(0);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// output the number of tryptic peptides (no cut before proline) </span></div>
<div class="line">  protease.<a class="code" href="a08057.html#ad92ecbad776494178a2a3de0ab7eed19">setEnzyme</a>(<span class="stringliteral">&quot;Trypsin&quot;</span>);</div>
<div class="line">  cout &lt;&lt; protease.<a class="code" href="a08057.html#ad0d24e15a1dd611eceb19eb9c421c579">peptideCount</a>(<a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;ACKPDE&quot;</span>)) &lt;&lt; <span class="stringliteral">&quot; &quot;</span></div>
<div class="line">       &lt;&lt; protease.<a class="code" href="a08057.html#ad0d24e15a1dd611eceb19eb9c421c579">peptideCount</a>(<a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;ACRPDEKA&quot;</span>))</div>
<div class="line">       &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// digest C-terminally amidated peptide </span></div>
<div class="line">  vector&lt;AASequence&gt; products;</div>
<div class="line">  protease.<a class="code" href="a08057.html#af4f5ef513607033cc9e96f776474eed3">digest</a>(<a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;ARCDRE.(Amidated)&quot;</span>), products);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// output digestion products</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">const</span> <a class="code" href="a07865.html">AASequence</a>&amp; p : products)</div>
<div class="line">  {</div>
<div class="line">    cout &lt;&lt; p.toString() &lt;&lt; <span class="stringliteral">&quot; &quot;</span>;</div>
<div class="line">  }</div>
<div class="line">  cout &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// allow many miss-cleavages</span></div>
<div class="line">  protease.<a class="code" href="a07909.html#a6ea268a321b164aaa6246c0909bed97f">setMissedCleavages</a>(10);</div>
<div class="line">  protease.<a class="code" href="a08057.html#af4f5ef513607033cc9e96f776474eed3">digest</a>(<a class="code" href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">AASequence::fromString</a>(<span class="stringliteral">&quot;ARCDRE.(Amidated)&quot;</span>), products);</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// output digestion products</span></div>
<div class="line">  <span class="keywordflow">for</span> (<span class="keyword">const</span> <a class="code" href="a07865.html">AASequence</a>&amp; p : products)</div>
<div class="line">  {</div>
<div class="line">    cout &lt;&lt; p.toString() &lt;&lt; <span class="stringliteral">&quot; &quot;</span>;</div>
<div class="line">  }</div>
<div class="line">  cout &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// ... many more</span></div>
<div class="line">  <span class="keywordflow">return</span> 0;</div>
<div class="line">}</div>
<div class="line"> </div>
</div><!-- fragment --> <h1><a class="anchor" id="tutorial_tooldev"></a>
Tool development</h1>
<h2><a class="anchor" id="tutorial_topp"></a>
TOPP-Tool</h2>
<p>TOPP (The OpenMS Pipeline) tools are small command line applications built using the OpenMS library. They act as building blocks for complex analysis workflows and may perform e.g. simple signal processing tasks like filtering, up to more complex tasks like protein inference and quantitation over several MS runs. Common to all TOPP tools is a command line interface allowing automatic integration into workflow engines like KNIME. They are the preferred way to integrate novel methods as application into OpenMS. When we first create a novel TOPP tool it is considered unstable. To set it apart from the stable and well tested tools it gets first created as TOPP Util (note: the name “util” has historic reasons and may be changed to unstable tools in the future). If it is well tested it will be promoted to a stable Tool in future OpenMS versions.</p>
<p>Imagine that you want to create a new tool that allows filtering of sequence databases. What you usually would first do is check if such or similar functionality has already been implemented in any of the &gt;150 TOPP tools. If you are unsure which one to use, just ask on the mailing list, the gitter chat or contact one of the developers directly. The following subsection demonstrates how the original “DatabaseFilter” tool was created from scratch an integrated into OpenMS. Basically any tool you want to integrate needs to follow the steps outlined below.</p>
<p>But let’s first get started by defining what our tool should actually do: The DatabaseFilter tool should provide functionality to reduce a fasta database by filtering its entries based on different criteria. A simple criterion could be the length of a protein. To make the task a bit more interesting and to show other parts of the OpenMS library, we will start with a bit more complex filtering step that keeps all entries from the fasta database that have been identified in a peptide search (e.g., using X!Tandem, Mascot or MSGF+). This functionality might come in handy if the size of large databases needs to be reduced to a manageable size. In addition, we want the user to be able to choose between keeping and removing matching protein id.</p>
<h2><a class="anchor" id="tutorial_create"></a>
Create and register a minimal tool in OpenMS</h2>
<ul>
<li>Create an empty file <a class="el" href="a05567.html">src/utils/DatabaseFilter.cpp</a></li>
<li>Add the scaffold code for a minimal TOPP tool. Text in bold will later be adapted to our DatabaseFilter tool.</li>
</ul>
<p><b>Example: Tutorial_Template.cpp</b> <br  />
 Template for OpenMS tool development</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">//                   OpenMS -- Open-Source Mass Spectrometry</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,</span></div>
<div class="line"><span class="comment">// ETH Zurich, and Freie Universitaet Berlin 2002-2018.</span></div>
<div class="line"><span class="comment">//</span></div>
<div class="line"><span class="comment">// This software is released under a three-clause BSD license:</span></div>
<div class="line"><span class="comment">//  * Redistributions of source code must retain the above copyright</span></div>
<div class="line"><span class="comment">//    notice, this list of conditions and the following disclaimer.</span></div>
<div class="line"><span class="comment">//  * Redistributions in binary form must reproduce the above copyright</span></div>
<div class="line"><span class="comment">//    notice, this list of conditions and the following disclaimer in the</span></div>
<div class="line"><span class="comment">//    documentation and/or other materials provided with the distribution.</span></div>
<div class="line"><span class="comment">//  * Neither the name of any author or any participating institution</span></div>
<div class="line"><span class="comment">//    may be used to endorse or promote products derived from this software</span></div>
<div class="line"><span class="comment">//    without specific prior written permission.</span></div>
<div class="line"><span class="comment">// For a full list of authors, refer to the file AUTHORS.</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS &quot;AS IS&quot;</span></div>
<div class="line"><span class="comment">// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE</span></div>
<div class="line"><span class="comment">// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE</span></div>
<div class="line"><span class="comment">// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING</span></div>
<div class="line"><span class="comment">// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,</span></div>
<div class="line"><span class="comment">// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,</span></div>
<div class="line"><span class="comment">// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;</span></div>
<div class="line"><span class="comment">// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,</span></div>
<div class="line"><span class="comment">// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR</span></div>
<div class="line"><span class="comment">// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF</span></div>
<div class="line"><span class="comment">// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</span></div>
<div class="line"><span class="comment">//</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// $Maintainer: Maintainer $</span></div>
<div class="line"><span class="comment">// $Authors: Author1, Author2 $</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03251.html">OpenMS/APPLICATIONS/TOPPBase.h</a>&gt;</span></div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// Doxygen docu</span></div>
<div class="line"><span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"><span class="comment">// We do not want this class to show up in the docu:</span></div>
<div class="line"><span class="comment"></span> </div>
<div class="line"><span class="keyword">class </span>TOPPNewTool :</div>
<div class="line">  <span class="keyword">public</span> <a class="code" href="a07857.html">TOPPBase</a></div>
<div class="line">{</div>
<div class="line"><span class="keyword">public</span>:</div>
<div class="line">  TOPPNewTool() :</div>
<div class="line">    <a class="code" href="a07857.html">TOPPBase</a>(<span class="stringliteral">&quot;NewTool&quot;</span>, <span class="stringliteral">&quot;Template for Tool creation&quot;</span>, false)</div>
<div class="line">  {</div>
<div class="line"> </div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line"><span class="keyword">protected</span>:</div>
<div class="line"> </div>
<div class="line">  <span class="comment">// this function will be used to register the tool parameters</span></div>
<div class="line">  <span class="comment">// it gets automatically called on tool execution</span></div>
<div class="line">  <span class="keywordtype">void</span> registerOptionsAndFlags_()</div>
<div class="line">  {</div>
<div class="line"> </div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">  <span class="comment">// the main_ function is called after all parameters are read</span></div>
<div class="line">  ExitCodes main_(<span class="keywordtype">int</span>, <span class="keyword">const</span> <span class="keywordtype">char</span> **)</div>
<div class="line">  {</div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">    <span class="comment">// parsing parameters</span></div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">    <span class="comment">// reading input</span></div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">    <span class="comment">// calculations</span></div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">    <span class="comment">// writing output</span></div>
<div class="line">    <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line">  }</div>
<div class="line"> </div>
<div class="line">};</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line"><span class="comment">// the actual main function needed to create an executable</span></div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>(<span class="keywordtype">int</span> argc, <span class="keyword">const</span> <span class="keywordtype">char</span> ** argv)</div>
<div class="line">{</div>
<div class="line">  TOPPNewTool tool;</div>
<div class="line">  <span class="keywordflow">return</span> tool.main(argc, argv);</div>
<div class="line">}</div>
<div class="line"> </div>
</div><!-- fragment --><ul>
<li>Now add a line with DatabaseFilter.cpp to src/utils/executables.cmake. This registers the novel tool in the OpenMS build system.</li>
<li>Then add the tool to getUtilList() in src/openms/source/APPLICATIONS/ToolHandler.cpp This creates a manual (doxygen) page with the information –help output of the tool (using TOPPDocumenter). This page must be included at the end of the doxygen documentation of your tool (see other tools for an example).</li>
<li>Add yourself as Maintainer/Author</li>
<li>Write the basic documentation (doxygen docu). You probably need to refine it later but you can already insert the correct Toolname etc..</li>
</ul>
<h2><a class="anchor" id="tutorial_param_def"></a>
Define tool parameters</h2>
<p>Define tool parameters Each TOPP tool defines a set of parameters that will be available from the command line, KNIME, and other workflow systems. This is done in the void registerOptionsAndFlags_() method. In our case we want to read a protein database (fasta format), a file containing identification data (idXML format), and an option to switch between keeping (whitelisting) and removing (blacklisting) entries based on the filter result. This is our input. The reduced database forms the output and should be written to a protein database in fasta format. This is easily done by adding following lines to: <br  />
<br  />
 <b>Example: Tutorial_Final.cpp</b> <br  />
 Registration of tool parameters</p>
<div class="fragment"><div class="line"> </div>
<div class="line"> <span class="keywordtype">void</span> registerOptionsAndFlags_()<span class="keyword"> override</span></div>
<div class="line"><span class="keyword"> </span>{</div>
<div class="line">   registerInputFile_(<span class="stringliteral">&quot;in&quot;</span>, <span class="stringliteral">&quot;&lt;file&gt;&quot;</span>, <span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;Input FASTA file, containing a database.&quot;</span>);</div>
<div class="line">   setValidFormats_(<span class="stringliteral">&quot;in&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;fasta&quot;</span>));</div>
<div class="line">   registerInputFile_(<span class="stringliteral">&quot;id&quot;</span>, <span class="stringliteral">&quot;&lt;file&gt;&quot;</span>, <span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;Input file containing identified peptides and proteins.&quot;</span>);</div>
<div class="line">   setValidFormats_(<span class="stringliteral">&quot;id&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;idXML,mzid&quot;</span>));</div>
<div class="line">   registerStringOption_(<span class="stringliteral">&quot;method&quot;</span>, <span class="stringliteral">&quot;&lt;choice&gt;&quot;</span>, <span class="stringliteral">&quot;whitelist&quot;</span>, <span class="stringliteral">&quot;Switch between white-/blacklisting of protein IDs&quot;</span>, <span class="keyword">false</span>);</div>
<div class="line">   setValidStrings_(<span class="stringliteral">&quot;method&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;whitelist,blacklist&quot;</span>));</div>
<div class="line">   registerOutputFile_(<span class="stringliteral">&quot;out&quot;</span>, <span class="stringliteral">&quot;&lt;file&gt;&quot;</span>, <span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;Output FASTA file where the reduced database will be written to.&quot;</span>);</div>
<div class="line">   setValidFormats_(<span class="stringliteral">&quot;out&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;fasta&quot;</span>));</div>
<div class="line"> }</div>
<div class="line"> </div>
</div><!-- fragment --><p> Functions, classes and references can be checked in the OpenMS / TOPP documentation (<a href="ftp://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/index.html">ftp://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/index.html</a>)</p>
<h2><a class="anchor" id="tutorial_param_read"></a>
Read tool parameters</h2>
<p>After a tool is executed, the registered parameters are available in the main_ function of the TOPP tool and can be read using the getStringOption_ method. Special methods for integers, lists and floating point parameters exist and are in the TOPPBase documentation but are not needed for this example. <br  />
<br  />
 <b>Example: Tutorial_Final.cpp</b> <br  />
</p>
<div class="fragment"><div class="line"> </div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">   <span class="comment">// parsing parameters</span></div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">   String in(getStringOption_(<span class="stringliteral">&quot;in&quot;</span>));</div>
<div class="line">   String ids(getStringOption_(<span class="stringliteral">&quot;id&quot;</span>));</div>
<div class="line">   String method(getStringOption_(<span class="stringliteral">&quot;method&quot;</span>));</div>
<div class="line">   <span class="keywordtype">bool</span> whitelist = (method == <span class="stringliteral">&quot;whitelist&quot;</span>);</div>
<div class="line">   String out(getStringOption_(<span class="stringliteral">&quot;out&quot;</span>));</div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_read"></a>
Read Input Files</h2>
<p>First the different file formats and data structures for peptide identifications have to be included at the top of the file. <br  />
<br  />
 <b>Example: Tutorial_Final.cpp</b> <br  />
 Add essential includes</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03893.html">OpenMS/FORMAT/FileHandler.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03980.html">OpenMS/FORMAT/IdXMLFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03887.html">OpenMS/FORMAT/FASTAFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04037.html">OpenMS/FORMAT/MzIdentMLFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03896.html">OpenMS/FORMAT/FileTypes.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04502.html">OpenMS/METADATA/PeptideIdentification.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03251.html">OpenMS/APPLICATIONS/TOPPBase.h</a>&gt;</span></div>
<div class="line"> </div>
</div><!-- fragment --><p> <b> Read the input files </b> <br  />
</p>
<div class="fragment"><div class="line"> </div>
<div class="line">   vector&lt;FASTAFile::FASTAEntry&gt; db;</div>
<div class="line">   FASTAFile().load(in, db);</div>
<div class="line"> </div>
</div><!-- fragment --><p> Note: both peptide_identifications and protein_identifications contain protein accessions. The difference between them is that protein_identifications only contain the inferred set of protein accessions while peptide_identifications contains all protein accessions the peptides map to. We consider only the larger set of protein accessions stored in the peptide identifications. In principle, it would be easy to add another parameter that adds a filter for the inferred accessions stored in protein_identifications.</p>
<h2><a class="anchor" id="tutorial_add"></a>
Add the tool functionality</h2>
<p>First, the accessions are extracted from the IdXML file. Here knowledge of the data structure is needed to extract the protein accessions. The class PeptideIdentification stores general information about a single identified spectrum (e.g., retention time, precursor mass-to-charge). A vector of PeptideHits is stored in each PeptideIdentification object and represent the potentially multiple PSMs of a single spectrum. They can be returned by calling .getHits(). Each peptide sequence stored in a PeptideHit may map to one or multiple proteins. This peptide to protein mapping information is stored in a vector of PeptideEvidence accessible by .getPepitdeEvidences(). From each of these evidences we can extract the protein accession with .getProteinAccession().</p>
<p>To store all proteins accessions in the set id_accessions, we write: <br  />
<br  />
 <b>Example: Tutorial_Final.cpp</b> <br  />
 Store protein accessions</p>
<div class="fragment"><div class="line"> </div>
<div class="line"> <span class="keywordtype">void</span> filterByProteinIDs_(<span class="keyword">const</span> vector&lt;FASTAFile::FASTAEntry&gt;&amp; db, <span class="keyword">const</span> vector&lt;PeptideIdentification&gt;&amp; peptide_identifications, <span class="keywordtype">bool</span> whitelist, vector&lt;FASTAFile::FASTAEntry&gt;&amp; db_new)</div>
<div class="line"> {</div>
<div class="line">   set&lt;String&gt; id_accessions;</div>
<div class="line">   <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> i = 0; i != peptide_identifications.size(); ++i)</div>
<div class="line">   {</div>
<div class="line">     <span class="keyword">const</span> PeptideIdentification&amp; <span class="keywordtype">id</span> = peptide_identifications[i];</div>
<div class="line">     <span class="keyword">const</span> vector&lt;PeptideHit&gt;&amp; hits = <span class="keywordtype">id</span>.getHits();</div>
<div class="line">     <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> <a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a> = 0; <a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a> != hits.size(); ++<a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a>)</div>
<div class="line">     {</div>
<div class="line">       <span class="keyword">const</span> vector&lt;PeptideEvidence&gt;&amp; evidences = hits[<a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a>].getPeptideEvidences();</div>
<div class="line">       <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> m = 0; m != evidences.size(); ++m)</div>
<div class="line">       {</div>
<div class="line">         <span class="keyword">const</span> String&amp; id_accession = evidences[m].getProteinAccession();</div>
<div class="line">         id_accessions.insert(id_accession);</div>
<div class="line">       }</div>
<div class="line">     }</div>
<div class="line">   }</div>
<div class="line"> </div>
</div><!-- fragment --><p> Now that we assembled the set of all protein accessions we are ready to compare them to the fasta_accessions. If they are similar and the method whitelist or they are different and the method blacklist was chosen, the fasta entries are copied to the new fasta database. <br  />
<br  />
 <b>Example: Tutorial_Final.cpp</b> <br  />
 Add method functionality </p><div class="fragment"><div class="line"> </div>
<div class="line">   <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> i = 0; i != db.size() ; ++i)</div>
<div class="line">   {</div>
<div class="line">     <span class="keyword">const</span> String&amp; fasta_accession = db[i].identifier;</div>
<div class="line">     <span class="keyword">const</span> <span class="keywordtype">bool</span> found = id_accessions.find(fasta_accession) != id_accessions.end();</div>
<div class="line">     <span class="keywordflow">if</span> ((found &amp;&amp; whitelist) || (!found &amp;&amp; !whitelist)) <span class="comment">//either found in the whitelist or not found in the blacklist</span></div>
<div class="line">     {</div>
<div class="line">       db_new.push_back(db[i]);</div>
<div class="line">     }</div>
<div class="line">   }</div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_write"></a>
Write Output Files</h2>
<p><b>Example: Tutorial_Final.cpp</b> <br  />
 Write the output</p>
<div class="fragment"><div class="line"> </div>
<div class="line">   FASTAFile().store(out, db_new);</div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_test"></a>
Adding TOPP tests</h2>
<p>Testing your tools is essential and required to promote your experimental util to an official TOPP tool. It is not mandatory to provide a test for a util but appreciated. For this test a .fasta and a compatible .idXML file have to be added to /src/tests/topp/. Further the test procedure has to be added to CMakeLists.txt in the same folder. <br  />
<br  />
 <b>Example: Tutorial_Test.cpp</b> <br  />
 Add tests</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="preprocessor"># DatabaseFilter test:</span></div>
<div class="line">add_test(<span class="stringliteral">&quot;UTILS_DatabaseFilter_1&quot;</span> ${TOPP_BIN_PATH}/DatabaseFilter -<a class="code" href="a05829.html#a69f942d7289d5ed2ffe3a00b12235d04">test</a> -in ${DATA_DIR_TOPP}/DatabaseFilter_1.fasta -accession ${DATA_DIR_TOPP}/DatabaseFilter_1.idXML -out DatabaseFilter_1_out.fasta.tmp)</div>
<div class="line">add_test(<span class="stringliteral">&quot;UTILS_DatabaseFilter_1_out&quot;</span> ${DIFF} -in1 DatabaseFilter_1_out.fasta.tmp -in2 ${DATA_DIR_TOPP}/DatabaseFilter_1_out.fasta )</div>
<div class="line">set_tests_properties(<span class="stringliteral">&quot;UTILS_DatabaseFilter_1_out&quot;</span> PROPERTIES DEPENDS <span class="stringliteral">&quot;UTILS_DatabaseFilter_1&quot;</span>)</div>
<div class="line">add_test(<span class="stringliteral">&quot;UTILS_DatabaseFilter_2&quot;</span> ${TOPP_BIN_PATH}/DatabaseFilter -<a class="code" href="a05829.html#a69f942d7289d5ed2ffe3a00b12235d04">test</a> -in ${DATA_DIR_TOPP}/DatabaseFilter_1.fasta -accession ${DATA_DIR_TOPP}/DatabaseFilter_1.idXML -out DatabaseFilter_2_out.fasta.tmp -method blacklist)</div>
<div class="line">add_test(<span class="stringliteral">&quot;UTILS_DatabaseFilter_2_out&quot;</span> ${DIFF} -in1 DatabaseFilter_2_out.fasta.tmp -in2 ${DATA_DIR_TOPP}/DatabaseFilter_2_out.fasta )</div>
<div class="line">set_tests_properties(<span class="stringliteral">&quot;UTILS_DatabaseFilter_2_out&quot;</span> PROPERTIES DEPENDS <span class="stringliteral">&quot;UTILS_DatabaseFilter_2&quot;</span>)</div>
<div class="line"> </div>
</div><!-- fragment --><p> These tests run the program with the given parameters and then call a diff tool to compare the generated output to the expected output.</p>
<h2><a class="anchor" id="tutorial_doc"></a>
Finish documentation</h2>
<p>We add it to the UTILS docu page (in <a class="el" href="a02624.html">doc/doxygen/public/UTILS.doxygen</a>). Later (when we have a working application) we will write an application test (this is optional but recommended for Utils. For Tools it is mandatory). See TOPP tools above and add the test to the bottom of src/tests/topp/CMakeLists.txt.</p>
<h2><a class="anchor" id="tutorial_polish"></a>
Polish your code</h2>
<p>This is how a util should look after code polishing: Here, the support for different formats was extended (idXML and MZIdentML). Since different filter criteria may be introduced in the future, the structure was slightly changed with a function for the filtering by ID (filterByProteinIDs_) - in order to allow higher flexibility when adding new a functionality later on. <br  />
<br  />
 <b>Example: Tutorial_final.cpp</b> <br  />
 Polish your code - add additional functionality</p>
<div class="fragment"><div class="line"> </div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">//                   OpenMS -- Open-Source Mass Spectrometry</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,</span></div>
<div class="line"><span class="comment">// ETH Zurich, and Freie Universitaet Berlin 2002-2018.</span></div>
<div class="line"><span class="comment">//</span></div>
<div class="line"><span class="comment">// This software is released under a three-clause BSD license:</span></div>
<div class="line"><span class="comment">//  * Redistributions of source code must retain the above copyright</span></div>
<div class="line"><span class="comment">//    notice, this list of conditions and the following disclaimer.</span></div>
<div class="line"><span class="comment">//  * Redistributions in binary form must reproduce the above copyright</span></div>
<div class="line"><span class="comment">//    notice, this list of conditions and the following disclaimer in the</span></div>
<div class="line"><span class="comment">//    documentation and/or other materials provided with the distribution.</span></div>
<div class="line"><span class="comment">//  * Neither the name of any author or any participating institution</span></div>
<div class="line"><span class="comment">//    may be used to endorse or promote products derived from this software</span></div>
<div class="line"><span class="comment">//    without specific prior written permission.</span></div>
<div class="line"><span class="comment">// For a full list of authors, refer to the file AUTHORS.</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS &quot;AS IS&quot;</span></div>
<div class="line"><span class="comment">// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE</span></div>
<div class="line"><span class="comment">// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE</span></div>
<div class="line"><span class="comment">// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING</span></div>
<div class="line"><span class="comment">// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,</span></div>
<div class="line"><span class="comment">// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,</span></div>
<div class="line"><span class="comment">// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;</span></div>
<div class="line"><span class="comment">// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,</span></div>
<div class="line"><span class="comment">// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR</span></div>
<div class="line"><span class="comment">// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF</span></div>
<div class="line"><span class="comment">// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</span></div>
<div class="line"><span class="comment">//</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"><span class="comment">// $Maintainer: Oliver Alka $</span></div>
<div class="line"><span class="comment">// $Authors: Oliver Alka $</span></div>
<div class="line"><span class="comment">// --------------------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03893.html">OpenMS/FORMAT/FileHandler.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03980.html">OpenMS/FORMAT/IdXMLFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03887.html">OpenMS/FORMAT/FASTAFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04037.html">OpenMS/FORMAT/MzIdentMLFile.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03896.html">OpenMS/FORMAT/FileTypes.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a04502.html">OpenMS/METADATA/PeptideIdentification.h</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="a03251.html">OpenMS/APPLICATIONS/TOPPBase.h</a>&gt;</span></div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line"><span class="keyword">using namespace </span><a class="code" href="a05822.html">OpenMS</a>;</div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"> </div>
<div class="line"><span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"><span class="comment">//Doxygen docu</span></div>
<div class="line"><span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"><span class="comment">// We do not want this class to show up in the docu:</span></div>
<div class="line"><span class="comment"></span> </div>
<div class="line"><span class="keyword">class </span>TOPPDatabaseFilter :</div>
<div class="line"> <span class="keyword">public</span> <a class="code" href="a07857.html">TOPPBase</a></div>
<div class="line">{</div>
<div class="line"><span class="keyword">public</span>:</div>
<div class="line"> TOPPDatabaseFilter() :</div>
<div class="line">   <a class="code" href="a07857.html">TOPPBase</a>(<span class="stringliteral">&quot;DatabaseFilter&quot;</span>, <span class="stringliteral">&quot;Filters a protein database (FASTA format) based on identified proteins&quot;</span>, false)</div>
<div class="line"> {</div>
<div class="line"> }</div>
<div class="line"> </div>
<div class="line"><span class="keyword">protected</span>:</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line"> <span class="keywordtype">void</span> registerOptionsAndFlags_()<span class="keyword"> override</span></div>
<div class="line"><span class="keyword"> </span>{</div>
<div class="line">   registerInputFile_(<span class="stringliteral">&quot;in&quot;</span>, <span class="stringliteral">&quot;&lt;file&gt;&quot;</span>, <span class="stringliteral">&quot;&quot;</span>,<span class="stringliteral">&quot;Input FASTA file, containing a database.&quot;</span>);</div>
<div class="line">   setValidFormats_(<span class="stringliteral">&quot;in&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;fasta&quot;</span>));</div>
<div class="line">   registerInputFile_(<span class="stringliteral">&quot;id&quot;</span>, <span class="stringliteral">&quot;&lt;file&gt;&quot;</span>, <span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;Input file containing identified peptides and proteins.&quot;</span>);</div>
<div class="line">   setValidFormats_(<span class="stringliteral">&quot;id&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;idXML,mzid&quot;</span>));</div>
<div class="line">   registerStringOption_(<span class="stringliteral">&quot;method&quot;</span>, <span class="stringliteral">&quot;&lt;choice&gt;&quot;</span>, <span class="stringliteral">&quot;whitelist&quot;</span>, <span class="stringliteral">&quot;Switch between white-/blacklisting of protein IDs&quot;</span>, <span class="keyword">false</span>);</div>
<div class="line">   setValidStrings_(<span class="stringliteral">&quot;method&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;whitelist,blacklist&quot;</span>));</div>
<div class="line">   registerOutputFile_(<span class="stringliteral">&quot;out&quot;</span>, <span class="stringliteral">&quot;&lt;file&gt;&quot;</span>, <span class="stringliteral">&quot;&quot;</span>, <span class="stringliteral">&quot;Output FASTA file where the reduced database will be written to.&quot;</span>);</div>
<div class="line">   setValidFormats_(<span class="stringliteral">&quot;out&quot;</span>, ListUtils::create&lt;String&gt;(<span class="stringliteral">&quot;fasta&quot;</span>));</div>
<div class="line"> }</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line"> <span class="keywordtype">void</span> filterByProteinIDs_(<span class="keyword">const</span> vector&lt;FASTAFile::FASTAEntry&gt;&amp; db, <span class="keyword">const</span> vector&lt;PeptideIdentification&gt;&amp; peptide_identifications, <span class="keywordtype">bool</span> whitelist, vector&lt;FASTAFile::FASTAEntry&gt;&amp; db_new)</div>
<div class="line"> {</div>
<div class="line">   set&lt;String&gt; id_accessions;</div>
<div class="line">   <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> i = 0; i != peptide_identifications.size(); ++i)</div>
<div class="line">   {</div>
<div class="line">     <span class="keyword">const</span> <a class="code" href="a10645.html">PeptideIdentification</a>&amp; <span class="keywordtype">id</span> = peptide_identifications[i];</div>
<div class="line">     <span class="keyword">const</span> vector&lt;PeptideHit&gt;&amp; hits = <span class="keywordtype">id</span>.getHits();</div>
<div class="line">     <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> <a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a> = 0; <a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a> != hits.size(); ++<a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a>)</div>
<div class="line">     {</div>
<div class="line">       <span class="keyword">const</span> vector&lt;PeptideEvidence&gt;&amp; evidences = hits[<a class="code" href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">k</a>].getPeptideEvidences();</div>
<div class="line">       <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> m = 0; m != evidences.size(); ++m)</div>
<div class="line">       {</div>
<div class="line">         <span class="keyword">const</span> <a class="code" href="a08769.html">String</a>&amp; id_accession = evidences[m].getProteinAccession();</div>
<div class="line">         id_accessions.insert(id_accession);</div>
<div class="line">       }</div>
<div class="line">     }</div>
<div class="line">   }</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   <a class="code" href="a03512.html#ac0313369bb5cec887b8c393508f17a9d">OPENMS_LOG_INFO</a> &lt;&lt; <span class="stringliteral">&quot;Protein IDs: &quot;</span> &lt;&lt; id_accessions.size() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   <span class="keywordflow">for</span> (<a class="code" href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">Size</a> i = 0; i != db.size() ; ++i)</div>
<div class="line">   {</div>
<div class="line">     <span class="keyword">const</span> <a class="code" href="a08769.html">String</a>&amp; fasta_accession = db[i].identifier;</div>
<div class="line">     <span class="keyword">const</span> <span class="keywordtype">bool</span> found = id_accessions.find(fasta_accession) != id_accessions.end();</div>
<div class="line">     <span class="keywordflow">if</span> ((found &amp;&amp; whitelist) || (!found &amp;&amp; !whitelist)) <span class="comment">//either found in the whitelist or not found in the blacklist</span></div>
<div class="line">     {</div>
<div class="line">       db_new.push_back(db[i]);</div>
<div class="line">     }</div>
<div class="line">   }</div>
<div class="line"> </div>
<div class="line"> }</div>
<div class="line"> </div>
<div class="line"> ExitCodes main_(<span class="keywordtype">int</span>, <span class="keyword">const</span> <span class="keywordtype">char</span> **)<span class="keyword"> override</span></div>
<div class="line"><span class="keyword"> </span>{</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">   <span class="comment">// parsing parameters</span></div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">   <a class="code" href="a08769.html">String</a> in(getStringOption_(<span class="stringliteral">&quot;in&quot;</span>));</div>
<div class="line">   <a class="code" href="a08769.html">String</a> ids(getStringOption_(<span class="stringliteral">&quot;id&quot;</span>));</div>
<div class="line">   <a class="code" href="a08769.html">String</a> method(getStringOption_(<span class="stringliteral">&quot;method&quot;</span>));</div>
<div class="line">   <span class="keywordtype">bool</span> whitelist = (method == <span class="stringliteral">&quot;whitelist&quot;</span>);</div>
<div class="line">   <a class="code" href="a08769.html">String</a> out(getStringOption_(<span class="stringliteral">&quot;out&quot;</span>));</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">   <span class="comment">// reading input</span></div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   vector&lt;FASTAFile::FASTAEntry&gt; db;</div>
<div class="line">   <a class="code" href="a09253.html">FASTAFile</a>().<a class="code" href="a09253.html#a874b02f68f4805124f9f144fb50b8018">load</a>(in, db);</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   <span class="comment">// Check if no filter criteria was given</span></div>
<div class="line">   <span class="comment">// If you add a new filter please check if it was set here as well</span></div>
<div class="line">   <span class="keywordflow">if</span> (ids.empty())</div>
<div class="line">   {</div>
<div class="line">     <a class="code" href="a09253.html">FASTAFile</a>().<a class="code" href="a09253.html#a92ac2327eb6ca69916f33ea9649fb75d">store</a>(out, db);</div>
<div class="line">   }</div>
<div class="line"> </div>
<div class="line">   vector&lt;FASTAFile::FASTAEntry&gt; db_new;</div>
<div class="line"> </div>
<div class="line">   <span class="keywordflow">if</span> (!ids.empty()) <span class="comment">// filter by protein IDs</span></div>
<div class="line">   {</div>
<div class="line">     <a class="code" href="a09265.html">FileHandler</a> fh;</div>
<div class="line">     <a class="code" href="a09269.html#a1d1cfd8ffb84e947f82999c682b666a7">FileTypes::Type</a> ids_type = fh.<a class="code" href="a09265.html#a851b569c3e65a4eeabdd10425f2e6402">getType</a>(ids);</div>
<div class="line">     vector&lt;ProteinIdentification&gt; protein_identifications;</div>
<div class="line">     vector&lt;PeptideIdentification&gt; peptide_identifications;</div>
<div class="line"> </div>
<div class="line">     <span class="keywordflow">if</span> (ids_type == <a class="code" href="a09269.html#a1d1cfd8ffb84e947f82999c682b666a7ab11f34894c9d7062f12a006a68c52859">FileTypes::IDXML</a>)</div>
<div class="line">     {</div>
<div class="line">       <a class="code" href="a09437.html">IdXMLFile</a>().<a class="code" href="a09437.html#a914edf92cb5c95df15eab679bf67691a">load</a>(ids, protein_identifications, peptide_identifications);</div>
<div class="line">     }</div>
<div class="line">     <span class="keywordflow">else</span> <span class="keywordflow">if</span> (ids_type == <a class="code" href="a09269.html#a1d1cfd8ffb84e947f82999c682b666a7a17bf484ac6de6611479496206fd7b49a">FileTypes::MZIDENTML</a>)</div>
<div class="line">     {</div>
<div class="line">       <a class="code" href="a09533.html">MzIdentMLFile</a>().<a class="code" href="a09533.html#ad2ed8d7ca3ca2f4fb1efa287b4657f38">load</a>(ids, protein_identifications, peptide_identifications);</div>
<div class="line">     }</div>
<div class="line">     <span class="keywordflow">else</span></div>
<div class="line">     {</div>
<div class="line">       writeLog_(<span class="stringliteral">&quot;Error: Unknown input file type given. Aborting!&quot;</span>);</div>
<div class="line">       printUsage_();</div>
<div class="line">       <span class="keywordflow">return</span> ILLEGAL_PARAMETERS;</div>
<div class="line">     }</div>
<div class="line"> </div>
<div class="line">     <a class="code" href="a03512.html#ac0313369bb5cec887b8c393508f17a9d">OPENMS_LOG_INFO</a> &lt;&lt; <span class="stringliteral">&quot;Identifications: &quot;</span> &lt;&lt; ids.size() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">     <span class="comment">// run filter</span></div>
<div class="line">     filterByProteinIDs_(db, peptide_identifications, whitelist, db_new);</div>
<div class="line">   }</div>
<div class="line"> </div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line">   <span class="comment">// writing output</span></div>
<div class="line">   <span class="comment">//-------------------------------------------------------------</span></div>
<div class="line"> </div>
<div class="line">   <a class="code" href="a03512.html#ac0313369bb5cec887b8c393508f17a9d">OPENMS_LOG_INFO</a> &lt;&lt; <span class="stringliteral">&quot;Database entries (before / after): &quot;</span> &lt;&lt; db.size() &lt;&lt; <span class="stringliteral">&quot; / &quot;</span> &lt;&lt; db_new.size() &lt;&lt; endl;</div>
<div class="line"> </div>
<div class="line">   <a class="code" href="a09253.html">FASTAFile</a>().<a class="code" href="a09253.html#a92ac2327eb6ca69916f33ea9649fb75d">store</a>(out, db_new);</div>
<div class="line"> </div>
<div class="line"> </div>
<div class="line">   <span class="keywordflow">return</span> EXECUTION_OK;</div>
<div class="line"> }</div>
<div class="line"> </div>
<div class="line">};</div>
<div class="line"> </div>
<div class="line"><span class="keywordtype">int</span> <a class="code" href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a>(<span class="keywordtype">int</span> argc, <span class="keyword">const</span> <span class="keywordtype">char</span> ** argv)</div>
<div class="line">{</div>
<div class="line"> TOPPDatabaseFilter tool;</div>
<div class="line"> <span class="keywordflow">return</span> tool.main(argc, argv);</div>
<div class="line">}</div>
<div class="line"> </div>
<div class="line"> </div>
</div><!-- fragment --> <h2><a class="anchor" id="tutorial_pull"></a>
Open a pull request</h2>
<p>Afterwards you can commit your changes to a new branch “feature/DatabaseFilter” of your OpenMS clone on github and submit a pull request on your github page. After a short review process by the OpenMS Team, the tool will be added the OpenMS Library.</p>
<h1><a class="anchor" id="tutorial_appendix"></a>
Appendix</h1>
<h2><a class="anchor" id="tutorial_d_dim"></a>
D-dimensional data points</h2>
<p>The d-dimensional data points are needed in special cases only, e.g. in template classes that operate in any number of dimensions. The base class of the d-dimensional data points is DPeak. The methods to access the position are getPosition and setPosition. Note that the one-dimensional and two-dimensional data points also have the methods getPosition and setPosition. They are needed in order to be able to write algorithms that can operate on all data point types. It is, however, recommended not to use these members unless you really write such a generic algorithm.</p>
<h2><a class="anchor" id="tutorial_ext_project"></a>
OpenMS as external project</h2>
<p>If OpenMS TOPP_tools and UTILS_tools are not sufficient for a certain scenario, you can either request changes to OpenMS or modify/extend your own fork of OpenMS. A third alternative is using OpenMS as a dependency while not touching OpenMS itself. Once you've finished your new tool, and it runs on the development machine, you're done. If you want to develop with OpenMS as external project have a look the example code ( /share/OpenMS/examples/external_code/). </p>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<div class="ttc" id="aa10081_html"><div class="ttname"><a href="a10081.html">OpenMS::MSExperiment</a></div><div class="ttdoc">In-Memory representation of a mass spectrometry experiment.</div><div class="ttdef"><b>Definition:</b> MSExperiment.h:77</div></div>
<div class="ttc" id="aa05216_html_a217dbf8b442f20279ea00b898af96f52"><div class="ttname"><a href="a05216.html#a217dbf8b442f20279ea00b898af96f52">main</a></div><div class="ttdeci">int main(int argc, const char **argv)</div><div class="ttdef"><b>Definition:</b> INIFileEditor.cpp:73</div></div>
<div class="ttc" id="aa03365_html"><div class="ttname"><a href="a03365.html">ResidueDB.h</a></div></div>
<div class="ttc" id="aa07865_html_a63b98c65a63f87f41a8a74ac1e04138b"><div class="ttname"><a href="a07865.html#a63b98c65a63f87f41a8a74ac1e04138b">OpenMS::AASequence::getPrefix</a></div><div class="ttdeci">AASequence getPrefix(Size index) const</div><div class="ttdoc">returns a peptide sequence of the first index residues</div></div>
<div class="ttc" id="aa08061_html_ade734ac92287f1969e1f38d97fd34cbe"><div class="ttname"><a href="a08061.html#ade734ac92287f1969e1f38d97fd34cbe">OpenMS::Residue::getAverageWeight</a></div><div class="ttdeci">double getAverageWeight(ResidueType res_type=Full) const</div><div class="ttdoc">returns average weight of the residue</div></div>
<div class="ttc" id="aa03278_html"><div class="ttname"><a href="a03278.html">ElementDB.h</a></div></div>
<div class="ttc" id="aa10725_html_ac69ba691dca00671a379aa956816cac5"><div class="ttname"><a href="a10725.html#ac69ba691dca00671a379aa956816cac5">OpenMS::SpectrumSettings::getPrecursors</a></div><div class="ttdeci">const std::vector&lt; Precursor &gt; &amp; getPrecursors() const</div><div class="ttdoc">returns a const reference to the precursors</div></div>
<div class="ttc" id="aa10081_html_a878abca09cb3f6810e634b72b99007eb"><div class="ttname"><a href="a10081.html#a878abca09cb3f6810e634b72b99007eb">OpenMS::MSExperiment::areaBegin</a></div><div class="ttdeci">AreaIterator areaBegin(CoordinateType min_rt, CoordinateType max_rt, CoordinateType min_mz, CoordinateType max_mz)</div><div class="ttdoc">Returns an area iterator for area.</div></div>
<div class="ttc" id="aa08069_html_afca32dec9307255ca6a01424b0f15452"><div class="ttname"><a href="a08069.html#afca32dec9307255ca6a01424b0f15452">OpenMS::ResidueModification::getFullId</a></div><div class="ttdeci">const String &amp; getFullId() const</div><div class="ttdoc">returns the full identifier of the mod (Unimod accession + origin, if available)</div></div>
<div class="ttc" id="aa04223_html"><div class="ttname"><a href="a04223.html">MSChromatogram.h</a></div></div>
<div class="ttc" id="aa05830_html_a93a8dc068053ec44ee7e45864ef969f7"><div class="ttname"><a href="a05830.html#a93a8dc068053ec44ee7e45864ef969f7">OpenMS::Constants::k</a></div><div class="ttdeci">const double k</div></div>
<div class="ttc" id="aa03251_html"><div class="ttname"><a href="a03251.html">TOPPBase.h</a></div></div>
<div class="ttc" id="aa10677_html_a86f5c6d7c17c3aa94ec734e2a4fb50fa"><div class="ttname"><a href="a10677.html#a86f5c6d7c17c3aa94ec734e2a4fb50fa">OpenMS::ProteinIdentification::setHits</a></div><div class="ttdeci">void setHits(const std::vector&lt; ProteinHit &gt; &amp;hits)</div><div class="ttdoc">Sets the protein hits.</div></div>
<div class="ttc" id="aa09269_html_a1d1cfd8ffb84e947f82999c682b666a7"><div class="ttname"><a href="a09269.html#a1d1cfd8ffb84e947f82999c682b666a7">OpenMS::FileTypes::Type</a></div><div class="ttdeci">Type</div><div class="ttdoc">Actual file types enum.</div><div class="ttdef"><b>Definition:</b> FileTypes.h:58</div></div>
<div class="ttc" id="aa08745_html"><div class="ttname"><a href="a08745.html">OpenMS::Param</a></div><div class="ttdoc">Management and storage of parameters / INI files.</div><div class="ttdef"><b>Definition:</b> Param.h:73</div></div>
<div class="ttc" id="aa09437_html"><div class="ttname"><a href="a09437.html">OpenMS::IdXMLFile</a></div><div class="ttdoc">Used to load and store idXML files.</div><div class="ttdef"><b>Definition:</b> IdXMLFile.h:63</div></div>
<div class="ttc" id="aa08581_html_a56a6f29cf9fa7e13a8df5c871afe4b94"><div class="ttname"><a href="a08581.html#a56a6f29cf9fa7e13a8df5c871afe4b94">OpenMS::DateTime::getDate</a></div><div class="ttdeci">void getDate(UInt &amp;month, UInt &amp;day, UInt &amp;year) const</div><div class="ttdoc">Fills the arguments with the date.</div></div>
<div class="ttc" id="aa09537_html"><div class="ttname"><a href="a09537.html">OpenMS::MzMLFile</a></div><div class="ttdoc">File adapter for MzML files.</div><div class="ttdef"><b>Definition:</b> MzMLFile.h:55</div></div>
<div class="ttc" id="aa08061_html_aba14af7abcdff896b4673851c4b6248f"><div class="ttname"><a href="a08061.html#aba14af7abcdff896b4673851c4b6248f">OpenMS::Residue::getName</a></div><div class="ttdeci">const String &amp; getName() const</div><div class="ttdoc">returns the name of the residue</div></div>
<div class="ttc" id="aa10149_html_a75a60db5a9ee9fc609b6d4caaf4d82b3"><div class="ttname"><a href="a10149.html#a75a60db5a9ee9fc609b6d4caaf4d82b3">OpenMS::RangeManager::getMaxInt</a></div><div class="ttdeci">double getMaxInt() const</div><div class="ttdoc">Returns the maximum intensity.</div><div class="ttdef"><b>Definition:</b> RangeManager.h:129</div></div>
<div class="ttc" id="aa08133_html"><div class="ttname"><a href="a08133.html">OpenMS::TheoreticalSpectrumGenerator</a></div><div class="ttdoc">Generates theoretical spectra for peptides with various options.</div><div class="ttdef"><b>Definition:</b> TheoreticalSpectrumGenerator.h:68</div></div>
<div class="ttc" id="aa07865_html_a2aa42ac42db18219a6f89ecb7bde389a"><div class="ttname"><a href="a07865.html#a2aa42ac42db18219a6f89ecb7bde389a">OpenMS::AASequence::getFormula</a></div><div class="ttdeci">EmpiricalFormula getFormula(Residue::ResidueType type=Residue::Full, Int charge=0) const</div><div class="ttdoc">returns the formula of the peptide</div></div>
<div class="ttc" id="aa10685_html_a6b3f6ef19caf97c201fba99444bf0663"><div class="ttname"><a href="a10685.html#a6b3f6ef19caf97c201fba99444bf0663">OpenMS::ProteinIdentification::SearchParameters::charges</a></div><div class="ttdeci">String charges</div><div class="ttdoc">The allowed charges for the search.</div><div class="ttdef"><b>Definition:</b> ProteinIdentification.h:227</div></div>
<div class="ttc" id="aa10081_html_afb936fe46d35a9d29e36552a68c1ef2a"><div class="ttname"><a href="a10081.html#afb936fe46d35a9d29e36552a68c1ef2a">OpenMS::MSExperiment::areaEnd</a></div><div class="ttdeci">AreaIterator areaEnd()</div><div class="ttdoc">Returns an invalid area iterator marking the end of an area.</div></div>
<div class="ttc" id="aa08029_html_aef99622c54e40b2bae02a5ceb5c81eef"><div class="ttname"><a href="a08029.html#aef99622c54e40b2bae02a5ceb5c81eef">OpenMS::ModificationsDB::getInstance</a></div><div class="ttdeci">static ModificationsDB * getInstance(OpenMS::String unimod_file=&quot;CHEMISTRY/unimod.xml&quot;, OpenMS::String psimod_file=&quot;CHEMISTRY/PSI-MOD.obo&quot;, OpenMS::String xlmod_file=&quot;CHEMISTRY/XLMOD.obo&quot;)</div><div class="ttdoc">Returns a pointer to the modifications DB (singleton)</div><div class="ttdef"><b>Definition:</b> ModificationsDB.h:77</div></div>
<div class="ttc" id="aa10149_html_aeab815807dd490278441926c58ebf3b8"><div class="ttname"><a href="a10149.html#aeab815807dd490278441926c58ebf3b8">OpenMS::RangeManager::getMinInt</a></div><div class="ttdeci">double getMinInt() const</div><div class="ttdoc">Returns the minimum intensity.</div><div class="ttdef"><b>Definition:</b> RangeManager.h:123</div></div>
<div class="ttc" id="aa10645_html_aee2ef4fb4e6965786d394761543a27b3"><div class="ttname"><a href="a10645.html#aee2ef4fb4e6965786d394761543a27b3">OpenMS::PeptideIdentification::getScoreType</a></div><div class="ttdeci">const String &amp; getScoreType() const</div><div class="ttdoc">returns the peptide score type</div></div>
<div class="ttc" id="aa03344_html"><div class="ttname"><a href="a03344.html">ModificationsDB.h</a></div></div>
<div class="ttc" id="aa09269_html_a1d1cfd8ffb84e947f82999c682b666a7ab11f34894c9d7062f12a006a68c52859"><div class="ttname"><a href="a09269.html#a1d1cfd8ffb84e947f82999c682b666a7ab11f34894c9d7062f12a006a68c52859">OpenMS::FileTypes::IDXML</a></div><div class="ttdoc">OpenMS identification format (.idXML)</div><div class="ttdef"><b>Definition:</b> FileTypes.h:66</div></div>
<div class="ttc" id="aa10125_html_a396c5ebcac2e70f9b9a478bb2456985b"><div class="ttname"><a href="a10125.html#a396c5ebcac2e70f9b9a478bb2456985b">OpenMS::Peak2D::setRT</a></div><div class="ttdeci">void setRT(CoordinateType coordinate)</div><div class="ttdoc">Mutable access to the RT coordinate (index 0)</div><div class="ttdef"><b>Definition:</b> Peak2D.h:214</div></div>
<div class="ttc" id="aa10097_html_add06770712321b6e489bbdbcd123c888"><div class="ttname"><a href="a10097.html#add06770712321b6e489bbdbcd123c888">OpenMS::MSSpectrum::MZEnd</a></div><div class="ttdeci">Iterator MZEnd(CoordinateType mz)</div><div class="ttdoc">Binary search for peak range end (returns the past-the-end iterator)</div></div>
<div class="ttc" id="aa07897_html_a3a98b87fe586720f67f6b18c41778772"><div class="ttname"><a href="a07897.html#a3a98b87fe586720f67f6b18c41778772">OpenMS::Element::getMonoWeight</a></div><div class="ttdeci">double getMonoWeight() const</div><div class="ttdoc">returns the mono isotopic weight of the element</div></div>
<div class="ttc" id="aa07925_html"><div class="ttname"><a href="a07925.html">OpenMS::CoarseIsotopePatternGenerator</a></div><div class="ttdoc">Isotope pattern generator for coarse isotope distributions.</div><div class="ttdef"><b>Definition:</b> CoarseIsotopePatternGenerator.h:96</div></div>
<div class="ttc" id="aa04499_html"><div class="ttname"><a href="a04499.html">PeptideHit.h</a></div></div>
<div class="ttc" id="aa09437_html_a914edf92cb5c95df15eab679bf67691a"><div class="ttname"><a href="a09437.html#a914edf92cb5c95df15eab679bf67691a">OpenMS::IdXMLFile::load</a></div><div class="ttdeci">void load(const String &amp;filename, std::vector&lt; ProteinIdentification &gt; &amp;protein_ids, std::vector&lt; PeptideIdentification &gt; &amp;peptide_ids)</div><div class="ttdoc">Loads the identifications of an idXML file without identifier.</div></div>
<div class="ttc" id="aa05782_html_gaf9ecec2d692138fab9167164a457cbd4"><div class="ttname"><a href="a05782.html#gaf9ecec2d692138fab9167164a457cbd4">OpenMS::Size</a></div><div class="ttdeci">size_t Size</div><div class="ttdoc">Size type e.g. used as variable which can hold result of size()</div><div class="ttdef"><b>Definition:</b> Types.h:127</div></div>
<div class="ttc" id="aa07905_html_a26b2839c6b49fe42966074e2375553d7"><div class="ttname"><a href="a07905.html#a26b2839c6b49fe42966074e2375553d7">OpenMS::EmpiricalFormula::toString</a></div><div class="ttdeci">String toString() const</div><div class="ttdoc">returns the formula as a string (charges are not included)</div></div>
<div class="ttc" id="aa10617_html_af26eef0efdc5d34a5a280d2bb42f79e9"><div class="ttname"><a href="a10617.html#af26eef0efdc5d34a5a280d2bb42f79e9">OpenMS::PeptideHit::setScore</a></div><div class="ttdeci">void setScore(double score)</div><div class="ttdoc">sets the PSM score</div></div>
<div class="ttc" id="aa09945_html"><div class="ttname"><a href="a09945.html">OpenMS::ChromatogramPeak</a></div><div class="ttdoc">A 1-dimensional raw data point or peak for chromatograms.</div><div class="ttdef"><b>Definition:</b> ChromatogramPeak.h:54</div></div>
<div class="ttc" id="aa10097_html"><div class="ttname"><a href="a10097.html">OpenMS::MSSpectrum</a></div><div class="ttdoc">The representation of a 1D spectrum.</div><div class="ttdef"><b>Definition:</b> MSSpectrum.h:67</div></div>
<div class="ttc" id="aa09533_html"><div class="ttname"><a href="a09533.html">OpenMS::MzIdentMLFile</a></div><div class="ttdoc">File adapter for MzIdentML files.</div><div class="ttdef"><b>Definition:</b> MzIdentMLFile.h:67</div></div>
<div class="ttc" id="aa10645_html_a566e1d015a01fa37751e0a77c64d70a8"><div class="ttname"><a href="a10645.html#a566e1d015a01fa37751e0a77c64d70a8">OpenMS::PeptideIdentification::setMZ</a></div><div class="ttdeci">void setMZ(double mz)</div><div class="ttdoc">sets the MZ of the MS2 spectrum</div></div>
<div class="ttc" id="aa10149_html_ab76aaa57fa843ab12a5239af35775eb9"><div class="ttname"><a href="a10149.html#ab76aaa57fa843ab12a5239af35775eb9">OpenMS::RangeManager::getMax</a></div><div class="ttdeci">const PositionType &amp; getMax() const</div><div class="ttdoc">Returns the maximum position.</div><div class="ttdef"><b>Definition:</b> RangeManager.h:117</div></div>
<div class="ttc" id="aa10081_html_ae3d9a0d5a931f4978e44abb667dec531"><div class="ttname"><a href="a10081.html#ae3d9a0d5a931f4978e44abb667dec531">OpenMS::MSExperiment::getPrecursorSpectrum</a></div><div class="ttdeci">ConstIterator getPrecursorSpectrum(ConstIterator iterator) const</div><div class="ttdoc">Returns the precursor spectrum of the scan pointed to by iterator.</div></div>
<div class="ttc" id="aa07909_html_a6ea268a321b164aaa6246c0909bed97f"><div class="ttname"><a href="a07909.html#a6ea268a321b164aaa6246c0909bed97f">OpenMS::EnzymaticDigestion::setMissedCleavages</a></div><div class="ttdeci">void setMissedCleavages(Size missed_cleavages)</div><div class="ttdoc">Sets the number of missed cleavages for the digestion (default is 0). This setting is ignored when lo...</div></div>
<div class="ttc" id="aa08581_html"><div class="ttname"><a href="a08581.html">OpenMS::DateTime</a></div><div class="ttdoc">DateTime Class.</div><div class="ttdef"><b>Definition:</b> DateTime.h:54</div></div>
<div class="ttc" id="aa04037_html"><div class="ttname"><a href="a04037.html">MzIdentMLFile.h</a></div></div>
<div class="ttc" id="aa07905_html"><div class="ttname"><a href="a07905.html">OpenMS::EmpiricalFormula</a></div><div class="ttdoc">Representation of an empirical formula.</div><div class="ttdef"><b>Definition:</b> EmpiricalFormula.h:82</div></div>
<div class="ttc" id="aa08305_html"><div class="ttname"><a href="a08305.html">OpenMS::Exception::InvalidSize</a></div><div class="ttdoc">Invalid UInt exception.</div><div class="ttdef"><b>Definition:</b> Exception.h:303</div></div>
<div class="ttc" id="aa07897_html_a45fe2b2630d5ff57c151320f156a1b34"><div class="ttname"><a href="a07897.html#a45fe2b2630d5ff57c151320f156a1b34">OpenMS::Element::getSymbol</a></div><div class="ttdeci">const String &amp; getSymbol() const</div><div class="ttdoc">returns symbol of the element</div></div>
<div class="ttc" id="aa08057_html_af4f5ef513607033cc9e96f776474eed3"><div class="ttname"><a href="a08057.html#af4f5ef513607033cc9e96f776474eed3">OpenMS::ProteaseDigestion::digest</a></div><div class="ttdeci">Size digest(const AASequence &amp;protein, std::vector&lt; AASequence &gt; &amp;output, Size min_length=1, Size max_length=0) const</div><div class="ttdoc">: Performs the enzymatic digestion of a protein.</div></div>
<div class="ttc" id="aa10657_html_a2b8bf5f3febdb76461bdea9ff39d6b6d"><div class="ttname"><a href="a10657.html#a2b8bf5f3febdb76461bdea9ff39d6b6d">OpenMS::ProteinHit::setSequence</a></div><div class="ttdeci">void setSequence(const String &amp;sequence)</div><div class="ttdoc">sets the protein sequence</div></div>
<div class="ttc" id="aa03980_html"><div class="ttname"><a href="a03980.html">IdXMLFile.h</a></div></div>
<div class="ttc" id="aa07961_html"><div class="ttname"><a href="a07961.html">OpenMS::IsotopeDistribution</a></div><div class="ttdef"><b>Definition:</b> IsotopeDistribution.h:64</div></div>
<div class="ttc" id="aa08061_html"><div class="ttname"><a href="a08061.html">OpenMS::Residue</a></div><div class="ttdoc">Representation of a residue.</div><div class="ttdef"><b>Definition:</b> Residue.h:61</div></div>
<div class="ttc" id="aa10097_html_a24f081fe74bbbf773533ad19e96d8358"><div class="ttname"><a href="a10097.html#a24f081fe74bbbf773533ad19e96d8358">OpenMS::MSSpectrum::setMSLevel</a></div><div class="ttdeci">void setMSLevel(UInt ms_level)</div><div class="ttdoc">Sets the MS level.</div></div>
<div class="ttc" id="aa10677_html"><div class="ttname"><a href="a10677.html">OpenMS::ProteinIdentification</a></div><div class="ttdoc">Representation of a protein identification run.</div><div class="ttdef"><b>Definition:</b> ProteinIdentification.h:71</div></div>
<div class="ttc" id="aa10677_html_a82e595befae76c1f7c22918420d2d9e8"><div class="ttname"><a href="a10677.html#a82e595befae76c1f7c22918420d2d9e8">OpenMS::ProteinIdentification::setSearchEngineVersion</a></div><div class="ttdeci">void setSearchEngineVersion(const String &amp;search_engine_version)</div><div class="ttdoc">Sets the search engine version.</div></div>
<div class="ttc" id="aa04229_html"><div class="ttname"><a href="a04229.html">MSSpectrum.h</a></div></div>
<div class="ttc" id="aa03359_html"><div class="ttname"><a href="a03359.html">ProteaseDigestion.h</a></div></div>
<div class="ttc" id="aa07897_html_a94ec5aa6fb849e4859971fc78afa68de"><div class="ttname"><a href="a07897.html#a94ec5aa6fb849e4859971fc78afa68de">OpenMS::Element::getAverageWeight</a></div><div class="ttdeci">double getAverageWeight() const</div><div class="ttdoc">returns the average weight of the element</div></div>
<div class="ttc" id="aa07897_html_aba14af7abcdff896b4673851c4b6248f"><div class="ttname"><a href="a07897.html#aba14af7abcdff896b4673851c4b6248f">OpenMS::Element::getName</a></div><div class="ttdeci">const String &amp; getName() const</div><div class="ttdoc">returns the name of the element</div></div>
<div class="ttc" id="aa09253_html_a874b02f68f4805124f9f144fb50b8018"><div class="ttname"><a href="a09253.html#a874b02f68f4805124f9f144fb50b8018">OpenMS::FASTAFile::load</a></div><div class="ttdeci">static void load(const String &amp;filename, std::vector&lt; FASTAEntry &gt; &amp;data)</div><div class="ttdoc">loads a FASTA file given by 'filename' and stores the information in 'data'</div></div>
<div class="ttc" id="aa03887_html"><div class="ttname"><a href="a03887.html">FASTAFile.h</a></div></div>
<div class="ttc" id="aa03275_html"><div class="ttname"><a href="a03275.html">Element.h</a></div></div>
<div class="ttc" id="aa03533_html"><div class="ttname"><a href="a03533.html">Types.h</a></div></div>
<div class="ttc" id="aa04502_html"><div class="ttname"><a href="a04502.html">PeptideIdentification.h</a></div></div>
<div class="ttc" id="aa10677_html_afb0f58ddeea0e6f58da132fae1421563"><div class="ttname"><a href="a10677.html#afb0f58ddeea0e6f58da132fae1421563">OpenMS::ProteinIdentification::setSearchParameters</a></div><div class="ttdeci">void setSearchParameters(const SearchParameters &amp;search_parameters)</div><div class="ttdoc">Sets the search parameters.</div></div>
<div class="ttc" id="aa08029_html_aaaa3cb3bf4886e9ab4d0ceb2905224a8"><div class="ttname"><a href="a08029.html#aaaa3cb3bf4886e9ab4d0ceb2905224a8">OpenMS::ModificationsDB::getModification</a></div><div class="ttdeci">const ResidueModification * getModification(Size index) const</div><div class="ttdoc">Returns the modification with the given index. note: out-of-bounds check is only performed in debug m...</div></div>
<div class="ttc" id="aa03512_html_ac0313369bb5cec887b8c393508f17a9d"><div class="ttname"><a href="a03512.html#ac0313369bb5cec887b8c393508f17a9d">OPENMS_LOG_INFO</a></div><div class="ttdeci">#define OPENMS_LOG_INFO</div><div class="ttdoc">Macro if a information, e.g. a status should be reported.</div><div class="ttdef"><b>Definition:</b> LogStream.h:465</div></div>
<div class="ttc" id="aa08133_html_a8bf29e7f36f5d0f3a533cb590554a894"><div class="ttname"><a href="a08133.html#a8bf29e7f36f5d0f3a533cb590554a894">OpenMS::TheoreticalSpectrumGenerator::getSpectrum</a></div><div class="ttdeci">virtual void getSpectrum(PeakSpectrum &amp;spec, const AASequence &amp;peptide, Int min_charge, Int max_charge) const</div><div class="ttdoc">Generates a spectrum for a peptide sequence, with the ion types that are set in the tool parameters.</div></div>
<div class="ttc" id="aa09269_html_a1d1cfd8ffb84e947f82999c682b666a7a17bf484ac6de6611479496206fd7b49a"><div class="ttname"><a href="a09269.html#a1d1cfd8ffb84e947f82999c682b666a7a17bf484ac6de6611479496206fd7b49a">OpenMS::FileTypes::MZIDENTML</a></div><div class="ttdoc">mzIdentML (HUPO PSI AnalysisXML followup format) (.mzid)</div><div class="ttdef"><b>Definition:</b> FileTypes.h:77</div></div>
<div class="ttc" id="aa10657_html_a64df7d8252c536e76dd3ef6468fc61b8"><div class="ttname"><a href="a10657.html#a64df7d8252c536e76dd3ef6468fc61b8">OpenMS::ProteinHit::setScore</a></div><div class="ttdeci">void setScore(const double score)</div><div class="ttdoc">sets the score of the protein hit</div></div>
<div class="ttc" id="aa10097_html_a86f7b200260209865e730a2e4fcaa628"><div class="ttname"><a href="a10097.html#a86f7b200260209865e730a2e4fcaa628">OpenMS::MSSpectrum::sortByPosition</a></div><div class="ttdeci">void sortByPosition()</div><div class="ttdoc">Lexicographically sorts the peaks by their position.</div></div>
<div class="ttc" id="aa07897_html"><div class="ttname"><a href="a07897.html">OpenMS::Element</a></div><div class="ttdoc">Representation of an element.</div><div class="ttdef"><b>Definition:</b> Element.h:53</div></div>
<div class="ttc" id="aa10057_html"><div class="ttname"><a href="a10057.html">OpenMS::FeatureMap</a></div><div class="ttdoc">A container for features.</div><div class="ttdef"><b>Definition:</b> FeatureMap.h:95</div></div>
<div class="ttc" id="aa10617_html_a7daed733ac80f8002f3a1fbdc97c21e1"><div class="ttname"><a href="a10617.html#a7daed733ac80f8002f3a1fbdc97c21e1">OpenMS::PeptideHit::setRank</a></div><div class="ttdeci">void setRank(UInt newrank)</div><div class="ttdoc">sets the PSM rank</div></div>
<div class="ttc" id="aa10657_html"><div class="ttname"><a href="a10657.html">OpenMS::ProteinHit</a></div><div class="ttdoc">Representation of a protein hit.</div><div class="ttdef"><b>Definition:</b> ProteinHit.h:57</div></div>
<div class="ttc" id="aa05829_html_a69f942d7289d5ed2ffe3a00b12235d04"><div class="ttname"><a href="a05829.html#a69f942d7289d5ed2ffe3a00b12235d04">OpenMS::Internal::ClassTest::test</a></div><div class="ttdeci">bool test</div><div class="ttdoc">Status of the current subsection.</div></div>
<div class="ttc" id="aa10125_html_a2694f0775c46d9076355cb85d9815425"><div class="ttname"><a href="a10125.html#a2694f0775c46d9076355cb85d9815425">OpenMS::Peak2D::setMZ</a></div><div class="ttdeci">void setMZ(CoordinateType coordinate)</div><div class="ttdoc">Mutable access to the m/z coordinate (index 1)</div><div class="ttdef"><b>Definition:</b> Peak2D.h:202</div></div>
<div class="ttc" id="aa10685_html"><div class="ttname"><a href="a10685.html">OpenMS::ProteinIdentification::SearchParameters</a></div><div class="ttdoc">Search parameters of the DB search.</div><div class="ttdef"><b>Definition:</b> ProteinIdentification.h:221</div></div>
<div class="ttc" id="aa10645_html_afe08b9bf9f9599bffa043dad0feeea43"><div class="ttname"><a href="a10645.html#afe08b9bf9f9599bffa043dad0feeea43">OpenMS::PeptideIdentification::getMZ</a></div><div class="ttdeci">double getMZ() const</div><div class="ttdoc">returns the MZ of the MS2 spectrum</div></div>
<div class="ttc" id="aa10073_html"><div class="ttname"><a href="a10073.html">OpenMS::MSChromatogram</a></div><div class="ttdoc">The representation of a chromatogram.</div><div class="ttdef"><b>Definition:</b> MSChromatogram.h:54</div></div>
<div class="ttc" id="aa07865_html_a3a77e574f42bcdb282c1df6a82a6cc5c"><div class="ttname"><a href="a07865.html#a3a77e574f42bcdb282c1df6a82a6cc5c">OpenMS::AASequence::setModification</a></div><div class="ttdeci">void setModification(Size index, const String &amp;modification)</div></div>
<div class="ttc" id="aa08065_html_aa01a846d5ba4f9ea4128685f55247967"><div class="ttname"><a href="a08065.html#aa01a846d5ba4f9ea4128685f55247967">OpenMS::ResidueDB::getResidue</a></div><div class="ttdeci">const Residue * getResidue(const String &amp;name) const</div><div class="ttdoc">returns a pointer to the residue with name, 3 letter code or 1 letter code name</div></div>
<div class="ttc" id="aa10389_html_aab74cfd047d44d3b939b327a16e4d6f1"><div class="ttname"><a href="a10389.html#aab74cfd047d44d3b939b327a16e4d6f1">OpenMS::ChromatogramSettings::getPrecursor</a></div><div class="ttdeci">const Precursor &amp; getPrecursor() const</div><div class="ttdoc">returns a const reference to the precursors</div></div>
<div class="ttc" id="aa07865_html_aa41dfefaa0fdd58959b3a3210c8173e8"><div class="ttname"><a href="a07865.html#aa41dfefaa0fdd58959b3a3210c8173e8">OpenMS::AASequence::fromString</a></div><div class="ttdeci">static AASequence fromString(const String &amp;s, bool permissive=true)</div><div class="ttdoc">create AASequence object by parsing an OpenMS string</div></div>
<div class="ttc" id="aa00001_html_a12329"><div class="ttname"><a href="a00001.html#a12329">int</a></div></div>
<div class="ttc" id="aa10685_html_ae525bb5a71aa3ae9d811588db4652c00"><div class="ttname"><a href="a10685.html#ae525bb5a71aa3ae9d811588db4652c00">OpenMS::ProteinIdentification::SearchParameters::db</a></div><div class="ttdeci">String db</div><div class="ttdoc">The used database.</div><div class="ttdef"><b>Definition:</b> ProteinIdentification.h:224</div></div>
<div class="ttc" id="aa03896_html"><div class="ttname"><a href="a03896.html">FileTypes.h</a></div></div>
<div class="ttc" id="aa09945_html_a98fe0c3fba1788864d35d8ef6159459f"><div class="ttname"><a href="a09945.html#a98fe0c3fba1788864d35d8ef6159459f">OpenMS::ChromatogramPeak::setRT</a></div><div class="ttdeci">void setRT(CoordinateType rt)</div><div class="ttdoc">Mutable access to RT.</div><div class="ttdef"><b>Definition:</b> ChromatogramPeak.h:122</div></div>
<div class="ttc" id="aa04211_html"><div class="ttname"><a href="a04211.html">FeatureMap.h</a></div></div>
<div class="ttc" id="aa03281_html"><div class="ttname"><a href="a03281.html">EmpiricalFormula.h</a></div></div>
<div class="ttc" id="aa08069_html_a8c9a8ccd7f43d0592f5b996d232f1eb9"><div class="ttname"><a href="a08069.html#a8c9a8ccd7f43d0592f5b996d232f1eb9">OpenMS::ResidueModification::getDiffMonoMass</a></div><div class="ttdeci">double getDiffMonoMass() const</div><div class="ttdoc">returns the diff monoisotopic mass if set</div></div>
<div class="ttc" id="aa10657_html_aba1fff7a176ff80d53f932769e606980"><div class="ttname"><a href="a10657.html#aba1fff7a176ff80d53f932769e606980">OpenMS::ProteinHit::setAccession</a></div><div class="ttdeci">void setAccession(const String &amp;accession)</div><div class="ttdoc">sets the accession of the protein</div></div>
<div class="ttc" id="aa10389_html_afa370200c7cf12d530364dec51830515"><div class="ttname"><a href="a10389.html#afa370200c7cf12d530364dec51830515">OpenMS::ChromatogramSettings::setNativeID</a></div><div class="ttdeci">void setNativeID(const String &amp;native_id)</div><div class="ttdoc">sets the native identifier for the spectrum, used by the acquisition software.</div></div>
<div class="ttc" id="aa10677_html_a15a14866884b8b4dea8950b2151a0963"><div class="ttname"><a href="a10677.html#a15a14866884b8b4dea8950b2151a0963">OpenMS::ProteinIdentification::setSearchEngine</a></div><div class="ttdeci">void setSearchEngine(const String &amp;search_engine)</div><div class="ttdoc">Sets the search engine type.</div></div>
<div class="ttc" id="aa07865_html_ab197aa53161e28b8f340ae72c9458f8a"><div class="ttname"><a href="a07865.html#ab197aa53161e28b8f340ae72c9458f8a">OpenMS::AASequence::getSuffix</a></div><div class="ttdeci">AASequence getSuffix(Size index) const</div><div class="ttdoc">returns a peptide sequence of the last index residues</div></div>
<div class="ttc" id="aa10421_html"><div class="ttname"><a href="a10421.html">OpenMS::DataArrays::StringDataArray</a></div><div class="ttdoc">String data array class.</div><div class="ttdef"><b>Definition:</b> DataArrays.h:59</div></div>
<div class="ttc" id="aa07865_html_a26b2839c6b49fe42966074e2375553d7"><div class="ttname"><a href="a07865.html#a26b2839c6b49fe42966074e2375553d7">OpenMS::AASequence::toString</a></div><div class="ttdeci">String toString() const</div><div class="ttdoc">returns the peptide as string with modifications embedded in brackets</div></div>
<div class="ttc" id="aa10081_html_ab45dae688fc5d8983727abffa4389003"><div class="ttname"><a href="a10081.html#ab45dae688fc5d8983727abffa4389003">OpenMS::MSExperiment::end</a></div><div class="ttdeci">Iterator end()</div><div class="ttdef"><b>Definition:</b> MSExperiment.h:167</div></div>
<div class="ttc" id="aa10149_html_a3cc15077963df5fb4e2b0e68fdac3fda"><div class="ttname"><a href="a10149.html#a3cc15077963df5fb4e2b0e68fdac3fda">OpenMS::RangeManager::getMin</a></div><div class="ttdeci">const PositionType &amp; getMin() const</div><div class="ttdoc">Returns the minimum position.</div><div class="ttdef"><b>Definition:</b> RangeManager.h:111</div></div>
<div class="ttc" id="aa10081_html_a1c97663e0b6a5cc44a3b189b89210bac"><div class="ttname"><a href="a10081.html#a1c97663e0b6a5cc44a3b189b89210bac">OpenMS::MSExperiment::addSpectrum</a></div><div class="ttdeci">void addSpectrum(const MSSpectrum &amp;spectrum)</div><div class="ttdoc">adds a spectrum to the list</div></div>
<div class="ttc" id="aa03488_html"><div class="ttname"><a href="a03488.html">Exception.h</a></div></div>
<div class="ttc" id="aa04505_html"><div class="ttname"><a href="a04505.html">Precursor.h</a></div></div>
<div class="ttc" id="aa08069_html"><div class="ttname"><a href="a08069.html">OpenMS::ResidueModification</a></div><div class="ttdoc">Representation of a modification.</div><div class="ttdef"><b>Definition:</b> ResidueModification.h:76</div></div>
<div class="ttc" id="aa07857_html"><div class="ttname"><a href="a07857.html">OpenMS::TOPPBase</a></div><div class="ttdoc">Base class for TOPP applications.</div><div class="ttdef"><b>Definition:</b> TOPPBase.h:144</div></div>
<div class="ttc" id="aa09533_html_ad2ed8d7ca3ca2f4fb1efa287b4657f38"><div class="ttname"><a href="a09533.html#ad2ed8d7ca3ca2f4fb1efa287b4657f38">OpenMS::MzIdentMLFile::load</a></div><div class="ttdeci">void load(const String &amp;filename, std::vector&lt; ProteinIdentification &gt; &amp;poid, std::vector&lt; PeptideIdentification &gt; &amp;peid)</div><div class="ttdoc">Loads the identifications from a MzIdentML file.</div></div>
<div class="ttc" id="aa08769_html"><div class="ttname"><a href="a08769.html">OpenMS::String</a></div><div class="ttdoc">A more convenient string class.</div><div class="ttdef"><b>Definition:</b> String.h:58</div></div>
<div class="ttc" id="aa08057_html"><div class="ttname"><a href="a08057.html">OpenMS::ProteaseDigestion</a></div><div class="ttdoc">Class for the enzymatic digestion of proteins.</div><div class="ttdef"><b>Definition:</b> ProteaseDigestion.h:60</div></div>
<div class="ttc" id="aa04361_html"><div class="ttname"><a href="a04361.html">ChromatogramSettings.h</a></div></div>
<div class="ttc" id="aa10645_html_af4bee28529a0258718cd7269369886f9"><div class="ttname"><a href="a10645.html#af4bee28529a0258718cd7269369886f9">OpenMS::PeptideIdentification::setRT</a></div><div class="ttdeci">void setRT(double rt)</div><div class="ttdoc">sets the RT of the MS2 spectrum where the identification occurred</div></div>
<div class="ttc" id="aa03290_html"><div class="ttname"><a href="a03290.html">CoarseIsotopePatternGenerator.h</a></div></div>
<div class="ttc" id="aa10097_html_af4bee28529a0258718cd7269369886f9"><div class="ttname"><a href="a10097.html#af4bee28529a0258718cd7269369886f9">OpenMS::MSSpectrum::setRT</a></div><div class="ttdeci">void setRT(double rt)</div><div class="ttdoc">Sets the absolute retention time (in seconds)</div></div>
<div class="ttc" id="aa03368_html"><div class="ttname"><a href="a03368.html">ResidueModification.h</a></div></div>
<div class="ttc" id="aa10645_html_a1529ca20d43500a280e162a759dd993f"><div class="ttname"><a href="a10645.html#a1529ca20d43500a280e162a759dd993f">OpenMS::PeptideIdentification::getRT</a></div><div class="ttdeci">double getRT() const</div><div class="ttdoc">returns the RT of the MS2 spectrum where the identification occurred</div></div>
<div class="ttc" id="aa08745_html_a432099944e11263aa20aaf8ea1389790"><div class="ttname"><a href="a08745.html#a432099944e11263aa20aaf8ea1389790">OpenMS::Param::setValue</a></div><div class="ttdeci">void setValue(const String &amp;key, const DataValue &amp;value, const String &amp;description=&quot;&quot;, const StringList &amp;tags=StringList())</div><div class="ttdoc">Sets a value.</div></div>
<div class="ttc" id="aa04226_html"><div class="ttname"><a href="a04226.html">MSExperiment.h</a></div></div>
<div class="ttc" id="aa10081_html_aee0a1a68d0fab63eefa6e14010b4a7f0"><div class="ttname"><a href="a10081.html#aee0a1a68d0fab63eefa6e14010b4a7f0">OpenMS::MSExperiment::ConstIterator</a></div><div class="ttdeci">std::vector&lt; SpectrumType &gt;::const_iterator ConstIterator</div><div class="ttdoc">Non-mutable iterator.</div><div class="ttdef"><b>Definition:</b> MSExperiment.h:113</div></div>
<div class="ttc" id="aa08057_html_ad92ecbad776494178a2a3de0ab7eed19"><div class="ttname"><a href="a08057.html#ad92ecbad776494178a2a3de0ab7eed19">OpenMS::ProteaseDigestion::setEnzyme</a></div><div class="ttdeci">void setEnzyme(const String &amp;name)</div><div class="ttdoc">Sets the enzyme for the digestion (by name)</div></div>
<div class="ttc" id="aa08589_html_a1decb59b26f2579931e3b4850df53d98"><div class="ttname"><a href="a08589.html#a1decb59b26f2579931e3b4850df53d98">OpenMS::DefaultParamHandler::getParameters</a></div><div class="ttdeci">const Param &amp; getParameters() const</div><div class="ttdoc">Non-mutable access to the parameters.</div></div>
<div class="ttc" id="aa10645_html_a24d0fe4bb2a1dd9a237cc580cb48d318"><div class="ttname"><a href="a10645.html#a24d0fe4bb2a1dd9a237cc580cb48d318">OpenMS::PeptideIdentification::getHits</a></div><div class="ttdeci">const std::vector&lt; PeptideHit &gt; &amp; getHits() const</div><div class="ttdoc">returns the peptide hits as const</div></div>
<div class="ttc" id="aa10097_html_ade02ee1e066e024a94db1c1753f11991"><div class="ttname"><a href="a10097.html#ade02ee1e066e024a94db1c1753f11991">OpenMS::MSSpectrum::getStringDataArrays</a></div><div class="ttdeci">const StringDataArrays &amp; getStringDataArrays() const</div><div class="ttdoc">Returns a const reference to the string meta data arrays.</div></div>
<div class="ttc" id="aa09537_html_af242617eaf1d0f50a8ab0648823a4f23"><div class="ttname"><a href="a09537.html#af242617eaf1d0f50a8ab0648823a4f23">OpenMS::MzMLFile::load</a></div><div class="ttdeci">void load(const String &amp;filename, PeakMap &amp;map)</div><div class="ttdoc">Loads a map from a MzML file. Spectra and chromatograms are sorted by default (this can be disabled u...</div></div>
<div class="ttc" id="aa04514_html"><div class="ttname"><a href="a04514.html">ProteinIdentification.h</a></div></div>
<div class="ttc" id="aa07865_html"><div class="ttname"><a href="a07865.html">OpenMS::AASequence</a></div><div class="ttdoc">Representation of a peptide/protein sequence.</div><div class="ttdef"><b>Definition:</b> AASequence.h:113</div></div>
<div class="ttc" id="aa10645_html_a915462ec4a7a731fdd71f28e8a3463b1"><div class="ttname"><a href="a10645.html#a915462ec4a7a731fdd71f28e8a3463b1">OpenMS::PeptideIdentification::setScoreType</a></div><div class="ttdeci">void setScoreType(const String &amp;type)</div><div class="ttdoc">sets the peptide score type</div></div>
<div class="ttc" id="aa08065_html"><div class="ttname"><a href="a08065.html">OpenMS::ResidueDB</a></div><div class="ttdoc">residue data base which holds residues</div><div class="ttdef"><b>Definition:</b> ResidueDB.h:60</div></div>
<div class="ttc" id="aa08069_html_aca9fce25db8a5971b38cbce06abf44a0"><div class="ttname"><a href="a08069.html#aca9fce25db8a5971b38cbce06abf44a0">OpenMS::ResidueModification::getMonoMass</a></div><div class="ttdeci">double getMonoMass() const</div><div class="ttdoc">return the monoisotopic mass, if set</div></div>
<div class="ttc" id="aa04184_html"><div class="ttname"><a href="a04184.html">ChromatogramPeak.h</a></div></div>
<div class="ttc" id="aa10037_html"><div class="ttname"><a href="a10037.html">OpenMS::Feature</a></div><div class="ttdoc">An LC-MS feature.</div><div class="ttdef"><b>Definition:</b> Feature.h:70</div></div>
<div class="ttc" id="aa04040_html"><div class="ttname"><a href="a04040.html">MzMLFile.h</a></div></div>
<div class="ttc" id="aa08061_html_a78b142edbd067ad26f26373afa189db0"><div class="ttname"><a href="a08061.html#a78b142edbd067ad26f26373afa189db0">OpenMS::Residue::getThreeLetterCode</a></div><div class="ttdeci">const String &amp; getThreeLetterCode() const</div><div class="ttdoc">returns the name of the residue as three letter code (String of size 3)</div></div>
<div class="ttc" id="aa10057_html_ae8d7d7b28cd7807d573d0f75da4201f0"><div class="ttname"><a href="a10057.html#ae8d7d7b28cd7807d573d0f75da4201f0">OpenMS::FeatureMap::updateRanges</a></div><div class="ttdeci">void updateRanges() override</div><div class="ttdoc">Updates minimum and maximum position/intensity.</div></div>
<div class="ttc" id="aa08589_html_abc8067511cfc949ae942ea2b08588f81"><div class="ttname"><a href="a08589.html#abc8067511cfc949ae942ea2b08588f81">OpenMS::DefaultParamHandler::setParameters</a></div><div class="ttdeci">void setParameters(const Param &amp;param)</div><div class="ttdoc">Sets the parameters.</div></div>
<div class="ttc" id="aa10645_html_a00f1456779f764c2738c35be2c04f6e8"><div class="ttname"><a href="a10645.html#a00f1456779f764c2738c35be2c04f6e8">OpenMS::PeptideIdentification::setIdentifier</a></div><div class="ttdeci">void setIdentifier(const String &amp;id)</div><div class="ttdoc">sets the identifier</div></div>
<div class="ttc" id="aa08069_html_abae0525ed84c3fc9a6a14abe8f2f8ec9"><div class="ttname"><a href="a08069.html#abae0525ed84c3fc9a6a14abe8f2f8ec9">OpenMS::ResidueModification::getOrigin</a></div><div class="ttdeci">char getOrigin() const</div><div class="ttdoc">Returns the origin (i.e. modified amino acid)</div></div>
<div class="ttc" id="aa10617_html_aca40a4630f5f8a0568899b74a62dcd4b"><div class="ttname"><a href="a10617.html#aca40a4630f5f8a0568899b74a62dcd4b">OpenMS::PeptideHit::setCharge</a></div><div class="ttdeci">void setCharge(Int charge)</div><div class="ttdoc">sets the charge of the peptide</div></div>
<div class="ttc" id="aa07901_html_a2af9395d26b51383a18fa548f9789eb9"><div class="ttname"><a href="a07901.html#a2af9395d26b51383a18fa548f9789eb9">OpenMS::ElementDB::getElement</a></div><div class="ttdeci">const Element * getElement(const String &amp;name) const</div></div>
<div class="ttc" id="aa07865_html_a448ebefe744de451f7cce4b02160fa87"><div class="ttname"><a href="a07865.html#a448ebefe744de451f7cce4b02160fa87">OpenMS::AASequence::getAverageWeight</a></div><div class="ttdeci">double getAverageWeight(Residue::ResidueType type=Residue::Full, Int charge=0) const</div><div class="ttdoc">returns the average weight of the peptide</div></div>
<div class="ttc" id="aa07865_html_a6cd2e12e525408ce2615d7877dfaa08a"><div class="ttname"><a href="a07865.html#a6cd2e12e525408ce2615d7877dfaa08a">OpenMS::AASequence::getMonoWeight</a></div><div class="ttdeci">double getMonoWeight(Residue::ResidueType type=Residue::Full, Int charge=0) const</div></div>
<div class="ttc" id="aa09265_html_a851b569c3e65a4eeabdd10425f2e6402"><div class="ttname"><a href="a09265.html#a851b569c3e65a4eeabdd10425f2e6402">OpenMS::FileHandler::getType</a></div><div class="ttdeci">static FileTypes::Type getType(const String &amp;filename)</div><div class="ttdoc">Tries to determine the file type (by name or content)</div></div>
<div class="ttc" id="aa05786_html_ga4ed94a7147b86680101341d95e24253c"><div class="ttname"><a href="a05786.html#ga4ed94a7147b86680101341d95e24253c">OpenMS::Math::sum</a></div><div class="ttdeci">static double sum(IteratorType begin, IteratorType end)</div><div class="ttdoc">Calculates the sum of a range of values.</div><div class="ttdef"><b>Definition:</b> StatisticFunctions.h:120</div></div>
<div class="ttc" id="aa10677_html_a00f1456779f764c2738c35be2c04f6e8"><div class="ttname"><a href="a10677.html#a00f1456779f764c2738c35be2c04f6e8">OpenMS::ProteinIdentification::setIdentifier</a></div><div class="ttdeci">void setIdentifier(const String &amp;id)</div><div class="ttdoc">Sets the identifier.</div></div>
<div class="ttc" id="aa03362_html"><div class="ttname"><a href="a03362.html">Residue.h</a></div></div>
<div class="ttc" id="aa09253_html_a92ac2327eb6ca69916f33ea9649fb75d"><div class="ttname"><a href="a09253.html#a92ac2327eb6ca69916f33ea9649fb75d">OpenMS::FASTAFile::store</a></div><div class="ttdeci">static void store(const String &amp;filename, const std::vector&lt; FASTAEntry &gt; &amp;data)</div><div class="ttdoc">stores the data given by 'data' at the file 'filename'</div></div>
<div class="ttc" id="aa07901_html"><div class="ttname"><a href="a07901.html">OpenMS::ElementDB</a></div><div class="ttdoc">Stores elements.</div><div class="ttdef"><b>Definition:</b> ElementDB.h:69</div></div>
<div class="ttc" id="aa08581_html_a27fa24aa34288a3934c3d5a936463973"><div class="ttname"><a href="a08581.html#a27fa24aa34288a3934c3d5a936463973">OpenMS::DateTime::now</a></div><div class="ttdeci">static DateTime now()</div><div class="ttdoc">Returns the current date and time.</div></div>
<div class="ttc" id="aa10109_html_ab9c4ce5d172a75d6de2c6f5f1847c9ab"><div class="ttname"><a href="a10109.html#ab9c4ce5d172a75d6de2c6f5f1847c9ab">OpenMS::Peak1D::setMZ</a></div><div class="ttdeci">void setMZ(CoordinateType mz)</div><div class="ttdoc">Mutable access to m/z.</div><div class="ttdef"><b>Definition:</b> Peak1D.h:121</div></div>
<div class="ttc" id="aa10645_html_ab4d15b02dbc53727d0b6dae927ad6688"><div class="ttname"><a href="a10645.html#ab4d15b02dbc53727d0b6dae927ad6688">OpenMS::PeptideIdentification::setHits</a></div><div class="ttdeci">void setHits(const std::vector&lt; PeptideHit &gt; &amp;hits)</div><div class="ttdoc">Sets the peptide hits.</div></div>
<div class="ttc" id="aa10617_html"><div class="ttname"><a href="a10617.html">OpenMS::PeptideHit</a></div><div class="ttdoc">Representation of a peptide hit.</div><div class="ttdef"><b>Definition:</b> PeptideHit.h:54</div></div>
<div class="ttc" id="aa03893_html"><div class="ttname"><a href="a03893.html">FileHandler.h</a></div></div>
<div class="ttc" id="aa03650_html"><div class="ttname"><a href="a03650.html">String.h</a></div></div>
<div class="ttc" id="aa08061_html_a61ccbfa578837af65174c4ee27101b5d"><div class="ttname"><a href="a08061.html#a61ccbfa578837af65174c4ee27101b5d">OpenMS::Residue::getOneLetterCode</a></div><div class="ttdeci">const String &amp; getOneLetterCode() const</div><div class="ttdoc">returns the name as one letter code (String of size 1)</div></div>
<div class="ttc" id="aa09265_html"><div class="ttname"><a href="a09265.html">OpenMS::FileHandler</a></div><div class="ttdoc">Facilitates file handling by file type recognition.</div><div class="ttdef"><b>Definition:</b> FileHandler.h:62</div></div>
<div class="ttc" id="aa10097_html_a808327302a2ba3fb630ea132d32db942"><div class="ttname"><a href="a10097.html#a808327302a2ba3fb630ea132d32db942">OpenMS::MSSpectrum::MZBegin</a></div><div class="ttdeci">Iterator MZBegin(CoordinateType mz)</div><div class="ttdoc">Binary search for peak range begin.</div></div>
<div class="ttc" id="aa10645_html"><div class="ttname"><a href="a10645.html">OpenMS::PeptideIdentification</a></div><div class="ttdoc">Represents the peptide hits for a spectrum.</div><div class="ttdef"><b>Definition:</b> PeptideIdentification.h:62</div></div>
<div class="ttc" id="aa08061_html_af294687b8b378a19671bcd0cc5e3d01f"><div class="ttname"><a href="a08061.html#af294687b8b378a19671bcd0cc5e3d01f">OpenMS::Residue::getMonoWeight</a></div><div class="ttdeci">double getMonoWeight(ResidueType res_type=Full) const</div><div class="ttdoc">returns monoisotopic weight of the residue</div></div>
<div class="ttc" id="aa09253_html"><div class="ttname"><a href="a09253.html">OpenMS::FASTAFile</a></div><div class="ttdoc">This class serves for reading in and writing FASTA files.</div><div class="ttdef"><b>Definition:</b> FASTAFile.h:64</div></div>
<div class="ttc" id="aa10617_html_a209d1e10f38a766c3f73c9666e89e14d"><div class="ttname"><a href="a10617.html#a209d1e10f38a766c3f73c9666e89e14d">OpenMS::PeptideHit::setSequence</a></div><div class="ttdeci">void setSequence(const AASequence &amp;sequence)</div><div class="ttdoc">sets the peptide sequence</div></div>
<div class="ttc" id="aa10645_html_a98447dc4641b69db36fcc5087cfe73e6"><div class="ttname"><a href="a10645.html#a98447dc4641b69db36fcc5087cfe73e6">OpenMS::PeptideIdentification::setHigherScoreBetter</a></div><div class="ttdeci">void setHigherScoreBetter(bool value)</div><div class="ttdoc">sets the peptide score orientation</div></div>
<div class="ttc" id="aa10677_html_ae0728c54c04bcc99a4c6ebfa961ed2c5"><div class="ttname"><a href="a10677.html#ae0728c54c04bcc99a4c6ebfa961ed2c5">OpenMS::ProteinIdentification::setDateTime</a></div><div class="ttdeci">void setDateTime(const DateTime &amp;date)</div><div class="ttdoc">Sets the date of the protein identification run.</div></div>
<div class="ttc" id="aa03257_html"><div class="ttname"><a href="a03257.html">AASequence.h</a></div></div>
<div class="ttc" id="aa08061_html_a7651af21f9cf8ed6445415903fc6cb48a7894cbe7cb5a3c5508b8becc24ca040e"><div class="ttname"><a href="a08061.html#a7651af21f9cf8ed6445415903fc6cb48a7894cbe7cb5a3c5508b8becc24ca040e">OpenMS::Residue::BIon</a></div><div class="ttdoc">MS:1001224 N-terminus up to the peptide bond.</div><div class="ttdef"><b>Definition:</b> Residue.h:157</div></div>
<div class="ttc" id="aa10109_html"><div class="ttname"><a href="a10109.html">OpenMS::Peak1D</a></div><div class="ttdoc">A 1-dimensional raw data point or peak.</div><div class="ttdef"><b>Definition:</b> Peak1D.h:54</div></div>
<div class="ttc" id="aa03398_html"><div class="ttname"><a href="a03398.html">TheoreticalSpectrumGenerator.h</a></div></div>
<div class="ttc" id="aa08057_html_ad0d24e15a1dd611eceb19eb9c421c579"><div class="ttname"><a href="a08057.html#ad0d24e15a1dd611eceb19eb9c421c579">OpenMS::ProteaseDigestion::peptideCount</a></div><div class="ttdeci">Size peptideCount(const AASequence &amp;protein)</div><div class="ttdoc">Returns the number of peptides a digestion of protein would yield under the current enzyme and missed...</div></div>
<div class="ttc" id="aa07901_html_ae141930c87ab8851ec484bc1fd0ccb90"><div class="ttname"><a href="a07901.html#ae141930c87ab8851ec484bc1fd0ccb90">OpenMS::ElementDB::getInstance</a></div><div class="ttdeci">static const ElementDB * getInstance()</div><div class="ttdef"><b>Definition:</b> ElementDB.h:79</div></div>
<div class="ttc" id="aa09945_html_ab875c2479c6cd2a0189e4684bfa4f97d"><div class="ttname"><a href="a09945.html#ab875c2479c6cd2a0189e4684bfa4f97d">OpenMS::ChromatogramPeak::setIntensity</a></div><div class="ttdeci">void setIntensity(IntensityType intensity)</div><div class="ttdoc">Mutable access to the data point intensity (height)</div><div class="ttdef"><b>Definition:</b> ChromatogramPeak.h:113</div></div>
<div class="ttc" id="aa10677_html_a915462ec4a7a731fdd71f28e8a3463b1"><div class="ttname"><a href="a10677.html#a915462ec4a7a731fdd71f28e8a3463b1">OpenMS::ProteinIdentification::setScoreType</a></div><div class="ttdeci">void setScoreType(const String &amp;type)</div><div class="ttdoc">Sets the protein score type.</div></div>
<div class="ttc" id="aa10389_html_a397cbc084d939f590f1bfaa5eb7be4e2"><div class="ttname"><a href="a10389.html#a397cbc084d939f590f1bfaa5eb7be4e2">OpenMS::ChromatogramSettings::getProduct</a></div><div class="ttdeci">const Product &amp; getProduct() const</div><div class="ttdoc">returns a const reference to the products</div></div>
<div class="ttc" id="aa10653_html_a566e1d015a01fa37751e0a77c64d70a8"><div class="ttname"><a href="a10653.html#a566e1d015a01fa37751e0a77c64d70a8">OpenMS::Product::setMZ</a></div><div class="ttdeci">void setMZ(double mz)</div><div class="ttdoc">sets the target m/z</div></div>
<div class="ttc" id="aa08065_html_a416f4888cc1f2b8acb849cc5347b4342"><div class="ttname"><a href="a08065.html#a416f4888cc1f2b8acb849cc5347b4342">OpenMS::ResidueDB::getInstance</a></div><div class="ttdeci">static ResidueDB * getInstance()</div><div class="ttdoc">this member function serves as a replacement of the constructor</div></div>
<div class="ttc" id="aa05822_html"><div class="ttname"><a href="a05822.html">OpenMS</a></div><div class="ttdoc">Main OpenMS namespace.</div><div class="ttdef"><b>Definition:</b> FeatureDeconvolution.h:46</div></div>
<div class="ttc" id="aa10585_html_a135a7e818125b31198a8a65d9ac7a3d1"><div class="ttname"><a href="a10585.html#a135a7e818125b31198a8a65d9ac7a3d1">OpenMS::MetaInfoInterface::setMetaValue</a></div><div class="ttdeci">void setMetaValue(const String &amp;name, const DataValue &amp;value)</div><div class="ttdoc">Sets the DataValue corresponding to a name.</div></div>
<div class="ttc" id="aa10081_html_a2387033802383edbdc95f9bbb12a707e"><div class="ttname"><a href="a10081.html#a2387033802383edbdc95f9bbb12a707e">OpenMS::MSExperiment::begin</a></div><div class="ttdeci">Iterator begin()</div><div class="ttdef"><b>Definition:</b> MSExperiment.h:157</div></div>
<div class="ttc" id="aa08061_html_a47fd4b1ed0d95b8649da8dc5748fbc30"><div class="ttname"><a href="a08061.html#a47fd4b1ed0d95b8649da8dc5748fbc30">OpenMS::Residue::getFormula</a></div><div class="ttdeci">EmpiricalFormula getFormula(ResidueType res_type=Full) const</div><div class="ttdoc">returns the empirical formula of the residue</div></div>
<div class="ttc" id="aa07865_html_a9c074ddb7e1c4f2539770fdaf31fe853"><div class="ttname"><a href="a07865.html#a9c074ddb7e1c4f2539770fdaf31fe853">OpenMS::AASequence::toUnmodifiedString</a></div><div class="ttdeci">String toUnmodifiedString() const</div><div class="ttdoc">returns the peptide as string without any modifications or (e.g., &quot;PEPTIDER&quot;)</div></div>
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:25 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
