<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: OpenPepXL</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">OpenPepXL </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Search for peptide pairs linked with a labeled cross-linker</p>
<p>This tool performs a search for cross-links in the given mass spectra. It uses linked MS1 features to pair up MS2 spectra and uses these pairs to find the fragment peaks that contain the linker and those that do not.</p>
<p>It executes the following steps in order: </p><ul>
<li>
Processing of spectra: deisotoping and filtering </li>
<li>
Digesting and preprocessing the protein database, building a peptide pair index dependent on the precursor masses of the MS2 spectra </li>
<li>
Generating theoretical spectra of cross-linked peptides and aligning the experimental spectra against those </li>
<li>
Scoring of cross-link spectrum matches </li>
<li>
Using PeptideIndexer to map the peptides to all possible source proteins </li>
</ul>
<p>See below for available parameters and more functionality.</p>
<h3>Input: MS2 spectra, linked features from FeatureFinderMultiplex and fasta database of proteins expected to be cross-linked in the sample</h3>
<p>The spectra should be provided as one PeakMap. If you have multiple files, e.g. for multiple fractions, you should run this tool on each file separately. The database should be provided as a vector of FASTAEntrys containing the target and decoy proteins. A ConsensusMap, that links the MS1 feature pairs from heavy and light cross-linkers is also required. This can be generated by the tool FeatureFinderMultiplex. Setting up FeatureFinderMultiplex: In the FeatureFinderMultiplex parameters you have to change the mass of one of the labels to the difference between the light and heavy (e.g. change the mass of Arg6 to 12.075321 for labeled DSS) in the advanced options. The parameter -labels should have one empty label ( [] ) and the label you adapted (e.g. [][Arg6]). For the other settings refer to the documentation of FeatureFinderMultiplex.</p>
<h3>Parameters</h3>
<p>The parameters for fixed and variable modifications refer to additional modifications beside the cross-linker. The linker used in the experiment has to be described using the cross-linker specific parameters. Only one mass is allowed for a cross-linker, that links two peptides (cross_linker:mass_light), while multiple masses are possible for mono-links of the same cross-linking reagent. Mono-links are cross-linkers, that are linked to one peptide by one of their two reactive groups. The masses refer to the light version of the linker. The parameter cross_linker:mass_iso_shift defines the difference between the light and heavy versions of the cross-linker and the mono-links. The parameters cross_linker:residue1 and cross_linker:residue2 are used to enumerate the amino acids, that each end of the linker can react with. This way any heterobifunctional cross-linker can be defined. To define a homobifunctional cross-linker, these two parameters should have the same value. The parameter cross_linker:name is used to solve ambiguities arising from different cross-linkers having the same mass after the linking reaction (see section on output for clarification).</p>
<h3>Output: XL-MS Identifications with scores and linked positions in the proteins</h3>
<p>The input parameters protein_ids and peptide_ids are filled with XL-MS search parameters and IDs</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="2"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> OpenPepXL <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1">-  </td><td valign="middle" align="center" rowspan="1">-   </td></tr>
</table>
</center><p>Search for peptide pairs linked with a labeled cross-linker</p>
<p>This tool performs a search for cross-links in the given mass spectra. It uses linked MS1 features to pair up MS2 spectra and uses these pairs to find the fragment peaks that contain the linker and those that do not.</p>
<p>It executes the following steps in order: </p><ul>
<li>
Reading of MS2 spectra from the given mzML file </li>
<li>
Processing of spectra: deisotoping and filtering </li>
<li>
Digesting and preprocessing the protein database, building a peptide pair index dependent on the precursor masses of the MS2 spectra </li>
<li>
Generating theoretical spectra of cross-linked peptides and aligning the experimental spectra against those </li>
<li>
Scoring of cross-link spectrum matches </li>
<li>
Using PeptideIndexer to map the peptides to all possible source proteins </li>
<li>
Writing out the results in idXML, mzid according to mzIdentML 1.2 specifications and/or in the xQuest output format </li>
</ul>
<p>See below or have a look at the INI file (via "OpenPepXL -write_ini myini.ini") for available parameters and more functionality.</p>
<h3>Input: MS2 spectra, linked features from FeatureFinderMultiplex and fasta database of proteins expected to be cross-linked in the sample</h3>
<p>The spectra should be provided as one mzML file. If you have multiple files, e.g. for multiple fractions, you should run this tool on each file separately. The database can either be provided as one merged file containing targets and decoys or as two separate files. A consensusXML file, that links the MS1 feature pairs from heavy and light cross-linkers is also required. This file can be generated by the tool FeatureFinderMultiplex. Setting up FeatureFinderMultiplex: In the FeatureFinderMultiplex parameters you have to change the mass of one of the labels to the difference between the light and heavy (e.g. change the mass of Arg6 to 12.075321 for labeled DSS) in the advanced options. The parameter -labels should have one empty label ( [] ) and the label you adapted (e.g. [][Arg6]). For the other settings refer to the documentation of FeatureFinderMultiplex.</p>
<h3>Parameters</h3>
<p>The parameters for fixed and variable modifications refer to additional modifications beside the cross-linker. The linker used in the experiment has to be described using the cross-linker specific parameters. Only one mass is allowed for a cross-linker, that links two peptides (-cross_linker:mass_light), while multiple masses are possible for mono-links of the same cross-linking reagent. Mono-links are cross-linkers, that are linked to one peptide by one of their two reactive groups. The masses refer to the light version of the linker. The parameter -cross_linker:mass_iso_shift defines the difference between the light and heavy versions of the cross-linker and the mono-links. The parameters -cross_linker:residue1 and -cross_linker:residue2 are used to enumerate the amino acids, that each end of the linker can react with. This way any heterobifunctional cross-linker can be defined. To define a homobifunctional cross-linker, these two parameters should have the same value. The parameter -cross_linker:name is used to solve ambiguities arising from different cross-linkers having the same mass after the linking reaction (see section on output for clarification).</p>
<h3>Output: XL-MS Identifications with scores and linked positions in the proteins</h3>
<p>There are three file formats for output of data possible. idXML is the internal format of <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a>, and is recommended for post-processing using other TOPP tools like XFDR or TOPPView. The second format xquest.xml is the output format of xQuest, which is a popular XL-MS ID tool. This format is compatible with a number of post-processing and visulization tools, like xProphet for FDR estimation (Leitner, A. et al., 2014, Nature protocols) and through the xQuest Results Viewer also the XlinkAnalyzer for visualization and analysis using protein structures (Kosinski, J. et al., 2015, Journal of structural biology). The third format is mzIdentML according to the specifications for XL-MS ID data in version 1.2 (Vizca√≠no, J. A. et al., 2017, Mol Cell Proteomics). This is a standardized long term storage format and compatible with complete submissions to the PRIDE database, that is part of the ProteomeXchange consortium. The specification includes the XLMOD database of cross-linking reagents, and if the provided cross-link mass matches one from the database, its accession and name are used. If the name is provided with the -cross_linker:name parameter, it is used to solve ambiguities arising from different cross-linkers having the same mass after the linking reaction (e.g. DSS and BS3). It is also used as the name of the linker, if no matching masses are found in the database.</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="2"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> OpenPepXL <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1">-  </td><td valign="middle" align="center" rowspan="1">-   </td></tr>
</table>
</center><p><b>The command line parameters of this tool are:</b> </p><pre class="fragment"></pre><p> <b>INI file documentation of this tool:</b>  </div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:27 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
