<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: Quality Control</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16241.html">TOPP tutorial</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">Quality Control </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h1><a class="anchor" id="TOPP_example_qualitycontrol_section"></a>
Quality Control</h1>
<p><b>Goal:</b> Check the quality of the data (currently, on label-free workflow data is supported).</p>
<p>The QualityControl TOPP tool computes and collects data which allow to compute QC metrics to check the quality of LC-MS data. Depending on the given input data this tool collects data for metrics (see section 'Metrics' below). New metavalues will be added to existing data and the information will be written out in mzTab format. This mzTab file can then be processed using your custom scripts or via the R package (see <a href="https://github.com/cbielow/PTXQC/" target="blank_">PTXQC</a>).</p>
<h1><a class="anchor" id="TOPP_example_qualitycontrol_section2"></a>
Workflow</h1>
<div class="image">
<img src="TOPP_qualitycontrol.png" alt=""/>
</div>
 <p>An example workflow can be found in OpenMS/share/OpenMS/examples/TOPPAS/QualityControl.toppas.</p>
<h1><a class="anchor" id="TOPP_example_qualitycontrol_section3"></a>
Metrics</h1>
<p>This Table shows what each of the included metrics does, what they need to be executed and what they add to the data or what they return.</p>
<h2><a class="anchor" id="TOPP_example_qualitycontrol_section3_subsection"></a>
Input Data</h2>
<p>PostFDR FeatureXML: A FeatureXML after FDR filtering. <br  />
 Contaminants Fasta file: A Fasta file containing contaminant proteins. <br  />
 Raw mzML file: An unchanged mzML file. <br  />
 InternalCalibration MzML file: An MzML file after internal calibration. <br  />
 TrafoXML file: The RT alignment function as obtained from a MapAligner.</p>
<table border="1">
<tr>
<th><b><br  />
Contaminants</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> Contaminants Fasta file, PostFDR FeatureXML  </td><td><b> Output</b>   </td></tr>
<tr>
<td><b>Description:</b> <br  />
 The Contaminants metric takes the contaminants database and digests the protein sequences with the digestion enzyme that is given in the featureXML. Afterwards it checks whether each of all peptide sequences of the featureXML (including the unassigned PeptideIdentifications) is registered in the contaminants database.  </td><td><p class="starttd"><b> Changes in files:</b> <br  />
 Metavalue: <br  />
 </p><ul>
<li>
'is_contaminant' set to '1' or to '0' if the peptide is found in the contaminant database or not and sets a '0' if not.<br  />
  </li>
</ul>
<p><b> Other outputs:</b> <br  />
 Returns:<br  />
 </p><ul>
<li>
Contaminant ratio of all peptides </li>
<li>
Contaminant ratio of all assigned peptides </li>
<li>
Contaminant ratio of all unassigned peptides </li>
<li>
Intensity ratio of all contaminants in the assigned peptides </li>
<li>
Number of empty features, Number of all found features </li>
</ul>
<p class="endtd"></p>
</td></tr>
<tr>
<th><br  />
<b>FragmentMassError</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> PostFDR FeatureXML, raw mzML file  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 The FragmentMassError metric computes a list of fragment mass errors for each annotated MS2 spectrum in ppm and Da. Afterwards it calculates the mass delta between observed and theoretical peaks. <br  />
 </p>
<p class="endtd"></p>
</td><td><b> Changes in files:</b> <br  />
 Metavalue: <br  />
 <ul>
<li>
'fragment_mass_error_ppm' set to the fragment mass error in parts per million </li>
<li>
'fragment_mass_error_da' set to the fragment mass error in Dalton </li>
</ul>
<b> Other Output:</b> <br  />
 Returns: <br  />
 <ul>
<li>
Average and variance of fragment mass errors in ppm </li>
</ul>
</td></tr>
<tr>
<th><br  />
<b>MissedCleavages</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> PostFDR FeatureXML  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 This MissedCleavages metric counts the number of MissedCleavages per PeptideIdentification given a FeatureMap and returns an agglomeration statistic (observed counts). Additionally the first PeptideHit of each PeptideIdentification in the FeatureMap is augmented with metavalues. </p>
<p class="endtd"></p>
</td><td><b> Changes in files:</b> <br  />
 Metavalue:<br  />
 <ul>
<li>
'missed_cleavages' </li>
</ul>
<b> Other Output:</b> <br  />
 Returns: <br  />
 <ul>
<li>
Frequency map of missed cleavages as key/value pairs. </li>
</ul>
</td></tr>
<tr>
<th><br  />
<b>MS2IdentificationRate</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> PostFDR FeatureXML, raw mzML file  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 The MS2IdentificationRate metric calculates the Rate of the MS2 identification as follows: The number of all PeptideIdentifications are counted and that number is divided by the total number of MS2 spectra. </p>
<p class="endtd"></p>
</td><td><b> Changes in files:</b> <br  />
 This metric does not change anything in the data.<br  />
 <b> Other Output:</b> <br  />
 Returns: <br  />
 <ul>
<li>
Number of PeptideIdentifications </li>
<li>
Number of MS2 spectra </li>
<li>
Ratio of #pepID/#MS2 </li>
</ul>
</td></tr>
<tr>
<th><br  />
<b>MzCalibration</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> PostFDR FeatureXML<br  />
 <b>Optional input data:</b> InternalCalibration MzML file  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 This metric adds new metavalues to the first (best) hit of each PeptideIdentification. For this metric it is also possible to use this without an MzML File, but then only uncalibrated m/z error (ppm) will be reported. However for full functionality a PeakMap/MSExperiment with original m/z-values before m/z calibration generated by InternalCalibration has to be given. </p>
<p class="endtd"></p>
</td><td><b> Changes in files:</b> <br  />
 Metavalues: <br  />
 <ul>
<li>
'mz_raw'set to m/z value of original experiment </li>
<li>
'mz_ref' set to m/z value of calculated reference </li>
<li>
'uncalibrated_mz_error_ppm' set to uncalibrated m/z error in parts per million </li>
<li>
'calibrated_mz_error_ppm' set to calibrated m/z error in parts per million </li>
</ul>
<b> Other Output:</b> <br  />
 No additional output.   </td></tr>
<tr>
<th><br  />
<b>RTAlignment</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> PostFDR FeatureXML, trafoXML file  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 The RTAlignment metric checks what the retention time was before the alignment and how it is after the alignment. These two values are added to the metavalues in the PeptideIdentification. </p>
<p class="endtd"></p>
</td><td><b> Changes in files:</b> <br  />
 Metavalues: <br  />
 <ul>
<li>
'rt_align'set to retention time after alignment </li>
<li>
'rt_raw' set to retention time before alignment </li>
</ul>
<b> Other Output:</b> <br  />
 No additional output.   </td></tr>
<tr>
<th><br  />
<b>TIC</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> raw mzML file  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 This TIC metric calculates the total ion count of an MSExperiment if a bin size in RT seconds greater than 0 is given. All MS1 abundances within a bin are summed up. </p>
<p class="endtd"></p>
</td><td><b> Changes in files:</b> <br  />
 This metric does not change anything in the data.<br  />
 <b> Other Output:</b> <br  />
 Returns:<br  />
 <ul>
<li>
TIC chromatograms </li>
</ul>
</td></tr>
<tr>
<th><br  />
<b>TopNoverRT</b> </th><th></th></tr>
<tr>
<td><b>Required input data:</b> PostFDR FeatureXML, raw mzML file  </td><td><b>Output</b>   </td></tr>
<tr>
<td><p class="starttd"><b>Description:</b> <br  />
 The TopNoverRT metric calculates the ScanEventNumber (number of the MS2 scans after the MS1 scan) and adds them as the new metavalue 'ScanEventNumber' to the PeptideIdentifications. It finds all unidentified MS2-Spectra and adds corresponding 'empty' PeptideIdentifications without sequence as placeholders to the unassigned PeptideIdentification list. Furthermore it adds the metavalue 'identified' to the PeptideIdentification. </p>
<p class="endtd"></p>
</td><td><p class="starttd"><b> Changes in files:</b> <br  />
 Metavalues: <br  />
 </p><ul>
<li>
'ScanEventNumber' set to the calculated value </li>
<li>
'identified' set to '+' or '-' </li>
</ul>
<p>If provided: </p><ul>
<li>
'FWHM' set to RT peak width for all assigned PIs </li>
<li>
'ion_injection_time'set to injection time from MS2 spectrum </li>
<li>
'activation_method'set to activation method from MS2 spectrum </li>
<li>
'total_ion_count'set to summed intensity from MS2 spectrum </li>
<li>
'base_peak_intensity'set to highest intensity from MS2 spectrum </li>
</ul>
<p>Additionally: <br  />
 </p><ul>
<li>
Adds empty PeptideIdentifications </li>
</ul>
<p><b> Other Output:</b> <br  />
 No additional output.  </p>
<p class="endtd"></p>
</td></tr>
</table>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:25 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
