<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: UTILS documentation</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">UTILS documentation </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Besides TOPP, OpenMS offers a range of other tools. They are not included in TOPP as they are not part of typical analysis pipelines, but they still might be very helpful to you.</p>
<p>Just as the TOPP tools, they can be found in the <code>OpenMS/bin</code> folder.</p>
<p>The UTILS tools are divided into several subgroups:</p>
<p><b>Maintenance</b></p><ul>
<li><a class="el" href="a16381.html">INIUpdater</a> - Updates INI and TOPPAS files from previous versions of OpenMS as parameters and storage method might have change</li>
</ul>
<p><b>Signal Processing and Preprocessing</b></p><ul>
<li><a class="el" href="a16383.html">LowMemPeakPickerHiRes</a> - A tool for peak detection on streamed profile data.</li>
<li><a class="el" href="a16384.html">LowMemPeakPickerHiResRandomAccess</a> - A tool for peak detection on streamed profile data.</li>
<li><a class="el" href="a16403.html">PeakPickerIterative</a> - A tool for peak detection in profile data.</li>
</ul>
<p><b>File Handling</b></p><ul>
<li><a class="el" href="a16363.html">CVInspector</a> - A tool for visualization and validation of PSI mapping and CV files.</li>
<li><a class="el" href="a16374.html">FuzzyDiff</a> - Compares two files, tolerating numeric differences.</li>
<li><a class="el" href="a16379.html">IDSplitter</a> - Splits protein/peptide identifications off of annotated data files.</li>
<li><a class="el" href="a16395.html">MzMLSplitter</a> - Splits an mzML file into multiple parts.</li>
<li><a class="el" href="a16418.html">SemanticValidator</a> - SemanticValidator for analysisXML and mzML files.</li>
<li><a class="el" href="a16428.html">XMLValidator</a> - Validates XML files against an XSD schema.</li>
</ul>
<p><b>Algorithm Evaluation</b></p><ul>
<li><a class="el" href="a16373.html">FFEval</a> - Evaluation tool for feature detection algorithms.</li>
<li><a class="el" href="a16382.html">LabeledEval</a> - Evaluation tool for isotope-labeled quantitation experiments.</li>
<li><a class="el" href="a16417.html">RTEvaluation</a> - Application that evaluates true positives (TP), true negatives (TN), false positives (FP), and false negatives (FN) for an idXML file with predicted RTs.</li>
<li><a class="el" href="a16427.html">TransformationEvaluation</a> - Simple evaluation of transformations (e.g. RT transformations produced by a MapAligner tool).</li>
</ul>
<p><b>Protein/Peptide Identification</b></p><ul>
<li><a class="el" href="a16364.html">DatabaseFilter</a> - Filters a protein database in FASTA format according to one or multiple filtering criteria.</li>
<li><a class="el" href="a16365.html">DecoyDatabase</a> - Creates decoy peptide databases from normal ones.</li>
<li><a class="el" href="a16367.html">Digestor</a> - Digests a protein database in-silico.</li>
<li><a class="el" href="a16368.html">DigestorMotif</a> - Digests a protein database in-silico (optionally allowing only peptides with a specific motif) and produces statistical data for all peptides.</li>
<li><a class="el" href="a16369.html">Epifany</a> - Bayesian protein inference based on PSM probabilities.</li>
<li><a class="el" href="a16376.html">IDExtractor</a> - Extracts n peptides randomly or best n from idXML files.</li>
<li><a class="el" href="a16377.html">IDMassAccuracy</a> - Calculates a distribution of the mass error from given mass spectra and IDs.</li>
<li><a class="el" href="a16378.html">IDScoreSwitcher</a> - Switches between different scores of peptide or protein hits in identification data.</li>
<li><a class="el" href="a16391.html">MSFraggerAdapter</a> - Peptide Identification with MSFragger.</li>
<li><a class="el" href="a16396.html">NovorAdapter</a> - De novo sequencing from tandem mass spectrometry data.</li>
<li><a class="el" href="a16405.html">PSMFeatureExtractor</a> - Creates search engine specific features for PercolatorAdapter input.</li>
<li><a class="el" href="a16419.html">SequenceCoverageCalculator</a> - Prints information about idXML files.</li>
<li><a class="el" href="a16422.html">SpecLibCreator</a> - Creates an MSP-formatted spectral library.</li>
<li><a class="el" href="a16423.html">SpectraSTSearchAdapter</a> - An interface to the 'SEARCH' mode of the SpectraST program (external, beta).</li>
</ul>
<p><b>Cross-linking</b></p><ul>
<li>UTILS_OpenPepXL - Search for peptide pairs linked with a labeled cross-linker.</li>
<li>UTILS_OpenPepXLLF - Search for cross-linked peptide pairs in tandem MS spectra.</li>
<li><a class="el" href="a16415.html">RNPxlSearch</a> - Annotates RNA-to-peptide cross-links in MS/MS spectra.</li>
<li><a class="el" href="a16416.html">RNPxlXICFilter</a> - Removes MS2 spectra from treatment based on the fold change between control and treatment for RNA-to-peptide cross-linking experiments.</li>
<li>UTILS_XFDR - Calculates false discovery rate estimates on cross-link identifications.</li>
</ul>
<p><b>Quantitation</b></p><ul>
<li><a class="el" href="a16370.html">ERPairFinder</a> - Evaluates pair ratios on enhanced resolution (zoom) scans.</li>
<li><a class="el" href="a16371.html">FeatureFinderMetaboIdent</a> - Detects features in MS1 data corresponding to small molecule identifications.</li>
<li><a class="el" href="a16372.html">FeatureFinderSuperHirn</a> - Finds features using the SuperHirn algorithm (can handle centroided or profile data, see .ini file).</li>
<li>UTILS_ProteomicLFQ - A standard Proteomics LFQ pipeline.</li>
<li><a class="el" href="a16386.html">UTILS_MetaboliteAdductDecharger</a> - Decharges and merges different feature charge variants of the same small molecule.</li>
<li><a class="el" href="a16394.html">MultiplexResolver</a> - Resolves conflicts between identifications and quantifications in multiplex data.</li>
</ul>
<p><b>Metabolite Identification</b></p><ul>
<li><a class="el" href="a16359.html">AccurateMassSearch</a> - Finds potential HMDB IDs within the given mass error window.</li>
<li><a class="el" href="a16387.html">MetaboliteSpectralMatcher</a> - Identifies small molecules from tandem MS spectra.</li>
<li><a class="el" href="a16421.html">SiriusAdapter</a> - De novo metabolite identification.</li>
</ul>
<p><b>Targeted Experiments and OpenSWATH</b></p><ul>
<li><a class="el" href="a16362.html">CorrelateMassTraces</a> - Identifies precursor mass traces and tries to correlate them with fragment ion mass traces in SWATH maps.</li>
<li><a class="el" href="a16389.html">MRMPairFinder</a> - Evaluates labeled pair ratios on MRM features.</li>
<li><a class="el" href="a16398.html">OpenSwathDIAPreScoring</a> - SWATH (data-independent acquisition) pre-scoring.</li>
<li><a class="el" href="a16399.html">OpenSwathFileSplitter</a> - A tool for splitting a single SWATH / DIA file into a set of files, each containing one SWATH window.</li>
<li><a class="el" href="a16400.html">OpenSwathMzMLFileCacher</a> - Caching of large mzML files.</li>
<li><a class="el" href="a16401.html">OpenSwathRewriteToFeatureXML</a> - Rewrites results from mProphet back into featureXML.</li>
</ul>
<p><b>RNA</b></p><ul>
<li><a class="el" href="a16397.html">NucleicAcidSearchEngine</a> - Search MzML files for oligonucleotides and their modifications.</li>
<li><a class="el" href="a16413.html">RNADigestor</a> - Digests an RNA sequence database in-silico.</li>
<li><a class="el" href="a16414.html">RNAMassCalculator</a> - Calculates masses and mass-to-charge ratios of RNA sequences.</li>
</ul>
<p><b>Quality Control</b></p><ul>
<li><a class="el" href="a16406.html">QCCalculator</a> - Calculates basic quality parameters from MS experiments and compiles data for subsequent QC into a qcML file.</li>
<li><a class="el" href="a16407.html">QCEmbedder</a> - Embeds tables or plots generated externally as attachments to existing quality parameters in qcML files.</li>
<li><a class="el" href="a16408.html">QCExporter</a> - Extracts several quality parameter from several runs/sets from a qcML file into a tabular (text) format - counterpart to QCImporter.</li>
<li><a class="el" href="a16409.html">QCExtractor</a> - Extracts a table attachment of a given quality parameter from a qcML file as tabular (text) format.</li>
<li><a class="el" href="a16410.html">QCImporter</a> - Imports several quality parameters from a tabular (text) format into a qcML file - counterpart to QCExporter.</li>
<li><a class="el" href="a16411.html">QCMerger</a> - Merges two qcML files together.</li>
<li><a class="el" href="a16412.html">QCShrinker</a> - Removes extra verbose table attachments from a qcML file that are not needed anymore, e.g. for a final report.</li>
</ul>
<p><b>Misc</b></p><ul>
<li><a class="el" href="a16361.html">ClusterMassTraces</a> - Cluster mass traces occurring in the same map together.</li>
<li><a class="el" href="a16366.html">DeMeanderize</a> - Orders the spectra of MALDI spotting plates correctly.</li>
<li><a class="el" href="a16380.html">ImageCreator</a> - Creates images from MS1 data (with MS2 data points indicated as dots).</li>
<li><a class="el" href="a16385.html">MassCalculator</a> - Calculates masses and mass-to-charge ratios of peptide sequences.</li>
<li><a class="el" href="a16388.html">MetaProSIP</a> - Performs proteinSIP on peptide features for elemental flux analysis.</li>
<li><a class="el" href="a16392.html">MSSimulator</a> - A highly configurable simulator for mass spectrometry experiments.</li>
<li><a class="el" href="a16424.html">SvmTheoreticalSpectrumGeneratorTrainer</a> - A trainer for SVM models as input for SvmTheoreticalSpectrumGenerator.</li>
<li><a class="el" href="a16426.html">TICCalculator</a> - Calculates the TIC of a raw mass spectrometric file.</li>
<li><a class="el" href="a16393.html">MSstatsConverter</a> - Converter to input for MSstats.</li>
</ul>
<p><b>Deprecated</b></p><ul>
<li><a class="el" href="a16375.html">IDDecoyProbability</a> - Estimates peptide probabilities using a decoy search strategy. WARNING: This utility is deprecated. </li>
</ul>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:27 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
