<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: PrecursorIonSelector</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16240.html">TOPP documentation</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">PrecursorIonSelector </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>A tool for precursor ion selection based on identification results.</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="3"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> PrecursorIonSelector <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1"><a class="el" href="a16271.html">FeatureFinderCentroided</a>  </td><td valign="middle" align="center" rowspan="2">-   </td></tr>
</table>
</center><p>This tool provides a precursor ion selection based on previous MS/MS identifications.</p>
<p>Different strategies can be chosen: </p><table class="doxtable">
<tr>
<td><b>DEX</b> </td><td>Dynamic exclusion of features with m/z matching predicted tryptic peptides masses of already identified proteins.  </td></tr>
<tr>
<td><b>SPS</b> </td><td>Selection based on score reflecting the feature's suitability for fragmentation.  </td></tr>
<tr>
<td><b>Downshift</b> </td><td>Similar to DEX, but features are not excluded, only ranked down in the feature list  </td></tr>
<tr>
<td><b>Upshift</b> </td><td>Features with m/z matching predicted tryptic peptide masses of unidentified proteins are ranked up.  </td></tr>
<tr>
<td><b>IPS</b> </td><td>Combination of Down- and Upshift.  </td></tr>
<tr>
<td><b>ILP_IPS</b> </td><td>Iterative precursor ion selection using LP formulations.  </td></tr>
</table>
<p>This method is described in: Zerck, A. and Nordhoff, E. and Resemann, A. and Mirgorodskaya, E. and Suckau, D. and Reinert, K. and Lehrach, H. and Gobom, J.: An iterative strategy for precursor ion selection for LC-MS/MS based shotgun proteomics, J Prot Res, 2009, 8 (7), 3239-3251.</p>
<p>Given the feature map of the LC-MS run and the identification results the tool determines the next precursors. The precursors are ranked depending on the chosen strategy.</p>
<p>It is also possible run a simulation of selection strategies on a complete LC-MS/MS run, e.g. to determine what would have been the most efficient strategy.</p>
<dl class="section note"><dt>Note</dt><dd>Currently mzIdentML (mzid) is not directly supported as an input/output format of this tool. Convert mzid files to/from idXML using <a class="el" href="a16291.html">IDFileConverter</a> if necessary.</dd></dl>
<p><b>The command line parameters of this tool are:</b> </p><pre class="fragment">

PrecursorIonSelector -- PrecursorIonSelector

Full documentation: http://www.openms.de/documentation/TOPP_PrecursorIonSelector.html

Version: 2.4.0-HEAD-2019-12-09 Dec  9 2019, 13:47:09, Revision: af8a11e

To cite OpenMS:

  Rost HL, Sachsenberg T, Aiche S, Bielow C et al.. OpenMS: a flexible open-source software platform for mass spectrometry data analysis. Nat Meth. 2016; 13, 9: 741-748. doi:10.1038/nmeth.3959.



Usage:

  PrecursorIonSelector &lt;options&gt;



This tool has algorithm parameters that are not shown here! Please check the ini file for a detailed descript

ion or use the --helphelp option.



Options (mandatory options marked with '*'):

  -in &lt;input file&gt;*            Input feature map file (featureXML) (valid formats: 'featureXML')

  -out &lt;output file&gt;           Modified feature map (valid formats: 'featureXML')

  -next_feat &lt;output file&gt;     Feature map (featureXML) file with the selected precursors (valid formats: 

                               'featureXML')

  -ids &lt;id file&gt;*              File containing results of identification (valid formats: 'idXML')

  -num_precursors &lt;Int&gt;        Number of precursors to be selected (default: '1')

  -raw_data &lt;file&gt;             Input profile data. (valid formats: 'mzML')

  -load_preprocessing          The preprocessed db is loaded from file, not calculated.

  -store_preprocessing         The preprocessed db is stored.

  -simulation                  Simulate the whole LC-MS/MS run.

  -sim_results &lt;output file&gt;   File containing the results of the simulation run (valid formats: 'txt')

  -db_path &lt;db-file&gt;           Db file (valid formats: 'fasta')

  -rt_model &lt;rt-model-file&gt;    SVM Model for RTPredict (valid formats: 'txt')

  -dt_model &lt;dt-model-file&gt;    SVM Model for PTPredict (valid formats: 'txt')

  -fixed_modifications &lt;mods&gt;  The modifications i.e. Carboxymethyl (C)

                               

                               

Common TOPP options:

  -ini &lt;file&gt;                  Use the given TOPP INI file

  -threads &lt;n&gt;                 Sets the number of threads allowed to be used by the TOPP tool (default: '1')

  -write_ini &lt;file&gt;            Writes the default configuration file

  --help                       Shows options

  --helphelp                   Shows all options (including advanced)



The following configuration subsections are valid:

 - algorithm   Settings for the compound list creation and rescoring.



You can write an example INI file using the '-write_ini' option.

Documentation of subsection parameters can be found in the doxygen documentation or the INIFileEditor.

For more information, please consult the online documentation for this tool:

  - http://www.openms.de/documentation/TOPP_PrecursorIonSelector.html



</pre><p> <b>INI file documentation of this tool:</b> <div class="ini_global">
<div class="legend">
<b>Legend:</b><br>
 <div class="item item_required">required parameter</div>
 <div class="item item_advanced">advanced parameter</div>
</div>
  <div class="node"><span class="node_name">+PrecursorIonSelector</span><span class="node_description">PrecursorIonSelector</span></div>
    <div class="item item_advanced"><span class="item_name" style="padding-left:16px;">version</span><span class="item_value">2.4.0-HEAD-2019-12-09</span>
<span class="item_description">Version of the tool that generated this parameters file.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>    <div class="node"><span class="node_name">++1</span><span class="node_description">Instance '1' section for 'PrecursorIonSelector'</span></div>
      <div class="item"><span class="item_name item_required" style="padding-left:24px;">in</span><span class="item_value"></span>
<span class="item_description">Input feature map file (featureXML)</span><span class="item_tags">input file</span><span class="item_restrictions">*.featureXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out</span><span class="item_value"></span>
<span class="item_description">modified feature map</span><span class="item_tags">output file</span><span class="item_restrictions">*.featureXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">next_feat</span><span class="item_value"></span>
<span class="item_description">feature map (featureXML) file with the selected precursors</span><span class="item_tags">output file</span><span class="item_restrictions">*.featureXML</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">ids</span><span class="item_value"></span>
<span class="item_description">file containing results of identification</span><span class="item_tags">input file</span><span class="item_restrictions">*.idXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">num_precursors</span><span class="item_value">1</span>
<span class="item_description">number of precursors to be selected</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">raw_data</span><span class="item_value"></span>
<span class="item_description">Input profile data.</span><span class="item_tags">input file</span><span class="item_restrictions">*.mzML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">load_preprocessing</span><span class="item_value">false</span>
<span class="item_description">The preprocessed db is loaded from file, not calculated.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">store_preprocessing</span><span class="item_value">false</span>
<span class="item_description">The preprocessed db is stored.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">simulation</span><span class="item_value">false</span>
<span class="item_description">Simulate the whole LC-MS/MS run.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">sim_results</span><span class="item_value"></span>
<span class="item_description">File containing the results of the simulation run</span><span class="item_tags">output file</span><span class="item_restrictions">*.txt</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">db_path</span><span class="item_value"></span>
<span class="item_description">db file</span><span class="item_tags">input file</span><span class="item_restrictions">*.fasta</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">rt_model</span><span class="item_value"></span>
<span class="item_description">SVM Model for RTPredict</span><span class="item_tags">input file</span><span class="item_restrictions">*.txt</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">dt_model</span><span class="item_value"></span>
<span class="item_description">SVM Model for PTPredict</span><span class="item_tags">input file</span><span class="item_restrictions">*.txt</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">solver</span><span class="item_value">GLPK</span>
<span class="item_description">LP solver type</span><span class="item_tags"></span><span class="item_restrictions">GLPK,COINOR</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">fixed_modifications</span><span class="item_value">[]</span>
<span class="item_description">the modifications i.e. Carboxymethyl (C)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">log</span><span class="item_value"></span>
<span class="item_description">Name of log file (created only when specified)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">debug</span><span class="item_value">0</span>
<span class="item_description">Sets the debug level</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">threads</span><span class="item_value">1</span>
<span class="item_description">Sets the number of threads allowed to be used by the TOPP tool</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">no_progress</span><span class="item_value">false</span>
<span class="item_description">Disables progress logging to command line</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">force</span><span class="item_value">false</span>
<span class="item_description">Overwrite tool specific checks.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">test</span><span class="item_value">false</span>
<span class="item_description">Enables the test mode (needed for internal use only)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="node"><span class="node_name">+++algorithm</span><span class="node_description">Settings for the compound list creation and rescoring.</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">type</span><span class="item_value">IPS</span>
<span class="item_description">Strategy for precursor ion selection.</span><span class="item_tags"></span><span class="item_restrictions">ILP_IPS,IPS,SPS,Upshift,Downshift,DEX</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">max_iteration</span><span class="item_value">100</span>
<span class="item_description">Maximal number of iterations.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">rt_bin_capacity</span><span class="item_value">10</span>
<span class="item_description">Maximal number of precursors per rt bin.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">step_size</span><span class="item_value">1</span>
<span class="item_description">Maximal number of precursors per iteration.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">peptide_min_prob</span><span class="item_value">0.2</span>
<span class="item_description">Minimal peptide probability.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">sequential_spectrum_order</span><span class="item_value">false</span>
<span class="item_description">If true, precursors are selected sequentially with respect to their RT.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>        <div class="node"><span class="node_name">++++MIPFormulation</span><span class="node_description"></span></div>
          <div class="node"><span class="node_name">+++++thresholds</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">min_protein_probability</span><span class="item_value">0.2</span>
<span class="item_description">Minimal protein probability for a protein to be considered in the ILP</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_protein_id_probability</span><span class="item_value">0.95</span>
<span class="item_description">Minimal protein probability for a protein to be considered identified.</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_pt_weight</span><span class="item_value">0.5</span>
<span class="item_description">Minimal pt weight of a precursor</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_mz</span><span class="item_value">500.0</span>
<span class="item_description">Minimal mz to be considered in protein based LP formulation.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">max_mz</span><span class="item_value">5000.0</span>
<span class="item_description">Minimal mz to be considered in protein based LP formulation.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_pred_pep_prob</span><span class="item_value">0.5</span>
<span class="item_description">Minimal predicted peptide probability of a precursor</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_rt_weight</span><span class="item_value">0.5</span>
<span class="item_description">Minimal rt weight of a precursor</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">use_peptide_rule</span><span class="item_value">false</span>
<span class="item_description">Use peptide rule instead of minimal protein id probability</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_peptide_ids</span><span class="item_value">2</span>
<span class="item_description">If use_peptide_rule is true, this parameter sets the minimal number of peptide ids for a protein id</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_peptide_probability</span><span class="item_value">0.95</span>
<span class="item_description">If use_peptide_rule is true, this parameter sets the minimal probability for a peptide to be safely identified</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>          <div class="node"><span class="node_name">+++++combined_ilp</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">k1</span><span class="item_value">0.2</span>
<span class="item_description">combined ilp: weight for z_i</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">k2</span><span class="item_value">0.2</span>
<span class="item_description">combined ilp: weight for x_j,s*int_j,s</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">k3</span><span class="item_value">0.4</span>
<span class="item_description">combined ilp: weight for -x_j,s*w_j,s</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">scale_matching_probs</span><span class="item_value">true</span>
<span class="item_description">flag if detectability * rt_weight shall be scaled to cover all [0,1]</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>          <div class="node"><span class="node_name">+++++feature_based</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">no_intensity_normalization</span><span class="item_value">false</span>
<span class="item_description">Flag indicating if intensities shall be scaled to be in [0,1]. This is done for each feature separately, so that the feature's maximal intensity in a spectrum is set to 1.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">max_number_precursors_per_feature</span><span class="item_value">1</span>
<span class="item_description">The maximal number of precursors per feature.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>        <div class="node"><span class="node_name">++++Preprocessing</span><span class="node_description"></span></div>
          <div class="item"><span class="item_name" style="padding-left:40px;">precursor_mass_tolerance</span><span class="item_value">10.0</span>
<span class="item_description">Precursor mass tolerance which is used to query the peptide database for peptides</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">precursor_mass_tolerance_unit</span><span class="item_value">ppm</span>
<span class="item_description">Precursor mass tolerance unit.</span><span class="item_tags"></span><span class="item_restrictions">ppm,Da</span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">preprocessed_db_path</span><span class="item_value"></span>
<span class="item_description">Path where the preprocessed database should be stored</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">preprocessed_db_pred_rt_path</span><span class="item_value"></span>
<span class="item_description">Path where the predicted rts of the preprocessed database should be stored</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">preprocessed_db_pred_dt_path</span><span class="item_value"></span>
<span class="item_description">Path where the predicted rts of the preprocessed database should be stored</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">max_peptides_per_run</span><span class="item_value">100000</span>
<span class="item_description">Number of peptides for that the pt and rt are parallely predicted.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">missed_cleavages</span><span class="item_value">1</span>
<span class="item_description">Number of allowed missed cleavages.</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">taxonomy</span><span class="item_value"></span>
<span class="item_description">Taxonomy</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">tmp_dir</span><span class="item_value"></span>
<span class="item_description">Absolute path to tmp data directory used to store files needed for rt and dt prediction.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">store_peptide_sequences</span><span class="item_value">false</span>
<span class="item_description">Flag if peptide sequences should be stored.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="node"><span class="node_name">+++++rt_settings</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">min_rt</span><span class="item_value">960.0</span>
<span class="item_description">Minimal RT in the experiment (in seconds)</span><span class="item_tags"></span><span class="item_restrictions">1.0:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">max_rt</span><span class="item_value">3840.0</span>
<span class="item_description">Maximal RT in the experiment (in seconds)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">rt_step_size</span><span class="item_value">30.0</span>
<span class="item_description">Time between two consecutive spectra (in seconds)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">gauss_mean</span><span class="item_value">-1.0</span>
<span class="item_description">mean of the gauss curve</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">gauss_sigma</span><span class="item_value">3.0</span>
<span class="item_description">std of the gauss curve</span><span class="item_tags"></span><span class="item_restrictions"> </span></div></div>
<p>For the parameters of the algorithm section see the algorithm's documentation: <br  />
 <a class="el" href="a07621.html">OpenMS::PrecursorIonSelection</a> <br  />
</p>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:29 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
