<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: MSSimulator</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16243.html">UTILS documentation</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">MSSimulator </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>A highly configurable simulator for mass spectrometry experiments.</p>
<p>This implementation is described in </p>
<p>Bielow C, Aiche S, Andreotti S, Reinert K<br  />
 MSSimulator: Simulation of Mass Spectrometry Data<br  />
 Journal of Proteome Research (2011), DOI: 10.1021/pr200155f<br  />
 </p>
<p>The most important features are: </p><ul>
<li>
Simulation of Capillary electrophoresis and HPLC as separation step </li>
<li>
Simulation of MS spectra </li>
<li>
Simulation of MS/MS spectra with configurable precursor-selection strategy </li>
<li>
Simulation of iTRAQ labels </li>
<li>
Simulation of different noise models and instrument types (resolution, peak shape) </li>
</ul>
<p>Look at the INI file (via "MSSimulator -write_ini myini.ini") to see the available parameters and more functionality.</p>
<h3>Input: FASTA files</h3>
<p>Protein sequences (including amino acid modifications) can be provided as FASTA file. We allow a special tag in the description of each entry to specify protein abundance. If you want to create a complex FASTA file with a Gaussian protein abundance model in log space, see our Python script shipping with your <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> installation (e.g., &lt;OpenMS-dir&gt;/share/OpenMS/examples/simulation/FASTAProteinAbundanceSampling.py). It supports (random) sampling from a large FASTA file, protein weight filtering and adds an intensity tag to each entry.</p>
<p>If multiplexed data is simulated (like SILAC or iTRAQ) you need to supply multiple FASTA input files. For the label-free setting, all FASTA input files will be merged into one, before simulation.</p>
<p>For MS/MS simulation only a test model is shipped with OpenMS.<br  />
 Please find trained models at: <a href="http://sourceforge.net/projects/open-ms/files/Supplementary/Simulation/">http://sourceforge.net/projects/open-ms/files/Supplementary/Simulation/</a>. </p>
<p>To specify intensity values for certain proteins, add an abundance tag for the corresponding protein in the FASTA input file:<br  />
</p><ul>
<li>add '[# &lt;key&gt;=&lt;value&gt; #]' at the end of the &gt; line to specify intensity For RT control (disable digestion, to make this work!) <ul>
<li>
rt (subjected to small local error by randomization) </li>
<li>
RT (used as is without local error) </li>
</ul>
</li>
</ul>
<p>For amino acid modifications, insert their name at the respective amino acid residues. The modifications are fixed. If you need variable modifications, you have to add the desired combinatorial variants (presence/absence of one or all modifications) to the FASTA file. Valid modification names are listed in many TOPP/UTILS, e.g <a class="el" href="a16314.html">MSGFPlusAdapter</a> 's <em>-fixed_modifications</em> parameter.</p>
<p>e.g. </p><div class="fragment"><div class="line">&gt;seq1 optional comment [# intensity=567.4 #]</div>
<div class="line">(Acetyl).M(Oxidation)ASQYLATARHGC(Carbamidomethyl)FLPRHRDTGILP</div>
<div class="line">&gt;seq2 optional comment [# intensity=117.4, RT=405.3 #]</div>
<div class="line">QKRPSQRHGLATAC(Carbamidomethyl)RHGTGGGDRAT.(Dehydrated)</div>
</div><!-- fragment --><p><b>The command line parameters of this tool are:</b> </p><pre class="fragment">

MSSimulator -- A highly configurable simulator for mass spectrometry experiments.

Full documentation: http://www.openms.de/documentation/UTILS_MSSimulator.html

Version: 2.4.0-HEAD-2019-12-09 Dec  9 2019, 13:47:09, Revision: af8a11e

To cite OpenMS:

  Rost HL, Sachsenberg T, Aiche S, Bielow C et al.. OpenMS: a flexible open-source software platform for mass spectrometry data analysis. Nat Meth. 2016; 13, 9: 741-748. doi:10.1038/nmeth.3959.

To cite MSSimulator:

  Bielow C, Aiche S, Andreotti S, Reinert K. MSSimulator: Simulation of Mass Spectrometry Data. J. Proteome Res. 2011; 10, 7:2922-2929. doi:10.1021/pr200155f.



Usage:

  MSSimulator &lt;options&gt;



This tool has algorithm parameters that are not shown here! Please check the ini file for a detailed descript

ion or use the --helphelp option.



Options (mandatory options marked with '*'):

  -in &lt;files&gt;*       Input protein sequences (valid formats: 'FASTA')

  -out &lt;file&gt;        Output: simulated MS raw (profile) data (valid formats: 'mzML')

  -out_pm &lt;file&gt;     Output: ground-truth picked (centroided) MS data (valid formats: 'mzML')

  -out_fm &lt;file&gt;     Output: ground-truth features (valid formats: 'featureXML')

  -out_cm &lt;file&gt;     Output: ground-truth features, grouping ESI charge variants of each parent peptide (vali

                     d formats: 'consensusXML')

  -out_lcm &lt;file&gt;    Output: ground-truth features, grouping labeled variants (valid formats: 'consensusXML')

  -out_cntm &lt;file&gt;   Output: ground-truth features caused by contaminants (valid formats: 'featureXML')

  -out_id &lt;file&gt;     Output: ground-truth MS2 peptide identifications (valid formats: 'idXML')

                     

Common UTIL options:

  -ini &lt;file&gt;        Use the given TOPP INI file

  -threads &lt;n&gt;       Sets the number of threads allowed to be used by the TOPP tool (default: '1')

  -write_ini &lt;file&gt;  Writes the default configuration file

  --help             Shows options

  --helphelp         Shows all options (including advanced)



The following configuration subsections are valid:

 - algorithm   Algorithm parameters section



You can write an example INI file using the '-write_ini' option.

Documentation of subsection parameters can be found in the doxygen documentation or the INIFileEditor.

For more information, please consult the online documentation for this tool:

  - http://www.openms.de/documentation/UTILS_MSSimulator.html



</pre><p> <b>INI file documentation of this tool:</b> <div class="ini_global">
<div class="legend">
<b>Legend:</b><br>
 <div class="item item_required">required parameter</div>
 <div class="item item_advanced">advanced parameter</div>
</div>
  <div class="node"><span class="node_name">+MSSimulator</span><span class="node_description">A highly configurable simulator for mass spectrometry experiments.</span></div>
    <div class="item item_advanced"><span class="item_name" style="padding-left:16px;">version</span><span class="item_value">2.4.0-HEAD-2019-12-09</span>
<span class="item_description">Version of the tool that generated this parameters file.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>    <div class="node"><span class="node_name">++1</span><span class="node_description">Instance '1' section for 'MSSimulator'</span></div>
      <div class="item"><span class="item_name item_required" style="padding-left:24px;">in</span><span class="item_value">[]</span>
<span class="item_description">Input protein sequences</span><span class="item_tags">input file</span><span class="item_restrictions">*.FASTA</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out</span><span class="item_value"></span>
<span class="item_description">output: simulated MS raw (profile) data</span><span class="item_tags">output file</span><span class="item_restrictions">*.mzML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out_pm</span><span class="item_value"></span>
<span class="item_description">output: ground-truth picked (centroided) MS data</span><span class="item_tags">output file</span><span class="item_restrictions">*.mzML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out_fm</span><span class="item_value"></span>
<span class="item_description">output: ground-truth features</span><span class="item_tags">output file</span><span class="item_restrictions">*.featureXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out_cm</span><span class="item_value"></span>
<span class="item_description">output: ground-truth features, grouping ESI charge variants of each parent peptide</span><span class="item_tags">output file</span><span class="item_restrictions">*.consensusXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out_lcm</span><span class="item_value"></span>
<span class="item_description">output: ground-truth features, grouping labeled variants</span><span class="item_tags">output file</span><span class="item_restrictions">*.consensusXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out_cntm</span><span class="item_value"></span>
<span class="item_description">output: ground-truth features caused by contaminants</span><span class="item_tags">output file</span><span class="item_restrictions">*.featureXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">out_id</span><span class="item_value"></span>
<span class="item_description">output: ground-truth MS2 peptide identifications</span><span class="item_tags">output file</span><span class="item_restrictions">*.idXML</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">log</span><span class="item_value"></span>
<span class="item_description">Name of log file (created only when specified)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">debug</span><span class="item_value">0</span>
<span class="item_description">Sets the debug level</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">threads</span><span class="item_value">1</span>
<span class="item_description">Sets the number of threads allowed to be used by the TOPP tool</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">no_progress</span><span class="item_value">false</span>
<span class="item_description">Disables progress logging to command line</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">force</span><span class="item_value">false</span>
<span class="item_description">Overwrite tool specific checks.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">test</span><span class="item_value">false</span>
<span class="item_description">Enables the test mode (needed for internal use only)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="node"><span class="node_name">+++algorithm</span><span class="node_description">Algorithm parameters section</span></div>
        <div class="node"><span class="node_name">++++MSSim</span><span class="node_description"></span></div>
          <div class="node"><span class="node_name">+++++Digestion</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">enzyme</span><span class="item_value">Trypsin</span>
<span class="item_description">Enzyme to use for digestion (select 'no cleavage' to skip digestion)</span><span class="item_tags"></span><span class="item_restrictions">Trypsin,Arg-C,PepsinA + P,glutamyl endopeptidase,Asp-N/B,CNBr,TrypChymo,Asp-N_ambic,Alpha-lytic protease,proline endopeptidase,2-iodobenzoate,Chymotrypsin/P,Lys-N,Lys-C,V8-E,Chymotrypsin,leukocyte elastase,proline-endopeptidase/HKR,Glu-C+P,Formic_acid,Trypsin/P,PepsinA,iodosobenzoate,Asp-N,Arg-C/P,Lys-C/P,staphylococcal protease/D,V8-DE,elastase-trypsin-chymotrypsin,no cleavage,Clostripain/P,cyanogen-bromide,unspecific cleavage</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">model</span><span class="item_value">naive</span>
<span class="item_description">The cleavage model to use for digestion. 'Trained' is based on a log likelihood model (see DOI:10.1021/pr060507u).</span><span class="item_tags"></span><span class="item_restrictions">trained,naive</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_peptide_length</span><span class="item_value">3</span>
<span class="item_description">Minimum peptide length after digestion (shorter ones will be discarded)</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>            <div class="node"><span class="node_name">++++++model_trained</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">threshold</span><span class="item_value">0.5</span>
<span class="item_description">Model threshold for calling a cleavage. Higher values increase the number of cleavages. -2 will give no cleavages, +4 almost full cleavage.</span><span class="item_tags"></span><span class="item_restrictions">-2.0:4.0</span></div>            <div class="node"><span class="node_name">++++++model_naive</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">missed_cleavages</span><span class="item_value">1</span>
<span class="item_description">Maximum number of missed cleavages considered. All possible resulting peptides will be created.</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>          <div class="node"><span class="node_name">+++++RT</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">rt_column</span><span class="item_value">HPLC</span>
<span class="item_description">Modelling of an RT or CE column</span><span class="item_tags"></span><span class="item_restrictions">none,HPLC,CE</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">auto_scale</span><span class="item_value">true</span>
<span class="item_description">Scale predicted RT's/MT's to given 'total_gradient_time'? If 'true', for CE this means that 'CE:lenght_d', 'CE:length_total', 'CE:voltage' have no influence.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">total_gradient_time</span><span class="item_value">2500.0</span>
<span class="item_description">The duration [s] of the gradient.</span><span class="item_tags"></span><span class="item_restrictions">1.0e-05:&#8734;</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">sampling_rate</span><span class="item_value">2.0</span>
<span class="item_description">Time interval [s] between consecutive scans</span><span class="item_tags"></span><span class="item_restrictions">0.01:60.0</span></div>            <div class="node"><span class="node_name">++++++scan_window</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">min</span><span class="item_value">500.0</span>
<span class="item_description">Start of RT Scan Window [s]</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">max</span><span class="item_value">1500.0</span>
<span class="item_description">End of RT Scan Window [s]</span><span class="item_tags"></span><span class="item_restrictions">1.0:&#8734;</span></div>            <div class="node"><span class="node_name">++++++variation</span><span class="node_description">Random component that simulates technical/biological variation</span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">feature_stddev</span><span class="item_value">3</span>
<span class="item_description">Standard deviation of shift in retention time [s] from predicted model (applied to every single feature independently)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">affine_offset</span><span class="item_value">0</span>
<span class="item_description">Global offset in retention time [s] from predicted model</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">affine_scale</span><span class="item_value">1</span>
<span class="item_description">Global scaling in retention time from predicted model</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++column_condition</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">distortion</span><span class="item_value">0</span>
<span class="item_description">Distortion of the elution profiles. Good presets are 0 for a perfect elution profile, 1 for a slightly distorted elution profile etc... For trapping instruments (e.g. Orbitrap) distortion should be >4.</span><span class="item_tags"></span><span class="item_restrictions">0:10</span></div>            <div class="node"><span class="node_name">++++++profile_shape</span><span class="node_description"></span></div>
              <div class="node"><span class="node_name">+++++++width</span><span class="node_description">Width of the EGH elution shape, i.e. the sigma^2 parameter, which is computed using 'value' + rnd_cauchy('variance')</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">value</span><span class="item_value">9.0</span>
<span class="item_description">Width of the Exponential Gaussian Hybrid distribution shape of the elution profile. This does not correspond directly to the width in [s].</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">variance</span><span class="item_value">1.6</span>
<span class="item_description">Random component of the width (set to 0 to disable randomness), i.e. scale parameter for the lorentzian variation of the variance (Note: The scale parameter has to be >= 0).</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="node"><span class="node_name">+++++++skewness</span><span class="node_description">Skewness of the EGH elution shape, i.e. the tau parameter, which is computed using 'value' + rnd_cauchy('variance')</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">value</span><span class="item_value">0.1</span>
<span class="item_description">Asymmetric component of the EGH. Higher absolute(!) values lead to more skewness (negative values cause fronting, positive values cause tailing). Tau parameter of the EGH, i.e. time constant of the exponential decay of the Exponential Gaussian Hybrid distribution shape of the elution profile.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">variance</span><span class="item_value">0.3</span>
<span class="item_description">Random component of skewness (set to 0 to disable randomness), i.e. scale parameter for the lorentzian variation of the time constant (Note: The scale parameter has to be > 0).</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="node"><span class="node_name">++++++HPLC</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">model_file</span><span class="item_value">examples/simulation/RTPredict.model</span>
<span class="item_description">SVM model for retention time prediction</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++CE</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">pH</span><span class="item_value">3.0</span>
<span class="item_description">pH of buffer</span><span class="item_tags"></span><span class="item_restrictions">0.0:14.0</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">alpha</span><span class="item_value">0.5</span>
<span class="item_description">Exponent Alpha used to calculate mobility</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">mu_eo</span><span class="item_value">0.0</span>
<span class="item_description">Electroosmotic flow</span><span class="item_tags"></span><span class="item_restrictions">0.0:5.0</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">lenght_d</span><span class="item_value">70.0</span>
<span class="item_description">Length of capillary [cm] from injection site to MS</span><span class="item_tags"></span><span class="item_restrictions">0.0:1000.0</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">length_total</span><span class="item_value">75.0</span>
<span class="item_description">Total length of capillary [cm]</span><span class="item_tags"></span><span class="item_restrictions">0.0:1000.0</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">voltage</span><span class="item_value">1000.0</span>
<span class="item_description">Voltage applied to capillary</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>          <div class="node"><span class="node_name">+++++Detectability</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">dt_simulation_on</span><span class="item_value">false</span>
<span class="item_description">Modelling detectibility enabled? This can serve as a filter to remove peptides which ionize badly, thus reducing peptide count</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">min_detect</span><span class="item_value">0.5</span>
<span class="item_description">Minimum peptide detectability accepted. Peptides with a lower score will be removed</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">dt_model_file</span><span class="item_value">examples/simulation/DTPredict.model</span>
<span class="item_description">SVM model for peptide detectability prediction</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="node"><span class="node_name">+++++Ionization</span><span class="node_description"></span></div>
            <div class="node"><span class="node_name">++++++esi</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">ionized_residues</span><span class="item_value">[Arg, Lys, His]</span>
<span class="item_description">List of residues (as three letter code) that will be considered during ES ionization. The N-term is always assumed to carry a charge. This parameter will be ignored during MALDI ionization</span><span class="item_tags"></span><span class="item_restrictions">Ala,Cys,Asp,Glu,Phe,Gly,His,Ile,Lys,Leu,Met,Asn,Pro,Gln,Arg,Sec,Ser,Thr,Val,Trp,Tyr</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">charge_impurity</span><span class="item_value">[H+:1]</span>
<span class="item_description">List of charged ions that contribute to charge with weight of occurrence (their sum is scaled to 1 internally), e.g. ['H:1'] or ['H:0.7' 'Na:0.3'], ['H:4' 'Na:1'] (which internally translates to ['H:0.8' 'Na:0.2'])</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item item_advanced"><span class="item_name" style="padding-left:56px;">max_impurity_set_size</span><span class="item_value">3</span>
<span class="item_description">Maximal #combinations of charge impurities allowed (each generating one feature) per charge state. E.g. assuming charge=3 and this parameter is 2, then we could choose to allow '3H+, 2H+Na+' features (given a certain 'charge_impurity' constraints), but no '3H+, 2H+Na+, 3Na+'</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">ionization_probability</span><span class="item_value">0.8</span>
<span class="item_description">Probability for the binomial distribution of the ESI charge states</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++maldi</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">ionization_probabilities</span><span class="item_value">[0.9, 0.1, 0.0]</span>
<span class="item_description">List of probabilities for different charge states (starting at charge=1, 2, ...) during MALDI ionization (the list must sum up to 1.0)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++mz</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">lower_measurement_limit</span><span class="item_value">200.0</span>
<span class="item_description">Lower m/z detector limit</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">upper_measurement_limit</span><span class="item_value">1200.0</span>
<span class="item_description">Upper m/z detector limit</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>          <div class="node"><span class="node_name">+++++RawSignal</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">enabled</span><span class="item_value">true</span>
<span class="item_description">Enable RAW signal simulation? (select 'false' if you only need feature-maps)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">peak_shape</span><span class="item_value">Gaussian</span>
<span class="item_description">Peak Shape used around each isotope peak (be aware that the area under the curve is constant for both types, but the maximal height will differ (~ 2:3 = Lorentz:Gaussian) due to the wider base of the Lorentzian</span><span class="item_tags"></span><span class="item_restrictions">Gaussian,Lorentzian</span></div>            <div class="node"><span class="node_name">++++++resolution</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">value</span><span class="item_value">50000</span>
<span class="item_description">Instrument resolution at 400 Th</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">type</span><span class="item_value">linear</span>
<span class="item_description">How does resolution change with increasing m/z?! QTOFs usually show 'constant' behavior, FTs have linear degradation, and on Orbitraps the resolution decreases with square root of mass</span><span class="item_tags"></span><span class="item_restrictions">constant,linear,sqrt</span></div>            <div class="node"><span class="node_name">++++++baseline</span><span class="node_description">Baseline modeling for MALDI ionization</span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">scaling</span><span class="item_value">0.0</span>
<span class="item_description">Scale of baseline. Set to 0 to disable simulation of baseline</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">shape</span><span class="item_value">0.5</span>
<span class="item_description">The baseline is modeled by an exponential probability density function (pdf) with f(x) = shape*e^(- shape*x)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="node"><span class="node_name">++++++mz</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">sampling_points</span><span class="item_value">3</span>
<span class="item_description">Number of raw data points per FWHM of the peak</span><span class="item_tags"></span><span class="item_restrictions">2:&#8734;</span></div>            <div class="node"><span class="node_name">++++++contaminants</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">file</span><span class="item_value">examples/simulation/contaminants.csv</span>
<span class="item_description">Contaminants file with sum formula and absolute RT interval. See 'OpenMS/examples/simulation/contaminants.txt' for details</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++variation</span><span class="node_description">Random components that simulate biological and technical variations of the simulated data</span></div>
              <div class="node"><span class="node_name">+++++++mz</span><span class="node_description">Shifts in mass to charge dimension of the simulated signals</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">error_mean</span><span class="item_value">0.0</span>
<span class="item_description">Average systematic m/z error (in Da)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">error_stddev</span><span class="item_value">0.0</span>
<span class="item_description">Standard deviation for m/z errors. Set to 0 to disable simulation of m/z errors</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="node"><span class="node_name">+++++++intensity</span><span class="node_description">Variations in intensity to model randomness in feature intensity</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">scale</span><span class="item_value">100.0</span>
<span class="item_description">Constant scale factor of the feature intensity. Set to 1.0 to get the real intensity values provided in the FASTA file</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">scale_stddev</span><span class="item_value">0.0</span>
<span class="item_description">Standard deviation of peak intensity (relative to the scaled peak height). Set to 0 to get simple rescaled intensities</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>            <div class="node"><span class="node_name">++++++noise</span><span class="node_description">Parameters modeling noise in mass spectrometry measurements</span></div>
              <div class="node"><span class="node_name">+++++++shot</span><span class="node_description">Parameters of Poisson and Exponential for shot noise modeling (set :rate OR :mean = 0 to disable)</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">rate</span><span class="item_value">0.0</span>
<span class="item_description">Poisson rate of shot noise per unit m/z (random peaks in m/z, where the number of peaks per unit m/z follows a Poisson distribution). Set this to 0 to disable simulation of shot noise</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">intensity-mean</span><span class="item_value">1.0</span>
<span class="item_description">Shot noise intensity mean (exponentially distributed with given mean)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="node"><span class="node_name">+++++++white</span><span class="node_description">Parameters of Gaussian distribution for white noise modeling (set :mean AND :stddev = 0 to disable). No new peaks are generated; only intensity of existing ones is changed</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">mean</span><span class="item_value">0.0</span>
<span class="item_description">Mean value of white noise (Gaussian) being added to each *measured* signal intensity</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">stddev</span><span class="item_value">0.0</span>
<span class="item_description">Standard deviation of white noise being added to each *measured* signal intensity</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="node"><span class="node_name">+++++++detector</span><span class="node_description">Parameters of Gaussian distribution for detector noise modeling (set :mean AND :stddev = 0 to disable). If enabled, ALL possible m/z positions (up to sampling frequency of detector) will receive an intensity increase/decrease according to the specified Gaussian intensity distribution (similar to a noisy baseline)</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">mean</span><span class="item_value">0.0</span>
<span class="item_description">Mean intensity value of the detector noise (Gaussian distribution)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">stddev</span><span class="item_value">0.0</span>
<span class="item_description">Standard deviation of the detector noise (Gaussian distribution)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="node"><span class="node_name">+++++RawTandemSignal</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">status</span><span class="item_value">disabled</span>
<span class="item_description">Create Tandem-MS scans?</span><span class="item_tags"></span><span class="item_restrictions">disabled,precursor,MS^E</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">tandem_mode</span><span class="item_value">0</span>
<span class="item_description">Algorithm to generate the tandem-MS spectra. 0 - fixed intensities, 1 - SVC prediction (abundant/missing), 2 - SVR prediction of peak intensity <br></span><span class="item_tags"></span><span class="item_restrictions">0:2</span></div>            <div class="item"><span class="item_name" style="padding-left:48px;">svm_model_set_file</span><span class="item_value">examples/simulation/SvmModelSet.model</span>
<span class="item_description">File containing the filenames of SVM Models for different charge variants</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++Precursor</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">ms2_spectra_per_rt_bin</span><span class="item_value">5</span>
<span class="item_description">Number of allowed MS/MS spectra in a retention time bin.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">min_mz_peak_distance</span><span class="item_value">2.0</span>
<span class="item_description">The minimal distance (in Th) between two peaks for concurrent selection for fragmentation. Also used to define the m/z width of an exclusion window (distance +/- from m/z of precursor). If you set this lower than the isotopic envelope of a peptide, you might get multiple fragment spectra pointing to the same precursor.</span><span class="item_tags"></span><span class="item_restrictions">1.0e-04:&#8734;</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">mz_isolation_window</span><span class="item_value">2.0</span>
<span class="item_description">All peaks within a mass window (in Th) of a selected peak are also selected for fragmentation.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">exclude_overlapping_peaks</span><span class="item_value">false</span>
<span class="item_description">If true, overlapping or nearby peaks (within 'min_mz_peak_distance') are excluded for selection.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">charge_filter</span><span class="item_value">[2, 3]</span>
<span class="item_description">Charges considered for MS2 fragmentation.</span><span class="item_tags"></span><span class="item_restrictions">1:5</span></div>              <div class="node"><span class="node_name">+++++++Exclusion</span><span class="node_description"></span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">use_dynamic_exclusion</span><span class="item_value">false</span>
<span class="item_description">If true dynamic exclusion is applied.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">exclusion_time</span><span class="item_value">100.0</span>
<span class="item_description">The time (in seconds) a feature is excluded.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="node"><span class="node_name">+++++++ProteinBasedInclusion</span><span class="node_description"></span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">max_list_size</span><span class="item_value">1000</span>
<span class="item_description">The maximal number of precursors in the inclusion list.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>                <div class="node"><span class="node_name">++++++++rt</span><span class="node_description"></span></div>
                  <div class="item"><span class="item_name" style="padding-left:72px;">min_rt</span><span class="item_value">960.0</span>
<span class="item_description">Minimal rt in seconds.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">max_rt</span><span class="item_value">3840.0</span>
<span class="item_description">Maximal rt in seconds.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">rt_step_size</span><span class="item_value">30.0</span>
<span class="item_description">rt step size in seconds.</span><span class="item_tags"></span><span class="item_restrictions">1.0:&#8734;</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">rt_window_size</span><span class="item_value">100</span>
<span class="item_description">rt window size in seconds.</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>                <div class="node"><span class="node_name">++++++++thresholds</span><span class="node_description"></span></div>
                  <div class="item"><span class="item_name" style="padding-left:72px;">min_protein_id_probability</span><span class="item_value">0.95</span>
<span class="item_description">Minimal protein probability for a protein to be considered identified.</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">min_pt_weight</span><span class="item_value">0.5</span>
<span class="item_description">Minimal pt weight of a precursor</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">min_mz</span><span class="item_value">500.0</span>
<span class="item_description">Minimal mz to be considered in protein based LP formulation.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">max_mz</span><span class="item_value">5000.0</span>
<span class="item_description">Minimal mz to be considered in protein based LP formulation.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">use_peptide_rule</span><span class="item_value">false</span>
<span class="item_description">Use peptide rule instead of minimal protein id probability</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">min_peptide_ids</span><span class="item_value">2</span>
<span class="item_description">If use_peptide_rule is true, this parameter sets the minimal number of peptide ids for a protein id</span><span class="item_tags"></span><span class="item_restrictions">1:&#8734;</span></div>                  <div class="item"><span class="item_name" style="padding-left:72px;">min_peptide_probability</span><span class="item_value">0.95</span>
<span class="item_description">If use_peptide_rule is true, this parameter sets the minimal probability for a peptide to be safely identified</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="node"><span class="node_name">++++++MS_E</span><span class="node_description"></span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">add_single_spectra</span><span class="item_value">false</span>
<span class="item_description">If true, the MS2 spectra for each peptide signal are included in the output (might be a lot). They will have a meta value 'MSE_DebugSpectrum' attached, so they can be filtered out. Native MS_E spectra will have 'MSE_Spectrum' instead.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>            <div class="node"><span class="node_name">++++++TandemSim</span><span class="node_description"></span></div>
              <div class="node"><span class="node_name">+++++++Simple</span><span class="node_description"></span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">add_isotopes</span><span class="item_value">false</span>
<span class="item_description">If set to 'true' isotope peaks of the product ion peaks are added</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">max_isotope</span><span class="item_value">2</span>
<span class="item_description">Defines the maximal isotopic peak which is added if 'add_isotopes' is 'true'</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_metainfo</span><span class="item_value">false</span>
<span class="item_description">Adds the type of peaks as metainfo to the peaks, like y8+, [M-H2O+2H]++</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_losses</span><span class="item_value">false</span>
<span class="item_description">Adds common losses to those ion expect to have them, only water and ammonia loss is considered</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_precursor_peaks</span><span class="item_value">false</span>
<span class="item_description">Adds peaks of the unfragmented precursor ion to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_all_precursor_charges</span><span class="item_value">false</span>
<span class="item_description">Adds precursor peaks with all charges in the given range</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_abundant_immonium_ions</span><span class="item_value">false</span>
<span class="item_description">Add most abundant immonium ions</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_first_prefix_ion</span><span class="item_value">false</span>
<span class="item_description">If set to true e.g. b1 ions are added</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_y_ions</span><span class="item_value">true</span>
<span class="item_description">Add peaks of y-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_b_ions</span><span class="item_value">true</span>
<span class="item_description">Add peaks of b-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_a_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of a-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_c_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of c-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_x_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of  x-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_z_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of z-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">y_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the y-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">b_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the b-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">a_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the a-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">c_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the c-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">x_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the x-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">z_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the z-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">relative_loss_intensity</span><span class="item_value">0.1</span>
<span class="item_description">Intensity of loss ions, in relation to the intact ion intensity</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">precursor_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the precursor peak</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">precursor_H2O_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the H2O loss peak of the precursor</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">precursor_NH3_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the NH3 loss peak of the precursor</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="node"><span class="node_name">+++++++SVM</span><span class="node_description"></span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">add_isotopes</span><span class="item_value">false</span>
<span class="item_description">If set to 1 isotope peaks of the product ion peaks are added</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">max_isotope</span><span class="item_value">2</span>
<span class="item_description">Defines the maximal isotopic peak which is added, add_isotopes must be set to 1</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_metainfo</span><span class="item_value">false</span>
<span class="item_description">Adds the type of peaks as metainfo to the peaks, like y8+, [M-H2O+2H]++</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">add_first_prefix_ion</span><span class="item_value">false</span>
<span class="item_description">If set to true e.g. b1 ions are added</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_y_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of y-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_y2_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of y-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_b_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of b-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_b2_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of b-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_a_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of a-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_c_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of c-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_x_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of  x-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_z_ions</span><span class="item_value">false</span>
<span class="item_description">Add peaks of z-ions to the spectrum</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">hide_losses</span><span class="item_value">false</span>
<span class="item_description">Adds common losses to those ion expect to have them, only water and ammonia loss is considered</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">y_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the y-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">b_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the b-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">a_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the a-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">c_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the c-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">x_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the x-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">z_intensity</span><span class="item_value">1.0</span>
<span class="item_description">Intensity of the z-ions</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">relative_loss_intensity</span><span class="item_value">0.1</span>
<span class="item_description">Intensity of loss ions, in relation to the intact ion intensity</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>          <div class="node"><span class="node_name">+++++Global</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">ionization_type</span><span class="item_value">ESI</span>
<span class="item_description">Type of Ionization (MALDI or ESI)</span><span class="item_tags"></span><span class="item_restrictions">MALDI,ESI</span></div>          <div class="node"><span class="node_name">+++++Labeling</span><span class="node_description"></span></div>
            <div class="item"><span class="item_name" style="padding-left:48px;">type</span><span class="item_value">labelfree</span>
<span class="item_description">Select the labeling type you want for your experiment</span><span class="item_tags"></span><span class="item_restrictions">ICPL,SILAC,itraq,labelfree,o18</span></div>            <div class="node"><span class="node_name">++++++ICPL</span><span class="node_description">ICPL labeling on MS1 level of lysines and n-term (on protein or peptide level) with either two or three channels.</span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">ICPL_fixed_rtshift</span><span class="item_value">0.0</span>
<span class="item_description">Fixed retention time shift between labeled pairs. If set to 0.0 only the retention times, computed by the RT model step are used.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">label_proteins</span><span class="item_value">true</span>
<span class="item_description">Enables protein-labeling. (select 'false' if you only need peptide-labeling)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>              <div class="item item_advanced"><span class="item_name" style="padding-left:56px;">ICPL_light_channel_label</span><span class="item_value">UniMod:365</span>
<span class="item_description">UniMod Id of the light channel ICPL label.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item item_advanced"><span class="item_name" style="padding-left:56px;">ICPL_medium_channel_label</span><span class="item_value">UniMod:687</span>
<span class="item_description">UniMod Id of the medium channel ICPL label.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item item_advanced"><span class="item_name" style="padding-left:56px;">ICPL_heavy_channel_label</span><span class="item_value">UniMod:364</span>
<span class="item_description">UniMod Id of the heavy channel ICPL label.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++SILAC</span><span class="node_description">SILAC labeling on MS1 level with up to 3 channels and custom modifications.</span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">fixed_rtshift</span><span class="item_value">1.0e-04</span>
<span class="item_description">Fixed retention time shift between labeled peptides. If set to 0.0 only the retention times computed by the RT model step are used.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>              <div class="node"><span class="node_name">+++++++medium_channel</span><span class="node_description">Modifications for the medium SILAC channel.</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">modification_lysine</span><span class="item_value">UniMod:481</span>
<span class="item_description">Modification of Lysine in the medium SILAC channel</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">modification_arginine</span><span class="item_value">UniMod:188</span>
<span class="item_description">Modification of Arginine in the medium SILAC channel</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="node"><span class="node_name">+++++++heavy_channel</span><span class="node_description">Modifications for the heavy SILAC channel. If you want to use only 2 channels, just leave the Labels as they are and provide only 2 input files.</span></div>
                <div class="item"><span class="item_name" style="padding-left:64px;">modification_lysine</span><span class="item_value">UniMod:259</span>
<span class="item_description">Modification of Lysine in the heavy SILAC channel. If left empty, two channelSILAC is assumed.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>                <div class="item"><span class="item_name" style="padding-left:64px;">modification_arginine</span><span class="item_value">UniMod:267</span>
<span class="item_description">Modification of Arginine in the heavy SILAC channel. If left empty, two-channel SILAC is assumed.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>            <div class="node"><span class="node_name">++++++itraq</span><span class="node_description">iTRAQ labeling on MS2 level with up to 4 (4plex) or 8 (8plex) channels.</span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">iTRAQ</span><span class="item_value">4plex</span>
<span class="item_description">4plex or 8plex iTRAQ?</span><span class="item_tags"></span><span class="item_restrictions">4plex,8plex</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">reporter_mass_shift</span><span class="item_value">0.1</span>
<span class="item_description">Allowed shift (uniformly distributed - left to right) in Da from the expected position (of e.g. 114.1, 115.1)</span><span class="item_tags"></span><span class="item_restrictions">0.0:0.5</span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">channel_active_4plex</span><span class="item_value">[114:myReference]</span>
<span class="item_description">Four-plex only: Each channel that was used in the experiment and its description (114-117) in format <channel>:<name>, e.g. "114:myref","115:liver".</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">channel_active_8plex</span><span class="item_value">[113:myReference]</span>
<span class="item_description">Eight-plex only: Each channel that was used in the experiment and its description (113-121) in format <channel>:<name>, e.g. "113:myref","115:liver","118:lung".</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item item_advanced"><span class="item_name" style="padding-left:56px;">isotope_correction_values_4plex</span><span class="item_value">[114:0.0/1.0/5.9/0.2, 115:0.0/2.0/5.6/0.1, 116:0.0/3.0/4.5/0.1, 117:0.1/4.0/3.5/0.1]</span>
<span class="item_description">override default values (see Documentation); use the following format: <channel>:<-2Da>/<-1Da>/<+1Da>/<+2Da> ; e.g. '114:0/0.3/4/0' , '116:0.1/0.3/3/0.2' </span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item item_advanced"><span class="item_name" style="padding-left:56px;">isotope_correction_values_8plex</span><span class="item_value">[113:0.0/0.0/6.89/0.22, 114:0.0/0.94/5.9/0.16, 115:0.0/1.88/4.9/0.1, 116:0.0/2.82/3.9/0.07, 117:0.06/3.77/2.99/0.0, 118:0.09/4.71/1.88/0.0, 119:0.14/5.66/0.87/0.0, 121:0.27/7.44/0.18/0.0]</span>
<span class="item_description">override default values (see Documentation); use the following format: <channel>:<-2Da>/<-1Da>/<+1Da>/<+2Da> ; e.g. '113:0/0.3/4/0' , '116:0.1/0.3/3/0.2' </span><span class="item_tags"></span><span class="item_restrictions"> </span></div>              <div class="item"><span class="item_name" style="padding-left:56px;">Y_contamination</span><span class="item_value">0.3</span>
<span class="item_description">Efficiency of labeling tyrosine ('Y') residues. 0=off, 1=full labeling</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>            <div class="node"><span class="node_name">++++++o18</span><span class="node_description">18O labeling on MS1 level with 2 channels, requiring trypsin digestion.</span></div>
              <div class="item"><span class="item_name" style="padding-left:56px;">labeling_efficiency</span><span class="item_value">1.0</span>
<span class="item_description">Describes the distribution of the labeled peptide over the different states (unlabeled, mono- and di-labeled)</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>        <div class="node"><span class="node_name">++++RandomNumberGenerators</span><span class="node_description">Parameters for generating the random aspects (e.g. noise) in the simulated data. The generation is separated into two parts, the technical part, like noise in the raw signal, and the biological part, like systematic deviations in the predicted retention times</span></div>
          <div class="item"><span class="item_name" style="padding-left:40px;">biological</span><span class="item_value">random</span>
<span class="item_description">Controls the 'biological' randomness of the generated data (e.g. systematic effects like deviations in RT). If set to 'random' each experiment will look different. If set to 'reproducible' each experiment will have the same outcome (given that the input data is the same)</span><span class="item_tags"></span><span class="item_restrictions">reproducible,random</span></div>          <div class="item"><span class="item_name" style="padding-left:40px;">technical</span><span class="item_value">random</span>
<span class="item_description">Controls the 'technical' randomness of the generated data (e.g. noise in the raw signal). If set to 'random' each experiment will look different. If set to 'reproducible' each experiment will have the same outcome (given that the input data is the same)</span><span class="item_tags"></span><span class="item_restrictions">reproducible,random</span></div></div>
 </div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:29 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
