<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: MascotAdapter</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16240.html">TOPP documentation</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">MascotAdapter </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Identifies peptides in MS/MS spectra via Mascot.</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="2"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> MascotAdapter <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1">any signal-/preprocessing tool <br  />
 (in mzML format) </td><td valign="middle" align="center" rowspan="1"><a class="el" href="a16292.html">IDFilter</a> or <br  />
 any protein/peptide processing tool  </td></tr>
</table>
</center><p>This wrapper application serves for getting peptide identifications for MS/MS spectra. It uses a local installation of the Mascot server to generate the identifications. A second wrapper (MascotAdapterOnline) is available which is able to perform identifications by communicating with a Mascot server over the network. So, it is not necessary to execute MascotAdapterOnline on the same machine as Mascot.</p>
<p>The minimal version of Mascot supported with this server is 2.1.</p>
<p>This wrapper can be executed in three different modes: </p><ol>
<li>
<p class="startli">The whole process of ProteinIdentification via Mascot is executed. Inputfile is a mzData file containing the MS/MS spectra for which the identifications are to be found. The results are written as a idXML output file. This mode is selected by default. </p>
<p class="endli"></p>
</li>
<li>
<p class="startli">Only the first part of the ProteinIdentification process is performed. This means that the MS/MS data is transformed into Mascot Generic Format (mgf) which can be used directly with Mascot. Being in the cgi directory of the Mascot directory calling a Mascot process should look like the following:</p>
<div class="fragment"><div class="line">./nph-mascot.exe 1 -commandline -f outputfilename &lt; inputfilename </div>
</div><!-- fragment --><p class="interli">Consult your Mascot reference manual for further details.</p>
<p class="interli">This mode is selected by the <b>-mascot_in</b> option in the command line. </p>
<p class="endli"></p>
</li>
<li>
<p class="startli">Only the second part of the ProteinIdentification process is performed. This means that the outputfile of the Mascot server is translated into idXML.</p>
<p class="endli">This mode is selected by the <b>-mascot_out</b> option in the command line.  </p>
</li>
</ol>
<p><br  />
 If your Mascot server is installed on the same computer as the TOPP applications the MascotAdapter can be executed in mode 1. Otherwise the Mascot engine has to be executed manually assisted by mode 2 and mode 3. The ProteinIdentification steps then look like:</p>
<ul>
<li>
execute MascotAdapter in mode 2 <div class="fragment"><div class="line">./MascotAdapter -in mzDataFile -out mascotGenericFormatFile -mascot_in </div>
</div><!-- fragment -->  </li>
<li>
copy mascotGenericFormatFile to your Mascot server  </li>
<li>
call your Mascot server process: <div class="fragment"><div class="line">./nph-mascot.exe 1 -commandline -f mascotOutFile &lt; mascotGenericFormatFile </div>
</div><!-- fragment -->  </li>
<li>
call the script to export your outfile in mascot xml <div class="fragment"><div class="line">./export_dat.pl do_export=1 export_format=XML file=mascotOutFile _sigthreshold=0</div>
<div class="line">           _showsubset=1 show_same_sets=1 show_unassigned=0 prot_score=0 pep_exp_z=0 pep_score=0</div>
<div class="line">           pep_homol=0 pep_ident=0 pep_seq=1 show_header=1 show_queries=1 pep_rank=0 &gt; mascotXMLFile </div>
</div><!-- fragment -->  </li>
<li>
copy mascotXMLFile to the server on which the TOPP applications are installed  </li>
<li>
execute MascotAdapter in mode 3 <div class="fragment"><div class="line">./MascotAdapter -in mascotXMLFile -out IdXMLFile -mascot_out </div>
</div><!-- fragment -->  </li>
</ul>
<p>For mode 1 you have to specify the directory in which the Mascot server is installed. This is done by setting the option <b>mascot_dir</b> in the ini file. Furthermore you have to specify a folder in which the user has write permissions. This is done by setting the option <b>temp_data_directory</b> in the ini file. Two temporary files will be created in this directory during execution but deleted at the end of execution. <br  />
</p>
<dl class="section note"><dt>Note</dt><dd>Currently mzIdentML (mzid) is not directly supported as an input/output format of this tool. Convert mzid files to/from idXML using <a class="el" href="a16291.html">IDFileConverter</a> if necessary.</dd></dl>
<p><b>The command line parameters of this tool are:</b> </p><pre class="fragment">

MascotAdapter -- Annotates MS/MS spectra using Mascot.

Full documentation: http://www.openms.de/documentation/TOPP_MascotAdapter.html

Version: 2.4.0-HEAD-2019-12-09 Dec  9 2019, 13:47:09, Revision: af8a11e

To cite OpenMS:

  Rost HL, Sachsenberg T, Aiche S, Bielow C et al.. OpenMS: a flexible open-source software platform for mass spectrometry data analysis. Nat Meth. 2016; 13, 9: 741-748. doi:10.1038/nmeth.3959.



Usage:

  MascotAdapter &lt;options&gt;



Options (mandatory options marked with '*'):

  -in &lt;file&gt;*                      Input file in mzData format.

                                   Note: In mode 'mascot_out' a Mascot results file (.mascotXML) is read

  -out &lt;file&gt;*                     Output file in idXML format.

                                   Note: In mode 'mascot_in' Mascot generic format is written.

  -mascot_in                       If this flag is set the MascotAdapter will read in mzData and write Mascot

                                   generic format

  -mascot_out                      If this flag is set the MascotAdapter will read in a Mascot results file 

                                   (.mascotXML) and write idXML

  -instrument &lt;i&gt;                  The instrument that was used to measure the spectra (default: 'Default')

  -precursor_mass_tolerance &lt;tol&gt;  The precursor mass tolerance (default: '2.0')

  -peak_mass_tolerance &lt;tol&gt;       The peak mass tolerance (default: '1.0')

  -taxonomy &lt;tax&gt;                  The taxonomy (default: 'All entries' valid: 'All entries', '. . Archaea 

                                   (Archaeobacteria)', '. . Eukaryota (eucaryotes)', '. . . . Alveolata (alve

                                   olates)', '. . . . . . Plasmodium falciparum (malaria parasite)', '. . .

                                   . . . Other Alveolata', '. . . . Metazoa (Animals)', '. . . . . . Caenorha

                                   bditis elegans', '. . . . . . Drosophila (fruit flies)', '. . . . . . Chor

                                   data (vertebrates and relatives)', '. . . . . . . . bony vertebrates', '.

                                   . . . . . . . . . lobe-finned fish and tetrapod clade', '. . . . . . . .

                                   . . . . Mammalia (mammals)', '. . . . . . . . . . . . . . Primates', '. .

                                   ...

                                   ilable')

  -modifications &lt;mods&gt;            The modifications i.e. Carboxymethyl (C)

  -variable_modifications &lt;mods&gt;   The variable modifications i.e. Carboxymethyl (C)

  -charges [1+ 2+ ...]             The different charge states (default: '[1+ 2+ 3+]')

  -db &lt;name&gt;                       The database to search in (default: 'MSDB')

  -hits &lt;num&gt;                      The number of hits to report (default: 'AUTO')

  -cleavage &lt;enz&gt;                  The enzyme descriptor to the enzyme used for digestion. (Trypsin is defaul

                                   t, None would be best for peptide input or unspecific digestion, for more

                                   please refer to your mascot server). (default: 'Trypsin' valid: 'Trypsin',

                                   'Arg-C', 'Asp-N', 'Asp-N_ambic', 'Chymotrypsin', 'CNBr', 'CNBr+Trypsin',

                                   'Formic_acid', 'Lys-C', 'Lys-C/P', 'PepsinA', 'Tryp-CNBr', 'TrypChymo',

                                   'Trypsin/P', 'V8-DE', 'V8-E', 'semiTrypsin', 'LysC+AspN', 'None')

  -missed_cleavages &lt;num&gt;          Number of allowed missed cleavages (default: '0' min: '0')

  -sig_threshold &lt;num&gt;             Significance threshold (default: '0.05')

  -pep_homol &lt;num&gt;                 Peptide homology threshold (default: '1.0')

  -pep_ident &lt;num&gt;                 Peptide ident threshold (default: '1.0')

  -pep_rank &lt;num&gt;                  Peptide rank (default: '1')

  -prot_score &lt;num&gt;                Protein score (default: '1.0')

  -pep_score &lt;num&gt;                 Peptide score (default: '1.0')

  -pep_exp_z &lt;num&gt;                 Peptide expected charge (default: '1')

  -show_unassigned &lt;num&gt;           Show_unassigned (default: '1')

  -first_dim_rt &lt;num&gt;              Additional information which is added to every peptide identification as 

                                   metavalue if set &gt; 0 (default: '0.0')

  -boundary &lt;string&gt;               MIME boundary for mascot output format

  -mass_type &lt;type&gt;                Mass type (default: 'Monoisotopic' valid: 'Monoisotopic', 'Average')

  -mascot_directory &lt;dir&gt;          The directory in which mascot is located

  -temp_data_directory &lt;dir&gt;       A directory in which some temporary files can be stored

                                   

Common TOPP options:

  -ini &lt;file&gt;                      Use the given TOPP INI file

  -threads &lt;n&gt;                     Sets the number of threads allowed to be used by the TOPP tool (default: 

                                   '1')

  -write_ini &lt;file&gt;                Writes the default configuration file

  --help                           Shows options

  --helphelp                       Shows all options (including advanced)



</pre><p> <b>INI file documentation of this tool:</b> <div class="ini_global">
<div class="legend">
<b>Legend:</b><br>
 <div class="item item_required">required parameter</div>
 <div class="item item_advanced">advanced parameter</div>
</div>
  <div class="node"><span class="node_name">+MascotAdapter</span><span class="node_description">Annotates MS/MS spectra using Mascot.</span></div>
    <div class="item item_advanced"><span class="item_name" style="padding-left:16px;">version</span><span class="item_value">2.4.0-HEAD-2019-12-09</span>
<span class="item_description">Version of the tool that generated this parameters file.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>    <div class="node"><span class="node_name">++1</span><span class="node_description">Instance '1' section for 'MascotAdapter'</span></div>
      <div class="item"><span class="item_name item_required" style="padding-left:24px;">in</span><span class="item_value"></span>
<span class="item_description">input file in mzData format.<br>Note: In mode 'mascot_out' a Mascot results file (.mascotXML) is read</span><span class="item_tags">input file</span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">out</span><span class="item_value"></span>
<span class="item_description">output file in idXML format.<br>Note: In mode 'mascot_in' Mascot generic format is written.</span><span class="item_tags">output file</span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">mascot_in</span><span class="item_value">false</span>
<span class="item_description">if this flag is set the MascotAdapter will read in mzData and write Mascot generic format</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">mascot_out</span><span class="item_value">false</span>
<span class="item_description">if this flag is set the MascotAdapter will read in a Mascot results file (.mascotXML) and write idXML</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">instrument</span><span class="item_value">Default</span>
<span class="item_description">the instrument that was used to measure the spectra</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">precursor_mass_tolerance</span><span class="item_value">2.0</span>
<span class="item_description">the precursor mass tolerance</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">peak_mass_tolerance</span><span class="item_value">1.0</span>
<span class="item_description">the peak mass tolerance</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">taxonomy</span><span class="item_value">All entries</span>
<span class="item_description">the taxonomy</span><span class="item_tags"></span><span class="item_restrictions">All entries,. . Archaea (Archaeobacteria),. . Eukaryota (eucaryotes),. . . . Alveolata (alveolates),. . . . . . Plasmodium falciparum (malaria parasite),. . . . . . Other Alveolata,. . . . Metazoa (Animals),. . . . . . Caenorhabditis elegans,. . . . . . Drosophila (fruit flies),. . . . . . Chordata (vertebrates and relatives),. . . . . . . . bony vertebrates,. . . . . . . . . . lobe-finned fish and tetrapod clade,. . . . . . . . . . . . Mammalia (mammals),. . . . . . . . . . . . . . Primates,. . . . . . . . . . . . . . . . Homo sapiens (human),. . . . . . . . . . . . . . . . Other primates,. . . . . . . . . . . . . . Rodentia (Rodents),. . . . . . . . . . . . . . . . Mus.,. . . . . . . . . . . . . . . . . . Mus musculus (house mouse),. . . . . . . . . . . . . . . . Rattus,. . . . . . . . . . . . . . . . Other rodentia,. . . . . . . . . . . . . . Other mammalia,. . . . . . . . . . . . Xenopus laevis (African clawed frog),. . . . . . . . . . . . Other lobe-finned fish and tetrapod clade,. . . . . . . . . . Actinopterygii (ray-finned fishes),. . . . . . . . . . . . Takifugu rubripes (Japanese Pufferfish),. . . . . . . . . . . . Danio rerio (zebra fish),. . . . . . . . . . . . Other Actinopterygii,. . . . . . . . Other Chordata,. . . . . . Other Metazoa,. . . . Dictyostelium discoideum,. . . . Fungi,. . . . . . Saccharomyces Cerevisiae (baker's yeast),. . . . . . Schizosaccharomyces pombe (fission yeast),. . . . . . Pneumocystis carinii,. . . . . . Other Fungi,. . . . Viridiplantae (Green Plants),. . . . . . Arabidopsis thaliana (thale cress),. . . . . . Oryza sativa (rice),. . . . . . Other green plants,. . . . Other Eukaryota,. . Bacteria (Eubacteria),. . . . Actinobacteria (class),. . . . . . Mycobacterium tuberculosis complex,. . . . . . Other Actinobacteria (class),. . . . Firmicutes (gram-positive bacteria),. . . . . . Bacillus subtilis,. . . . . . Mycoplasma,. . . . . . Streptococcus Pneumoniae,. . . . . . Streptomyces coelicolor,. . . . . . Other Firmicutes,. . . . Proteobacteria (purple bacteria),. . . . . . Agrobacterium tumefaciens,. . . . . . Campylobacter jejuni,. . . . . . Escherichia coli,. . . . . . Neisseria meningitidis,. . . . . . Salmonella,. . . . . . Other Proteobacteria,. . . . Other Bacteria,. . Viruses,. . . . Hepatitis C virus,. . . . Other viruses,. . Other (includes plasmids and artificial sequences),. . unclassified,. . Species information unavailable</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">modifications</span><span class="item_value">[]</span>
<span class="item_description">the modifications i.e. Carboxymethyl (C)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">variable_modifications</span><span class="item_value">[]</span>
<span class="item_description">the variable modifications i.e. Carboxymethyl (C)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">charges</span><span class="item_value">[1+, 2+, 3+]</span>
<span class="item_description">the different charge states</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">db</span><span class="item_value">MSDB</span>
<span class="item_description">the database to search in</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">hits</span><span class="item_value">AUTO</span>
<span class="item_description">the number of hits to report</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">cleavage</span><span class="item_value">Trypsin</span>
<span class="item_description">The enzyme descriptor to the enzyme used for digestion. (Trypsin is default, None would be best for peptide input or unspecific digestion, for more please refer to your mascot server).</span><span class="item_tags"></span><span class="item_restrictions">Trypsin,Arg-C,Asp-N,Asp-N_ambic,Chymotrypsin,CNBr,CNBr+Trypsin,Formic_acid,Lys-C,Lys-C/P,PepsinA,Tryp-CNBr,TrypChymo,Trypsin/P,V8-DE,V8-E,semiTrypsin,LysC+AspN,None</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">missed_cleavages</span><span class="item_value">0</span>
<span class="item_description">number of allowed missed cleavages</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">sig_threshold</span><span class="item_value">0.05</span>
<span class="item_description">significance threshold</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">pep_homol</span><span class="item_value">1.0</span>
<span class="item_description">peptide homology threshold</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">pep_ident</span><span class="item_value">1.0</span>
<span class="item_description">peptide ident threshold</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">pep_rank</span><span class="item_value">1</span>
<span class="item_description">peptide rank</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">prot_score</span><span class="item_value">1.0</span>
<span class="item_description">protein score</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">pep_score</span><span class="item_value">1.0</span>
<span class="item_description">peptide score</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">pep_exp_z</span><span class="item_value">1</span>
<span class="item_description">peptide expected charge</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">show_unassigned</span><span class="item_value">1</span>
<span class="item_description">show_unassigned</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">first_dim_rt</span><span class="item_value">0.0</span>
<span class="item_description">additional information which is added to every peptide identification as metavalue if set > 0</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">boundary</span><span class="item_value"></span>
<span class="item_description">MIME boundary for mascot output format</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">mass_type</span><span class="item_value">Monoisotopic</span>
<span class="item_description">mass type</span><span class="item_tags"></span><span class="item_restrictions">Monoisotopic,Average</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">mascot_directory</span><span class="item_value"></span>
<span class="item_description">the directory in which mascot is located</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">temp_data_directory</span><span class="item_value"></span>
<span class="item_description">a directory in which some temporary files can be stored</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">log</span><span class="item_value"></span>
<span class="item_description">Name of log file (created only when specified)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">debug</span><span class="item_value">0</span>
<span class="item_description">Sets the debug level</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">threads</span><span class="item_value">1</span>
<span class="item_description">Sets the number of threads allowed to be used by the TOPP tool</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">no_progress</span><span class="item_value">false</span>
<span class="item_description">Disables progress logging to command line</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">force</span><span class="item_value">false</span>
<span class="item_description">Overwrite tool specific checks.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">test</span><span class="item_value">false</span>
<span class="item_description">Enables the test mode (needed for internal use only)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div></div>
<p>You can specify the Mascot parameters <b>precursor_mass_tolerance</b> (the peptide mass tolerance), <b>peak_mass_tolerance</b> (the MS/MS tolerance), <b>taxonomy</b> (restriction to a certain subset of the database), <b>modifications</b>, <b>variable_modifications</b>, <b>charges</b> (the possible charge variants), <b>db</b> (database where the peptides are searched in), <b>hits</b> (number of hits), <b>cleavage</b> (the cleavage enzyme), <b>missed_cleavages</b> (number of missed cleavages) and <b>mass_type</b> (Monoisotopic or Average) via the ini file.</p>
<p><br  />
 Known problems with Mascot server execution: </p><ul>
<li>
<p class="startli">getting error message: "FATAL_ERROR: M00327
         The ms-monitor daemon/service is not running, please start it." </p>
<p class="endli"></p>
</li>
<li>
Possible explanations:  <ul>
<li>
Your ms-monitor is really not running =&gt; consult your Mascot reference manual for details about starting the Mascot server.  </li>
<li>
(Suppose you have Mascot installed in directory mascot.) mascot/data/mascot.control is not writable for the current user. This has to be changed. Otherwise you will not be able to use the Mascot server via the shell and receive the above error message.<br  />
 =&gt; Change write permissions of the file mascot/data/mascot.control such that the current user has write permissions to it.  </li>
</ul>
</li>
</ul>
<dl class="todo"><dt><b><a class="el" href="a05777.html#_todo000044">Todo:</a></b></dt><dd>This adapter is using antiquated internal methods and needs to be updated! E.g. use <a class="el" href="a03998.html">MascotGenericFile.h</a> instead of <a class="el" href="a04001.html">MascotInfile.h</a>.... </dd></dl>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:28 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
