<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: PeptideIndexer</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16240.html">TOPP documentation</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">PeptideIndexer </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Refreshes the protein references for all peptide hits from an idXML file and adds target/decoy information.</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="2"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> PeptideIndexer <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1"><a class="el" href="a16292.html">IDFilter</a> or <br  />
 any protein/peptide processing tool  </td><td valign="middle" align="center" rowspan="1"><a class="el" href="a16270.html">FalseDiscoveryRate</a>   </td></tr>
</table>
</center><p>PeptideIndexer refreshes target/decoy information and mapping of peptides to proteins. The target/decoy information is crucial for the <a class="el" href="a16270.html">FalseDiscoveryRate</a> tool. (For FDR calculations, "target+decoy" peptide hits count as target hits.)</p>
<p>PeptideIndexer allows for ambiguous amino acids (B|J|Z|X) in the protein database, but not in the peptide sequences. For the latter only I/L can be treated as equivalent (see 'IL_equivalent' flag), but 'J' is not allowed.</p>
<p>Enzyme cutting rules and partial specificity can be specified.</p>
<p>Resulting protein hits appear in the order of the FASTA file, except for orphaned proteins, which will appear first with an empty target_decoy metavalue. Duplicate protein accessions &amp; sequences will not raise a warning, but create multiple hits (PeptideIndexer scans over the FASTA file once for efficiency reasons, and thus might not see all accessions &amp; sequences at once).</p>
<p>All peptide and protein hits are annotated with target/decoy information, using the meta value "target_decoy". For proteins the possible values are "target" and "decoy", depending on whether the protein accession contains the decoy pattern (parameter <code>decoy_string</code>) as a suffix or prefix, respectively (see parameter <code>prefix</code>).</p>
<p>Peptide hits are annotated with metavalue 'protein_references', and if matched to at least one protein also with metavalue 'target_decoy'. The possible values for 'target_decoy' are "target", "decoy" and "target+decoy", depending on whether the peptide sequence is found only in target proteins, only in decoy proteins, or in both. The metavalue is not present, if the peptide is unmatched.</p>
<p>Runtime: PeptideIndexer is usually very fast (loading and storing the data takes the most time) and search speed can be further improved (linearly), but using more threads. Avoid allowing too many (&gt;=4) ambiguous amino acids if your database contains long stretches of 'X' (exponential search space).</p>
<p>PeptideIndexer supports relative database filenames, which (when not found in the current working directory) are looked up in the directories specified by <code>OpenMS.ini:id_db_dir</code> (see <a class="el" href="a16207.html">TOPP for Advanced Users</a>).</p>
<p>Further details can be found in the underlying PeptideIndexing implementation.</p>
<dl class="section note"><dt>Note</dt><dd>Currently mzIdentML (mzid) is not directly supported as an input/output format of this tool. Convert mzid files to/from idXML using <a class="el" href="a16291.html">IDFileConverter</a> if necessary.</dd></dl>
<p><b>The command line parameters of this tool are:</b> </p><pre class="fragment">

PeptideIndexer -- Refreshes the protein references for all peptide hits.

Full documentation: http://www.openms.de/documentation/TOPP_PeptideIndexer.html

Version: 2.4.0-HEAD-2019-12-09 Dec  9 2019, 13:47:09, Revision: af8a11e

To cite OpenMS:

  Rost HL, Sachsenberg T, Aiche S, Bielow C et al.. OpenMS: a flexible open-source software platform for mass spectrometry data analysis. Nat Meth. 2016; 13, 9: 741-748. doi:10.1038/nmeth.3959.



Usage:

  PeptideIndexer &lt;options&gt;



Options (mandatory options marked with '*'):

  -in &lt;file&gt;*                      Input idXML file containing the identifications. (valid formats: 'idXML')

  -fasta &lt;file&gt;*                   Input sequence database in FASTA format. Non-existing relative filenames 

                                   are looked up via 'OpenMS.ini:id_db_dir' (valid formats: 'fasta')

  -out &lt;file&gt;*                     Output idXML file. (valid formats: 'idXML')

  -decoy_string &lt;text&gt;             String that was appended (or prefixed - see 'decoy_string_position' flag 

                                   below) to the accessions in the protein database to indicate decoy protein

                                   s. If empty (default), it's determined automatically (checking for common

                                   terms, both as prefix and suffix).

  -decoy_string_position &lt;choice&gt;  Is the 'decoy_string' prepended (prefix) or appended (suffix) to the prote

                                   in accession? (ignored if decoy_string is empty) (default: 'prefix' valid:

                                   'prefix', 'suffix')

  -missing_decoy_action &lt;choice&gt;   Action to take if NO peptide was assigned to a decoy protein (which indica

                                   tes wrong database or decoy string): 'error' (exit with error, no output),

                                   'warn' (exit with success, warning message), 'silent' (no action is taken

                                   , not even a warning) (default: 'error' valid: 'error', 'warn', 'silent')

  -write_protein_sequence          If set, the protein sequences are stored as well.

  -write_protein_description       If set, the protein description is stored as well.

  -keep_unreferenced_proteins      If set, protein hits which are not referenced by any peptide are kept.

  -allow_unmatched                 If set, unmatched peptide sequences are allowed. By default (i.e. if this 

                                   flag is not set) the program terminates with an error on unmatched peptide

                                   s.

  -aaa_max &lt;number&gt;                Maximal number of ambiguous amino acids (AAAs) allowed when matching to a 

                                   protein database with AAAs. AAAs are B, J, Z and X! (default: '3' min:

                                   '0' max: '10')

  -mismatches_max &lt;number&gt;         Maximal number of mismatched (mm) amino acids allowed when matching to a 

                                   protein database. The required runtime is exponential in the number of

                                   mm's; apply with care. MM's are allowed in addition to AAA's. (default:

                                   '0' min: '0' max: '10')

  -IL_equivalent                   Treat the isobaric amino acids isoleucine ('I') and leucine ('L') as equiv

                                   alent (indistinguishable). Also occurences of 'J' will be treated as 'I'

                                   thus avoiding ambiguous matching.



enzyme:

  -enzyme:name &lt;choice&gt;            Enzyme which determines valid cleavage sites - e.g. trypsin cleaves after 

                                   lysine (K) or arginine (R), but not before proline (P). (default: 'Trypsin

                                   ' valid: 'Trypsin', 'Arg-C', 'iodosobenzoate', 'Formic_acid', 'V8-E', 'sta

                                   phylococcal protease/D', 'Chymotrypsin', 'Alpha-lytic protease', '2-iodobe

                                   nzoate', 'Glu-C+P', 'Asp-N', 'proline-endopeptidase/HKR', 'Chymotrypsin/P'

                                   , 'Asp-N_ambic', 'PepsinA + P', 'Lys-N', 'Arg-C/P', 'PepsinA', 'TrypChymo'

                                   , 'leukocyte elastase', 'Asp-N/B', 'CNBr', 'V8-DE', 'proline endopeptidase

                                   ', 'glutamyl endopeptidase', 'Lys-C', 'Trypsin/P', 'Lys-C/P', 'no cleavage

                                   ', 'cyanogen-bromide', 'unspecific cleavage', 'Clostripain/P', 'elastase-t

                                   rypsin-chymotrypsin')

  -enzyme:specificity &lt;choice&gt;     Specificity of the enzyme.

                                   'full': both internal cleavage sites must match.

                                   'semi': one of two internal cleavage sites must match.

                                   'none': allow all peptide hits no matter their context. Therefore, the

                                   enzyme chosen does not play a role here (default: 'full' valid: 'full',

                                   'semi', 'none')



                                   

Common TOPP options:

  -ini &lt;file&gt;                      Use the given TOPP INI file

  -threads &lt;n&gt;                     Sets the number of threads allowed to be used by the TOPP tool (default: 

                                   '1')

  -write_ini &lt;file&gt;                Writes the default configuration file

  --help                           Shows options

  --helphelp                       Shows all options (including advanced)



</pre><p> <b>INI file documentation of this tool:</b> <div class="ini_global">
<div class="legend">
<b>Legend:</b><br>
 <div class="item item_required">required parameter</div>
 <div class="item item_advanced">advanced parameter</div>
</div>
  <div class="node"><span class="node_name">+PeptideIndexer</span><span class="node_description">Refreshes the protein references for all peptide hits.</span></div>
    <div class="item item_advanced"><span class="item_name" style="padding-left:16px;">version</span><span class="item_value">2.4.0-HEAD-2019-12-09</span>
<span class="item_description">Version of the tool that generated this parameters file.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>    <div class="node"><span class="node_name">++1</span><span class="node_description">Instance '1' section for 'PeptideIndexer'</span></div>
      <div class="item"><span class="item_name item_required" style="padding-left:24px;">in</span><span class="item_value"></span>
<span class="item_description">Input idXML file containing the identifications.</span><span class="item_tags">input file</span><span class="item_restrictions">*.idXML</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">fasta</span><span class="item_value"></span>
<span class="item_description">Input sequence database in FASTA format. Non-existing relative filenames are looked up via 'OpenMS.ini:id_db_dir'</span><span class="item_tags">input file</span><span class="item_restrictions">*.fasta</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">out</span><span class="item_value"></span>
<span class="item_description">Output idXML file.</span><span class="item_tags">output file</span><span class="item_restrictions">*.idXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">decoy_string</span><span class="item_value"></span>
<span class="item_description">String that was appended (or prefixed - see 'decoy_string_position' flag below) to the accessions in the protein database to indicate decoy proteins. If empty (default), it's determined automatically (checking for common terms, both as prefix and suffix).</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">decoy_string_position</span><span class="item_value">prefix</span>
<span class="item_description">Is the 'decoy_string' prepended (prefix) or appended (suffix) to the protein accession? (ignored if decoy_string is empty)</span><span class="item_tags"></span><span class="item_restrictions">prefix,suffix</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">missing_decoy_action</span><span class="item_value">error</span>
<span class="item_description">Action to take if NO peptide was assigned to a decoy protein (which indicates wrong database or decoy string): 'error' (exit with error, no output), 'warn' (exit with success, warning message), 'silent' (no action is taken, not even a warning)</span><span class="item_tags"></span><span class="item_restrictions">error,warn,silent</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">write_protein_sequence</span><span class="item_value">false</span>
<span class="item_description">If set, the protein sequences are stored as well.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">write_protein_description</span><span class="item_value">false</span>
<span class="item_description">If set, the protein description is stored as well.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">keep_unreferenced_proteins</span><span class="item_value">false</span>
<span class="item_description">If set, protein hits which are not referenced by any peptide are kept.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">allow_unmatched</span><span class="item_value">false</span>
<span class="item_description">If set, unmatched peptide sequences are allowed. By default (i.e. if this flag is not set) the program terminates with an error on unmatched peptides.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">aaa_max</span><span class="item_value">3</span>
<span class="item_description">Maximal number of ambiguous amino acids (AAAs) allowed when matching to a protein database with AAAs. AAAs are B, J, Z and X!</span><span class="item_tags"></span><span class="item_restrictions">0:10</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">mismatches_max</span><span class="item_value">0</span>
<span class="item_description">Maximal number of mismatched (mm) amino acids allowed when matching to a protein database. The required runtime is exponential in the number of mm's; apply with care. MM's are allowed in addition to AAA's.</span><span class="item_tags"></span><span class="item_restrictions">0:10</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">IL_equivalent</span><span class="item_value">false</span>
<span class="item_description">Treat the isobaric amino acids isoleucine ('I') and leucine ('L') as equivalent (indistinguishable). Also occurences of 'J' will be treated as 'I' thus avoiding ambiguous matching.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">log</span><span class="item_value"></span>
<span class="item_description">Name of log file (created only when specified)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">debug</span><span class="item_value">0</span>
<span class="item_description">Sets the debug level</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">threads</span><span class="item_value">1</span>
<span class="item_description">Sets the number of threads allowed to be used by the TOPP tool</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">no_progress</span><span class="item_value">false</span>
<span class="item_description">Disables progress logging to command line</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">force</span><span class="item_value">false</span>
<span class="item_description">Overwrite tool specific checks.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">test</span><span class="item_value">false</span>
<span class="item_description">Enables the test mode (needed for internal use only)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="node"><span class="node_name">+++enzyme</span><span class="node_description"></span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">name</span><span class="item_value">Trypsin</span>
<span class="item_description">Enzyme which determines valid cleavage sites - e.g. trypsin cleaves after lysine (K) or arginine (R), but not before proline (P).</span><span class="item_tags"></span><span class="item_restrictions">Asp-N,Asp-N_ambic,Chymotrypsin,Formic_acid,PepsinA,TrypChymo,Asp-N/B,Arg-C/P,CNBr,Lys-C,Lys-N,Chymotrypsin/P,Lys-C/P,Trypsin,Arg-C,Trypsin/P,V8-DE,leukocyte elastase,PepsinA + P,elastase-trypsin-chymotrypsin,Glu-C+P,proline endopeptidase,iodosobenzoate,proline-endopeptidase/HKR,glutamyl endopeptidase,cyanogen-bromide,V8-E,Alpha-lytic protease,2-iodobenzoate,staphylococcal protease/D,Clostripain/P,no cleavage,unspecific cleavage</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">specificity</span><span class="item_value">full</span>
<span class="item_description">Specificity of the enzyme.<br>  'full': both internal cleavage sites must match.<br>  'semi': one of two internal cleavage sites must match.<br>  'none': allow all peptide hits no matter their context. Therefore, the enzyme chosen does not play a role here</span><span class="item_tags"></span><span class="item_restrictions">full,semi,none</span></div></div>
 </div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:29 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
