<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: MSFraggerAdapter</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="a16239.html">User Quickstart Guide</a></li><li class="navelem"><a class="el" href="a16243.html">UTILS documentation</a></li>  </ul>
</div>
</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">MSFraggerAdapter </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Peptide Identification with MSFragger</p>
<center> <table class="doxtable">
<tr>
<td align="center" bgcolor="#EBEBEB">pot. predecessor tools  </td><td valign="middle" rowspan="2"><img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> MSFraggerAdapter <img class="formulaInl" alt="$ \longrightarrow $" src="form_34.png"/> </td><td align="center" bgcolor="#EBEBEB">pot. successor tools   </td></tr>
<tr>
<td valign="middle" align="center" rowspan="1">any signal-/preprocessing tool <br  />
 (in mzML format) </td><td valign="middle" align="center" rowspan="1"><a class="el" href="a16292.html">IDFilter</a> or <br  />
 any protein/peptide processing tool  </td></tr>
</table>
</center><p><em>MSFragger</em> must be installed before this adapter can be used.</p>
<p>All MSFragger parameters (as specified in the fragger.params file) have been transcribed to parameters of this <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> util. It is not possible to provide an explicit fragger.params file to avoid redundancy with the ini file. This adapter creates an fragger.params file prior to calling MSFragger. If the fragger.params file should be inspected, set the -debug option to 2. MSFraggerAdapter will print the path to the working directory to standard out.</p>
<p>MSFragger can process multiple input files (mzML, mzXML) one after another. The number of output files specified must match the number of input spectra files. The output file is then matched to the input file by index. The default parameters of the adapter are the same as given by the official MSFragger manual.</p>
<p>Please cite: Andy T Kong, Felipe V Leprevost, Dmitry M Avtonomov, Dattatreya Mellacheruvu &amp; Alexey I Nesvizhskii MSFragger: ultrafast and comprehensive peptide identification in mass spectrometry–based proteomics Nature Methods volume 14, pages 513–520 (2017) doi:10.1038/nmeth.4256</p>
<p><b>The command line parameters of this tool are:</b> </p><pre class="fragment">

MSFraggerAdapter -- Peptide Identification with MSFragger

Full documentation: http://www.openms.de/documentation/UTILS_MSFraggerAdapter.html

Version: 2.4.0-HEAD-2019-12-09 Dec  9 2019, 13:47:09, Revision: af8a11e

To cite OpenMS:

  Rost HL, Sachsenberg T, Aiche S, Bielow C et al.. OpenMS: a flexible open-source software platform for mass spectrometry data analysis. Nat Meth. 2016; 13, 9: 741-748. doi:10.1038/nmeth.3959.

To cite MSFraggerAdapter:

  Kong AT, Leprevost FV, Avtonomov DM, Mellacheruvu D, Nesvizhskii AI. MSFragger: ultrafast and comprehensive peptide identification in mass spectrometry–based proteomics. Nature Methods volume 14, pages 513–520 (2017). doi:doi:10.1038/nmeth.4256.



Usage:

  MSFraggerAdapter &lt;options&gt;



Options (mandatory options marked with '*'):

  -java_executable &lt;file&gt;                                                  The Java executable. Usually Java 

                                                                           is on the system PATH. If Java is

                                                                           not found, use this parameter to

                                                                           specify the full path to Java

  -java_heapmemory &lt;num&gt;                                                   Maximum Java heap size (in MB) 

                                                                           (default: '3500')

  -executable &lt;path_to_executable&gt;                                         Path to the MSFragger executable 

                                                                           to use; may be empty if the execut

                                                                           able is globally available.

  -in &lt;file&gt;*                                                              Input File with specta for MSFragg

                                                                           er (valid formats: 'mzML', 'mzXML'

                                                                           )

  -out &lt;file&gt;*                                                             MSFragger output file (valid forma

                                                                           ts: 'idXML')

  -opt_out &lt;file&gt;                                                          MSFragger optional output file 

                                                                           (valid formats: 'pepXML')

  -database &lt;path_to_fasta&gt;*                                               Protein FASTA database file path 

                                                                           (valid formats: 'FASTA', 'fasta',

                                                                           'fa', 'fas')



Search Tolerances:

  -tolerance:precursor_mass_tolerance &lt;precursor_mass_tolerance&gt;           Precursor mass tolerance (window 

                                                                           is +/- this value) (default: '20.0

                                                                           ' min: '0.0')

  -tolerance:precursor_mass_unit &lt;precursor_mass_unit&gt;                     Unit of precursor mass tolerance 

                                                                           (default: 'ppm' valid: 'Da', 'ppm'

                                                                           )

  -tolerance:precursor_true_tolerance &lt;precursor_true_tolerance&gt;           True precursor mass tolerance (win

                                                                           dow is +/- this value). Used for

                                                                           tie breaker of results (in spectra

                                                                           lly ambiguous cases) and zero bin

                                                                           boosting in open searches (0 disab

                                                                           les these features). This option

                                                                           is STRONGLY recommended for open

                                                                           searches. (default: '0.0' min:

                                                                           '0.0')

  -tolerance:precursor_true_unit &lt;precursor_true_unit&gt;                     Unit of precursor true tolerance 

                                                                           (default: 'ppm' valid: 'Da', 'ppm'

                                                                           )

  -tolerance:fragment_mass_tolerance &lt;fragment_mass_tolerance&gt;             Fragment mass tolerance (window 

                                                                           is +/- this value) (default: '20.0

                                                                           ' min: '0.0')

  -tolerance:fragment_mass_unit &lt;fragment_mass_unit&gt;                       Unit of fragment mass tolerance 

                                                                           (default: 'ppm' valid: 'Da', 'ppm'

                                                                           )

  -tolerance:isotope_error &lt;isotope_error&gt;                                 Isotope correction for MS/MS event

                                                                           s triggered on isotopic peaks.

                                                                           Should be set to 0 (disabled) for

                                                                           open search or 0/1/2 for correctio

                                                                           n of narrow window searches. Shift

                                                                           s the precursor mass window to

                                                                           multiples of this value multiplied

                                                                           by the mass of C13-C12. (default:

                                                                           '0' valid: '0', '1', '2')



In-Silico Digestion Parameters:

  -digest:search_enzyme_name &lt;search_enzyme_name&gt;                          Name of the enzyme to be written 

                                                                           to the pepXML file (default: 'Tryp

                                                                           sin' valid: 'Arg-C', 'Trypsin',

                                                                           'iodosobenzoate', 'Trypsin/P',

                                                                           'Asp-N/B', 'Chymotrypsin/P', 'Chym

                                                                           otrypsin', 'Formic_acid', 'PepsinA

                                                                           ', 'TrypChymo', 'Asp-N', 'Lys-C/P'

                                                                           , 'V8-DE', 'Lys-C', 'Lys-N', 'V8-E

                                                                           ...

                                                                           ain/P')

  -digest:search_enzyme_cutafter &lt;search_enzyme_cutafter&gt;                  Residues after which the enzyme 

                                                                           cuts (specified as a string of

                                                                           amino acids) (default: 'KR')

  -digest:search_enzyme_nocutbefore &lt;search_enzyme_nocutbefore&gt;            Residues that the enzyme will not 

                                                                           cut before (default: 'P')

  -digest:num_enzyme_termini &lt;num_enzyme_termini&gt;                          Number of enzyme termini (non-enzy

                                                                           matic (0), semi (1), fully (2)

                                                                           (default: 'fully' valid: 'non-enzy

                                                                           matic', 'semi', 'fully')

  -digest:allowed_missed_cleavage &lt;allowed_missed_cleavage&gt;                Allowed number of missed cleavages

                                                                           (default: '2' valid: '0', '1',

                                                                           '2', '3', '4', '5')

  -digest:min_length &lt;digest_min_length&gt;                                   Minimum length of peptides to be 

                                                                           generated during in-silico digesti

                                                                           on (default: '7' min: '0')

  -digest:max_length &lt;digest_max_length&gt;                                   Maximum length of peptides to be 

                                                                           generated during in-silico digesti

                                                                           on (default: '64' min: '0')

  -digest:mass_range_min &lt;digest_mass_range_min&gt;                           Min mass of peptides to be generat

                                                                           ed (Da) (default: '500.0' min:

                                                                           '0.0')

  -digest:mass_range_max &lt;digest_mass_range_max&gt;                           Max mass of peptides to be generat

                                                                           ed (Da) (default: '5000.0' min:

                                                                           '0.0')



Variable Modification Parameters:

  -varmod:clip_nterm_m                                                     Specifies the trimming of a protei

                                                                           n N-terminal methionine as a varia

                                                                           ble modification

  -varmod:masses &lt;varmod1_mass .. varmod7_mass&gt;                            Masses for variable modifications

  -varmod:syntaxes &lt;varmod1_syntax .. varmod7_syntax&gt;                      Syntax Strings for variable modifi

                                                                           cations

  -varmod:enable_common                                                    Enable common variable modificatio

                                                                           ns (15.9949 M and 42.0106 [^)

  -varmod:not_allow_multiple_variable_mods_on_residue                      Do not allow any one amino acid 

                                                                           to be modified by multiple variabl

                                                                           e modifications

  -varmod:max_variable_mods_per_mod &lt;max_variable_mods_per_mod&gt;            Maximum number of residues that 

                                                                           can be occupied by each variable

                                                                           modification (default: '2' valid:

                                                                           '0', '1', '2', '3', '4', '5')

  -varmod:max_variable_mods_combinations &lt;max_variable_mods_combinations&gt;  Maximum allowed number of modified

                                                                           variably modified peptides from

                                                                           each peptide sequence, (maximum

                                                                           of 65534). If a greater number

                                                                           than the maximum is generated,

                                                                           only the unmodified peptide is

                                                                           considered (default: '5000' min:

                                                                           '0' max: '65534')



Spectrum Processing Parameters:

  -spectrum:minimum_peaks &lt;minimum_peaks&gt;                                  Minimum number of peaks in experim

                                                                           ental spectrum for matching (defau

                                                                           lt: '10' min: '0')

  -spectrum:use_topn_peaks &lt;use_topN_peaks&gt;                                Pre-process experimental spectrum 

                                                                           to only use top N peaks (default:

                                                                           '50' min: '0')

  -spectrum:minimum_ratio &lt;minimum_ratio&gt;                                  Filters out all peaks in experimen

                                                                           tal spectrum less intense than

                                                                           this multiple of the base peak

                                                                           intensity (default: '0.0' min:

                                                                           '0.0' max: '1.0')

  -spectrum:clear_mz_range_min &lt;clear_mz_range_min&gt;                        Removes peaks in this m/z range 

                                                                           prior to matching (minimum value).

                                                                           Useful for iTRAQ/TMT experiments

                                                                           (i.e. 0.0 150.0) (default: '0.0'

                                                                           min: '0.0')

  -spectrum:clear_mz_range_max &lt;clear_mz_range_max&gt;                        Removes peaks in this m/z range 

                                                                           prior to matching (maximum value).

                                                                           Useful for iTRAQ/TMT experiments

                                                                           (i.e. 0.0 150.0) (default: '0.0'

                                                                           min: '0.0')

  -spectrum:max_fragment_charge &lt;max_fragment_charge&gt;                      Maximum charge state for theoretic

                                                                           al fragments to match (default:

                                                                           '2' valid: '1', '2', '3', '4')

  -spectrum:override_charge                                                Ignores precursor charge and uses 

                                                                           charge state specified in precurso

                                                                           r_charge range (parameters: spectr

                                                                           um:precursor_charge_min and spectr

                                                                           um:precursor_charge_max)

  -spectrum:precursor_charge_min &lt;precursor_charge_min&gt;                    Min charge of precursor charge 

                                                                           range to consider. If specified,

                                                                           also spectrum:override_charge must

                                                                           be set) (default: '1' min: '0')

  -spectrum:precursor_charge_max &lt;precursor_charge_max&gt;                    Max charge of precursor charge 

                                                                           range to consider. If specified,

                                                                           also spectrum:override_charge must

                                                                           be set) (default: '4' min: '0')



Open Search Features:

  -search:track_zero_topn &lt;track_zero_topn&gt;                                Track top N unmodified peptide 

                                                                           results separately from main resul

                                                                           ts internally for boosting feature

                                                                           s. Should be set to a number great

                                                                           er than search:output_report_topN

                                                                           if zero bin boosting is desired

                                                                           (default: '0' min: '0')

  -search:zero_bin_accept_expect &lt;zero_bin_accept_expect&gt;                  Ranks a zero-bin hit above all 

                                                                           non-zero-bin hit if it has expecta

                                                                           tion less than this value (default

                                                                           : '0.0' min: '0.0')

  -search:zero_bin_mult_expect &lt;zero_bin_mult_expect&gt;                      Multiplies expect value of PSMs 

                                                                           in the zero-bin during results

                                                                           ordering (set to less than 1 for

                                                                           boosting) (default: '1.0' min:

                                                                           '0.0')

  -search:add_topn_complementary &lt;add_topn_complementary&gt;                  Inserts complementary ions corresp

                                                                           onding to the top N most intense

                                                                           fragments in each experimental

                                                                           spectrum. Useful for recovery of

                                                                           modified peptides near C-terminus

                                                                           in open search. 0 disables this

                                                                           option (default: '0' min: '0')

  -search:min_fragments_modeling &lt;min_fragments_modeling&gt;                  Minimum number of matched peaks 

                                                                           in PSM for inclusion in statistica

                                                                           l modeling (default: '3' min: '0')

  -search:min_matched_fragments &lt;min_matched_fragments&gt;                    Minimum number of matched peaks 

                                                                           for PSM to be reported. MSFragger

                                                                           recommends a minimum of 4 for narr

                                                                           ow window searching and 6 for open

                                                                           searches (default: '4' min: '0')

  -search:output_report_topn &lt;output_report_topn&gt;                          Reports top N PSMs per input spect

                                                                           rum (default: '1' min: '0')

  -search:output_max_expect &lt;output_max_expect&gt;                            Suppresses reporting of PSM if 

                                                                           top hit has expectation greater

                                                                           than this threshold (default: '50.

                                                                           0' min: '0.0')



Static Modification Parameters:

  -statmod:add_cterm_peptide &lt;add_cterm_peptide&gt;                           Statically add mass in Da to C-ter

                                                                           minal of peptide (default: '0.0'

                                                                           min: '0.0')

  -statmod:add_nterm_peptide &lt;add_nterm_peptide&gt;                           Statically add mass in Da to N-ter

                                                                           minal of peptide (default: '0.0'

                                                                           min: '0.0')

  -statmod:add_cterm_protein &lt;add_cterm_protein&gt;                           Statically add mass in Da to C-ter

                                                                           minal of protein (default: '0.0'

                                                                           min: '0.0')

  -statmod:add_nterm_protein &lt;add_nterm_protein&gt;                           Statically add mass in Da to N-ter

                                                                           minal of protein (default: '0.0'

                                                                           min: '0.0')



                                                                           

Common UTIL options:

  -ini &lt;file&gt;                                                              Use the given TOPP INI file

  -threads &lt;n&gt;                                                             Sets the number of threads allowed

                                                                           to be used by the TOPP tool (defa

                                                                           ult: '1')

  -write_ini &lt;file&gt;                                                        Writes the default configuration 

                                                                           file

  --help                                                                   Shows options

  --helphelp                                                               Shows all options (including advan

                                                                           ced)



</pre><p> <b>INI file documentation of this tool:</b> <div class="ini_global">
<div class="legend">
<b>Legend:</b><br>
 <div class="item item_required">required parameter</div>
 <div class="item item_advanced">advanced parameter</div>
</div>
  <div class="node"><span class="node_name">+MSFraggerAdapter</span><span class="node_description">Peptide Identification with MSFragger</span></div>
    <div class="item item_advanced"><span class="item_name" style="padding-left:16px;">version</span><span class="item_value">2.4.0-HEAD-2019-12-09</span>
<span class="item_description">Version of the tool that generated this parameters file.</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>    <div class="node"><span class="node_name">++1</span><span class="node_description">Instance '1' section for 'MSFraggerAdapter'</span></div>
      <div class="item"><span class="item_name" style="padding-left:24px;">java_executable</span><span class="item_value">java</span>
<span class="item_description">The Java executable. Usually Java is on the system PATH. If Java is not found, use this parameter to specify the full path to Java</span><span class="item_tags">input file</span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">java_heapmemory</span><span class="item_value">3500</span>
<span class="item_description">Maximum Java heap size (in MB)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">executable</span><span class="item_value"></span>
<span class="item_description">Path to the MSFragger executable to use; may be empty if the executable is globally available.</span><span class="item_tags">input file</span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">in</span><span class="item_value"></span>
<span class="item_description">Input File with specta for MSFragger</span><span class="item_tags">input file</span><span class="item_restrictions">*.mzML,*.mzXML</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">out</span><span class="item_value"></span>
<span class="item_description">MSFragger output file</span><span class="item_tags">output file</span><span class="item_restrictions">*.idXML</span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">opt_out</span><span class="item_value"></span>
<span class="item_description">MSFragger optional output file</span><span class="item_tags">output file</span><span class="item_restrictions">*.pepXML</span></div>      <div class="item"><span class="item_name item_required" style="padding-left:24px;">database</span><span class="item_value"></span>
<span class="item_description">Protein FASTA database file path</span><span class="item_tags">input file</span><span class="item_restrictions">*.FASTA,*.fasta,*.fa,*.fas</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">log</span><span class="item_value"></span>
<span class="item_description">Name of log file (created only when specified)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">debug</span><span class="item_value">0</span>
<span class="item_description">Sets the debug level</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item"><span class="item_name" style="padding-left:24px;">threads</span><span class="item_value">1</span>
<span class="item_description">Sets the number of threads allowed to be used by the TOPP tool</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">no_progress</span><span class="item_value">false</span>
<span class="item_description">Disables progress logging to command line</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">force</span><span class="item_value">false</span>
<span class="item_description">Overwrite tool specific checks.</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="item item_advanced"><span class="item_name" style="padding-left:24px;">test</span><span class="item_value">false</span>
<span class="item_description">Enables the test mode (needed for internal use only)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>      <div class="node"><span class="node_name">+++tolerance</span><span class="node_description">Search Tolerances</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">precursor_mass_tolerance</span><span class="item_value">20.0</span>
<span class="item_description">Precursor mass tolerance (window is +/- this value)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">precursor_mass_unit</span><span class="item_value">ppm</span>
<span class="item_description">Unit of precursor mass tolerance</span><span class="item_tags"></span><span class="item_restrictions">Da,ppm</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">precursor_true_tolerance</span><span class="item_value">0.0</span>
<span class="item_description">True precursor mass tolerance (window is +/- this value). Used for tie breaker of results (in spectrally ambiguous cases) and zero bin boosting in open searches (0 disables these features). This option is STRONGLY recommended for open searches.</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">precursor_true_unit</span><span class="item_value">ppm</span>
<span class="item_description">Unit of precursor true tolerance</span><span class="item_tags"></span><span class="item_restrictions">Da,ppm</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">fragment_mass_tolerance</span><span class="item_value">20.0</span>
<span class="item_description">Fragment mass tolerance (window is +/- this value)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">fragment_mass_unit</span><span class="item_value">ppm</span>
<span class="item_description">Unit of fragment mass tolerance</span><span class="item_tags"></span><span class="item_restrictions">Da,ppm</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">isotope_error</span><span class="item_value">0</span>
<span class="item_description">Isotope correction for MS/MS events triggered on isotopic peaks. Should be set to 0 (disabled) for open search or 0/1/2 for correction of narrow window searches. Shifts the precursor mass window to multiples of this value multiplied by the mass of C13-C12.</span><span class="item_tags"></span><span class="item_restrictions">0,1,2</span></div>      <div class="node"><span class="node_name">+++digest</span><span class="node_description">In-Silico Digestion Parameters</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">search_enzyme_name</span><span class="item_value">Trypsin</span>
<span class="item_description">Name of the enzyme to be written to the pepXML file</span><span class="item_tags"></span><span class="item_restrictions">unspecific cleavage,proline-endopeptidase/HKR,Glu-C+P,PepsinA + P,cyanogen-bromide,elastase-trypsin-chymotrypsin,no cleavage,staphylococcal protease/D,Clostripain/P,Asp-N_ambic,TrypChymo,Trypsin/P,Arg-C/P,Alpha-lytic protease,2-iodobenzoate,Chymotrypsin,CNBr,Formic_acid,Lys-C,Asp-N/B,V8-DE,V8-E,leukocyte elastase,Lys-C/P,Trypsin,glutamyl endopeptidase,proline endopeptidase,Asp-N,Lys-N,PepsinA,iodosobenzoate,Arg-C,Chymotrypsin/P</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">search_enzyme_cutafter</span><span class="item_value">KR</span>
<span class="item_description">Residues after which the enzyme cuts (specified as a string of amino acids)</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">search_enzyme_nocutbefore</span><span class="item_value">P</span>
<span class="item_description">Residues that the enzyme will not cut before</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">num_enzyme_termini</span><span class="item_value">fully</span>
<span class="item_description">Number of enzyme termini (non-enzymatic (0), semi (1), fully (2)</span><span class="item_tags"></span><span class="item_restrictions">non-enzymatic,semi,fully</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">allowed_missed_cleavage</span><span class="item_value">2</span>
<span class="item_description">Allowed number of missed cleavages</span><span class="item_tags"></span><span class="item_restrictions">0,1,2,3,4,5</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">min_length</span><span class="item_value">7</span>
<span class="item_description">Minimum length of peptides to be generated during in-silico digestion</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">max_length</span><span class="item_value">64</span>
<span class="item_description">Maximum length of peptides to be generated during in-silico digestion</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">mass_range_min</span><span class="item_value">500.0</span>
<span class="item_description">Min mass of peptides to be generated (Da)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">mass_range_max</span><span class="item_value">5000.0</span>
<span class="item_description">Max mass of peptides to be generated (Da)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>      <div class="node"><span class="node_name">+++varmod</span><span class="node_description">Variable Modification Parameters</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">clip_nterm_m</span><span class="item_value">false</span>
<span class="item_description">Specifies the trimming of a protein N-terminal methionine as a variable modification</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">masses</span><span class="item_value">[]</span>
<span class="item_description">Masses for variable modifications</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">syntaxes</span><span class="item_value">[]</span>
<span class="item_description">Syntax Strings for variable modifications</span><span class="item_tags"></span><span class="item_restrictions"> </span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">enable_common</span><span class="item_value">false</span>
<span class="item_description">Enable common variable modifications (15.9949 M and 42.0106 [^)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">not_allow_multiple_variable_mods_on_residue</span><span class="item_value">false</span>
<span class="item_description">Do not allow any one amino acid to be modified by multiple variable modifications</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">max_variable_mods_per_mod</span><span class="item_value">2</span>
<span class="item_description">Maximum number of residues that can be occupied by each variable modification</span><span class="item_tags"></span><span class="item_restrictions">0,1,2,3,4,5</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">max_variable_mods_combinations</span><span class="item_value">5000</span>
<span class="item_description">Maximum allowed number of modified variably modified peptides from each peptide sequence, (maximum of 65534). If a greater number than the maximum is generated, only the unmodified peptide is considered</span><span class="item_tags"></span><span class="item_restrictions">0:65534</span></div>      <div class="node"><span class="node_name">+++spectrum</span><span class="node_description">Spectrum Processing Parameters</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">minimum_peaks</span><span class="item_value">10</span>
<span class="item_description">Minimum number of peaks in experimental spectrum for matching</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">use_topn_peaks</span><span class="item_value">50</span>
<span class="item_description">Pre-process experimental spectrum to only use top N peaks</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">minimum_ratio</span><span class="item_value">0.0</span>
<span class="item_description">Filters out all peaks in experimental spectrum less intense than this multiple of the base peak intensity</span><span class="item_tags"></span><span class="item_restrictions">0.0:1.0</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">clear_mz_range_min</span><span class="item_value">0.0</span>
<span class="item_description">Removes peaks in this m/z range prior to matching (minimum value). Useful for iTRAQ/TMT experiments (i.e. 0.0 150.0)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">clear_mz_range_max</span><span class="item_value">0.0</span>
<span class="item_description">Removes peaks in this m/z range prior to matching (maximum value). Useful for iTRAQ/TMT experiments (i.e. 0.0 150.0)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">max_fragment_charge</span><span class="item_value">2</span>
<span class="item_description">Maximum charge state for theoretical fragments to match</span><span class="item_tags"></span><span class="item_restrictions">1,2,3,4</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">override_charge</span><span class="item_value">false</span>
<span class="item_description">Ignores precursor charge and uses charge state specified in precursor_charge range (parameters: spectrum:precursor_charge_min and spectrum:precursor_charge_max)</span><span class="item_tags"></span><span class="item_restrictions">true,false</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">precursor_charge_min</span><span class="item_value">1</span>
<span class="item_description">Min charge of precursor charge range to consider. If specified, also spectrum:override_charge must be set)</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">precursor_charge_max</span><span class="item_value">4</span>
<span class="item_description">Max charge of precursor charge range to consider. If specified, also spectrum:override_charge must be set)</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>      <div class="node"><span class="node_name">+++search</span><span class="node_description">Open Search Features</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">track_zero_topn</span><span class="item_value">0</span>
<span class="item_description">Track top N unmodified peptide results separately from main results internally for boosting features. Should be set to a number greater than search:output_report_topN if zero bin boosting is desired</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">zero_bin_accept_expect</span><span class="item_value">0.0</span>
<span class="item_description">Ranks a zero-bin hit above all non-zero-bin hit if it has expectation less than this value</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">zero_bin_mult_expect</span><span class="item_value">1.0</span>
<span class="item_description">Multiplies expect value of PSMs in the zero-bin during results ordering (set to less than 1 for boosting)</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">add_topn_complementary</span><span class="item_value">0</span>
<span class="item_description">Inserts complementary ions corresponding to the top N most intense fragments in each experimental spectrum. Useful for recovery of modified peptides near C-terminus in open search. 0 disables this option</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">min_fragments_modeling</span><span class="item_value">3</span>
<span class="item_description">Minimum number of matched peaks in PSM for inclusion in statistical modeling</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">min_matched_fragments</span><span class="item_value">4</span>
<span class="item_description">Minimum number of matched peaks for PSM to be reported. MSFragger recommends a minimum of 4 for narrow window searching and 6 for open searches</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">output_report_topn</span><span class="item_value">1</span>
<span class="item_description">Reports top N PSMs per input spectrum</span><span class="item_tags"></span><span class="item_restrictions">0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">output_max_expect</span><span class="item_value">50.0</span>
<span class="item_description">Suppresses reporting of PSM if top hit has expectation greater than this threshold</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>      <div class="node"><span class="node_name">+++statmod</span><span class="node_description">Static Modification Parameters</span></div>
        <div class="item"><span class="item_name" style="padding-left:32px;">add_cterm_peptide</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass in Da to C-terminal of peptide</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">add_nterm_peptide</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass in Da to N-terminal of peptide</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">add_cterm_protein</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass in Da to C-terminal of protein</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item"><span class="item_name" style="padding-left:32px;">add_nterm_protein</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass in Da to N-terminal of protein</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_G_glycine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to glycine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_A_alanine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to alanine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_S_serine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to serine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_P_proline</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to proline</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_V_valine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to valine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_T_threonine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to threonine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_C_cysteine</span><span class="item_value">57.021464000000002</span>
<span class="item_description">Statically add mass to cysteine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_L_leucine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to leucine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_I_isoleucine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to isoleucine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_N_asparagine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to asparagine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_D_aspartic_acid</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to aspartic_acid</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_Q_glutamine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to glutamine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_K_lysine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to lysine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_E_glutamic_acid</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to glutamic_acid</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_M_methionine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to methionine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_H_histidine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to histidine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_F_phenylalanine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to phenylalanine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_R_arginine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to arginine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_Y_tyrosine</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to tyrosine</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div>        <div class="item item_advanced"><span class="item_name" style="padding-left:32px;">add_W_tryptophan</span><span class="item_value">0.0</span>
<span class="item_description">Statically add mass to tryptophan</span><span class="item_tags"></span><span class="item_restrictions">0.0:&#8734;</span></div></div>
 </div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:06:29 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
