<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: Class Hierarchy</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">Class Hierarchy</div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock">This inheritance list is sorted roughly, but not completely, alphabetically:</div><div class="directory">
<div class="levels">[detail level <span onclick="javascript:toggleLevel(1);">1</span><span onclick="javascript:toggleLevel(2);">2</span><span onclick="javascript:toggleLevel(3);">3</span><span onclick="javascript:toggleLevel(4);">4</span><span onclick="javascript:toggleLevel(5);">5</span><span onclick="javascript:toggleLevel(6);">6</span>]</div><table class="directory">
<tr id="row_0_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08669.html" target="_self">_Alloc_base&lt; _Tp, _Alloc &gt;</a></td><td class="desc"></td></tr>
<tr id="row_1_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_1_" class="arrow" onclick="toggleFolder('1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08669.html" target="_self">_Alloc_base&lt; _Val, _Alloc &gt;</a></td><td class="desc"></td></tr>
<tr id="row_1_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08689.html" target="_self">KDTree&lt; __K, _Val, _Acc, _Dist, _Cmp, _Alloc &gt;</a></td><td class="desc"></td></tr>
<tr id="row_2_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_2_" class="arrow" onclick="toggleFolder('2_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08669.html" target="_self">_Alloc_base&lt; KDTreeFeatureNode, std::allocator&lt; _Node&lt; KDTreeFeatureNode &gt; &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_2_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08689.html" target="_self">KDTree&lt; 2, KDTreeFeatureNode &gt;</a></td><td class="desc"></td></tr>
<tr id="row_3_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_3_" class="arrow" onclick="toggleFolder('3_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08681.html" target="_self">_Base_iterator</a></td><td class="desc"></td></tr>
<tr id="row_3_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08677.html" target="_self">_Iterator&lt; _Val, _Ref, _Ptr &gt;</a></td><td class="desc"></td></tr>
<tr id="row_4_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08653.html" target="_self">_Bracket_accessor&lt; _Val &gt;</a></td><td class="desc"></td></tr>
<tr id="row_5_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08653.html" target="_self">_Bracket_accessor&lt; KDTreeFeatureNode &gt;</a></td><td class="desc"></td></tr>
<tr id="row_6_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_6_" class="arrow" onclick="toggleFolder('6_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08641.html" target="_self">_Node_base</a></td><td class="desc"></td></tr>
<tr id="row_6_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08645.html" target="_self">_Node&lt; KDTreeFeatureNode &gt;</a></td><td class="desc"></td></tr>
<tr id="row_6_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08645.html" target="_self">_Node&lt; _Val &gt;</a></td><td class="desc"></td></tr>
<tr id="row_7_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08649.html" target="_self">_Node_compare&lt; _Val, _Acc, _Cmp &gt;</a></td><td class="desc"></td></tr>
<tr id="row_8_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08685.html" target="_self">_Region&lt; __K, _Val, _SubVal, _Acc, _Cmp &gt;</a></td><td class="desc"></td></tr>
<tr id="row_9_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06633" target="_self">AAcid_</a></td><td class="desc"></td></tr>
<tr id="row_10_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07861.html" target="_self">AAIndex</a></td><td class="desc">Representation of selected AAIndex properties </td></tr>
<tr id="row_11_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07865.html" target="_self">AASequence</a></td><td class="desc">Representation of a peptide/protein sequence </td></tr>
<tr id="row_12_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07765.html#a07789" target="_self">OPXLDataStructs::AASeqWithMass</a></td><td class="desc">The <a class="el" href="a07765.html#a07789" title="The AASeqWithMass struct represents a normal peptide with its precomputed mass.">AASeqWithMass</a> struct represents a normal peptide with its precomputed mass </td></tr>
<tr id="row_13_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07793.html" target="_self">OPXLDataStructs::AASeqWithMassComparator</a></td><td class="desc">The <a class="el" href="a07793.html" title="The AASeqWithMassComparator is a comparator for AASeqWithMass objects.">AASeqWithMassComparator</a> is a comparator for <a class="el" href="a07765.html#a07789" title="The AASeqWithMass struct represents a normal peptide with its precomputed mass.">AASeqWithMass</a> objects </td></tr>
<tr id="row_14_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08429.html#a08433" target="_self">FuzzyStringComparator::AbortComparison</a></td><td class="desc">Internal exception class </td></tr>
<tr id="row_15_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07389.html" target="_self">AbsoluteQuantitationMethod</a></td><td class="desc"><a class="el" href="a07389.html" title="AbsoluteQuantitationMethod is a class to hold information about the quantitation method and for apply...">AbsoluteQuantitationMethod</a> is a class to hold information about the quantitation method and for applying and/or generating the quantitation method </td></tr>
<tr id="row_16_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10369.html" target="_self">AbsoluteQuantitationStandards</a></td><td class="desc"><a class="el" href="a10369.html" title="AbsoluteQuantitationStandards is a class to handle the relationship between runs, components,...">AbsoluteQuantitationStandards</a> is a class to handle the relationship between runs, components, and their actual concentrations </td></tr>
<tr id="row_17_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09089.html" target="_self">AbsoluteQuantitationStandardsFile</a></td><td class="desc">Load files containing runConcentration data </td></tr>
<tr id="row_18_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06909.html#a06913" target="_self">SiriusMSFile::AccessionInfo</a></td><td class="desc"></td></tr>
<tr id="row_19_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06605.html" target="_self">AccurateMassSearchResult</a></td><td class="desc"></td></tr>
<tr id="row_20_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09281.html" target="_self">AcqusHandler</a></td><td class="desc">Read-only acqus <a class="el" href="a10873.html" title="Basic file handling operations.">File</a> handler for XMass Analysis </td></tr>
<tr id="row_21_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06717" target="_self">AdaptedIdentity</a></td><td class="desc"></td></tr>
<tr id="row_22_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08517.html" target="_self">Adduct</a></td><td class="desc"></td></tr>
<tr id="row_23_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06601.html" target="_self">AdductInfo</a></td><td class="desc"></td></tr>
<tr id="row_24_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09513.html#a09517" target="_self">MSstatsFile::AggregatedConsensusInfo</a></td><td class="desc"></td></tr>
<tr id="row_25_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06673.html" target="_self">AhoCorasickAmbiguous</a></td><td class="desc">Extended Aho-Corasick algorithm capable of matching ambiguous amino acids in the pattern (i.e. proteins) </td></tr>
<tr id="row_26_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08657.html" target="_self">always_true&lt; _Tp &gt;</a></td><td class="desc"></td></tr>
<tr id="row_27_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09753.html" target="_self">PepXMLFile::AminoAcidModification</a></td><td class="desc"></td></tr>
<tr id="row_28_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09313" target="_self">MzIdentMLDOMHandler::AnalysisSoftware</a></td><td class="desc">Struct to hold the used analysis software for that file </td></tr>
<tr id="row_29_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10605.html" target="_self">MSQuantifications::AnalysisSummary</a></td><td class="desc"></td></tr>
<tr id="row_30_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11993.html#a12001" target="_self">NucleicAcidSearchEngine::AnnotatedHit</a></td><td class="desc"></td></tr>
<tr id="row_31_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06897.html" target="_self">SimpleSearchEngineAlgorithm::AnnotatedHit_</a></td><td class="desc">Slimmer structure as storing all scored candidates in <a class="el" href="a10617.html" title="Representation of a peptide hit.">PeptideHit</a> objects takes too much space </td></tr>
<tr id="row_32_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_32_" class="arrow" onclick="toggleFolder('32_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11425.html" target="_self">Annotation1DItem</a></td><td class="desc">An abstract class acting as an interface for the different 1D annotation items </td></tr>
<tr id="row_32_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11417.html" target="_self">Annotation1DCaret</a></td><td class="desc">An annotation item which paints a set of carets on the canvas </td></tr>
<tr id="row_32_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11421.html" target="_self">Annotation1DDistanceItem</a></td><td class="desc">An annotation item which represents a measured distance between two peaks </td></tr>
<tr id="row_32_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11429.html" target="_self">Annotation1DPeakItem</a></td><td class="desc">A peak annotation item </td></tr>
<tr id="row_32_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11433.html" target="_self">Annotation1DTextItem</a></td><td class="desc">An annotation item which represents an arbitrary text on the canvas </td></tr>
<tr id="row_33_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10053.html" target="_self">AnnotationStatistics</a></td><td class="desc"></td></tr>
<tr id="row_34_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10465.html" target="_self">AppliedProcessingStep</a></td><td class="desc">A processing step that was applied to a data item, possibly with associated scores </td></tr>
<tr id="row_35_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10609.html" target="_self">MSQuantifications::Assay</a></td><td class="desc"></td></tr>
<tr id="row_36_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09781.html" target="_self">QcMLFile::Attachment</a></td><td class="desc">Representation of an attachment </td></tr>
<tr id="row_37_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10289.html" target="_self">AveragePosition&lt; D &gt;</a></td><td class="desc">Maintain an average position by summing up positions with weights </td></tr>
<tr id="row_38_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11461.html" target="_self">AxisPainter</a></td><td class="desc">Draws a coordinate axis. It has only static methods, that's why the constructor is private </td></tr>
<tr id="row_39_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11465.html" target="_self">AxisTickCalculator</a></td><td class="desc">Calculates ticks for a given value range </td></tr>
<tr id="row_40_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11305.html" target="_self">BackgroundControl</a></td><td class="desc"></td></tr>
<tr id="row_41_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11309.html" target="_self">BackgroundIntensityBin</a></td><td class="desc"></td></tr>
<tr id="row_42_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09093.html" target="_self">Base64</a></td><td class="desc">Class to encode and decode <a class="el" href="a09093.html" title="Class to encode and decode Base64.">Base64</a> </td></tr>
<tr id="row_43_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; ObjectType &gt;</a></td><td class="desc">A base class for all visualizer classes </td></tr>
<tr id="row_44_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_44_" class="arrow" onclick="toggleFolder('44_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Acquisition &gt;</a></td><td class="desc"></td></tr>
<tr id="row_44_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11765.html" target="_self">AcquisitionVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10381.html" title="Information about one raw data spectrum that was combined with several other raw data spectra.">Acquisition</a> objects </td></tr>
<tr id="row_45_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_45_" class="arrow" onclick="toggleFolder('45_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; AcquisitionInfo &gt;</a></td><td class="desc"></td></tr>
<tr id="row_45_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11761.html" target="_self">AcquisitionInfoVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10385.html" title="Description of the combination of raw data to a single spectrum.">AcquisitionInfo</a> objects </td></tr>
<tr id="row_46_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_46_" class="arrow" onclick="toggleFolder('46_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; ContactPerson &gt;</a></td><td class="desc"></td></tr>
<tr id="row_46_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11777.html" target="_self">ContactPersonVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10393.html" title="Contact person information.">ContactPerson</a> objects </td></tr>
<tr id="row_47_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_47_" class="arrow" onclick="toggleFolder('47_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; DataProcessing &gt;</a></td><td class="desc"></td></tr>
<tr id="row_47_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11781.html" target="_self">DataProcessingVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10425.html" title="Description of the applied preprocessing steps.">DataProcessing</a> objects </td></tr>
<tr id="row_48_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_48_" class="arrow" onclick="toggleFolder('48_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Digestion &gt;</a></td><td class="desc"></td></tr>
<tr id="row_48_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11785.html" target="_self">DigestionVisualizer</a></td><td class="desc">Class that displays all meta information of digestion objects </td></tr>
<tr id="row_49_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_49_" class="arrow" onclick="toggleFolder('49_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; DocumentIdentifier &gt;</a></td><td class="desc"></td></tr>
<tr id="row_49_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11789.html" target="_self">DocumentIdentifierVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10433.html" title="Manage source document information.">DocumentIdentifier</a> objects </td></tr>
<tr id="row_50_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_50_" class="arrow" onclick="toggleFolder('50_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; ExperimentalSettings &gt;</a></td><td class="desc"></td></tr>
<tr id="row_50_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11793.html" target="_self">ExperimentalSettingsVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10453.html" title="Description of the experimental settings.">ExperimentalSettings</a> objects </td></tr>
<tr id="row_51_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_51_" class="arrow" onclick="toggleFolder('51_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Gradient &gt;</a></td><td class="desc"></td></tr>
<tr id="row_51_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11797.html" target="_self">GradientVisualizer</a></td><td class="desc"><a class="el" href="a11797.html" title="GradientVisualizer is a visualizer class for objects of type gradient.">GradientVisualizer</a> is a visualizer class for objects of type gradient </td></tr>
<tr id="row_52_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_52_" class="arrow" onclick="toggleFolder('52_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; HPLC &gt;</a></td><td class="desc"></td></tr>
<tr id="row_52_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11801.html" target="_self">HPLCVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10461.html" title="Representation of a HPLC experiment.">HPLC</a> objects </td></tr>
<tr id="row_53_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_53_" class="arrow" onclick="toggleFolder('53_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Instrument &gt;</a></td><td class="desc"></td></tr>
<tr id="row_53_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11809.html" target="_self">InstrumentVisualizer</a></td><td class="desc">Class that displays all meta information for an MS instrument </td></tr>
<tr id="row_54_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_54_" class="arrow" onclick="toggleFolder('54_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; InstrumentSettings &gt;</a></td><td class="desc"></td></tr>
<tr id="row_54_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11805.html" target="_self">InstrumentSettingsVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10561.html" title="Description of the settings a MS Instrument was run with.">InstrumentSettings</a> objects </td></tr>
<tr id="row_55_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_55_" class="arrow" onclick="toggleFolder('55_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; IonDetector &gt;</a></td><td class="desc"></td></tr>
<tr id="row_55_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11813.html" target="_self">IonDetectorVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10565.html" title="Description of a ion detector (part of a MS Instrument)">IonDetector</a> objects </td></tr>
<tr id="row_56_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_56_" class="arrow" onclick="toggleFolder('56_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; IonSource &gt;</a></td><td class="desc"></td></tr>
<tr id="row_56_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11817.html" target="_self">IonSourceVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10569.html" title="Description of an ion source (part of a MS Instrument)">IonSource</a> objects </td></tr>
<tr id="row_57_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_57_" class="arrow" onclick="toggleFolder('57_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; MassAnalyzer &gt;</a></td><td class="desc"></td></tr>
<tr id="row_57_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11821.html" target="_self">MassAnalyzerVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10573.html" title="Description of a mass analyzer (part of a MS Instrument)">MassAnalyzer</a> objects </td></tr>
<tr id="row_58_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_58_" class="arrow" onclick="toggleFolder('58_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; MetaInfoDescription &gt;</a></td><td class="desc"></td></tr>
<tr id="row_58_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11825.html" target="_self">MetaInfoDescriptionVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10581.html" title="Description of the meta data arrays of MSSpectrum.">MetaInfoDescription</a> objects </td></tr>
<tr id="row_59_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_59_" class="arrow" onclick="toggleFolder('59_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; MetaInfoInterface &gt;</a></td><td class="desc"></td></tr>
<tr id="row_59_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11829.html" target="_self">MetaInfoVisualizer</a></td><td class="desc"><a class="el" href="a11829.html" title="MetaInfoVisualizer is a visualizer class for all classes that use one MetaInfo object as member.">MetaInfoVisualizer</a> is a visualizer class for all classes that use one <a class="el" href="a10577.html" title="A Type-Name-Value tuple class.">MetaInfo</a> object as member </td></tr>
<tr id="row_60_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_60_" class="arrow" onclick="toggleFolder('60_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Modification &gt;</a></td><td class="desc"></td></tr>
<tr id="row_60_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11833.html" target="_self">ModificationVisualizer</a></td><td class="desc">Class that displays all meta information of modification objects </td></tr>
<tr id="row_61_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_61_" class="arrow" onclick="toggleFolder('61_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; PeptideHit &gt;</a></td><td class="desc"></td></tr>
<tr id="row_61_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11837.html" target="_self">PeptideHitVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10617.html" title="Representation of a peptide hit.">PeptideHit</a> objects </td></tr>
<tr id="row_62_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_62_" class="arrow" onclick="toggleFolder('62_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; PeptideIdentification &gt;</a></td><td class="desc"></td></tr>
<tr id="row_62_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11841.html" target="_self">PeptideIdentificationVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10645.html" title="Represents the peptide hits for a spectrum.">PeptideIdentification</a> objects </td></tr>
<tr id="row_63_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_63_" class="arrow" onclick="toggleFolder('63_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Precursor &gt;</a></td><td class="desc"></td></tr>
<tr id="row_63_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11845.html" target="_self">PrecursorVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10649.html" title="Precursor meta information.">Precursor</a> objects </td></tr>
<tr id="row_64_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_64_" class="arrow" onclick="toggleFolder('64_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Product &gt;</a></td><td class="desc"></td></tr>
<tr id="row_64_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11849.html" target="_self">ProductVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10653.html" title="Product meta information.">Product</a> objects </td></tr>
<tr id="row_65_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_65_" class="arrow" onclick="toggleFolder('65_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; ProteinHit &gt;</a></td><td class="desc"></td></tr>
<tr id="row_65_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11853.html" target="_self">ProteinHitVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10657.html" title="Representation of a protein hit.">ProteinHit</a> objects </td></tr>
<tr id="row_66_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_66_" class="arrow" onclick="toggleFolder('66_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; ProteinIdentification &gt;</a></td><td class="desc"></td></tr>
<tr id="row_66_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11857.html" target="_self">ProteinIdentificationVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10677.html" title="Representation of a protein identification run.">ProteinIdentification</a> objects </td></tr>
<tr id="row_67_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_67_" class="arrow" onclick="toggleFolder('67_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Sample &gt;</a></td><td class="desc"></td></tr>
<tr id="row_67_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11861.html" target="_self">SampleVisualizer</a></td><td class="desc">Class that displays all meta information of sample objects </td></tr>
<tr id="row_68_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_68_" class="arrow" onclick="toggleFolder('68_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; ScanWindow &gt;</a></td><td class="desc"></td></tr>
<tr id="row_68_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11865.html" target="_self">ScanWindowVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10697.html" title="Scan window description.">ScanWindow</a> objects </td></tr>
<tr id="row_69_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_69_" class="arrow" onclick="toggleFolder('69_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Software &gt;</a></td><td class="desc"></td></tr>
<tr id="row_69_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11869.html" target="_self">SoftwareVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10701.html" title="Description of the software used for processing.">Software</a> objects </td></tr>
<tr id="row_70_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_70_" class="arrow" onclick="toggleFolder('70_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; SourceFile &gt;</a></td><td class="desc"></td></tr>
<tr id="row_70_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11873.html" target="_self">SourceFileVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10705.html" title="Description of a file location, used to store the origin of (meta) data.">SourceFile</a> objects </td></tr>
<tr id="row_71_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_71_" class="arrow" onclick="toggleFolder('71_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; SpectrumSettings &gt;</a></td><td class="desc"></td></tr>
<tr id="row_71_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11877.html" target="_self">SpectrumSettingsVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10725.html" title="Representation of 1D spectrum settings.">SpectrumSettings</a> objects </td></tr>
<tr id="row_72_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_72_" class="arrow" onclick="toggleFolder('72_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11769.html" target="_self">BaseVisualizer&lt; Tagging &gt;</a></td><td class="desc"></td></tr>
<tr id="row_72_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11881.html" target="_self">TaggingVisualizer</a></td><td class="desc">Class that displays all meta information of tagging objects </td></tr>
<tr id="row_73_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_73_" class="arrow" onclick="toggleFolder('73_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>basic_string&lt; Char &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_73_0_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_73_0_" class="arrow" onclick="toggleFolder('73_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>string</b></td><td class="desc">STL class </td></tr>
<tr id="row_73_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08769.html" target="_self">String</a></td><td class="desc">A more convenient string class </td></tr>
<tr id="row_74_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_74_" class="arrow" onclick="toggleFolder('74_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10293.html" target="_self">BasicStatistics&lt; RealT &gt;</a></td><td class="desc">Calculates some basic statistical parameters of a distribution: sum, mean, variance, and provides the normal approximation </td></tr>
<tr id="row_74_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10285.html" target="_self">AsymmetricStatistics&lt; RealT &gt;</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> class for asymmetric distributions </td></tr>
<tr id="row_75_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10293.html" target="_self">BasicStatistics&lt; CoordinateType &gt;</a></td><td class="desc"></td></tr>
<tr id="row_76_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10217.html" target="_self">BilinearInterpolation&lt; Key, Value &gt;</a></td><td class="desc">Provides access to bilinearly interpolated values (and derivatives) from discrete data points. Values beyond the given range of data points are implicitly taken as zero </td></tr>
<tr id="row_77_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_77_" class="arrow" onclick="toggleFolder('77_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>binary_function</b></td><td class="desc"></td></tr>
<tr id="row_77_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09941.html" target="_self">BaseFeature::QualityLess</a></td><td class="desc">Compare by quality </td></tr>
<tr id="row_77_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09949.html" target="_self">ChromatogramPeak::IntensityLess</a></td><td class="desc">Comparator by intensity </td></tr>
<tr id="row_77_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09957.html" target="_self">ChromatogramPeak::PositionLess</a></td><td class="desc">Comparator by position. As this class has dimension 1, this is basically an alias for <a class="el" href="a09953.html" title="Comparator by RT position.">RTLess</a> </td></tr>
<tr id="row_77_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09953.html" target="_self">ChromatogramPeak::RTLess</a></td><td class="desc">Comparator by RT position </td></tr>
<tr id="row_77_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10013.html" target="_self">ConsensusFeature::MapsLess</a></td><td class="desc">Compare by the sets of consensus elements (lexicographically) </td></tr>
<tr id="row_77_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10009.html" target="_self">ConsensusFeature::SizeLess</a></td><td class="desc">Compare by <a class="el" href="a10005.html#acce0ab2cacc475b2434f24c65c91685a">size()</a>, the number of consensus elements </td></tr>
<tr id="row_77_6_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10001.html" target="_self">EqualInTolerance&lt; CompareType &gt;</a></td><td class="desc">Struct for binary predicate to consider equality with a certain tolerance </td></tr>
<tr id="row_77_7_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10045.html" target="_self">FeatureHandle::IndexLess</a></td><td class="desc">Comparator by map and unique id </td></tr>
<tr id="row_77_8_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09973.html" target="_self">LexicographicComparator&lt; Cmp1, Cmp2 &gt;</a></td><td class="desc">A wrapper class that combines two comparators lexicographically. Normally you should use the make-function <a class="el" href="a09973.html#ac4e734db514973176bfe65e0fa9905ae" title="Make-function to create a LexicographicComparator from two other comparators without the need to spec...">lexicographicComparator()</a> because then you do not need to specify the template arguments </td></tr>
<tr id="row_77_9_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10077.html" target="_self">MSChromatogram::MZLess</a></td><td class="desc">Comparator for the retention time </td></tr>
<tr id="row_77_10_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10101.html" target="_self">MSSpectrum::RTLess</a></td><td class="desc">Comparator for the retention time </td></tr>
<tr id="row_77_11_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08837.html" target="_self">MZTrafoModel::RTLess</a></td><td class="desc">Comparator by position. As this class has dimension 1, this is basically an alias for MZLess </td></tr>
<tr id="row_77_12_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09977.html" target="_self">PairComparatorFirstElement&lt; PairType &gt;</a></td><td class="desc">Class for comparison of std::pair using first ONLY e.g. for use with std::sort </td></tr>
<tr id="row_77_13_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09985.html" target="_self">PairComparatorFirstElementMore&lt; PairType &gt;</a></td><td class="desc">Class for comparison of std::pair using first ONLY e.g. for use with std::sort </td></tr>
<tr id="row_77_14_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09981.html" target="_self">PairComparatorSecondElement&lt; PairType &gt;</a></td><td class="desc">Class for comparison of std::pair using second ONLY e.g. for use with std::sort </td></tr>
<tr id="row_77_15_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09989.html" target="_self">PairComparatorSecondElementMore&lt; PairType &gt;</a></td><td class="desc">Class for comparison of std::pair using second ONLY e.g. for use with std::sort </td></tr>
<tr id="row_77_16_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09993.html" target="_self">PairMatcherFirstElement&lt; PairType &gt;</a></td><td class="desc">Class for comparison of std::pair using first ONLY e.g. for use with std::sort </td></tr>
<tr id="row_77_17_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09997.html" target="_self">PairMatcherSecondElement&lt; PairType &gt;</a></td><td class="desc">Struct for comparison of std::pair using second ONLY e.g. for use with std::sort </td></tr>
<tr id="row_77_18_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10113.html" target="_self">Peak1D::IntensityLess</a></td><td class="desc"></td></tr>
<tr id="row_77_19_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10117.html" target="_self">Peak1D::MZLess</a></td><td class="desc">Comparator by m/z position </td></tr>
<tr id="row_77_20_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10121.html" target="_self">Peak1D::PositionLess</a></td><td class="desc">Comparator by position. As this class has dimension 1, this is basically an alias for <a class="el" href="a10117.html" title="Comparator by m/z position.">MZLess</a> </td></tr>
<tr id="row_77_21_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10129.html" target="_self">Peak2D::IntensityLess</a></td><td class="desc"></td></tr>
<tr id="row_77_22_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10137.html" target="_self">Peak2D::MZLess</a></td><td class="desc">Comparator by m/z position </td></tr>
<tr id="row_77_23_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10141.html" target="_self">Peak2D::PositionLess</a></td><td class="desc">Comparator by position. Lexicographical comparison (first RT then m/z) is done </td></tr>
<tr id="row_77_24_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10133.html" target="_self">Peak2D::RTLess</a></td><td class="desc">Comparator by RT position </td></tr>
<tr id="row_77_25_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09965.html" target="_self">PointerComparator&lt; Cmp &gt;</a></td><td class="desc">Wrapper that takes a comparator for `something' and makes a comparator for <em>pointers</em> to `something' out of it. Normally you should use the make-function <a class="el" href="a09965.html#aeb071107569224c672893cd945dff7fe" title="Make-function to create a PointerComparator from another comparator without the need to specify the t...">pointerComparator()</a> because then you do not need to specify the template arguments </td></tr>
<tr id="row_77_26_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07629.html" target="_self">PrecursorIonSelection::SeqTotalScoreMore</a></td><td class="desc">Compare by score </td></tr>
<tr id="row_77_27_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07625.html" target="_self">PrecursorIonSelection::TotalScoreMore</a></td><td class="desc">Compare by score </td></tr>
<tr id="row_77_28_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07645.html" target="_self">PSLPFormulation::IndexLess</a></td><td class="desc"></td></tr>
<tr id="row_77_29_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07649.html" target="_self">PSLPFormulation::ScanLess</a></td><td class="desc"></td></tr>
<tr id="row_77_30_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07653.html" target="_self">PSLPFormulation::VariableIndexLess</a></td><td class="desc"></td></tr>
<tr id="row_77_31_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07085.html" target="_self">ReactionMonitoringTransition::ProductMZLess</a></td><td class="desc">Comparator by <a class="el" href="a10653.html" title="Product meta information.">Product</a> ion MZ </td></tr>
<tr id="row_77_32_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09969.html" target="_self">ReverseComparator&lt; Cmp &gt;</a></td><td class="desc">Wrapper that reverses (exchanges) the two arguments of a comparator. Normally you should use the make-function <a class="el" href="a09969.html#a74e80d32be63268d8a46039297e18cc6" title="Make-function to create a ReverseComparator from another comparator without the need to specify the t...">reverseComparator()</a> because then you do not need to specify the template arguments </td></tr>
<tr id="row_77_33_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06453.html" target="_self">OSSpectrumMeta::RTLess</a></td><td class="desc">Comparator for the retention time </td></tr>
<tr id="row_77_34_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11965.html" target="_self">RIALess</a></td><td class="desc"></td></tr>
<tr id="row_77_35_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11961.html" target="_self">SequenceLess</a></td><td class="desc"></td></tr>
<tr id="row_77_36_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11957.html" target="_self">SizeLess</a></td><td class="desc"></td></tr>
<tr id="row_78_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09361.html" target="_self">MzMLHandlerHelper::BinaryData</a></td><td class="desc">Representation for binary data in mzML </td></tr>
<tr id="row_79_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05840.html#a09893" target="_self">BinaryDataArray</a></td><td class="desc">The datastructures used by the <a class="el" href="a05820.html">OpenSwath</a> interfaces </td></tr>
<tr id="row_80_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08521.html" target="_self">BinaryTreeNode</a></td><td class="desc">Elements of a binary tree used to represent a hierarchical clustering process </td></tr>
<tr id="row_81_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07917.html" target="_self">EnzymaticDigestionLogModel::BindingSite_</a></td><td class="desc"></td></tr>
<tr id="row_82_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_82_" class="arrow" onclick="toggleFolder('82_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>BinInputStream</b></td><td class="desc"></td></tr>
<tr id="row_82_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09109.html" target="_self">Bzip2InputStream</a></td><td class="desc">Implements the BinInputStream class of the xerces-c library in order to read bzip2 compressed XML files </td></tr>
<tr id="row_82_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09277.html" target="_self">GzipInputStream</a></td><td class="desc">Implements the BinInputStream class of the xerces-c library in order to read gzip compressed XML files </td></tr>
<tr id="row_83_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08221.html" target="_self">BinnedSpectrum</a></td><td class="desc">This is a binned representation of a PeakSpectrum </td></tr>
<tr id="row_84_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06641.html" target="_self">BitsPerValue&lt; AAcid &gt;</a></td><td class="desc"></td></tr>
<tr id="row_85_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10977.html#a10981" target="_self">FeatureFinderAlgorithmIsotopeWavelet::BoxElement</a></td><td class="desc">Internally used data structure for the sweep line algorithm </td></tr>
<tr id="row_86_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11093.html#a11097" target="_self">IsotopeWaveletTransform&lt; PeakType &gt;::BoxElement</a></td><td class="desc">Internally used data structure </td></tr>
<tr id="row_87_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05844.html#a10221" target="_self">BSpline&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_88_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10225.html" target="_self">BSpline2d</a></td><td class="desc">B spline interpolation </td></tr>
<tr id="row_89_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05844.html#a05844" target="_self">BSpline&lt; double &gt;</a></td><td class="desc"></td></tr>
<tr id="row_90_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09105.html" target="_self">Bzip2Ifstream</a></td><td class="desc">Decompresses files which are compressed in the bzip2 format (*.bz2) </td></tr>
<tr id="row_91_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_91_" class="arrow" onclick="toggleFolder('91_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09113.html" target="_self">CachedmzML</a></td><td class="desc">An class that uses on-disk caching to read and write spectra and chromatograms </td></tr>
<tr id="row_91_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07141.html" target="_self">SpectrumAccessOpenMSCached</a></td><td class="desc">An implementation of the Spectrum Access interface using on-disk caching </td></tr>
<tr id="row_92_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08525.html" target="_self">CalibrationData</a></td><td class="desc">A helper class, holding all calibration points </td></tr>
<tr id="row_93_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11313.html" target="_self">CentroidData</a></td><td class="desc"></td></tr>
<tr id="row_94_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_94_" class="arrow" onclick="toggleFolder('94_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11317.html" target="_self">CentroidPeak</a></td><td class="desc"></td></tr>
<tr id="row_94_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11321.html" target="_self">DeconvPeak</a></td><td class="desc"></td></tr>
<tr id="row_95_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07429.html#a07433" target="_self">ItraqConstants::ChannelInfo</a></td><td class="desc">Stores information on an iTRAQ channel </td></tr>
<tr id="row_96_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08529.html" target="_self">ChargePair</a></td><td class="desc">Representation of a (putative) link between two Features, which stem from the same compound but have different charge (including different adduct ions (H+, Na+, ..) </td></tr>
<tr id="row_97_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09901.html" target="_self">Chromatogram</a></td><td class="desc">A single chromatogram </td></tr>
<tr id="row_98_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09345.html#a09353" target="_self">MzMLHandler::ChromatogramData</a></td><td class="desc">Data necessary to generate a single chromatogram </td></tr>
<tr id="row_99_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09897.html" target="_self">ChromatogramMeta</a></td><td class="desc">Identifying information for a chromatogram </td></tr>
<tr id="row_100_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09945.html" target="_self">ChromatogramPeak</a></td><td class="desc">A 1-dimensional raw data point or peak for chromatograms </td></tr>
<tr id="row_101_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09961.html" target="_self">ChromatogramTools</a></td><td class="desc">Conversion class to convert chromatograms </td></tr>
<tr id="row_102_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09117.html" target="_self">ChromeleonFile</a></td><td class="desc">Load Chromeleon <a class="el" href="a10461.html" title="Representation of a HPLC experiment.">HPLC</a> text file and save it into a `MSExperiment` </td></tr>
<tr id="row_103_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a07285" target="_self">ChromExtractParams</a></td><td class="desc"><a class="el" href="a07089.html" title="The ChromatogramExtractor extracts chromatograms from a spectra file.">ChromatogramExtractor</a> parameters </td></tr>
<tr id="row_104_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07841.html" target="_self">Citation</a></td><td class="desc">Stores Citations for individual TOPP tools </td></tr>
<tr id="row_105_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07921.html" target="_self">EnzymaticDigestionLogModel::CleavageModel_</a></td><td class="desc"></td></tr>
<tr id="row_106_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07777.html" target="_self">OPXLDataStructs::CLSMScoreComparator</a></td><td class="desc">Comparator to sort CrossLinkSpectrumMatches by the main score </td></tr>
<tr id="row_107_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08153.html" target="_self">ClusterAnalyzer</a></td><td class="desc">Bundles analyzing tools for a clustering (given as sequence of <a class="el" href="a08521.html" title="Elements of a binary tree used to represent a hierarchical clustering process.">BinaryTreeNode</a>'s) </td></tr>
<tr id="row_108_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_108_" class="arrow" onclick="toggleFolder('108_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08157.html" target="_self">ClusterFunctor</a></td><td class="desc">Base class for cluster functors </td></tr>
<tr id="row_108_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08149.html" target="_self">AverageLinkage</a></td><td class="desc"><a class="el" href="a08149.html" title="AverageLinkage ClusterMethod.">AverageLinkage</a> ClusterMethod </td></tr>
<tr id="row_108_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08177.html" target="_self">CompleteLinkage</a></td><td class="desc"><a class="el" href="a08177.html" title="CompleteLinkage ClusterMethod.">CompleteLinkage</a> ClusterMethod </td></tr>
<tr id="row_108_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08209.html" target="_self">SingleLinkage</a></td><td class="desc"><a class="el" href="a08209.html" title="SingleLinkage ClusterMethod.">SingleLinkage</a> ClusterMethod </td></tr>
<tr id="row_109_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08165.html" target="_self">ClusterHierarchical</a></td><td class="desc">Hierarchical clustering with generic clustering functions </td></tr>
<tr id="row_110_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08173.html" target="_self">ClusteringGrid</a></td><td class="desc">Data structure to store 2D data to be clustered e.g. (m/z, retention time) coordinates from multiplex filtering </td></tr>
<tr id="row_111_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06953.html" target="_self">ClusterProxyKD</a></td><td class="desc">Proxy for a (potential) cluster </td></tr>
<tr id="row_112_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08873.html" target="_self">CmpHypothesesByScore</a></td><td class="desc"></td></tr>
<tr id="row_113_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08869.html" target="_self">CmpMassTraceByMZ</a></td><td class="desc"></td></tr>
<tr id="row_114_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_114_" class="arrow" onclick="toggleFolder('114_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07353.html" target="_self">TargetedSpectraExtractor::Comparator</a></td><td class="desc"></td></tr>
<tr id="row_114_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07357.html" target="_self">TargetedSpectraExtractor::BinnedSpectrumComparator</a></td><td class="desc"></td></tr>
<tr id="row_115_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07001.html" target="_self">MapAlignmentAlgorithmSpectrumAlignment::Compare</a></td><td class="desc">Inner class necessary for using the sort algorithm </td></tr>
<tr id="row_116_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06617.html" target="_self">AccurateMassSearchEngine::CompareEntryAndMass_</a></td><td class="desc"></td></tr>
<tr id="row_117_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06649" target="_self">CompareTypeImpl&lt; AAcid, char &gt;</a></td><td class="desc"></td></tr>
<tr id="row_118_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06645" target="_self">CompareTypeImpl&lt; AAcid, uint8_t &gt;</a></td><td class="desc"></td></tr>
<tr id="row_119_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08533.html" target="_self">Compomer</a></td><td class="desc">Holds information on an edge connecting two features from a (putative) charge ladder </td></tr>
<tr id="row_120_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07209.html#a07221" target="_self">MRMFeatureQC::ComponentGroupPairQCs</a></td><td class="desc">Quality Controls (QCs) for multiple components (between or within component_groups) </td></tr>
<tr id="row_121_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07197.html#a07205" target="_self">MRMFeaturePicker::ComponentGroupParams</a></td><td class="desc">Structure to contain information about a component group with its parameters </td></tr>
<tr id="row_122_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07209.html#a07217" target="_self">MRMFeatureQC::ComponentGroupQCs</a></td><td class="desc">Quality Controls (QCs) within a component group </td></tr>
<tr id="row_123_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07197.html#a07201" target="_self">MRMFeaturePicker::ComponentParams</a></td><td class="desc">Structure to contain information about a single component with its parameters </td></tr>
<tr id="row_124_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07209.html#a07213" target="_self">MRMFeatureQC::ComponentQCs</a></td><td class="desc">Quality Controls (QCs) for individual components </td></tr>
<tr id="row_125_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06909.html#a06917" target="_self">SiriusMSFile::CompoundInfo</a></td><td class="desc"></td></tr>
<tr id="row_126_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07605.html#a07609" target="_self">MetaboTargetedAssay::CompoundSpectrumPair</a></td><td class="desc"><a class="el" href="a07605.html#a07609" title="CompoundSpectrumPair stores a pair of CompoundInfo and MSSpectrum.">CompoundSpectrumPair</a> stores a pair of CompoundInfo and <a class="el" href="a10097.html" title="The representation of a 1D spectrum.">MSSpectrum</a> </td></tr>
<tr id="row_127_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06861.html" target="_self">ConnectedComponent</a></td><td class="desc"></td></tr>
<tr id="row_128_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11329.html" target="_self">ConsensusIsotopePattern</a></td><td class="desc"></td></tr>
<tr id="row_129_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06929.html" target="_self">ConsensusMapNormalizerAlgorithmMedian</a></td><td class="desc">Algorithms of ConsensusMapNormalizer </td></tr>
<tr id="row_130_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06933.html" target="_self">ConsensusMapNormalizerAlgorithmQuantile</a></td><td class="desc">Algorithms of ConsensusMapNormalizer </td></tr>
<tr id="row_131_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06937.html" target="_self">ConsensusMapNormalizerAlgorithmThreshold</a></td><td class="desc">Algorithms of ConsensusMapNormalizer </td></tr>
<tr id="row_132_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07821.html" target="_self">ConsoleUtils</a></td><td class="desc"></td></tr>
<tr id="row_133_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08041.html" target="_self">NASequence::ConstIterator</a></td><td class="desc"><a class="el" href="a08041.html" title="ConstIterator of NASequence class.">ConstIterator</a> of <a class="el" href="a08037.html" title="Representation of a nucleic acid sequence.">NASequence</a> class </td></tr>
<tr id="row_134_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07869.html" target="_self">AASequence::ConstIterator</a></td><td class="desc"><a class="el" href="a07869.html" title="ConstIterator for AASequence.">ConstIterator</a> for <a class="el" href="a07865.html" title="Representation of a peptide/protein sequence.">AASequence</a> </td></tr>
<tr id="row_135_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08537.html" target="_self">ConstRefVector&lt; ContainerT &gt;</a></td><td class="desc">This vector holds pointer to the elements of another container </td></tr>
<tr id="row_136_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_136_" class="arrow" onclick="toggleFolder('136_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08541.html" target="_self">ConstRefVector&lt; ContainerT &gt;::ConstRefVectorConstIterator&lt; ValueT &gt;</a></td><td class="desc">ConstIterator for the <a class="el" href="a08537.html" title="This vector holds pointer to the elements of another container.">ConstRefVector</a> </td></tr>
<tr id="row_136_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08545.html" target="_self">ConstRefVector&lt; ContainerT &gt;::ConstRefVectorIterator&lt; ValueT &gt;</a></td><td class="desc">Mutable iterator for the <a class="el" href="a08537.html" title="This vector holds pointer to the elements of another container.">ConstRefVector</a> </td></tr>
<tr id="row_137_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10085.html" target="_self">MSExperiment::ContainerAdd_&lt; ContainerValueType, addMassTraces &gt;</a></td><td class="desc">Helper class to add either general data points in set2DData or use mass traces from meta values </td></tr>
<tr id="row_138_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10089.html" target="_self">MSExperiment::ContainerAdd_&lt; ContainerValueType, false &gt;</a></td><td class="desc"></td></tr>
<tr id="row_139_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10093.html" target="_self">MSExperiment::ContainerAdd_&lt; ContainerValueType, true &gt;</a></td><td class="desc"></td></tr>
<tr id="row_140_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10849.html#a10853" target="_self">RawMSSignalSimulation::ContaminantInfo</a></td><td class="desc"></td></tr>
<tr id="row_141_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10733.html#a10737" target="_self">Contaminants::ContaminantsSummary</a></td><td class="desc">Structure for storing results </td></tr>
<tr id="row_142_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_142_" class="arrow" onclick="toggleFolder('142_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11205.html" target="_self">ContinuousWaveletTransform</a></td><td class="desc">This class is the base class of the continuous wavelet transformation </td></tr>
<tr id="row_142_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11209.html" target="_self">ContinuousWaveletTransformNumIntegration</a></td><td class="desc">This class computes the continuous wavelet transformation using a marr wavelet </td></tr>
<tr id="row_143_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09129.html" target="_self">ControlledVocabulary</a></td><td class="desc">Representation of a controlled vocabulary </td></tr>
<tr id="row_144_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08549.html" target="_self">ConvexHull2D</a></td><td class="desc">A 2-dimensional hull representation in [counter]clockwise direction - depending on axis labelling </td></tr>
<tr id="row_145_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07765.html#a07773" target="_self">OPXLDataStructs::CrossLinkSpectrumMatch</a></td><td class="desc">The <a class="el" href="a07765.html#a07773" title="The CrossLinkSpectrumMatch struct represents a PSM between a ProteinProteinCrossLink and a spectrum i...">CrossLinkSpectrumMatch</a> struct represents a PSM between a <a class="el" href="a07769.html" title="The ProteinProteinCrossLink struct represents a cross-link between two peptides in OpenPepXL.">ProteinProteinCrossLink</a> and a spectrum in OpenPepXL </td></tr>
<tr id="row_146_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09145.html#a09149" target="_self">CsiFingerIdMzTabWriter::CsiAdapterHit</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> structure used in SiriusAdapter that is used for the conversion of the Csi:FingerID output to an mzTab </td></tr>
<tr id="row_147_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09145.html#a09153" target="_self">CsiFingerIdMzTabWriter::CsiAdapterIdentification</a></td><td class="desc"></td></tr>
<tr id="row_148_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09145.html#a09157" target="_self">CsiFingerIdMzTabWriter::CsiAdapterRun</a></td><td class="desc"></td></tr>
<tr id="row_149_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09145.html" target="_self">CsiFingerIdMzTabWriter</a></td><td class="desc"></td></tr>
<tr id="row_150_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10229.html" target="_self">CubicSpline2d</a></td><td class="desc">Cubic spline interpolation as described in R.L. Burden, J.D. Faires, Numerical Analysis, 4th ed. PWS-Kent, 1989, ISBN 0-53491-585-X, pp. 126-131 </td></tr>
<tr id="row_151_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07669.html" target="_self">CV</a></td><td class="desc"></td></tr>
<tr id="row_152_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08553.html" target="_self">CVMappingRule</a></td><td class="desc">Representation of a CV Mapping rule used by <a class="el" href="a08557.html" title="Representation of controlled vocabulary mapping rules (for PSI formats)">CVMappings</a> </td></tr>
<tr id="row_153_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08557.html" target="_self">CVMappings</a></td><td class="desc">Representation of controlled vocabulary mapping rules (for PSI formats) </td></tr>
<tr id="row_154_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08561.html" target="_self">CVMappingTerm</a></td><td class="desc">Representation of controlled vocabulary term </td></tr>
<tr id="row_155_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08565.html" target="_self">CVReference</a></td><td class="desc">Controlled Vocabulary Reference </td></tr>
<tr id="row_156_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10397.html" target="_self">CVTerm</a></td><td class="desc">Representation of controlled vocabulary term </td></tr>
<tr id="row_157_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09133.html" target="_self">ControlledVocabulary::CVTerm</a></td><td class="desc">Representation of a CV term </td></tr>
<tr id="row_158_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09853.html#a09857" target="_self">SemanticValidator::CVTerm</a></td><td class="desc">Representation of a parsed CV term </td></tr>
<tr id="row_159_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10801.html#a10805" target="_self">EGHFitter1D::Data</a></td><td class="desc">Helper struct (contains the size of an area and a raw data container) </td></tr>
<tr id="row_160_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10937.html#a10941" target="_self">EmgFitter1D::Data</a></td><td class="desc">Helper struct (contains the size of an area and a raw data container) </td></tr>
<tr id="row_161_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11217.html#a11221" target="_self">OptimizePeakDeconvolution::Data</a></td><td class="desc">Class containing the data needed for optimization </td></tr>
<tr id="row_162_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11229.html#a11233" target="_self">OptimizePick::Data</a></td><td class="desc"></td></tr>
<tr id="row_163_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11285.html#a11289" target="_self">TwoDOptimization::Data</a></td><td class="desc">Helper struct (contains the size of an area and a raw data container) </td></tr>
<tr id="row_164_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09337" target="_self">MzIdentMLDOMHandler::DatabaseInput</a></td><td class="desc">Struct to hold the information from the <a class="el" href="a09309.html#a09337" title="Struct to hold the information from the DatabaseInput xml tag.">DatabaseInput</a> xml tag </td></tr>
<tr id="row_165_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08853.html" target="_self">DataFilters::DataFilter</a></td><td class="desc">Representation of a peak/feature filter combining FilterType, FilterOperation and a value </td></tr>
<tr id="row_166_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08849.html" target="_self">DataFilters</a></td><td class="desc"><a class="el" href="a08853.html" title="Representation of a peak/feature filter combining FilterType, FilterOperation and a value.">DataFilter</a> array providing some convenience functions </td></tr>
<tr id="row_167_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07053.html" target="_self">TransformationModel::DataPoint</a></td><td class="desc">Coordinate pair (with optional annotation) </td></tr>
<tr id="row_168_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08569.html" target="_self">DataValue</a></td><td class="desc">Class to hold strings, numeric values, lists of strings and lists of numeric values </td></tr>
<tr id="row_169_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08741.html" target="_self">DaTrait</a></td><td class="desc"></td></tr>
<tr id="row_170_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09321" target="_self">MzIdentMLDOMHandler::DBSequence</a></td><td class="desc">Struct to hold the information from the <a class="el" href="a09309.html#a09321" title="Struct to hold the information from the DBSequence xml tag.">DBSequence</a> xml tag </td></tr>
<tr id="row_171_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08621.html" target="_self">DecoyHelper</a></td><td class="desc">Helper class for calculcations on decoy proteins </td></tr>
<tr id="row_172_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_172_" class="arrow" onclick="toggleFolder('172_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>default_dfs_visitor</b></td><td class="desc"></td></tr>
<tr id="row_172_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06765.html" target="_self">IDBoostGraph::dfs_ccsplit_visitor</a></td><td class="desc">A boost dfs visitor that copies connected components into a vector of graphs </td></tr>
<tr id="row_173_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_173_" class="arrow" onclick="toggleFolder('173_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>DefaultHandler</b></td><td class="desc"></td></tr>
<tr id="row_173_0_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_173_0_" class="arrow" onclick="toggleFolder('173_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09417.html" target="_self">XMLHandler</a></td><td class="desc">Base class for XML handlers </td></tr>
<tr id="row_173_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09125.html" target="_self">ConsensusXMLFile</a></td><td class="desc">This class provides Input functionality for ConsensusMaps and Output functionality for alignments and quantitation </td></tr>
<tr id="row_173_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09141.html" target="_self">CVMappingFile</a></td><td class="desc">Used to load CvMapping files </td></tr>
<tr id="row_173_0_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09261.html" target="_self">FeatureXMLFile</a></td><td class="desc">This class provides Input/Output functionality for feature maps </td></tr>
<tr id="row_173_0_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09437.html" target="_self">IdXMLFile</a></td><td class="desc">Used to load and store idXML files </td></tr>
<tr id="row_173_0_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09301.html" target="_self">MascotXMLHandler</a></td><td class="desc">Handler that is used for parsing MascotXML data </td></tr>
<tr id="row_173_0_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09305.html" target="_self">MzDataHandler</a></td><td class="desc"></td></tr>
<tr id="row_173_0_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09341.html" target="_self">MzIdentMLHandler</a></td><td class="desc">XML STREAM handler for <a class="el" href="a09533.html" title="File adapter for MzIdentML files.">MzIdentMLFile</a> </td></tr>
<tr id="row_173_0_7_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_173_0_7_" class="arrow" onclick="toggleFolder('173_0_7_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09345.html" target="_self">MzMLHandler</a></td><td class="desc">Handler for mzML file format </td></tr>
<tr id="row_173_0_7_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span id="arr_173_0_7_0_" class="arrow" onclick="toggleFolder('173_0_7_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09185.html" target="_self">MSDataWritingConsumer</a></td><td class="desc">Consumer class that writes MS data to disk using the mzML format </td></tr>
<tr id="row_173_0_7_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:80px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09193.html" target="_self">NoopMSDataWritingConsumer</a></td><td class="desc">Consumer class that perform no operation </td></tr>
<tr id="row_173_0_7_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:80px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09189.html" target="_self">PlainMSDataWritingConsumer</a></td><td class="desc">Consumer class that writes MS data to disk using the mzML format </td></tr>
<tr id="row_173_0_8_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09377.html" target="_self">MzQuantMLHandler</a></td><td class="desc">XML handler for <a class="el" href="a09541.html" title="File adapter for MzQuantML files.">MzQuantMLFile</a> </td></tr>
<tr id="row_173_0_9_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09381.html" target="_self">MzXMLHandler</a></td><td class="desc"></td></tr>
<tr id="row_173_0_10_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_173_0_10_" class="arrow" onclick="toggleFolder('173_0_10_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09389.html" target="_self">ParamXMLHandler</a></td><td class="desc">XML Handler for <a class="el" href="a08745.html" title="Management and storage of parameters / INI files.">Param</a> files </td></tr>
<tr id="row_173_0_10_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09401.html" target="_self">ToolDescriptionHandler</a></td><td class="desc">XML handler for <a class="el" href="a09821.html" title="File adapter for ToolDescriptor files.">ToolDescriptionFile</a> </td></tr>
<tr id="row_173_0_11_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09397.html" target="_self">PTMXMLHandler</a></td><td class="desc">Handler that is used for parsing PTMXML data </td></tr>
<tr id="row_173_0_12_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_173_0_12_" class="arrow" onclick="toggleFolder('173_0_12_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09853.html" target="_self">SemanticValidator</a></td><td class="desc">Semantically validates XML files using <a class="el" href="a08557.html" title="Representation of controlled vocabulary mapping rules (for PSI formats)">CVMappings</a> and a <a class="el" href="a09129.html" title="Representation of a controlled vocabulary.">ControlledVocabulary</a> </td></tr>
<tr id="row_173_0_12_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09837.html" target="_self">MzDataValidator</a></td><td class="desc">Semantically validates MzXML files </td></tr>
<tr id="row_173_0_12_1_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09841.html" target="_self">MzIdentMLValidator</a></td><td class="desc">Semantically validates MzXML files </td></tr>
<tr id="row_173_0_12_2_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09845.html" target="_self">MzMLValidator</a></td><td class="desc">Semantically validates MzXML files </td></tr>
<tr id="row_173_0_12_3_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09849.html" target="_self">MzQuantMLValidator</a></td><td class="desc">Semantically validates MzQuantML files </td></tr>
<tr id="row_173_0_12_4_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09861.html" target="_self">TraMLValidator</a></td><td class="desc">Semantically validates MzXML files </td></tr>
<tr id="row_173_0_13_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09405.html" target="_self">TraMLHandler</a></td><td class="desc">XML handler for <a class="el" href="a09825.html" title="File adapter for HUPO PSI TraML files.">TraMLFile</a> </td></tr>
<tr id="row_173_0_14_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09409.html" target="_self">UnimodXMLHandler</a></td><td class="desc">Handler that is used for parsing XTandemXML data </td></tr>
<tr id="row_173_0_15_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09425.html" target="_self">XQuestResultXMLHandler</a></td><td class="desc"><a class="el" href="a09417.html" title="Base class for XML handlers.">XMLHandler</a> for the result files of XQuest </td></tr>
<tr id="row_173_0_16_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09717.html" target="_self">OMSSAXMLFile</a></td><td class="desc">Used to load OMSSAXML files </td></tr>
<tr id="row_173_0_17_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09749.html" target="_self">PepXMLFile</a></td><td class="desc">Used to load and store PepXML files </td></tr>
<tr id="row_173_0_18_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09757.html" target="_self">PepXMLFileMascot</a></td><td class="desc">Used to load Mascot PepXML files </td></tr>
<tr id="row_173_0_19_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09765.html" target="_self">ProtXMLFile</a></td><td class="desc">Used to load (storing not supported, yet) ProtXML files </td></tr>
<tr id="row_173_0_20_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09773.html" target="_self">QcMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for QcML files used to load and store QcML files </td></tr>
<tr id="row_173_0_21_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09829.html" target="_self">TransformationXMLFile</a></td><td class="desc">Used to load and store TransformationXML files </td></tr>
<tr id="row_173_0_22_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09885.html" target="_self">XTandemXMLFile</a></td><td class="desc">Used to load XTandemXML files </td></tr>
<tr id="row_174_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_174_" class="arrow" onclick="toggleFolder('174_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08589.html" target="_self">DefaultParamHandler</a></td><td class="desc">A base class for all classes handling default parameters </td></tr>
<tr id="row_174_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_0_" class="arrow" onclick="toggleFolder('174_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10913.html" target="_self">BaseModel&lt; 1 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_174_0_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_174_0_0_" class="arrow" onclick="toggleFolder('174_0_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11073.html" target="_self">InterpolationModel</a></td><td class="desc">Abstract class for 1D-models that are approximated using linear interpolation </td></tr>
<tr id="row_174_0_0_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10921.html" target="_self">BiGaussModel</a></td><td class="desc">BiGaussian distribution approximated using linear interpolation </td></tr>
<tr id="row_174_0_0_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10813.html" target="_self">EGHModel</a></td><td class="desc">Exponential-Gaussian hybrid distribution model for elution profiles </td></tr>
<tr id="row_174_0_0_2_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10949.html" target="_self">EmgModel</a></td><td class="desc">Exponentially modified gaussian distribution model for elution profiles </td></tr>
<tr id="row_174_0_0_3_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10961.html" target="_self">ExtendedIsotopeModel</a></td><td class="desc">Extended isotope distribution approximated using linear interpolation </td></tr>
<tr id="row_174_0_0_4_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11061.html" target="_self">GaussModel</a></td><td class="desc">Normal distribution approximated using linear interpolation </td></tr>
<tr id="row_174_0_0_5_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11081.html" target="_self">IsotopeModel</a></td><td class="desc">Isotope distribution approximated using linear interpolation </td></tr>
<tr id="row_174_1_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_1_" class="arrow" onclick="toggleFolder('174_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10913.html" target="_self">BaseModel&lt; 2 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_174_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11185.html" target="_self">ProductModel&lt; 2 &gt;</a></td><td class="desc">The class template is only implemented for D=2 because we use <a class="el" href="a10125.html" title="A 2-dimensional raw data point or peak.">Peak2D</a> here </td></tr>
<tr id="row_174_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07385.html" target="_self">AbsoluteQuantitation</a></td><td class="desc"><a class="el" href="a07385.html" title="AbsoluteQuantitation is a class to support absolute or relative quantitation for targeted or untarget...">AbsoluteQuantitation</a> is a class to support absolute or relative quantitation for targeted or untargeted quantitation workflows (e.g., Isotope Dilution Mass Spectrometry) </td></tr>
<tr id="row_174_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06609.html" target="_self">AccurateMassSearchEngine</a></td><td class="desc">An algorithm to search for exact mass matches from a spectrum against a database (e.g. HMDB) </td></tr>
<tr id="row_174_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06681.html" target="_self">AScore</a></td><td class="desc">Implementation of the Ascore For a given peptide sequence and its MS/MS spectrum it identifies the most probable phosphorylation-site(s). For each phosphorylation site a probability score is calculated. The algorithm is implemented according to Beausoleil et al. (Nat. Biotechnol. 2006) </td></tr>
<tr id="row_174_5_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_5_" class="arrow" onclick="toggleFolder('174_5_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06921.html" target="_self">BaseGroupFinder</a></td><td class="desc">The base class of all element group finding algorithms </td></tr>
<tr id="row_174_5_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06981.html" target="_self">LabeledPairFinder</a></td><td class="desc">The <a class="el" href="a06981.html" title="The LabeledPairFinder allows the matching of labeled features (features with a fixed distance).">LabeledPairFinder</a> allows the matching of labeled features (features with a fixed distance) </td></tr>
<tr id="row_174_5_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07029.html" target="_self">QTClusterFinder</a></td><td class="desc">A variant of QT clustering for the detection of feature groups </td></tr>
<tr id="row_174_5_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07033.html" target="_self">SimplePairFinder</a></td><td class="desc">This class implements a simple point pair finding algorithm </td></tr>
<tr id="row_174_5_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07037.html" target="_self">StablePairFinder</a></td><td class="desc">This class implements a pair finding algorithm for consensus features </td></tr>
<tr id="row_174_6_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_6_" class="arrow" onclick="toggleFolder('174_6_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10821.html" target="_self">BaseLabeler</a></td><td class="desc">Abstract base class for all kinds of labeling techniques </td></tr>
<tr id="row_174_6_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10825.html" target="_self">ICPLLabeler</a></td><td class="desc">Simulate ICPL experiments </td></tr>
<tr id="row_174_6_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10829.html" target="_self">ITRAQLabeler</a></td><td class="desc">Simulate iTRAQ experiments </td></tr>
<tr id="row_174_6_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10833.html" target="_self">LabelFreeLabeler</a></td><td class="desc">Abstract base class for all kinds of labeling techniques </td></tr>
<tr id="row_174_6_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10837.html" target="_self">O18Labeler</a></td><td class="desc">Simulate O-18 experiments </td></tr>
<tr id="row_174_6_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10841.html" target="_self">SILACLabeler</a></td><td class="desc">Simulate SILAC experiments </td></tr>
<tr id="row_174_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10913.html" target="_self">BaseModel&lt; D &gt;</a></td><td class="desc">Abstract base class for all D-dimensional models </td></tr>
<tr id="row_174_8_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_8_" class="arrow" onclick="toggleFolder('174_8_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06925.html" target="_self">BaseSuperimposer</a></td><td class="desc">The base class of all superimposer algorithms </td></tr>
<tr id="row_174_8_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07021.html" target="_self">PoseClusteringAffineSuperimposer</a></td><td class="desc">A superimposer that uses a voting scheme, also known as pose clustering, to find a good affine transformation </td></tr>
<tr id="row_174_8_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07025.html" target="_self">PoseClusteringShiftSuperimposer</a></td><td class="desc">A superimposer that uses a voting scheme, also known as pose clustering, to find a good shift transformation </td></tr>
<tr id="row_174_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06685.html" target="_self">BasicProteinInferenceAlgorithm</a></td><td class="desc">Algorithm class that implements simple protein inference by aggregation of peptide scores. It has multiple parameter options like the aggregation method, when to distinguish peptidoforms, and if you want to use shared peptides ("use_shared_peptides"). First, the best PSM per spectrum is used, then only the best PSM per peptidoform is aggregated. Peptidoforms can optionally be distinguished via the treat_X_separate parameters: </td></tr>
<tr id="row_174_10_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06689.html" target="_self">BayesianProteinInferenceAlgorithm</a></td><td class="desc">Performs a Bayesian protein inference on Protein/Peptide identifications or <a class="el" href="a10021.html" title="A container for consensus elements.">ConsensusMap</a> (experimental) </td></tr>
<tr id="row_174_11_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08989.html" target="_self">BernNorm</a></td><td class="desc"><a class="el" href="a08989.html" title="BernNorm scales the peaks by ranking them and then scaling them according to rank.">BernNorm</a> scales the peaks by ranking them and then scaling them according to rank </td></tr>
<tr id="row_174_12_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_12_" class="arrow" onclick="toggleFolder('174_12_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08225.html" target="_self">BinnedSpectrumCompareFunctor</a></td><td class="desc">Base class for compare functors of BinnedSpectra </td></tr>
<tr id="row_174_12_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08213.html" target="_self">BinnedSharedPeakCount</a></td><td class="desc">Compare functor scoring the shared peaks for similarity measurement </td></tr>
<tr id="row_174_12_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08217.html" target="_self">BinnedSpectralContrastAngle</a></td><td class="desc">Compare functor scoring the spectral contrast angle for similarity measurement </td></tr>
<tr id="row_174_12_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08229.html" target="_self">BinnedSumAgreeingIntensities</a></td><td class="desc">Sum of agreeing intensities for similarity measurement </td></tr>
<tr id="row_174_13_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_13_" class="arrow" onclick="toggleFolder('174_13_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06553.html" target="_self">CompNovoIdentificationBase</a></td><td class="desc">Run with <a class="el" href="a06553.html" title="run with CompNovoIdentificationBase">CompNovoIdentificationBase</a> </td></tr>
<tr id="row_174_13_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06549.html" target="_self">CompNovoIdentification</a></td><td class="desc">Run with <a class="el" href="a06549.html" title="run with CompNovoIdentification">CompNovoIdentification</a> </td></tr>
<tr id="row_174_13_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06561.html" target="_self">CompNovoIdentificationCID</a></td><td class="desc">Run with <a class="el" href="a06561.html" title="run with CompNovoIdentificationCID">CompNovoIdentificationCID</a> </td></tr>
<tr id="row_174_14_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_14_" class="arrow" onclick="toggleFolder('174_14_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06569.html" target="_self">CompNovoIonScoringBase</a></td><td class="desc">Run with <a class="el" href="a06569.html" title="run with CompNovoIonScoringBase">CompNovoIonScoringBase</a> </td></tr>
<tr id="row_174_14_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06565.html" target="_self">CompNovoIonScoring</a></td><td class="desc">Run with <a class="el" href="a06565.html" title="run with CompNovoIonScoring">CompNovoIonScoring</a> </td></tr>
<tr id="row_174_14_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06577.html" target="_self">CompNovoIonScoringCID</a></td><td class="desc">Run with <a class="el" href="a06577.html" title="run with CompNovoIonScoringCID">CompNovoIonScoringCID</a> </td></tr>
<tr id="row_174_15_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_15_" class="arrow" onclick="toggleFolder('174_15_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06693.html" target="_self">ConsensusIDAlgorithm</a></td><td class="desc">Abstract base class for all ConsensusID algorithms (that calculate a consensus from multiple ID runs) </td></tr>
<tr id="row_174_15_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_174_15_0_" class="arrow" onclick="toggleFolder('174_15_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06705.html" target="_self">ConsensusIDAlgorithmIdentity</a></td><td class="desc">Abstract base class for ConsensusID algorithms that compare only identical sequences </td></tr>
<tr id="row_174_15_0_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06697.html" target="_self">ConsensusIDAlgorithmAverage</a></td><td class="desc">Calculates a consensus from multiple ID runs by averaging the search scores </td></tr>
<tr id="row_174_15_0_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06701.html" target="_self">ConsensusIDAlgorithmBest</a></td><td class="desc">Calculates a consensus from multiple ID runs by taking the best search score </td></tr>
<tr id="row_174_15_0_2_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06733.html" target="_self">ConsensusIDAlgorithmRanks</a></td><td class="desc">Calculates a consensus from multiple ID runs based on the ranks of the search hits </td></tr>
<tr id="row_174_15_0_3_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06741.html" target="_self">ConsensusIDAlgorithmWorst</a></td><td class="desc">Calculates a consensus from multiple ID runs by taking the worst search score (conservative approach) </td></tr>
<tr id="row_174_15_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_174_15_1_" class="arrow" onclick="toggleFolder('174_15_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06737.html" target="_self">ConsensusIDAlgorithmSimilarity</a></td><td class="desc">Abstract base class for ConsensusID algorithms that take peptide similarity into account </td></tr>
<tr id="row_174_15_1_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06709.html" target="_self">ConsensusIDAlgorithmPEPIons</a></td><td class="desc">Calculates a consensus from multiple ID runs based on PEPs and shared ions </td></tr>
<tr id="row_174_15_1_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06729.html" target="_self">ConsensusIDAlgorithmPEPMatrix</a></td><td class="desc">Calculates a consensus from multiple ID runs based on PEPs and sequence similarities </td></tr>
<tr id="row_174_16_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06745.html" target="_self">ConsensusMapMergerAlgorithm</a></td><td class="desc">Merges identification data in ConsensusMaps </td></tr>
<tr id="row_174_17_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06581.html" target="_self">DeNovoAlgorithm</a></td><td class="desc">Base class for ion scoring implementation for de novo algorithms </td></tr>
<tr id="row_174_18_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06585.html" target="_self">DeNovoIdentification</a></td><td class="desc">Base class for de novo identification </td></tr>
<tr id="row_174_19_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06589.html" target="_self">DeNovoIonScoring</a></td><td class="desc">Base class for ion scoring implementation for de novo algorithms </td></tr>
<tr id="row_174_20_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06597.html" target="_self">DeNovoPostScoring</a></td><td class="desc">Base class for ion scoring implementation for de novo algorithms </td></tr>
<tr id="row_174_21_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10793.html" target="_self">DetectabilitySimulation</a></td><td class="desc">Simulates peptide detectability </td></tr>
<tr id="row_174_22_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07161.html" target="_self">DiaPrescore</a></td><td class="desc">Scoring of an spectrum given library intensities of a transition group </td></tr>
<tr id="row_174_23_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07165.html" target="_self">DIAScoring</a></td><td class="desc">Scoring of an spectrum at the peak apex of an chromatographic elution peak </td></tr>
<tr id="row_174_24_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10797.html" target="_self">DigestSimulation</a></td><td class="desc">Simulates protein digestion </td></tr>
<tr id="row_174_25_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10933.html" target="_self">ElutionModelFitter</a></td><td class="desc">Helper class for fitting elution models to features </td></tr>
<tr id="row_174_26_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08861.html" target="_self">ElutionPeakDetection</a></td><td class="desc">Extracts chromatographic peaks from a mass trace </td></tr>
<tr id="row_174_27_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10233.html" target="_self">EmgGradientDescent</a></td><td class="desc">Compute the area, background and shape metrics of a peak </td></tr>
<tr id="row_174_28_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06749.html" target="_self">FalseDiscoveryRate</a></td><td class="desc">Calculates false discovery rates (FDR) from identifications </td></tr>
<tr id="row_174_29_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06537.html" target="_self">FeatureDeconvolution</a></td><td class="desc">An algorithm to decharge features (i.e. as found by <a class="el" href="a10965.html" title="The main feature finder class.">FeatureFinder</a>) </td></tr>
<tr id="row_174_30_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06941.html" target="_self">FeatureDistance</a></td><td class="desc">A functor class for the calculation of distances between features or consensus features </td></tr>
<tr id="row_174_31_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_31_" class="arrow" onclick="toggleFolder('174_31_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10973.html" target="_self">FeatureFinderAlgorithm</a></td><td class="desc">Abstract base class for <a class="el" href="a10965.html" title="The main feature finder class.">FeatureFinder</a> algorithms </td></tr>
<tr id="row_174_31_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10977.html" target="_self">FeatureFinderAlgorithmIsotopeWavelet</a></td><td class="desc">Implements the isotope wavelet feature finder </td></tr>
<tr id="row_174_31_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10985.html" target="_self">FeatureFinderAlgorithmMRM</a></td><td class="desc"><a class="el" href="a10973.html" title="Abstract base class for FeatureFinder algorithms.">FeatureFinderAlgorithm</a> for MRM experiments </td></tr>
<tr id="row_174_31_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10989.html" target="_self">FeatureFinderAlgorithmPicked</a></td><td class="desc"><a class="el" href="a10973.html" title="Abstract base class for FeatureFinder algorithms.">FeatureFinderAlgorithm</a> for picked peaks </td></tr>
<tr id="row_174_31_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11297.html" target="_self">FeatureFinderAlgorithmSH</a></td><td class="desc">The Superhirn <a class="el" href="a10973.html" title="Abstract base class for FeatureFinder algorithms.">FeatureFinderAlgorithm</a> </td></tr>
<tr id="row_174_32_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11025.html" target="_self">FeatureFinderIdentificationAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_33_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11049.html" target="_self">FeatureFinderMultiplexAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_34_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08877.html" target="_self">FeatureFindingMetabo</a></td><td class="desc">Method for the assembly of mass traces belonging to the same isotope pattern, i.e., that are compatible in retention times, mass-to-charge ratios, and isotope abundances </td></tr>
<tr id="row_174_35_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_35_" class="arrow" onclick="toggleFolder('174_35_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06949.html" target="_self">FeatureGroupingAlgorithm</a></td><td class="desc">Base class for all feature grouping algorithms </td></tr>
<tr id="row_174_35_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06957.html" target="_self">FeatureGroupingAlgorithmKD</a></td><td class="desc">A feature grouping algorithm for unlabeled data </td></tr>
<tr id="row_174_35_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06961.html" target="_self">FeatureGroupingAlgorithmLabeled</a></td><td class="desc">A map feature grouping algorithm for labeling techniques with two labels </td></tr>
<tr id="row_174_35_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06965.html" target="_self">FeatureGroupingAlgorithmQT</a></td><td class="desc">A feature grouping algorithm for unlabeled data </td></tr>
<tr id="row_174_35_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06969.html" target="_self">FeatureGroupingAlgorithmUnlabeled</a></td><td class="desc">A map feature grouping algorithm for unlabeled data </td></tr>
<tr id="row_174_36_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_36_" class="arrow" onclick="toggleFolder('174_36_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09001.html" target="_self">FilterFunctor</a></td><td class="desc">A <a class="el" href="a09001.html" title="A FilterFunctor extracts some spectrum characteristics for quality assessment.">FilterFunctor</a> extracts some spectrum characteristics for quality assessment </td></tr>
<tr id="row_174_36_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08993.html" target="_self">ComplementFilter</a></td><td class="desc">Total intensity of peak pairs that could result from complementing fragments of charge state 1 </td></tr>
<tr id="row_174_36_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09005.html" target="_self">GoodDiffFilter</a></td><td class="desc"><a class="el" href="a09005.html" title="GoodDiffFilter counts the number ob peak pairs whose m/z difference can be explained by a amino acid ...">GoodDiffFilter</a> counts the number ob peak pairs whose m/z difference can be explained by a amino acid loss </td></tr>
<tr id="row_174_36_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09009.html" target="_self">IntensityBalanceFilter</a></td><td class="desc"><a class="el" href="a09009.html" title="IntensityBalanceFilter divides the m/z-range into ten regions and sums the intensity in these regions...">IntensityBalanceFilter</a> divides the m/z-range into ten regions and sums the intensity in these regions </td></tr>
<tr id="row_174_36_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09013.html" target="_self">IsotopeDiffFilter</a></td><td class="desc"><a class="el" href="a09013.html" title="IsotopeDiffFilter returns total intensity of peak pairs that could result from isotope peaks.">IsotopeDiffFilter</a> returns total intensity of peak pairs that could result from isotope peaks </td></tr>
<tr id="row_174_36_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09033.html" target="_self">NeutralLossDiffFilter</a></td><td class="desc"><a class="el" href="a09033.html" title="NeutralLossDiffFilter returns the total intensity ob peak pairs whose m/z difference can be explained...">NeutralLossDiffFilter</a> returns the total intensity ob peak pairs whose m/z difference can be explained by a neutral loss </td></tr>
<tr id="row_174_36_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09077.html" target="_self">TICFilter</a></td><td class="desc"><a class="el" href="a09077.html" title="TICFilter calculates TIC.">TICFilter</a> calculates <a class="el" href="a10789.html">TIC</a> </td></tr>
<tr id="row_174_37_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_37_" class="arrow" onclick="toggleFolder('174_37_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11053.html" target="_self">Fitter1D</a></td><td class="desc">Abstract base class for all 1D-dimensional model fitter </td></tr>
<tr id="row_174_37_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_174_37_0_" class="arrow" onclick="toggleFolder('174_37_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11105.html" target="_self">LevMarqFitter1D</a></td><td class="desc">Abstract class for 1D-model fitter using Levenberg-Marquardt algorithm for parameter optimization </td></tr>
<tr id="row_174_37_0_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10801.html" target="_self">EGHFitter1D</a></td><td class="desc">Exponential-Gaussian hybrid distribution fitter (1-dim.) using Levenberg-Marquardt algorithm (Eigen implementation) for parameter optimization </td></tr>
<tr id="row_174_37_0_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10937.html" target="_self">EmgFitter1D</a></td><td class="desc">Exponentially modified gaussian distribution fitter (1-dim.) using Levenberg-Marquardt algorithm (Eigen implementation) for parameter optimization </td></tr>
<tr id="row_174_37_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_174_37_1_" class="arrow" onclick="toggleFolder('174_37_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11113.html" target="_self">MaxLikeliFitter1D</a></td><td class="desc">Abstract base class for all 1D-model fitters using maximum likelihood optimization </td></tr>
<tr id="row_174_37_1_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10917.html" target="_self">BiGaussFitter1D</a></td><td class="desc">BiGaussian distribution fitter (1-dim.) approximated using linear interpolation </td></tr>
<tr id="row_174_37_1_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10957.html" target="_self">ExtendedIsotopeFitter1D</a></td><td class="desc">Extended isotope distribution fitter (1-dim.) approximated using linear interpolation </td></tr>
<tr id="row_174_37_1_2_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11057.html" target="_self">GaussFitter1D</a></td><td class="desc">Gaussian distribution fitter (1-dim.) approximated using linear interpolation </td></tr>
<tr id="row_174_37_1_3_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11077.html" target="_self">IsotopeFitter1D</a></td><td class="desc">Isotope distribution fitter (1-dim.) approximated using linear interpolation </td></tr>
<tr id="row_174_38_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08973.html" target="_self">GaussFilter</a></td><td class="desc">This class represents a Gaussian lowpass-filter which works on uniform as well as on non-uniform profile data </td></tr>
<tr id="row_174_39_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06789.html" target="_self">IDDecoyProbability</a></td><td class="desc"><a class="el" href="a06789.html" title="IDDecoyProbability calculates probabilities using decoy approach.">IDDecoyProbability</a> calculates probabilities using decoy approach </td></tr>
<tr id="row_174_40_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06797.html" target="_self">IDMapper</a></td><td class="desc">Annotates an <a class="el" href="a10081.html" title="In-Memory representation of a mass spectrometry experiment.">MSExperiment</a>, <a class="el" href="a10057.html" title="A container for features.">FeatureMap</a> or <a class="el" href="a10021.html" title="A container for consensus elements.">ConsensusMap</a> with peptide identifications </td></tr>
<tr id="row_174_41_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06805.html" target="_self">IDMergerAlgorithm</a></td><td class="desc">Creates a new Protein ID run into which other runs can be inserted. Creates union of protein hits but concatenates PSMs. Checks search engine consistency of all inserted runs. It differs from the IDMerger tool, in that it is an algorithm class and it allows inserting multiple peptide hits per peptide sequence (not only the first occurrence) </td></tr>
<tr id="row_174_42_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06809.html" target="_self">IDRipper</a></td><td class="desc">Ripping protein/peptide identification according their file origin </td></tr>
<tr id="row_174_43_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07593.html" target="_self">InclusionExclusionList</a></td><td class="desc">Provides functionality for writing inclusion or exclusion lists </td></tr>
<tr id="row_174_44_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10817.html" target="_self">IonizationSimulation</a></td><td class="desc">Simulates Protein ionization </td></tr>
<tr id="row_174_45_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07393.html" target="_self">IsobaricChannelExtractor</a></td><td class="desc">Extracts individual channels from MS/MS spectra for isobaric labeling experiments </td></tr>
<tr id="row_174_46_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07409.html" target="_self">IsobaricQuantifier</a></td><td class="desc">Given the extracted channel intensities the <a class="el" href="a07409.html" title="Given the extracted channel intensities the IsobaricQuantifier corrects and normalizes the intensitie...">IsobaricQuantifier</a> corrects and normalizes the intensities for further processing </td></tr>
<tr id="row_174_47_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_47_" class="arrow" onclick="toggleFolder('174_47_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07421.html" target="_self">IsobaricQuantitationMethod</a></td><td class="desc">Abstract base class describing an isobaric quantitation method in terms of the used channels and an isotope correction matrix </td></tr>
<tr id="row_174_47_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07437.html" target="_self">ItraqEightPlexQuantitationMethod</a></td><td class="desc">ITRAQ 8 plex quantitation to be used with the IsobaricQuantitation </td></tr>
<tr id="row_174_47_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07441.html" target="_self">ItraqFourPlexQuantitationMethod</a></td><td class="desc">ITRAQ 4 plex quantitation to be used with the IsobaricQuantitation </td></tr>
<tr id="row_174_47_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07501.html" target="_self">TMTElevenPlexQuantitationMethod</a></td><td class="desc">TMT 11plex quantitation to be used with the IsobaricQuantitation </td></tr>
<tr id="row_174_47_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07505.html" target="_self">TMTSixPlexQuantitationMethod</a></td><td class="desc">TMT 6plex quantitation to be used with the IsobaricQuantitation </td></tr>
<tr id="row_174_47_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07509.html" target="_self">TMTSixteenPlexQuantitationMethod</a></td><td class="desc">TMT 16plex quantitation to be used with the IsobaricQuantitation </td></tr>
<tr id="row_174_47_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07513.html" target="_self">TMTTenPlexQuantitationMethod</a></td><td class="desc">TMT 10plex quantitation to be used with the IsobaricQuantitation </td></tr>
<tr id="row_174_48_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07445.html" target="_self">KDTreeFeatureMaps</a></td><td class="desc">Stores a set of features, together with a 2D tree for fast search </td></tr>
<tr id="row_174_49_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_49_" class="arrow" onclick="toggleFolder('174_49_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09021.html" target="_self">LinearResampler</a></td><td class="desc">Linear Resampling of raw data </td></tr>
<tr id="row_174_49_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09025.html" target="_self">LinearResamplerAlign</a></td><td class="desc">Linear Resampling of raw data with alignment </td></tr>
<tr id="row_174_50_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08981.html" target="_self">LowessSmoothing</a></td><td class="desc">LOWESS (locally weighted scatterplot smoothing) </td></tr>
<tr id="row_174_51_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06985.html" target="_self">MapAlignmentAlgorithmIdentification</a></td><td class="desc">A map alignment algorithm based on peptide identifications from MS2 spectra </td></tr>
<tr id="row_174_52_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06993.html" target="_self">MapAlignmentAlgorithmPoseClustering</a></td><td class="desc">A map alignment algorithm based on pose clustering </td></tr>
<tr id="row_174_53_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06997.html" target="_self">MapAlignmentAlgorithmSpectrumAlignment</a></td><td class="desc">A map alignment algorithm based on spectrum similarity (dynamic programming) </td></tr>
<tr id="row_174_54_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09029.html" target="_self">MarkerMower</a></td><td class="desc"><a class="el" href="a09029.html" title="MarkerMower uses PeakMarker to find peaks, those that are not marked get removed.">MarkerMower</a> uses <a class="el" href="a09053.html" title="PeakMarker marks peaks that seem to fulfill some criterion.">PeakMarker</a> to find peaks, those that are not marked get removed </td></tr>
<tr id="row_174_55_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09461.html" target="_self">MascotGenericFile</a></td><td class="desc">Read/write Mascot generic files (MGF) </td></tr>
<tr id="row_174_56_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09469.html" target="_self">MascotRemoteQuery</a></td><td class="desc">Class which handles the communication between <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> and the Mascot server </td></tr>
<tr id="row_174_57_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08017.html" target="_self">MassDecompositionAlgorithm</a></td><td class="desc">Mass decomposition algorithm, given a mass it suggests possible compositions </td></tr>
<tr id="row_174_58_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07173.html" target="_self">MasstraceCorrelator</a></td><td class="desc">Correlates individual masstraces found in mass spectrometric maps </td></tr>
<tr id="row_174_59_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08885.html" target="_self">MassTraceDetection</a></td><td class="desc">A mass trace extraction method that gathers peaks similar in m/z and moving along retention time </td></tr>
<tr id="row_174_60_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10349.html" target="_self">PosteriorErrorProbabilityModel</a></td><td class="desc">Implements a mixture model of the inverse gumbel and the gauss distribution or a gaussian mixture </td></tr>
<tr id="row_174_61_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06545.html" target="_self">MetaboliteFeatureDeconvolution</a></td><td class="desc">An algorithm to decharge small molecule features (i.e. as found by <a class="el" href="a10965.html" title="The main feature finder class.">FeatureFinder</a>) </td></tr>
<tr id="row_174_62_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06845.html" target="_self">MetaboliteSpectralMatching</a></td><td class="desc"></td></tr>
<tr id="row_174_63_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08821.html" target="_self">MorphologicalFilter</a></td><td class="desc">This class implements baseline filtering operations using methods from mathematical morphology </td></tr>
<tr id="row_174_64_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07185.html" target="_self">MRMDecoy</a></td><td class="desc">This class generates a <a class="el" href="a07661.html" title="A description of a targeted experiment containing precursor and production ions.">TargetedExperiment</a> object with decoys based on a <a class="el" href="a07661.html" title="A description of a targeted experiment containing precursor and production ions.">TargetedExperiment</a> object </td></tr>
<tr id="row_174_65_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07189.html" target="_self">MRMFeatureFilter</a></td><td class="desc">The <a class="el" href="a07189.html" title="The MRMFeatureFilter either flags components and/or transitions that do not pass the QC criteria or f...">MRMFeatureFilter</a> either flags components and/or transitions that do not pass the QC criteria or filters out components and/or transitions that do not pass the QC criteria </td></tr>
<tr id="row_174_66_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07193.html" target="_self">MRMFeatureFinderScoring</a></td><td class="desc">The MRMFeatureFinder finds and scores peaks of transitions that co-elute </td></tr>
<tr id="row_174_67_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07077.html" target="_self">MRMFragmentSelection</a></td><td class="desc">This class can select appropriate fragment ions of an MS/MS spectrum of a peptide </td></tr>
<tr id="row_174_68_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07613.html" target="_self">MRMMapping</a></td><td class="desc">A class to map targeted assays to chromatograms </td></tr>
<tr id="row_174_69_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07253.html" target="_self">MRMTransitionGroupPicker</a></td><td class="desc">The <a class="el" href="a07253.html" title="The MRMTransitionGroupPicker finds peaks in chromatograms that belong to the same precursors.">MRMTransitionGroupPicker</a> finds peaks in chromatograms that belong to the same precursors </td></tr>
<tr id="row_174_70_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09501.html" target="_self">MSPFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MSP files (NIST spectra library) </td></tr>
<tr id="row_174_71_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09505.html" target="_self">MSPGenericFile</a></td><td class="desc">Load MSP text file and save it into an `MSExperiment` </td></tr>
<tr id="row_174_72_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10845.html" target="_self">MSSim</a></td><td class="desc">Central class for simulation of mass spectrometry experiments </td></tr>
<tr id="row_174_73_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11137.html" target="_self">MultiplexDeltaMassesGenerator</a></td><td class="desc">Generates complete list of all possible mass shifts due to isotopic labelling </td></tr>
<tr id="row_174_74_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09041.html" target="_self">NLargest</a></td><td class="desc"><a class="el" href="a09041.html" title="NLargest removes all but the n largest peaks.">NLargest</a> removes all but the n largest peaks </td></tr>
<tr id="row_174_75_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09045.html" target="_self">Normalizer</a></td><td class="desc">Normalizes the peak intensities spectrum-wise </td></tr>
<tr id="row_174_76_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08049.html" target="_self">NucleicAcidSpectrumGenerator</a></td><td class="desc">Generates theoretical spectra for nucleic acid sequences </td></tr>
<tr id="row_174_77_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07617.html" target="_self">OfflinePrecursorIonSelection</a></td><td class="desc">Implements different algorithms for precursor ion selection </td></tr>
<tr id="row_174_78_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07757.html" target="_self">OpenPepXLAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_79_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07761.html" target="_self">OpenPepXLLFAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_80_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11217.html" target="_self">OptimizePeakDeconvolution</a></td><td class="desc">This class provides the deconvolution of peak regions using non-linear optimization </td></tr>
<tr id="row_174_81_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09049.html" target="_self">ParentPeakMower</a></td><td class="desc"><a class="el" href="a09049.html" title="ParentPeakMower gets rid of high peaks that could stem from unfragmented precursor ions.">ParentPeakMower</a> gets rid of high peaks that could stem from unfragmented precursor ions </td></tr>
<tr id="row_174_82_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07305.html" target="_self">PeakIntegrator</a></td><td class="desc">Compute the area, background and shape metrics of a peak </td></tr>
<tr id="row_174_83_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_83_" class="arrow" onclick="toggleFolder('174_83_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09053.html" target="_self">PeakMarker</a></td><td class="desc"><a class="el" href="a09053.html" title="PeakMarker marks peaks that seem to fulfill some criterion.">PeakMarker</a> marks peaks that seem to fulfill some criterion </td></tr>
<tr id="row_174_83_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08997.html" target="_self">ComplementMarker</a></td><td class="desc"><a class="el" href="a08997.html" title="ComplementMarker marks peak pairs which could represent y - b ion pairs.">ComplementMarker</a> marks peak pairs which could represent y - b ion pairs </td></tr>
<tr id="row_174_83_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09017.html" target="_self">IsotopeMarker</a></td><td class="desc"><a class="el" href="a09017.html" title="IsotopeMarker marks peak pairs which could represent an ion and its isotope.">IsotopeMarker</a> marks peak pairs which could represent an ion and its isotope </td></tr>
<tr id="row_174_83_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09037.html" target="_self">NeutralLossMarker</a></td><td class="desc"><a class="el" href="a09037.html" title="NeutralLossMarker marks peak pairs which could represent an ion an its neutral loss (water,...">NeutralLossMarker</a> marks peak pairs which could represent an ion an its neutral loss (water, ammonia) </td></tr>
<tr id="row_174_84_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11241.html" target="_self">PeakPickerCWT</a></td><td class="desc">This class implements a peak picking algorithm using wavelet techniques </td></tr>
<tr id="row_174_85_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11249.html" target="_self">PeakPickerHiRes</a></td><td class="desc">This class implements a fast peak-picking algorithm best suited for high resolution MS data (FT-ICR-MS, Orbitrap). In high resolution data, the signals of ions with similar mass-to-charge ratios (m/z) exhibit little or no overlapping and therefore allow for a clear separation. Furthermore, ion signals tend to show well-defined peak shapes with narrow peak width </td></tr>
<tr id="row_174_86_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11261.html" target="_self">PeakPickerIterative</a></td><td class="desc">This class implements a peak-picking algorithm for high-resolution MS data (specifically designed for TOF-MS data) </td></tr>
<tr id="row_174_87_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07321.html" target="_self">PeakPickerMRM</a></td><td class="desc">The <a class="el" href="a07321.html" title="The PeakPickerMRM finds peaks a single chromatogram.">PeakPickerMRM</a> finds peaks a single chromatogram </td></tr>
<tr id="row_174_88_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11273.html" target="_self">PeakPickerSH</a></td><td class="desc"></td></tr>
<tr id="row_174_89_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_89_" class="arrow" onclick="toggleFolder('174_89_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08237.html" target="_self">PeakSpectrumCompareFunctor</a></td><td class="desc">Base class for compare functors of spectra, that return a similarity value for two spectra </td></tr>
<tr id="row_174_89_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08233.html" target="_self">PeakAlignment</a></td><td class="desc">Make a <a class="el" href="a08233.html" title="make a PeakAlignment of two PeakSpectra">PeakAlignment</a> of two PeakSpectra </td></tr>
<tr id="row_174_89_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08241.html" target="_self">SpectraSTSimilarityScore</a></td><td class="desc">Similarity score of SpectraST </td></tr>
<tr id="row_174_89_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08249.html" target="_self">SpectrumAlignmentScore</a></td><td class="desc">Similarity score via spectra alignment </td></tr>
<tr id="row_174_89_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08253.html" target="_self">SpectrumCheapDPCorr</a></td><td class="desc"><a class="el" href="a08253.html" title="SpectrumCheapDPCorr calculates an optimal alignment on stick spectra.">SpectrumCheapDPCorr</a> calculates an optimal alignment on stick spectra </td></tr>
<tr id="row_174_89_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08257.html" target="_self">SpectrumPrecursorComparator</a></td><td class="desc"><a class="el" href="a08257.html" title="SpectrumPrecursorComparator compares just the parent mass of two spectra.">SpectrumPrecursorComparator</a> compares just the parent mass of two spectra </td></tr>
<tr id="row_174_89_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08261.html" target="_self">SteinScottImproveScore</a></td><td class="desc">Similarity score based of Stein &amp; Scott </td></tr>
<tr id="row_174_89_6_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08265.html" target="_self">ZhangSimilarityScore</a></td><td class="desc">Similarity score of Zhang </td></tr>
<tr id="row_174_90_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07453.html" target="_self">PeptideAndProteinQuant</a></td><td class="desc">Helper class for peptide and protein quantification based on feature data annotated with IDs </td></tr>
<tr id="row_174_91_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06849.html" target="_self">PeptideIndexing</a></td><td class="desc">Refreshes the protein references for all peptide hits in a vector of PeptideIdentifications and adds target/decoy information </td></tr>
<tr id="row_174_92_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07621.html" target="_self">PrecursorIonSelection</a></td><td class="desc">This class implements different precursor ion selection strategies </td></tr>
<tr id="row_174_93_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07633.html" target="_self">PrecursorIonSelectionPreprocessing</a></td><td class="desc">This class implements the database preprocessing needing for precursor ion selection </td></tr>
<tr id="row_174_94_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07473.html" target="_self">ProteinResolver</a></td><td class="desc">Helper class for peptide and protein quantification based on feature data annotated with IDs </td></tr>
<tr id="row_174_95_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06889.html" target="_self">ProtonDistributionModel</a></td><td class="desc">A proton distribution model to calculate the proton distribution over charged peptides </td></tr>
<tr id="row_174_96_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07637.html" target="_self">PSLPFormulation</a></td><td class="desc">Implements ILP formulation of precursor selection problems </td></tr>
<tr id="row_174_97_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07497.html" target="_self">QuantitativeExperimentalDesign</a></td><td class="desc">Merge files according to experimental design </td></tr>
<tr id="row_174_98_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10849.html" target="_self">RawMSSignalSimulation</a></td><td class="desc">Simulates MS signals for a given set of peptides </td></tr>
<tr id="row_174_99_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10857.html" target="_self">RawTandemMSSignalSimulation</a></td><td class="desc">Simulates tandem MS signals for a given set of peptides </td></tr>
<tr id="row_174_100_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10861.html" target="_self">RTSimulation</a></td><td class="desc">Simulates/Predicts retention times for peptides or peptide separation </td></tr>
<tr id="row_174_101_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08985.html" target="_self">SavitzkyGolayFilter</a></td><td class="desc">Computes the Savitzky-Golay filter coefficients using QR decomposition </td></tr>
<tr id="row_174_102_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09057.html" target="_self">Scaler</a></td><td class="desc"><a class="el" href="a09057.html" title="Scaler scales the peak by ranking the peaks and assigning intensity according to rank.">Scaler</a> scales the peak by ranking the peaks and assigning intensity according to rank </td></tr>
<tr id="row_174_103_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_103_" class="arrow" onclick="toggleFolder('174_103_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html" target="_self">SignalToNoiseEstimator&lt; Container &gt;</a></td><td class="desc">This class represents the abstract base class of a signal to noise estimator </td></tr>
<tr id="row_174_103_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08957.html" target="_self">SignalToNoiseEstimatorMeanIterative&lt; Container &gt;</a></td><td class="desc">Estimates the signal/noise (S/N) ratio of each data point in a scan based on an iterative scheme which discards high intensities </td></tr>
<tr id="row_174_103_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08961.html" target="_self">SignalToNoiseEstimatorMedian&lt; Container &gt;</a></td><td class="desc">Estimates the signal/noise (S/N) ratio of each data point in a scan by using the median (histogram based) </td></tr>
<tr id="row_174_104_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06893.html" target="_self">SimpleSearchEngineAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_105_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07573.html" target="_self">SimpleSVM</a></td><td class="desc">Simple interface to support vector machines for classification (via LIBSVM) </td></tr>
<tr id="row_174_106_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08089.html" target="_self">SimpleTSGXLMS</a></td><td class="desc">Generates theoretical spectra for cross-linked peptides </td></tr>
<tr id="row_174_107_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06901.html" target="_self">SiriusAdapterAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_108_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07325.html" target="_self">SONARScoring</a></td><td class="desc">Scoring of an spectrum using SONAR data </td></tr>
<tr id="row_174_109_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11633.html" target="_self">SpectraIdentificationViewWidget</a></td><td class="desc">Tabular visualization / selection of identified spectra </td></tr>
<tr id="row_174_110_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09061.html" target="_self">SpectraMerger</a></td><td class="desc">Merges blocks of MS or MS2 spectra </td></tr>
<tr id="row_174_111_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09065.html" target="_self">SpectraMerger::SpectraDistance_</a></td><td class="desc"></td></tr>
<tr id="row_174_112_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08245.html" target="_self">SpectrumAlignment</a></td><td class="desc">Aligns the peaks of two sorted spectra Method 1: Using a banded (width via 'tolerance' parameter) alignment if absolute tolerances are given. Scoring function is the m/z distance between peaks. Intensity does not play a role! </td></tr>
<tr id="row_174_113_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08105.html" target="_self">SpectrumAnnotator</a></td><td class="desc">Annotates spectra from identifications and theoretical spectra or identifications from spectra and theoretical spectra matching with various options </td></tr>
<tr id="row_174_114_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_114_" class="arrow" onclick="toggleFolder('174_114_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11669.html" target="_self">SpectrumCanvas</a></td><td class="desc">Base class for visualization canvas classes </td></tr>
<tr id="row_174_114_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11641.html" target="_self">Spectrum1DCanvas</a></td><td class="desc">Canvas for visualization of one or several spectra </td></tr>
<tr id="row_174_114_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11649.html" target="_self">Spectrum2DCanvas</a></td><td class="desc">Canvas for 2D-visualization of peak map, feature map and consensus map data </td></tr>
<tr id="row_174_114_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11657.html" target="_self">Spectrum3DCanvas</a></td><td class="desc">Canvas for 3D-visualization of peak map data </td></tr>
<tr id="row_174_115_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09069.html" target="_self">SqrtMower</a></td><td class="desc">Scales the intensity of peaks to the sqrt </td></tr>
<tr id="row_174_116_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08109.html" target="_self">SvmTheoreticalSpectrumGenerator</a></td><td class="desc">Simulates MS2 spectra with support vector machines </td></tr>
<tr id="row_174_117_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08129.html" target="_self">SvmTheoreticalSpectrumGeneratorTrainer</a></td><td class="desc">Train SVM models that are used by <a class="el" href="a08109.html" title="Simulates MS2 spectra with support vector machines.">SvmTheoreticalSpectrumGenerator</a> </td></tr>
<tr id="row_174_118_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07333.html" target="_self">SwathMapMassCorrection</a></td><td class="desc">A class containing correction functions for Swath MS maps </td></tr>
<tr id="row_174_119_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07345.html" target="_self">TargetedSpectraExtractor</a></td><td class="desc">This class filters, annotates, picks, and scores spectra (e.g., taken from a DDA experiment) based on a target list </td></tr>
<tr id="row_174_120_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08133.html" target="_self">TheoreticalSpectrumGenerator</a></td><td class="desc">Generates theoretical spectra for peptides with various options </td></tr>
<tr id="row_174_121_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08137.html" target="_self">TheoreticalSpectrumGeneratorXLMS</a></td><td class="desc">Generates theoretical spectra for cross-linked peptides </td></tr>
<tr id="row_174_122_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09073.html" target="_self">ThresholdMower</a></td><td class="desc"><a class="el" href="a09073.html" title="ThresholdMower removes all peaks below a threshold.">ThresholdMower</a> removes all peaks below a threshold </td></tr>
<tr id="row_174_123_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08845.html" target="_self">TOFCalibration</a></td><td class="desc">This class implements an external calibration for TOF data using external calibrant spectra </td></tr>
<tr id="row_174_124_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11449.html" target="_self">TOPPASBase</a></td><td class="desc">Main window of the TOPPAS tool </td></tr>
<tr id="row_174_125_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11453.html" target="_self">TOPPViewBase</a></td><td class="desc">Main window of TOPPView tool </td></tr>
<tr id="row_174_126_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_126_" class="arrow" onclick="toggleFolder('174_126_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11193.html" target="_self">TraceFitter</a></td><td class="desc">Abstract fitter for RT profile fitting </td></tr>
<tr id="row_174_126_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10925.html" target="_self">EGHTraceFitter</a></td><td class="desc">A RT Profile fitter using an Exponential Gaussian Hybrid background model </td></tr>
<tr id="row_174_126_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11065.html" target="_self">GaussTraceFitter</a></td><td class="desc">Fitter for RT profiles using a Gaussian background model </td></tr>
<tr id="row_174_127_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_127_" class="arrow" onclick="toggleFolder('174_127_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07365.html" target="_self">TransitionTSVFile</a></td><td class="desc">This class supports reading and writing of OpenSWATH transition lists </td></tr>
<tr id="row_174_127_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07361.html" target="_self">TransitionPQPFile</a></td><td class="desc">This class supports reading and writing of PQP files </td></tr>
<tr id="row_174_128_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11285.html" target="_self">TwoDOptimization</a></td><td class="desc">This class provides the two-dimensional optimization of the picked peak parameters </td></tr>
<tr id="row_174_129_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09081.html" target="_self">WindowMower</a></td><td class="desc"><a class="el" href="a09081.html" title="WindowMower augments the highest peaks in a sliding or jumping window.">WindowMower</a> augments the highest peaks in a sliding or jumping window </td></tr>
<tr id="row_174_130_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07813.html" target="_self">XFDRAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_174_131_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_131_" class="arrow" onclick="toggleFolder('174_131_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html" target="_self">SignalToNoiseEstimator&lt; ContainerT &gt;</a></td><td class="desc"></td></tr>
<tr id="row_174_131_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08961.html" target="_self">SignalToNoiseEstimatorMedian&lt; ContainerT &gt;</a></td><td class="desc"></td></tr>
<tr id="row_174_132_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_174_132_" class="arrow" onclick="toggleFolder('174_132_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html" target="_self">SignalToNoiseEstimator&lt; OpenMS::MSChromatogram &gt;</a></td><td class="desc"></td></tr>
<tr id="row_174_132_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08961.html" target="_self">SignalToNoiseEstimatorMedian&lt; OpenMS::MSChromatogram &gt;</a></td><td class="desc"></td></tr>
<tr id="row_175_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08857.html" target="_self">Deisotoper</a></td><td class="desc"></td></tr>
<tr id="row_176_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11133.html" target="_self">MultiplexDeltaMasses::DeltaMass</a></td><td class="desc">Mass shift with corresponding label set </td></tr>
<tr id="row_177_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08109.html#a08117" target="_self">SvmTheoreticalSpectrumGenerator::DescriptorSet</a></td><td class="desc">A set of descriptors for a single training row </td></tr>
<tr id="row_178_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_178_" class="arrow" onclick="toggleFolder('178_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07881.html" target="_self">DigestionEnzyme</a></td><td class="desc">Abstract base class for digestion enzymes </td></tr>
<tr id="row_178_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07889.html" target="_self">DigestionEnzymeProtein</a></td><td class="desc">Representation of a digestion enzyme for proteins (protease) </td></tr>
<tr id="row_178_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07893.html" target="_self">DigestionEnzymeRNA</a></td><td class="desc">Representation of a digestion enzyme for RNA (RNase) </td></tr>
<tr id="row_179_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07885.html" target="_self">DigestionEnzymeDB&lt; DigestionEnzymeType, InstanceType &gt;</a></td><td class="desc"><a class="el" href="a10429.html" title="Meta information about digestion of a sample.">Digestion</a> enzyme database (base class) </td></tr>
<tr id="row_180_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_180_" class="arrow" onclick="toggleFolder('180_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07885.html" target="_self">DigestionEnzymeDB&lt; DigestionEnzymeProtein, ProteaseDB &gt;</a></td><td class="desc"></td></tr>
<tr id="row_180_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08053.html" target="_self">ProteaseDB</a></td><td class="desc">Database for enzymes that digest proteins (proteases) </td></tr>
<tr id="row_181_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_181_" class="arrow" onclick="toggleFolder('181_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07885.html" target="_self">DigestionEnzymeDB&lt; DigestionEnzymeRNA, RNaseDB &gt;</a></td><td class="desc"></td></tr>
<tr id="row_181_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08081.html" target="_self">RNaseDB</a></td><td class="desc">Database for enzymes that digest RNA (RNases) </td></tr>
<tr id="row_182_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08941.html" target="_self">IDFilter::DigestionFilter</a></td><td class="desc">Is peptide evidence digestion product of some protein </td></tr>
<tr id="row_183_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_183_" class="arrow" onclick="toggleFolder('183_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08593.html" target="_self">DIntervalBase&lt; D &gt;</a></td><td class="desc">A base class for D-dimensional interval </td></tr>
<tr id="row_183_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08585.html" target="_self">DBoundingBox&lt; 2 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_183_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08605.html" target="_self">DRange&lt; 1 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_183_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08605.html" target="_self">DRange&lt; 2 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_183_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08605.html" target="_self">DRange&lt; 3 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_183_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08585.html" target="_self">DBoundingBox&lt; D &gt;</a></td><td class="desc">A D-dimensional bounding box </td></tr>
<tr id="row_183_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08605.html" target="_self">DRange&lt; D &gt;</a></td><td class="desc">A D-dimensional half-open interval </td></tr>
<tr id="row_184_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08597.html" target="_self">DistanceMatrix&lt; Value &gt;</a></td><td class="desc">A two-dimensional distance matrix, similar to <a class="el" href="a07417.html" title="A two-dimensional matrix. Similar to std::vector, but uses a binary operator(,) for element access.">OpenMS::Matrix</a> </td></tr>
<tr id="row_185_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06945.html" target="_self">FeatureDistance::DistanceParams_</a></td><td class="desc">Structure for storing distance parameters </td></tr>
<tr id="row_186_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_186_" class="arrow" onclick="toggleFolder('186_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10433.html" target="_self">DocumentIdentifier</a></td><td class="desc">Manage source document information </td></tr>
<tr id="row_186_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10021.html" target="_self">ConsensusMap</a></td><td class="desc">A container for consensus elements </td></tr>
<tr id="row_186_1_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_186_1_" class="arrow" onclick="toggleFolder('186_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10453.html" target="_self">ExperimentalSettings</a></td><td class="desc">Description of the experimental settings </td></tr>
<tr id="row_186_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10081.html" target="_self">MSExperiment</a></td><td class="desc">In-Memory representation of a mass spectrometry experiment </td></tr>
<tr id="row_186_1_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10601.html" target="_self">MSQuantifications</a></td><td class="desc"></td></tr>
<tr id="row_186_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10057.html" target="_self">FeatureMap</a></td><td class="desc">A container for features </td></tr>
<tr id="row_187_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10437.html" target="_self">DocumentIDTagger</a></td><td class="desc">Tags <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> file containers with a DocumentID </td></tr>
<tr id="row_188_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08697.html" target="_self">ListUtils::DoubleTolerancePredicate_</a></td><td class="desc">Predicate to check double equality with a given tolerance </td></tr>
<tr id="row_189_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a10033" target="_self">DPeak&lt; dimensions &gt;</a></td><td class="desc">Metafunction to choose among <a class="el" href="a10109.html" title="A 1-dimensional raw data point or peak.">Peak1D</a> respectively <a class="el" href="a10125.html" title="A 2-dimensional raw data point or peak.">Peak2D</a> through a template argument </td></tr>
<tr id="row_190_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08601.html" target="_self">DPosition&lt; D, TCoordinateType &gt;</a></td><td class="desc">Representation of a coordinate in D-dimensional space </td></tr>
<tr id="row_191_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08601.html" target="_self">DPosition&lt; 1 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_192_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08601.html" target="_self">DPosition&lt; 2 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_193_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08601.html" target="_self">DPosition&lt; 2, Int64 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_194_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08601.html" target="_self">DPosition&lt; D &gt;</a></td><td class="desc"></td></tr>
<tr id="row_195_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08601.html" target="_self">DPosition&lt; DIMENSION &gt;</a></td><td class="desc"></td></tr>
<tr id="row_196_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09241.html" target="_self">DTAFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for DTA files </td></tr>
<tr id="row_197_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09245.html" target="_self">EDTAFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for Enhanced DTA files </td></tr>
<tr id="row_198_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07897.html" target="_self">Element</a></td><td class="desc">Representation of an element </td></tr>
<tr id="row_199_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07901.html" target="_self">ElementDB</a></td><td class="desc">Stores elements </td></tr>
<tr id="row_200_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10237.html" target="_self">EmgGradientDescent_friend</a></td><td class="desc"></td></tr>
<tr id="row_201_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10953.html" target="_self">EmgScoring</a></td><td class="desc">Scoring of an elution peak using an exponentially modified gaussian distribution model </td></tr>
<tr id="row_202_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07905.html" target="_self">EmpiricalFormula</a></td><td class="desc">Representation of an empirical formula </td></tr>
<tr id="row_203_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_203_" class="arrow" onclick="toggleFolder('203_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11569.html" target="_self">EnhancedTabBarWidgetInterface</a></td><td class="desc">Widgets that are placed into an <a class="el" href="a11565.html" title="Convenience tab bar implementation.">EnhancedTabBar</a> must implement this interface </td></tr>
<tr id="row_203_0_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_203_0_" class="arrow" onclick="toggleFolder('203_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11673.html" target="_self">SpectrumWidget</a></td><td class="desc">Base class for spectrum widgets </td></tr>
<tr id="row_203_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11645.html" target="_self">Spectrum1DWidget</a></td><td class="desc">Widget for visualization of several spectra </td></tr>
<tr id="row_203_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11653.html" target="_self">Spectrum2DWidget</a></td><td class="desc">Widget for 2D-visualization of peak map and feature map data </td></tr>
<tr id="row_203_0_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11665.html" target="_self">Spectrum3DWidget</a></td><td class="desc">Widget for 3D-visualization of map data </td></tr>
<tr id="row_203_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11749.html" target="_self">TOPPASWidget</a></td><td class="desc">Widget visualizing and allowing to edit TOPP pipelines </td></tr>
<tr id="row_204_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_204_" class="arrow" onclick="toggleFolder('204_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07909.html" target="_self">EnzymaticDigestion</a></td><td class="desc">Class for the enzymatic digestion of sequences </td></tr>
<tr id="row_204_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08057.html" target="_self">ProteaseDigestion</a></td><td class="desc">Class for the enzymatic digestion of proteins </td></tr>
<tr id="row_204_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08085.html" target="_self">RNaseDigestion</a></td><td class="desc">Class for the enzymatic digestion of RNAs </td></tr>
<tr id="row_205_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07913.html" target="_self">EnzymaticDigestionLogModel</a></td><td class="desc">Class for the Log L model of enzymatic digestion of proteins </td></tr>
<tr id="row_206_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_206_" class="arrow" onclick="toggleFolder('206_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>ErrorHandler</b></td><td class="desc"></td></tr>
<tr id="row_206_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09865.html" target="_self">XMLValidator</a></td><td class="desc">Validator for XML files </td></tr>
<tr id="row_207_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08181.html" target="_self">EuclideanSimilarity</a></td><td class="desc">CompareFunctor for 2Dpoints </td></tr>
<tr id="row_208_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_208_" class="arrow" onclick="toggleFolder('208_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>exception</b></td><td class="desc">STL class </td></tr>
<tr id="row_208_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_208_0_" class="arrow" onclick="toggleFolder('208_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08269.html" target="_self">BaseException</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> base class </td></tr>
<tr id="row_208_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08161.html" target="_self">ClusterFunctor::InsufficientInput</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> thrown if not enough data (&lt;2) is used </td></tr>
<tr id="row_208_0_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08353.html" target="_self">BufferOverflow</a></td><td class="desc">Buffer overflow exception </td></tr>
<tr id="row_208_0_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08321.html" target="_self">ConversionError</a></td><td class="desc">Invalid conversion exception </td></tr>
<tr id="row_208_0_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08417.html" target="_self">DepletedIDPool</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> used if no more unique document ID's can be drawn from ID pool </td></tr>
<tr id="row_208_0_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08357.html" target="_self">DivisionByZero</a></td><td class="desc">Division by zero error exception </td></tr>
<tr id="row_208_0_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08405.html" target="_self">ElementNotFound</a></td><td class="desc"><a class="el" href="a07897.html" title="Representation of an element.">Element</a> could not be found exception </td></tr>
<tr id="row_208_0_6_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08297.html" target="_self">FailedAPICall</a></td><td class="desc">A call to an external library (other than <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a>) went wrong </td></tr>
<tr id="row_208_0_7_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08385.html" target="_self">FileEmpty</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> is empty </td></tr>
<tr id="row_208_0_8_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08377.html" target="_self">FileNameTooLong</a></td><td class="desc">Filename is too long to be writable/readable by the filesystem </td></tr>
<tr id="row_208_0_9_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08365.html" target="_self">FileNotFound</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> not found exception </td></tr>
<tr id="row_208_0_10_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08369.html" target="_self">FileNotReadable</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> not readable exception </td></tr>
<tr id="row_208_0_11_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08373.html" target="_self">FileNotWritable</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> not writable exception </td></tr>
<tr id="row_208_0_12_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08401.html" target="_self">IllegalArgument</a></td><td class="desc">A method or algorithm argument contains illegal values </td></tr>
<tr id="row_208_0_13_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08389.html" target="_self">IllegalPosition</a></td><td class="desc">Invalid 3-dimensional position exception </td></tr>
<tr id="row_208_0_14_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08325.html" target="_self">IllegalSelfOperation</a></td><td class="desc">Illegal self operation exception </td></tr>
<tr id="row_208_0_15_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08345.html" target="_self">IllegalTreeOperation</a></td><td class="desc">Illegal tree operation exception </td></tr>
<tr id="row_208_0_16_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08337.html" target="_self">IncompatibleIterators</a></td><td class="desc">Incompatible iterator exception </td></tr>
<tr id="row_208_0_17_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08293.html" target="_self">IndexOverflow</a></td><td class="desc">Int overflow exception </td></tr>
<tr id="row_208_0_18_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08285.html" target="_self">IndexUnderflow</a></td><td class="desc">Int underflow exception </td></tr>
<tr id="row_208_0_19_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08333.html" target="_self">InvalidIterator</a></td><td class="desc">Invalid iterator exception </td></tr>
<tr id="row_208_0_20_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08317.html" target="_self">InvalidParameter</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> indicating that an invalid parameter was handed over to an algorithm </td></tr>
<tr id="row_208_0_21_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08301.html" target="_self">InvalidRange</a></td><td class="desc">Invalid range exception </td></tr>
<tr id="row_208_0_22_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08305.html" target="_self">InvalidSize</a></td><td class="desc">Invalid UInt exception </td></tr>
<tr id="row_208_0_23_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08313.html" target="_self">InvalidValue</a></td><td class="desc">Invalid value exception </td></tr>
<tr id="row_208_0_24_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08381.html" target="_self">IOException</a></td><td class="desc">General <a class="el" href="a08381.html" title="General IOException.">IOException</a> </td></tr>
<tr id="row_208_0_25_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08281.html" target="_self">MissingInformation</a></td><td class="desc">Not all required information provided </td></tr>
<tr id="row_208_0_26_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08341.html" target="_self">NotImplemented</a></td><td class="desc">Not implemented exception </td></tr>
<tr id="row_208_0_27_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08329.html" target="_self">NullPointer</a></td><td class="desc">Null pointer argument is invalid exception </td></tr>
<tr id="row_208_0_28_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08361.html" target="_self">OutOfGrid</a></td><td class="desc">Out of grid exception </td></tr>
<tr id="row_208_0_29_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08349.html" target="_self">OutOfMemory</a></td><td class="desc">Out of memory exception </td></tr>
<tr id="row_208_0_30_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08309.html" target="_self">OutOfRange</a></td><td class="desc">Out of range exception </td></tr>
<tr id="row_208_0_31_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08393.html" target="_self">ParseError</a></td><td class="desc">Parse Error exception </td></tr>
<tr id="row_208_0_32_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08277.html" target="_self">Postcondition</a></td><td class="desc"><a class="el" href="a08277.html" title="Postcondition failed exception.">Postcondition</a> failed exception </td></tr>
<tr id="row_208_0_33_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08273.html" target="_self">Precondition</a></td><td class="desc"><a class="el" href="a08273.html" title="Precondition failed exception.">Precondition</a> failed exception </td></tr>
<tr id="row_208_0_34_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07853.html" target="_self">RequiredParameterNotGiven</a></td><td class="desc">A required parameter was not given </td></tr>
<tr id="row_208_0_35_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08289.html" target="_self">SizeUnderflow</a></td><td class="desc">UInt underflow exception </td></tr>
<tr id="row_208_0_36_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08413.html" target="_self">UnableToCalibrate</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> used if an error occurred while calibrating a dataset </td></tr>
<tr id="row_208_0_37_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08397.html" target="_self">UnableToCreateFile</a></td><td class="desc">Unable to create file exception </td></tr>
<tr id="row_208_0_38_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08409.html" target="_self">UnableToFit</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> used if an error occurred while fitting a model to a given dataset </td></tr>
<tr id="row_208_0_39_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07845.html" target="_self">UnregisteredParameter</a></td><td class="desc">An unregistered parameter was accessed </td></tr>
<tr id="row_208_0_40_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07849.html" target="_self">WrongParameterType</a></td><td class="desc">A parameter was accessed with the wrong type </td></tr>
<tr id="row_208_0_41_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11021.html" target="_self">FeatureFinderDefs::NoSuccessor</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> that is thrown if a method an invalid IndexPair is given </td></tr>
<tr id="row_208_0_42_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09421.html" target="_self">XMLHandler::EndParsingSoftly</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> that is thrown if the parsing is ended by some event (e.g. if only a prefix of the XML file is needed) </td></tr>
<tr id="row_208_0_43_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08721.html" target="_self">Map&lt; Key, T &gt;::IllegalKey</a></td><td class="desc"><a class="el" href="a08717.html" title="Map class based on the STL map (containing several convenience functions)">Map</a> illegal key exception </td></tr>
<tr id="row_208_0_44_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08169.html" target="_self">UnnormalizedComparator</a></td><td class="desc"><a class="el" href="a05827.html" title="Exception namespace">Exception</a> thrown if clustering is attempted without a normalized compare functor </td></tr>
<tr id="row_208_1_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_208_1_" class="arrow" onclick="toggleFolder('208_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>bad_alloc</b></td><td class="desc">STL class </td></tr>
<tr id="row_208_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08349.html" target="_self">OutOfMemory</a></td><td class="desc">Out of memory exception </td></tr>
<tr id="row_209_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10441.html" target="_self">ExperimentalDesign</a></td><td class="desc">Representation of the Experimental Design in <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a>. Instances can be loaded via the <a class="el" href="a09249.html" title="Provides means to load an ExperimentalDesign from a TSV file.">ExperimentalDesignFile</a> class </td></tr>
<tr id="row_210_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09249.html" target="_self">ExperimentalDesignFile</a></td><td class="desc">Provides means to load an <a class="el" href="a10441.html" title="Representation of the Experimental Design in OpenMS. Instances can be loaded via the ExperimentalDesi...">ExperimentalDesign</a> from a TSV file </td></tr>
<tr id="row_211_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07097.html" target="_self">ChromatogramExtractorAlgorithm::ExtractionCoordinates</a></td><td class="desc"></td></tr>
<tr id="row_212_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_212_" class="arrow" onclick="toggleFolder('212_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08425.html" target="_self">FactoryBase</a></td><td class="desc">Base class for Factory&lt;T&gt; </td></tr>
<tr id="row_212_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08421.html" target="_self">Factory&lt; FactoryProduct &gt;</a></td><td class="desc">Returns FactoryProduct* based on the name of the desired concrete FactoryProduct </td></tr>
<tr id="row_213_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a08609" target="_self">FASTAContainer&lt; TBackend &gt;</a></td><td class="desc">Template parameter for vector-based FASTA access </td></tr>
<tr id="row_214_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08613.html" target="_self">FASTAContainer&lt; TFI_File &gt;</a></td><td class="desc"><a class="el" href="a08613.html" title="FASTAContainer&lt;TFI_File&gt; will make FASTA entries available chunk-wise from start to end by loading it...">FASTAContainer&lt;TFI_File&gt;</a> will make FASTA entries available chunk-wise from start to end by loading it from a FASTA file. This avoids having to load the full file into memory. While loading, the container will memorize the file offsets of each entry, allowing to read an arbitrary i'th entry again from disk. If possible, only entries from the currently cached chunk should be queried, otherwise access will be slow </td></tr>
<tr id="row_215_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08617.html" target="_self">FASTAContainer&lt; TFI_Vector &gt;</a></td><td class="desc"><a class="el" href="a08617.html" title="FASTAContainer&lt;TFI_Vector&gt; simply takes an existing vector of FASTAEntries and provides the same inte...">FASTAContainer&lt;TFI_Vector&gt;</a> simply takes an existing vector of FASTAEntries and provides the same interface with a potentially huge speed benefit over <a class="el" href="a08613.html" title="FASTAContainer&lt;TFI_File&gt; will make FASTA entries available chunk-wise from start to end by loading it...">FASTAContainer&lt;TFI_File&gt;</a> since it does not need disk access, but at the cost of memory </td></tr>
<tr id="row_216_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09257.html" target="_self">FASTAFile::FASTAEntry</a></td><td class="desc">FASTA entry type (identifier, description and sequence) </td></tr>
<tr id="row_217_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09253.html" target="_self">FASTAFile</a></td><td class="desc">This class serves for reading in and writing FASTA files </td></tr>
<tr id="row_218_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11045.html" target="_self">FeatureFinderIdentificationAlgorithm::FeatureCompare</a></td><td class="desc">Comparison functor for features </td></tr>
<tr id="row_219_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10369.html#a10377" target="_self">AbsoluteQuantitationStandards::featureConcentration</a></td><td class="desc">Structure to hold a single component and its corresponding known concentration </td></tr>
<tr id="row_220_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09721.html" target="_self">FeatureFileOptions</a></td><td class="desc">Options for loading files containing features </td></tr>
<tr id="row_221_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11037.html" target="_self">FeatureFinderIdentificationAlgorithm::FeatureFilterPeptides</a></td><td class="desc">Predicate for filtering features by assigned peptides: </td></tr>
<tr id="row_222_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11033.html" target="_self">FeatureFinderIdentificationAlgorithm::FeatureFilterQuality</a></td><td class="desc">Predicate for filtering features by overall quality: </td></tr>
<tr id="row_223_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a10993" target="_self">FeatureFinderAlgorithmPickedHelperStructs</a></td><td class="desc">Wrapper struct for all the classes needed by the <a class="el" href="a10989.html" title="FeatureFinderAlgorithm for picked peaks.">FeatureFinderAlgorithmPicked</a> and the associated classes </td></tr>
<tr id="row_224_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11301.html" target="_self">FeatureFinderAlgorithmSHCtrl</a></td><td class="desc">A facade for various Superhirn <a class="el" href="a10965.html" title="The main feature finder class.">FeatureFinder</a> classes. Use <a class="el" href="a11297.html" title="The Superhirn FeatureFinderAlgorithm.">FeatureFinderAlgorithmSH</a> instead </td></tr>
<tr id="row_225_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_225_" class="arrow" onclick="toggleFolder('225_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11017.html" target="_self">FeatureFinderDefs</a></td><td class="desc">The purpose of this struct is to provide definitions of classes and typedefs which are used throughout all <a class="el" href="a10965.html" title="The main feature finder class.">FeatureFinder</a> classes </td></tr>
<tr id="row_225_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10965.html" target="_self">FeatureFinder</a></td><td class="desc">The main feature finder class </td></tr>
<tr id="row_225_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10985.html" target="_self">FeatureFinderAlgorithmMRM</a></td><td class="desc"><a class="el" href="a10973.html" title="Abstract base class for FeatureFinder algorithms.">FeatureFinderAlgorithm</a> for MRM experiments </td></tr>
<tr id="row_225_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10989.html" target="_self">FeatureFinderAlgorithmPicked</a></td><td class="desc"><a class="el" href="a10973.html" title="Abstract base class for FeatureFinder algorithms.">FeatureFinderAlgorithm</a> for picked peaks </td></tr>
<tr id="row_225_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11297.html" target="_self">FeatureFinderAlgorithmSH</a></td><td class="desc">The Superhirn <a class="el" href="a10973.html" title="Abstract base class for FeatureFinder algorithms.">FeatureFinderAlgorithm</a> </td></tr>
<tr id="row_225_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11053.html" target="_self">Fitter1D</a></td><td class="desc">Abstract base class for all 1D-dimensional model fitter </td></tr>
<tr id="row_226_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08865.html" target="_self">FeatureHypothesis</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> structure used in <a class="el" href="a08877.html">FeatureFindingMetabo</a> that keeps track of a feature hypothesis (isotope group hypothesis) </td></tr>
<tr id="row_227_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11337.html" target="_self">FeatureLCProfile</a></td><td class="desc"></td></tr>
<tr id="row_228_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06973.html" target="_self">FeatureMapping</a></td><td class="desc"></td></tr>
<tr id="row_229_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06973.html#a06977" target="_self">FeatureMapping::FeatureToMs2Indices</a></td><td class="desc"></td></tr>
<tr id="row_230_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11085.html#a11089" target="_self">IsotopeWavelet::fi_</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> union for fast computation of the power function </td></tr>
<tr id="row_231_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10873.html" target="_self">File</a></td><td class="desc">Basic file handling operations </td></tr>
<tr id="row_232_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09265.html" target="_self">FileHandler</a></td><td class="desc">Facilitates file handling by file type recognition </td></tr>
<tr id="row_233_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08797.html" target="_self">FileMapping</a></td><td class="desc">Maps input/output files to filenames for the external program </td></tr>
<tr id="row_234_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09269.html" target="_self">FileTypes</a></td><td class="desc">Centralizes the file types recognized by <a class="el" href="a09265.html" title="Facilitates file handling by file type recognition.">FileHandler</a> </td></tr>
<tr id="row_235_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07721.html" target="_self">FLASHDeconvAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_236_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07725.html" target="_self">FLASHDeconvHelperStructs</a></td><td class="desc">Wrapper struct for all the structs needed by the <a class="el" href="a11925.html">FLASHDeconv</a> </td></tr>
<tr id="row_237_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10741.html#a10745" target="_self">FragmentMassError::FMEStatistics</a></td><td class="desc">Structure for storing results: average and variance of all FragmentMassErrors in ppm </td></tr>
<tr id="row_238_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06857.html" target="_self">PeptideIndexing::FoundProteinFunctor</a></td><td class="desc"></td></tr>
<tr id="row_239_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07545.html" target="_self">RNPxlFragmentAnnotationHelper::FragmentAnnotationDetail_</a></td><td class="desc">Single fragment annotation </td></tr>
<tr id="row_240_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11341.html" target="_self">FTPeakDetectController</a></td><td class="desc"></td></tr>
<tr id="row_241_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10329.html" target="_self">GumbelMaxLikelihoodFitter::Functor&lt; _Scalar, NX, NY &gt;</a></td><td class="desc"></td></tr>
<tr id="row_242_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_242_" class="arrow" onclick="toggleFolder('242_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10329.html" target="_self">GumbelMaxLikelihoodFitter::Functor&lt; double &gt;</a></td><td class="desc"></td></tr>
<tr id="row_242_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10333.html" target="_self">GumbelMaxLikelihoodFitter::GumbelDistributionFunctor</a></td><td class="desc"></td></tr>
<tr id="row_243_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08429.html" target="_self">FuzzyStringComparator</a></td><td class="desc">Fuzzy comparison of strings, tolerates numeric differences </td></tr>
<tr id="row_244_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10301.html" target="_self">GammaDistributionFitter::GammaDistributionFitResult</a></td><td class="desc">Struct to represent the parameters of a gamma distribution </td></tr>
<tr id="row_245_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10297.html" target="_self">GammaDistributionFitter</a></td><td class="desc">Implements a fitter for the Gamma distribution </td></tr>
<tr id="row_246_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08977.html" target="_self">GaussFilterAlgorithm</a></td><td class="desc">This class represents a Gaussian lowpass-filter which works on uniform as well as on non-uniform profile data </td></tr>
<tr id="row_247_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10309.html" target="_self">GaussFitter::GaussFitResult</a></td><td class="desc">Struct of parameters of a Gaussian distribution </td></tr>
<tr id="row_248_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10305.html" target="_self">GaussFitter</a></td><td class="desc">Implements a fitter for Gaussian functions </td></tr>
<tr id="row_249_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html#a08953" target="_self">SignalToNoiseEstimator&lt; Container &gt;::GaussianEstimate</a></td><td class="desc">Protected struct to store parameters my, sigma for a Gaussian distribution </td></tr>
<tr id="row_250_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_250_" class="arrow" onclick="toggleFolder('250_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11109.html" target="_self">LevMarqFitter1D::GenericFunctor</a></td><td class="desc"></td></tr>
<tr id="row_250_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10809.html" target="_self">EGHFitter1D::EGHFitterFunctor</a></td><td class="desc"></td></tr>
<tr id="row_250_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10945.html" target="_self">EmgFitter1D::EgmFitterFunctor</a></td><td class="desc"></td></tr>
<tr id="row_251_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_251_" class="arrow" onclick="toggleFolder('251_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11197.html" target="_self">TraceFitter::GenericFunctor</a></td><td class="desc"></td></tr>
<tr id="row_251_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10929.html" target="_self">EGHTraceFitter::EGHTraceFunctor</a></td><td class="desc"></td></tr>
<tr id="row_251_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11069.html" target="_self">GaussTraceFitter::GaussTraceFunctor</a></td><td class="desc"></td></tr>
<tr id="row_252_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08933.html" target="_self">IDFilter::GetMatchingItems&lt; HitType, Entry &gt;</a></td><td class="desc">Builds a map index of data that have a <a class="el" href="a08769.html" title="A more convenient string class.">String</a> index to find matches and return the objects </td></tr>
<tr id="row_253_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08933.html" target="_self">IDFilter::GetMatchingItems&lt; OpenMS::PeptideEvidence, OpenMS::FASTAFile::FASTAEntry &gt;</a></td><td class="desc"></td></tr>
<tr id="row_254_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08449.html" target="_self">GlobalExceptionHandler</a></td><td class="desc"><a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> global exception handler </td></tr>
<tr id="row_255_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a03626.html#a08705" target="_self">glp_prob</a></td><td class="desc"></td></tr>
<tr id="row_256_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10457.html" target="_self">Gradient</a></td><td class="desc">Representation of a <a class="el" href="a10461.html" title="Representation of a HPLC experiment.">HPLC</a> gradient </td></tr>
<tr id="row_257_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08185.html" target="_self">GridBasedCluster</a></td><td class="desc">Basic data structure for clustering </td></tr>
<tr id="row_258_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08629.html" target="_self">GridFeature</a></td><td class="desc">Representation of a feature in a hash grid </td></tr>
<tr id="row_259_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10253.html" target="_self">GridSearch&lt; TupleTypes &gt;</a></td><td class="desc"></td></tr>
<tr id="row_260_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11605.html" target="_self">GUIHelpers</a></td><td class="desc">Class which holds static GUI-related helper functions </td></tr>
<tr id="row_261_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10317.html" target="_self">GumbelDistributionFitter::GumbelDistributionFitResult</a></td><td class="desc">Struct to represent the parameters of a gumbel distribution </td></tr>
<tr id="row_262_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10325.html" target="_self">GumbelMaxLikelihoodFitter::GumbelDistributionFitResult</a></td><td class="desc">Struct to represent the parameters of a gumbel distribution </td></tr>
<tr id="row_263_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10313.html" target="_self">GumbelDistributionFitter</a></td><td class="desc">Implements a fitter for the Gumbel distribution </td></tr>
<tr id="row_264_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10321.html" target="_self">GumbelMaxLikelihoodFitter</a></td><td class="desc">Implements a fitter for the Gumbel distribution </td></tr>
<tr id="row_265_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09273.html" target="_self">GzipIfstream</a></td><td class="desc">Decompresses files which are compressed in the gzip format (*.gzip) </td></tr>
<tr id="row_266_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08921.html" target="_self">IDFilter::HasDecoyAnnotation&lt; HitType &gt;</a></td><td class="desc">Is this a decoy hit? </td></tr>
<tr id="row_267_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08905.html" target="_self">IDFilter::HasGoodScore&lt; HitType &gt;</a></td><td class="desc">Is the score of this hit at least as good as the given value? </td></tr>
<tr id="row_268_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08777.html" target="_self">hash&lt; OpenMS::String &gt;</a></td><td class="desc"></td></tr>
<tr id="row_269_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08197.html" target="_self">HashGrid&lt; Cluster &gt;</a></td><td class="desc">Container for (2-dimensional coordinate, value) pairs </td></tr>
<tr id="row_270_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a12005.html" target="_self">NucleicAcidSearchEngine::HasInvalidLength</a></td><td class="desc"></td></tr>
<tr id="row_271_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08929.html" target="_self">IDFilter::HasMatchingAccession&lt; HitType &gt;</a></td><td class="desc">Given a list of protein accessions, do any occur in the annotation(s) of this hit? </td></tr>
<tr id="row_272_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08925.html" target="_self">IDFilter::HasMatchingAccessionUnordered&lt; HitType &gt;</a></td><td class="desc">Given a list of protein accessions, do any occur in the annotation(s) of this hit? </td></tr>
<tr id="row_273_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08917.html" target="_self">IDFilter::HasMaxMetaValue&lt; HitType &gt;</a></td><td class="desc">Does a meta value of this hit have at most the given value? </td></tr>
<tr id="row_274_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08909.html" target="_self">IDFilter::HasMaxRank&lt; HitType &gt;</a></td><td class="desc">Is the rank of this hit below or at the given cut-off? </td></tr>
<tr id="row_275_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08913.html" target="_self">IDFilter::HasMetaValue&lt; HitType &gt;</a></td><td class="desc">Is a meta value with given key and value set on this hit? </td></tr>
<tr id="row_276_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08945.html" target="_self">IDFilter::HasNoHits&lt; IdentificationType &gt;</a></td><td class="desc">Is the list of hits of this peptide/protein ID empty? </td></tr>
<tr id="row_277_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09429.html" target="_self">HDF5Connector</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for HDF5 files </td></tr>
<tr id="row_278_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06757.html" target="_self">HiddenMarkovModel</a></td><td class="desc">Hidden Markov Model implementation of PILIS </td></tr>
<tr id="row_279_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10337.html" target="_self">Histogram&lt; ValueType, BinSizeType &gt;</a></td><td class="desc">Representation of a histogram </td></tr>
<tr id="row_280_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06753.html" target="_self">HMMState</a></td><td class="desc">Hidden Markov Model State class for the Hidden Markov Model </td></tr>
<tr id="row_281_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06665" target="_self">Host&lt; Pattern&lt; TNeedle, FuzzyAC &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_282_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06669" target="_self">Host&lt; Pattern&lt; TNeedle, FuzzyAC &gt; const &gt;</a></td><td class="desc"></td></tr>
<tr id="row_283_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10461.html" target="_self">HPLC</a></td><td class="desc">Representation of a <a class="el" href="a10461.html" title="Representation of a HPLC experiment.">HPLC</a> experiment </td></tr>
<tr id="row_284_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07517.html" target="_self">HyperScore</a></td><td class="desc">An implementation of the X!Tandem <a class="el" href="a07517.html" title="An implementation of the X!Tandem HyperScore PSM scoring function.">HyperScore</a> PSM scoring function </td></tr>
<tr id="row_285_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09433.html" target="_self">IBSpectraFile</a></td><td class="desc">Implements the export of consensusmaps into the IBSpectra format used by isobar to load quantification results </td></tr>
<tr id="row_286_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09921.html" target="_self">IChromatogramsReader</a></td><td class="desc">The interface of read-access to a list of chromatograms </td></tr>
<tr id="row_287_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09929.html" target="_self">IChromatogramsWriter</a></td><td class="desc"></td></tr>
<tr id="row_288_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_288_" class="arrow" onclick="toggleFolder('288_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06425.html" target="_self">IDataFrameWriter</a></td><td class="desc"></td></tr>
<tr id="row_288_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06433.html" target="_self">CSVWriter</a></td><td class="desc"></td></tr>
<tr id="row_288_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06429.html" target="_self">DataMatrix</a></td><td class="desc"></td></tr>
<tr id="row_289_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06761.html" target="_self">IDBoostGraph</a></td><td class="desc">Creates and maintains a boost graph based on the <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> ID datastructures </td></tr>
<tr id="row_290_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06785.html" target="_self">IDConflictResolverAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_291_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10501.html" target="_self">IdentificationDataConverter</a></td><td class="desc"></td></tr>
<tr id="row_292_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10753.html#a10757" target="_self">Ms2IdentificationRate::IdentificationRateData</a></td><td class="desc">Structure for storing results </td></tr>
<tr id="row_293_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08901.html" target="_self">IDFilter</a></td><td class="desc">Collection of functions for filtering peptide and protein identifications </td></tr>
<tr id="row_294_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06817.html" target="_self">IDScoreGetterSetter</a></td><td class="desc">A class for extracting and reinserting IDScores from Peptide/ProteinIdentifications and from ConsensusMaps </td></tr>
<tr id="row_295_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07597.html" target="_self">InclusionExclusionList::IEWindow</a></td><td class="desc"></td></tr>
<tr id="row_296_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_296_" class="arrow" onclick="toggleFolder('296_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06465.html" target="_self">IFeature</a></td><td class="desc"></td></tr>
<tr id="row_296_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07117.html" target="_self">FeatureOpenMS</a></td><td class="desc">An implementation of the OpenSWATH <a class="el" href="a10037.html" title="An LC-MS feature.">Feature</a> Access interface using <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> </td></tr>
<tr id="row_296_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06481.html" target="_self">MockFeature</a></td><td class="desc">Mock object implementing <a class="el" href="a06465.html">IFeature</a> </td></tr>
<tr id="row_297_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06541.html" target="_self">ILPDCWrapper</a></td><td class="desc"></td></tr>
<tr id="row_298_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_298_" class="arrow" onclick="toggleFolder('298_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06469.html" target="_self">IMRMFeature</a></td><td class="desc"></td></tr>
<tr id="row_298_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07121.html" target="_self">MRMFeatureOpenMS</a></td><td class="desc">An implementation of the OpenSWATH MRM <a class="el" href="a10037.html" title="An LC-MS feature.">Feature</a> Access interface using <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> </td></tr>
<tr id="row_298_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06485.html" target="_self">MockMRMFeature</a></td><td class="desc">Mock object implementing <a class="el" href="a06469.html">IMRMFeature</a> </td></tr>
<tr id="row_299_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07969.html" target="_self">IMSAlphabet</a></td><td class="desc">Holds an indexed list of bio-chemical elements </td></tr>
<tr id="row_300_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07977.html" target="_self">IMSAlphabetParser&lt; AlphabetElementType, Container, InputSource &gt;</a></td><td class="desc">An abstract templatized parser to load the data that is used to initialize <code>Alphabet</code> objects </td></tr>
<tr id="row_301_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_301_" class="arrow" onclick="toggleFolder('301_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07977.html" target="_self">IMSAlphabetParser&lt;&gt;</a></td><td class="desc"></td></tr>
<tr id="row_301_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07981.html" target="_self">IMSAlphabetTextParser</a></td><td class="desc">Implements abstract <code>AlphabetParser</code> to read data from the plain text format </td></tr>
<tr id="row_302_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_302_" class="arrow" onclick="toggleFolder('302_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09913.html" target="_self">IMSDataConsumer</a></td><td class="desc">The interface of a consumer of spectra and chromatograms </td></tr>
<tr id="row_302_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_302_0_" class="arrow" onclick="toggleFolder('302_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09221.html" target="_self">FullSwathFileConsumer</a></td><td class="desc">Abstract base class which can consume spectra coming from SWATH experiment stored in a single file </td></tr>
<tr id="row_302_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09229.html" target="_self">CachedSwathFileConsumer</a></td><td class="desc">On-disk cached implementation of <a class="el" href="a09221.html" title="Abstract base class which can consume spectra coming from SWATH experiment stored in a single file.">FullSwathFileConsumer</a> </td></tr>
<tr id="row_302_0_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09233.html" target="_self">MzMLSwathFileConsumer</a></td><td class="desc">On-disk mzML implementation of <a class="el" href="a09221.html" title="Abstract base class which can consume spectra coming from SWATH experiment stored in a single file.">FullSwathFileConsumer</a> </td></tr>
<tr id="row_302_0_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09225.html" target="_self">RegularSwathFileConsumer</a></td><td class="desc">In-memory implementation of <a class="el" href="a09221.html" title="Abstract base class which can consume spectra coming from SWATH experiment stored in a single file.">FullSwathFileConsumer</a> </td></tr>
<tr id="row_302_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09161.html" target="_self">MSDataAggregatingConsumer</a></td><td class="desc">Aggregates spectra by retention time </td></tr>
<tr id="row_302_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09165.html" target="_self">MSDataCachedConsumer</a></td><td class="desc">Transforming and cached writing consumer of MS data </td></tr>
<tr id="row_302_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09169.html" target="_self">MSDataChainingConsumer</a></td><td class="desc">Consumer class that passes all consumed data through a set of operations </td></tr>
<tr id="row_302_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09173.html" target="_self">MSDataSqlConsumer</a></td><td class="desc">A data consumer that inserts MS data into a SQLite database </td></tr>
<tr id="row_302_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09177.html" target="_self">MSDataStoringConsumer</a></td><td class="desc">Consumer class that simply stores the data </td></tr>
<tr id="row_302_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09181.html" target="_self">MSDataTransformingConsumer</a></td><td class="desc">Transforming consumer of MS data </td></tr>
<tr id="row_302_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09185.html" target="_self">MSDataWritingConsumer</a></td><td class="desc">Consumer class that writes MS data to disk using the mzML format </td></tr>
<tr id="row_302_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09197.html" target="_self">NoopMSDataConsumer</a></td><td class="desc">Consumer class that performs no operation </td></tr>
<tr id="row_303_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07985.html" target="_self">IMSElement</a></td><td class="desc">Represents a chemical atom with name and isotope distribution </td></tr>
<tr id="row_304_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07989.html" target="_self">IMSIsotopeDistribution</a></td><td class="desc">Represents a distribution of isotopes restricted to the first K elements </td></tr>
<tr id="row_305_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09293.html" target="_self">IndexedMzMLDecoder</a></td><td class="desc">A class to analyze indexedmzML files and extract the offsets of individual tags </td></tr>
<tr id="row_306_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09441.html" target="_self">IndexedMzMLFileLoader</a></td><td class="desc">A class to load an indexedmzML file </td></tr>
<tr id="row_307_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09297.html" target="_self">IndexedMzMLHandler</a></td><td class="desc">A low-level class to read an indexedmzML file </td></tr>
<tr id="row_308_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07637.html#a07641" target="_self">PSLPFormulation::IndexTriple</a></td><td class="desc">Struct that holds the indices of the precursors in the feature map and the ilp formulation </td></tr>
<tr id="row_309_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07825.html" target="_self">INIUpdater</a></td><td class="desc"></td></tr>
<tr id="row_310_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08437.html" target="_self">FuzzyStringComparator::InputLine</a></td><td class="desc">Stores information about the current input line (i.e., stream for the line and the current position in the stream) </td></tr>
<tr id="row_311_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_311_" class="arrow" onclick="toggleFolder('311_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>InputSource</b></td><td class="desc"></td></tr>
<tr id="row_311_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09121.html" target="_self">CompressedInputSource</a></td><td class="desc">This class is based on xercesc::LocalFileInputSource </td></tr>
<tr id="row_312_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09445.html" target="_self">InspectInfile</a></td><td class="desc">Inspect input file adapter </td></tr>
<tr id="row_313_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09449.html" target="_self">InspectOutfile</a></td><td class="desc">Representation of an Inspect outfile </td></tr>
<tr id="row_314_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07065.html" target="_self">TransformationModelInterpolated::Interpolator</a></td><td class="desc">The class defines a generic interpolation technique used in the <a class="el" href="a07061.html" title="Interpolation model for transformations.">TransformationModelInterpolated</a> </td></tr>
<tr id="row_315_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11729.html" target="_self">TOPPASToolVertex::IOInfo</a></td><td class="desc">Stores the information for input/output files/lists </td></tr>
<tr id="row_316_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07169.html" target="_self">IonMobilityScoring</a></td><td class="desc">A class that calls the ion mobility scoring routines </td></tr>
<tr id="row_317_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06573.html" target="_self">CompNovoIonScoringBase::IonScore</a></td><td class="desc"></td></tr>
<tr id="row_318_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06593.html" target="_self">DeNovoIonScoring::IonScore</a></td><td class="desc"><a class="el" href="a06593.html" title="IonScore.">IonScore</a> </td></tr>
<tr id="row_319_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08113.html" target="_self">SvmTheoreticalSpectrumGenerator::IonType</a></td><td class="desc">Nested class </td></tr>
<tr id="row_320_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_320_" class="arrow" onclick="toggleFolder('320_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>ios_base</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_320_0_" class="arrow" onclick="toggleFolder('320_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>basic_ios&lt; Char &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_320_0_0_" class="arrow" onclick="toggleFolder('320_0_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>basic_istream&lt; Char &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span id="arr_320_0_0_0_" class="arrow" onclick="toggleFolder('320_0_0_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>basic_ifstream&lt; Char &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_0_0_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span id="arr_320_0_0_0_0_" class="arrow" onclick="toggleFolder('320_0_0_0_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>ifstream</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_0_0_0_0_" style="display:none;"><td class="entry"><span style="width:96px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09289.html" target="_self">FidHandler</a></td><td class="desc">Read-only fid <a class="el" href="a10873.html" title="Basic file handling operations.">File</a> handler for XMass Analysis </td></tr>
<tr id="row_320_0_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_320_0_1_" class="arrow" onclick="toggleFolder('320_0_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>basic_ostream&lt; Char &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span id="arr_320_0_1_0_" class="arrow" onclick="toggleFolder('320_0_1_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>ostream</b></td><td class="desc">STL class </td></tr>
<tr id="row_320_0_1_0_0_" style="display:none;"><td class="entry"><span style="width:80px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08473.html" target="_self">LogStream</a></td><td class="desc">Log Stream Class </td></tr>
<tr id="row_320_0_1_0_1_" style="display:none;"><td class="entry"><span style="width:80px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09809.html" target="_self">SVOutStream</a></td><td class="desc">Stream class for writing to comma/tab/...-separated values files </td></tr>
<tr id="row_321_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07473.html#a07489" target="_self">ProteinResolver::ISDGroup</a></td><td class="desc"></td></tr>
<tr id="row_322_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06825.html" target="_self">IDScoreGetterSetter::IsHitType&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_323_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06821.html" target="_self">IDScoreGetterSetter::IsIDType&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_324_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_324_" class="arrow" onclick="toggleFolder('324_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06477.html" target="_self">ISignalToNoise</a></td><td class="desc"></td></tr>
<tr id="row_324_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07129.html" target="_self">SignalToNoiseOpenMS&lt; ContainerT &gt;</a></td><td class="desc">An implementation of the OpenSWATH SignalToNoise Access interface using <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> </td></tr>
<tr id="row_324_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06493.html" target="_self">MockSignalToNoise</a></td><td class="desc">Mock object implementing <a class="el" href="a06477.html">ISignalToNoise</a> </td></tr>
<tr id="row_325_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07425.html" target="_self">IsobaricQuantitationMethod::IsobaricChannelInformation</a></td><td class="desc"><a class="el" href="a10969.html" title="Summary of fitting results.">Summary</a> of an isobaric quantitation channel </td></tr>
<tr id="row_326_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07401.html" target="_self">IsobaricIsotopeCorrector</a></td><td class="desc">Performs isotope impurity correction on the intensities extracted from an isobaric labeling experiment </td></tr>
<tr id="row_327_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07405.html" target="_self">IsobaricNormalizer</a></td><td class="desc">Performs median normalization on the extracted ratios of isobaric labeling experiment </td></tr>
<tr id="row_328_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07413.html" target="_self">IsobaricQuantifierStatistics</a></td><td class="desc">Statistics for quantitation performance and comparison of <a class="el" href="a05848.html">NNLS</a> vs. naive method (aka matrix inversion) </td></tr>
<tr id="row_329_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_329_" class="arrow" onclick="toggleFolder('329_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07933.html" target="_self">IsoSpecGeneratorWrapper</a></td><td class="desc">Interface for the IsoSpec algorithm - a generator of infinitely-resolved theoretical spectra </td></tr>
<tr id="row_329_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07949.html" target="_self">IsoSpecOrderedGeneratorWrapper</a></td><td class="desc">Generate the stream of configurations, ordered from most likely to least likely </td></tr>
<tr id="row_329_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07945.html" target="_self">IsoSpecThresholdGeneratorWrapper</a></td><td class="desc">Provides a threshold-based generator of isotopologues: generates all isotopologues more probable than a given probability threshold </td></tr>
<tr id="row_329_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07941.html" target="_self">IsoSpecTotalProbGeneratorWrapper</a></td><td class="desc">Generate a p-set of configurations for a given p (that is, a set of configurations such that their probabilities sum up to p). The p in normal usage will usually be close to 1 (e.g. 0.99) </td></tr>
<tr id="row_330_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_330_" class="arrow" onclick="toggleFolder('330_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07937.html" target="_self">IsoSpecWrapper</a></td><td class="desc">A convenience class for the IsoSpec algorithm - easier to use than the <a class="el" href="a07933.html" title="Interface for the IsoSpec algorithm - a generator of infinitely-resolved theoretical spectra.">IsoSpecGeneratorWrapper</a> classes </td></tr>
<tr id="row_330_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07957.html" target="_self">IsoSpecThresholdWrapper</a></td><td class="desc">A non-generator version of <a class="el" href="a07945.html" title="Provides a threshold-based generator of isotopologues: generates all isotopologues more probable than...">IsoSpecThresholdGeneratorWrapper</a> </td></tr>
<tr id="row_330_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07953.html" target="_self">IsoSpecTotalProbWrapper</a></td><td class="desc">Create a p-set of configurations for a given p (that is, a set of configurations such that their probabilities sum up to p). The p in normal usage will usually be close to 1 (e.g. 0.99) </td></tr>
<tr id="row_331_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08633.html" target="_self">IsotopeCluster</a></td><td class="desc">Stores information about an isotopic cluster (i.e. potential peptide charge variants) </td></tr>
<tr id="row_332_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07961.html" target="_self">IsotopeDistribution</a></td><td class="desc"></td></tr>
<tr id="row_333_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08881.html" target="_self">IsotopeDistributionCache</a></td><td class="desc">Pre-calculate isotope distributions for interesting mass ranges </td></tr>
<tr id="row_334_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11013.html" target="_self">FeatureFinderAlgorithmPickedHelperStructs::IsotopePattern</a></td><td class="desc">Helper structure for a found isotope pattern used in <a class="el" href="a10989.html" title="FeatureFinderAlgorithm for picked peaks.">FeatureFinderAlgorithmPicked</a> </td></tr>
<tr id="row_335_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_335_" class="arrow" onclick="toggleFolder('335_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07965.html" target="_self">IsotopePatternGenerator</a></td><td class="desc">Provides an interface for different isotope pattern generator methods </td></tr>
<tr id="row_335_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07925.html" target="_self">CoarseIsotopePatternGenerator</a></td><td class="desc">Isotope pattern generator for coarse isotope distributions </td></tr>
<tr id="row_335_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07929.html" target="_self">FineIsotopePatternGenerator</a></td><td class="desc">Isotope pattern generator for fine isotope distributions </td></tr>
<tr id="row_336_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11085.html" target="_self">IsotopeWavelet</a></td><td class="desc">Implements the isotope wavelet function </td></tr>
<tr id="row_337_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11093.html" target="_self">IsotopeWaveletTransform&lt; PeakType &gt;</a></td><td class="desc">A class implementing the isotope wavelet transform. If you just want to find features using the isotope wavelet, take a look at the <a class="el" href="a10977.html" title="Implements the isotope wavelet feature finder.">FeatureFinderAlgorithmIsotopeWavelet</a> class. Usually, you only have to consider the class at hand if you plan to change the basic implementation of the transform </td></tr>
<tr id="row_338_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11345.html" target="_self">IsotopicDist</a></td><td class="desc"></td></tr>
<tr id="row_339_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09917.html" target="_self">ISpectraReader</a></td><td class="desc">The interface of read-access to a list of spectra </td></tr>
<tr id="row_340_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09925.html" target="_self">ISpectraWriter</a></td><td class="desc"></td></tr>
<tr id="row_341_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_341_" class="arrow" onclick="toggleFolder('341_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06461.html" target="_self">ISpectrumAccess</a></td><td class="desc">The interface of a mass spectrometry experiment </td></tr>
<tr id="row_341_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07137.html" target="_self">SpectrumAccessOpenMS</a></td><td class="desc">An implementation of the OpenSWATH Spectrum Access interface using <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> </td></tr>
<tr id="row_341_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07141.html" target="_self">SpectrumAccessOpenMSCached</a></td><td class="desc">An implementation of the Spectrum Access interface using on-disk caching </td></tr>
<tr id="row_341_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07145.html" target="_self">SpectrumAccessOpenMSInMemory</a></td><td class="desc">An implementation of the OpenSWATH Spectrum Access interface completely in memory </td></tr>
<tr id="row_341_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07153.html" target="_self">SpectrumAccessSqMass</a></td><td class="desc">An implementation of the Spectrum Access interface using SQL files </td></tr>
<tr id="row_341_4_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_341_4_" class="arrow" onclick="toggleFolder('341_4_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07157.html" target="_self">SpectrumAccessTransforming</a></td><td class="desc">An abstract base class implementing a transforming wrapper around spectrum access </td></tr>
<tr id="row_341_4_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07149.html" target="_self">SpectrumAccessQuadMZTransforming</a></td><td class="desc">A transforming m/z wrapper around spectrum access using a quadratic equation </td></tr>
<tr id="row_342_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08045.html" target="_self">NASequence::Iterator</a></td><td class="desc"><a class="el" href="a08045.html" title="Iterator of NASequence class.">Iterator</a> of <a class="el" href="a08037.html" title="Representation of a nucleic acid sequence.">NASequence</a> class </td></tr>
<tr id="row_343_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_343_" class="arrow" onclick="toggleFolder('343_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>iterator</b></td><td class="desc"></td></tr>
<tr id="row_343_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08205.html" target="_self">HashGrid&lt; Cluster &gt;::ConstIterator</a></td><td class="desc">Constant element iterator for the hash grid </td></tr>
<tr id="row_343_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08201.html" target="_self">HashGrid&lt; Cluster &gt;::Iterator</a></td><td class="desc"><a class="el" href="a07897.html" title="Representation of an element.">Element</a> iterator for the hash grid </td></tr>
<tr id="row_343_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09933.html" target="_self">AreaIterator&lt; ValueT, ReferenceT, PointerT, SpectrumIteratorT, PeakIteratorT &gt;</a></td><td class="desc">Forward iterator for an area of peaks in an experiment </td></tr>
<tr id="row_343_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08817.html" target="_self">IntensityIteratorWrapper&lt; IteratorT &gt;</a></td><td class="desc">An iterator wrapper to access peak intensities instead of the peak itself </td></tr>
<tr id="row_344_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_344_" class="arrow" onclick="toggleFolder('344_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>set&lt; K &gt;::iterator</b></td><td class="desc">STL iterator class </td></tr>
<tr id="row_344_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10513.html" target="_self">IteratorWrapper&lt; DataProcessingSoftwares::iterator &gt;</a></td><td class="desc"></td></tr>
<tr id="row_344_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10513.html" target="_self">IteratorWrapper&lt; DataQueries::iterator &gt;</a></td><td class="desc"></td></tr>
<tr id="row_345_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07873.html" target="_self">AASequence::Iterator</a></td><td class="desc"><a class="el" href="a07873.html" title="Iterator class for AASequence.">Iterator</a> class for <a class="el" href="a07865.html" title="Representation of a peptide/protein sequence.">AASequence</a> </td></tr>
<tr id="row_346_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_346_" class="arrow" onclick="toggleFolder('346_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06473.html" target="_self">ITransitionGroup</a></td><td class="desc"></td></tr>
<tr id="row_346_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07125.html" target="_self">TransitionGroupOpenMS&lt; SpectrumT, TransitionT &gt;</a></td><td class="desc">An implementation of the OpenSWATH Transition Group Access interface using <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> </td></tr>
<tr id="row_346_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06489.html" target="_self">MockTransitionGroup</a></td><td class="desc">Mock object implementing <a class="el" href="a06473.html">ITransitionGroup</a> </td></tr>
<tr id="row_347_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07429.html" target="_self">ItraqConstants</a></td><td class="desc">Some constants used throughout iTRAQ classes </td></tr>
<tr id="row_348_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10885.html" target="_self">JavaInfo</a></td><td class="desc">Detect Java and retrieve information </td></tr>
<tr id="row_349_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07449.html" target="_self">KDTreeFeatureNode</a></td><td class="desc">A node of the kD-tree with pointer to corresponding data and index </td></tr>
<tr id="row_350_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09453.html" target="_self">KroenikFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for Kroenik (HardKloer sibling) files </td></tr>
<tr id="row_351_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11141.html" target="_self">MultiplexDeltaMassesGenerator::Label</a></td><td class="desc">Complete label information </td></tr>
<tr id="row_352_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11585.html" target="_self">LayerData</a></td><td class="desc">Class that stores the data for one layer </td></tr>
<tr id="row_353_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11349.html" target="_self">LCElutionPeak</a></td><td class="desc"></td></tr>
<tr id="row_354_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11353.html" target="_self">LCMS</a></td><td class="desc"></td></tr>
<tr id="row_355_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11365.html" target="_self">LCMSCData</a></td><td class="desc"></td></tr>
<tr id="row_356_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09457.html" target="_self">LibSVMEncoder</a></td><td class="desc">Serves for encoding sequences into feature vectors </td></tr>
<tr id="row_357_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06509.html" target="_self">LightCompound</a></td><td class="desc"></td></tr>
<tr id="row_358_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05820.html#a06505" target="_self">LightModification</a></td><td class="desc"></td></tr>
<tr id="row_359_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05820.html#a06513" target="_self">LightProtein</a></td><td class="desc"></td></tr>
<tr id="row_360_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06517.html" target="_self">LightTargetedExperiment</a></td><td class="desc"></td></tr>
<tr id="row_361_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06501.html" target="_self">LightTransition</a></td><td class="desc"></td></tr>
<tr id="row_362_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10257.html" target="_self">LinearInterpolation&lt; Key, Value &gt;</a></td><td class="desc">Provides access to linearly interpolated values (and derivatives) from discrete data points. Values beyond the given range of data points are implicitly taken as zero </td></tr>
<tr id="row_363_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10257.html" target="_self">LinearInterpolation&lt; double &gt;</a></td><td class="desc"></td></tr>
<tr id="row_364_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10341.html" target="_self">LinearRegression</a></td><td class="desc">This class offers functions to perform least-squares fits to a straight line model, <img class="formulaInl" alt="$ Y(c,x) = c_0 + c_1 x $" src="form_89.png"/> </td></tr>
<tr id="row_365_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10345.html" target="_self">LinearRegressionWithoutIntercept</a></td><td class="desc">This class offers functions to perform least-squares fits to a straight line model, <img class="formulaInl" alt="$ Y(c,x) = c_0 + c_1 x $" src="form_89.png"/> </td></tr>
<tr id="row_366_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_366_" class="arrow" onclick="toggleFolder('366_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>list&lt; T &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_366_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11437.html" target="_self">Annotations1DContainer</a></td><td class="desc">Container for annotations to content of <a class="el" href="a11641.html" title="Canvas for visualization of one or several spectra.">Spectrum1DCanvas</a> </td></tr>
<tr id="row_367_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08693.html" target="_self">ListUtils</a></td><td class="desc">Collection of utility functions for management of vectors </td></tr>
<tr id="row_368_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07373.html#a07377" target="_self">LocalLinearMap::LLMParam</a></td><td class="desc">Define parameters needed by the Local Linear Map (LLM) model </td></tr>
<tr id="row_369_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07373.html" target="_self">LocalLinearMap</a></td><td class="desc">Trained Local Linear Map (LLM) model for peak intensity prediction </td></tr>
<tr id="row_370_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08829.html" target="_self">InternalCalibration::LockMass</a></td><td class="desc">Helper class, describing a lock mass </td></tr>
<tr id="row_371_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08457.html#a08465" target="_self">LogStreamBuf::LogCacheStruct</a></td><td class="desc">Holds a counter of occurrences and an index for the occurrence sequence of the corresponding log message </td></tr>
<tr id="row_372_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08453.html" target="_self">LogConfigHandler</a></td><td class="desc">The <a class="el" href="a08453.html" title="The LogConfigHandler provides the functionality to configure the internal logging of OpenMS algorithm...">LogConfigHandler</a> provides the functionality to configure the internal logging of <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> algorithms that use the global instances of LogStream </td></tr>
<tr id="row_373_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11937.html" target="_self">FLASHDeconv::LogMzPeak</a></td><td class="desc"></td></tr>
<tr id="row_374_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07737.html" target="_self">FLASHDeconvHelperStructs::LogMzPeak</a></td><td class="desc"></td></tr>
<tr id="row_375_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08469.html" target="_self">LogStreamNotifier</a></td><td class="desc"></td></tr>
<tr id="row_376_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05823.html#a10241" target="_self">Looper&lt; param_index, grid_size, EvalResult, Tuple, TupleTypes &gt;</a></td><td class="desc"></td></tr>
<tr id="row_377_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10245.html" target="_self">Looper&lt; grid_size, grid_size, EvalResult, Tuple, TupleTypes... &gt;</a></td><td class="desc"></td></tr>
<tr id="row_378_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10249.html" target="_self">Looper&lt; param_index, grid_size, EvalResult, Tuple, FirstTupleType, TupleTypes... &gt;</a></td><td class="desc"></td></tr>
<tr id="row_379_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08089.html#a08101" target="_self">SimpleTSGXLMS::LossIndex</a></td><td class="desc"></td></tr>
<tr id="row_380_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08137.html#a08141" target="_self">TheoreticalSpectrumGeneratorXLMS::LossIndex</a></td><td class="desc"></td></tr>
<tr id="row_381_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08709.html" target="_self">LPWrapper</a></td><td class="desc"></td></tr>
<tr id="row_382_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06877.html" target="_self">PercolatorFeatureSetHelper::lq_PeptideEvidence</a></td><td class="desc">For accession dependent sorting of PeptideEvidences </td></tr>
<tr id="row_383_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06873.html" target="_self">PercolatorFeatureSetHelper::lq_ProteinHit</a></td><td class="desc">For accession dependent sorting of ProteinHits </td></tr>
<tr id="row_384_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_384_" class="arrow" onclick="toggleFolder('384_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>map&lt; K, T &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_384_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; char, const OpenMS::Residue * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; char, double &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; const OpenMS::BaseFeature *, std::vector&lt; size_t &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; const OpenMS::Residue *, char &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; double, std::vector&lt; OpenMS::MassDecomposition &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::EnzymaticDigestionLogModel::BindingSite_, OpenMS::EnzymaticDigestionLogModel::CleavageModel_ &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::HMMState *, double &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::HMMState *, OpenMS::Map&lt; OpenMS::HMMState *, double &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::HMMState *, OpenMS::Map&lt; OpenMS::HMMState *, Size &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::HMMState *, OpenMS::Map&lt; OpenMS::HMMState *, std::pair&lt; OpenMS::HMMState *, OpenMS::HMMState * &gt; &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_10_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::HMMState *, OpenMS::Map&lt; OpenMS::HMMState *, std::vector&lt; double &gt; &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_11_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::HMMState *, std::set&lt; OpenMS::HMMState * &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_12_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, const DigestionEnzymeProtein * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_13_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, const DigestionEnzymeRNA * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_14_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, const DigestionEnzymeType * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_15_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, const OpenMS::Element * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_16_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, IntList &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_17_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::AASequence &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_18_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::ControlledVocabulary::CVTerm &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_19_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::HMMState * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_20_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::Instrument &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_21_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::Map&lt; OpenMS::String, OpenMS::Map&lt; OpenMS::String, UInt &gt; &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_22_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::Map&lt; OpenMS::String, OpenMS::Residue * &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_23_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::Map&lt; OpenMS::String, std::pair&lt; OpenMS::String, OpenMS::String &gt; &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_24_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::Sample &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_25_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::Software &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_26_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::SourceFile &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_27_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, OpenMS::String &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_28_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, Size &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_29_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::set&lt; const OpenMS::Residue * &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_30_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::set&lt; const OpenMS::ResidueModification * &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_31_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::set&lt; OpenMS::String &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_32_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::vector&lt; DataProcessingPtr &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_33_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::vector&lt; OpenMS::CVMappingRule &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_34_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::vector&lt; OpenMS::CVTerm &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_35_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, std::vector&lt; OpenMS::Internal::SemanticValidator::CVTerm &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_36_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; OpenMS::String, UInt &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_37_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; PointType::CoordinateType, DBoundingBox&lt; 1 &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_38_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; QString, QList&lt; OpenMS::TOPPASResource &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_39_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; QString, QString &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_40_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; Size, OpenMS::Map&lt; Size, std::set&lt; OpenMS::String &gt; &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_41_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; Size, OpenMS::String &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_42_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; Size, Size &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_43_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; Size, std::vector&lt; double &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_44_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; UInt, const OpenMS::Element * &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_45_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; UInt, std::vector&lt; const OpenMS::ResidueModification * &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_46_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; UInt, UInt &gt;</a></td><td class="desc"></td></tr>
<tr id="row_384_47_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08717.html" target="_self">Map&lt; Key, T &gt;</a></td><td class="desc"><a class="el" href="a08717.html" title="Map class based on the STL map (containing several convenience functions)">Map</a> class based on the STL map (containing several convenience functions) </td></tr>
<tr id="row_385_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06989.html" target="_self">MapAlignmentAlgorithmKD</a></td><td class="desc">An efficient reference-free feature map alignment algorithm for unlabeled data </td></tr>
<tr id="row_386_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_386_" class="arrow" onclick="toggleFolder('386_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07005.html" target="_self">MapAlignmentEvaluationAlgorithm</a></td><td class="desc">Base class for all Caap evaluation algorithms </td></tr>
<tr id="row_386_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07009.html" target="_self">MapAlignmentEvaluationAlgorithmPrecision</a></td><td class="desc">Caap evaluation algorithm to obtain a precision value </td></tr>
<tr id="row_386_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07013.html" target="_self">MapAlignmentEvaluationAlgorithmRecall</a></td><td class="desc">Caap evaluation algorithm to obtain a recall value </td></tr>
<tr id="row_387_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07017.html" target="_self">MapAlignmentTransformer</a></td><td class="desc">This class collects functions for applying retention time transformations to data structures </td></tr>
<tr id="row_388_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10029.html" target="_self">MapConversion</a></td><td class="desc"></td></tr>
<tr id="row_389_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06609.html#a06613" target="_self">AccurateMassSearchEngine::MappingEntry_</a></td><td class="desc"></td></tr>
<tr id="row_390_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08801.html" target="_self">MappingParam</a></td><td class="desc">Filename mappings for all input/output files </td></tr>
<tr id="row_391_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07521.html#a07525" target="_self">ModifiedPeptideGenerator::MapToResidueType</a></td><td class="desc"></td></tr>
<tr id="row_392_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_392_" class="arrow" onclick="toggleFolder('392_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08001.html" target="_self">MassDecomposer&lt; ValueType, DecompositionValueType &gt;</a></td><td class="desc">An interface to handle decomposing of integer values/masses over a set of integer weights (alphabet) </td></tr>
<tr id="row_392_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07997.html" target="_self">IntegerMassDecomposer&lt; ValueType, DecompositionValueType &gt;</a></td><td class="desc">Implements <code><a class="el" href="a08001.html" title="An interface to handle decomposing of integer values/masses over a set of integer weights (alphabet).">MassDecomposer</a></code> interface using algorithm and data structures described in paper "Efficient Mass Decomposition" S. Böcker, Z. Lipták, ACM SAC-BIO, 2004 doi:10.1145/1066677.1066715 </td></tr>
<tr id="row_393_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08013.html" target="_self">MassDecomposition</a></td><td class="desc">Class represents a decomposition of a mass into amino acids </td></tr>
<tr id="row_394_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08725.html" target="_self">MassExplainer</a></td><td class="desc">Computes empirical formulas for given mass differences using a set of allowed elements </td></tr>
<tr id="row_395_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07745.html" target="_self">MassFeatureTrace</a></td><td class="desc"></td></tr>
<tr id="row_396_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07973.html" target="_self">IMSAlphabet::MassSortingCriteria_</a></td><td class="desc">Private class-functor to sort out elements in mass ascending order </td></tr>
<tr id="row_397_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11001.html" target="_self">FeatureFinderAlgorithmPickedHelperStructs::MassTrace</a></td><td class="desc">Helper struct for mass traces used in <a class="el" href="a10989.html" title="FeatureFinderAlgorithm for picked peaks.">FeatureFinderAlgorithmPicked</a> </td></tr>
<tr id="row_398_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10061.html" target="_self">MassTrace</a></td><td class="desc">A container type that gathers peaks similar in m/z and moving along retention time </td></tr>
<tr id="row_399_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a12021.html" target="_self">TDfeatureDeconvolution::MassTraceApex</a></td><td class="desc"></td></tr>
<tr id="row_400_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07349.html" target="_self">TargetedSpectraExtractor::Match</a></td><td class="desc"></td></tr>
<tr id="row_401_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08729.html" target="_self">MatchedIterator&lt; CONT_T, TRAIT, CONST_T &gt;</a></td><td class="desc">For each element in the reference container the closest peak in the target will be searched. If no match is found within the tolerance window, the peak will be skipped over </td></tr>
<tr id="row_402_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06421.html" target="_self">mean_and_stddev</a></td><td class="desc">Functor to compute the mean and stddev of sequence using the std::foreach algorithm </td></tr>
<tr id="row_403_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10905.html" target="_self">SysInfo::MemUsage</a></td><td class="desc">A convenience class to report either absolute or delta (between two timepoints) RAM usage </td></tr>
<tr id="row_404_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06829.html" target="_self">MessagePasserFactory&lt; Label &gt;</a></td><td class="desc"></td></tr>
<tr id="row_405_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07605.html" target="_self">MetaboTargetedAssay</a></td><td class="desc">This class provides methods for the extraction of targeted assays for metabolomics </td></tr>
<tr id="row_406_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10577.html" target="_self">MetaInfo</a></td><td class="desc">A Type-Name-Value tuple class </td></tr>
<tr id="row_407_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_407_" class="arrow" onclick="toggleFolder('407_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10585.html" target="_self">MetaInfoInterface</a></td><td class="desc">Interface for classes that can store arbitrary meta information (Type-Name-Value tuples) </td></tr>
<tr id="row_407_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10381.html" target="_self">Acquisition</a></td><td class="desc">Information about one raw data spectrum that was combined with several other raw data spectra </td></tr>
<tr id="row_407_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10385.html" target="_self">AcquisitionInfo</a></td><td class="desc">Description of the combination of raw data to a single spectrum </td></tr>
<tr id="row_407_2_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_2_" class="arrow" onclick="toggleFolder('407_2_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10389.html" target="_self">ChromatogramSettings</a></td><td class="desc">Representation of chromatogram settings, e.g. SRM/MRM chromatograms </td></tr>
<tr id="row_407_2_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10073.html" target="_self">MSChromatogram</a></td><td class="desc">The representation of a chromatogram </td></tr>
<tr id="row_407_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10021.html" target="_self">ConsensusMap</a></td><td class="desc">A container for consensus elements </td></tr>
<tr id="row_407_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10025.html" target="_self">ConsensusMap::ColumnHeader</a></td><td class="desc">Description of the columns in a consensus map </td></tr>
<tr id="row_407_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10393.html" target="_self">ContactPerson</a></td><td class="desc">Contact person information </td></tr>
<tr id="row_407_6_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_6_" class="arrow" onclick="toggleFolder('407_6_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10405.html" target="_self">CVTermList</a></td><td class="desc">Representation of controlled vocabulary term list </td></tr>
<tr id="row_407_6_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07589.html" target="_self">IncludeExcludeTarget</a></td><td class="desc">This class stores a SRM/MRM transition </td></tr>
<tr id="row_407_6_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10649.html" target="_self">Precursor</a></td><td class="desc"><a class="el" href="a10649.html" title="Precursor meta information.">Precursor</a> meta information </td></tr>
<tr id="row_407_6_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10653.html" target="_self">Product</a></td><td class="desc"><a class="el" href="a10653.html" title="Product meta information.">Product</a> meta information </td></tr>
<tr id="row_407_6_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07081.html" target="_self">ReactionMonitoringTransition</a></td><td class="desc">This class stores a SRM/MRM transition </td></tr>
<tr id="row_407_6_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_407_6_4_" class="arrow" onclick="toggleFolder('407_6_4_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10701.html" target="_self">Software</a></td><td class="desc">Description of the software used for processing </td></tr>
<tr id="row_407_6_4_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10469.html" target="_self">DataProcessingSoftware</a></td><td class="desc">Information about software used for data processing </td></tr>
<tr id="row_407_6_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10705.html" target="_self">SourceFile</a></td><td class="desc">Description of a file location, used to store the origin of (meta) data </td></tr>
<tr id="row_407_6_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07665.html" target="_self">Configuration</a></td><td class="desc"></td></tr>
<tr id="row_407_6_7_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07697.html" target="_self">Contact</a></td><td class="desc"></td></tr>
<tr id="row_407_6_8_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07705.html" target="_self">Instrument</a></td><td class="desc"></td></tr>
<tr id="row_407_6_9_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_407_6_9_" class="arrow" onclick="toggleFolder('407_6_9_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07681.html" target="_self">PeptideCompound</a></td><td class="desc">Base class to represent either a peptide or a compound </td></tr>
<tr id="row_407_6_9_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07685.html" target="_self">Compound</a></td><td class="desc">Represents a compound (small molecule) </td></tr>
<tr id="row_407_6_9_1_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07689.html" target="_self">Peptide</a></td><td class="desc">Represents a peptide (amino acid sequence) </td></tr>
<tr id="row_407_6_10_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07709.html" target="_self">Prediction</a></td><td class="desc"></td></tr>
<tr id="row_407_6_11_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07673.html" target="_self">Protein</a></td><td class="desc"></td></tr>
<tr id="row_407_6_12_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07701.html" target="_self">Publication</a></td><td class="desc"></td></tr>
<tr id="row_407_7_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_7_" class="arrow" onclick="toggleFolder('407_7_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10409.html" target="_self">CVTermListInterface</a></td><td class="desc">Interface to the controlled vocabulary term list </td></tr>
<tr id="row_407_7_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07713.html" target="_self">Interpretation</a></td><td class="desc"><a class="el" href="a10653.html" title="Product meta information.">Product</a> ion interpretation </td></tr>
<tr id="row_407_7_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07693.html" target="_self">Peptide::Modification</a></td><td class="desc"></td></tr>
<tr id="row_407_7_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07677.html" target="_self">RetentionTime</a></td><td class="desc">This class stores a retention time structure that is used in <a class="el" href="a07661.html" title="A description of a targeted experiment containing precursor and production ions.">TargetedExperiment</a> (representing a TraML file) </td></tr>
<tr id="row_407_7_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07717.html" target="_self">TraMLProduct</a></td><td class="desc">Represents a product ion </td></tr>
<tr id="row_407_8_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10425.html" target="_self">DataProcessing</a></td><td class="desc">Description of the applied preprocessing steps </td></tr>
<tr id="row_407_9_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10453.html" target="_self">ExperimentalSettings</a></td><td class="desc">Description of the experimental settings </td></tr>
<tr id="row_407_10_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10057.html" target="_self">FeatureMap</a></td><td class="desc">A container for features </td></tr>
<tr id="row_407_11_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10549.html" target="_self">Identification</a></td><td class="desc">Represents a object which can store the information of an analysisXML instance </td></tr>
<tr id="row_407_12_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10485.html" target="_self">IdentificationData</a></td><td class="desc">Representation of spectrum identification results and associated data </td></tr>
<tr id="row_407_13_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10473.html" target="_self">DataProcessingStep</a></td><td class="desc">Data processing step that is applied to the data (e.g. database search, PEP calculation, filtering, ConsensusID) </td></tr>
<tr id="row_407_14_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10477.html" target="_self">DataQuery</a></td><td class="desc">Search query, e.g. spectrum or feature </td></tr>
<tr id="row_407_15_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10481.html" target="_self">DBSearchParam</a></td><td class="desc">Parameters specific to a database search step </td></tr>
<tr id="row_407_16_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10517.html" target="_self">MoleculeParentMatch</a></td><td class="desc">Meta data for the association between an identified molecule (e.g. peptide) and a parent molecule (e.g. protein) </td></tr>
<tr id="row_407_17_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_17_" class="arrow" onclick="toggleFolder('407_17_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10541.html" target="_self">ScoredProcessingResult</a></td><td class="desc">Base class for ID data with scores and processing steps (and meta info) </td></tr>
<tr id="row_407_17_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10505.html" target="_self">IdentifiedCompound</a></td><td class="desc"></td></tr>
<tr id="row_407_17_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10509.html" target="_self">IdentifiedSequence&lt; SeqType &gt;</a></td><td class="desc">Identified sequence (peptide or oligonucleotide) </td></tr>
<tr id="row_407_17_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10521.html" target="_self">MoleculeQueryMatch</a></td><td class="desc">Meta data for a search hit (e.g. peptide-spectrum match) </td></tr>
<tr id="row_407_17_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10525.html" target="_self">ParentMolecule</a></td><td class="desc">Representation of a parent molecule that is identified only indirectly (e.g. a protein) </td></tr>
<tr id="row_407_17_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10533.html" target="_self">ParentMoleculeGrouping</a></td><td class="desc">Set of groups of ambiguously identified parent molecules (e.g. results of running a protein inference algorithm) </td></tr>
<tr id="row_407_17_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10537.html" target="_self">QueryMatchGroup</a></td><td class="desc">: Group of related (co-identified) molecule-query matches </td></tr>
<tr id="row_407_18_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10545.html" target="_self">ScoreType</a></td><td class="desc">Information about a score type </td></tr>
<tr id="row_407_19_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10553.html" target="_self">IdentificationHit</a></td><td class="desc">Represents a object which can store the information of an analysisXML instance </td></tr>
<tr id="row_407_20_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10557.html" target="_self">Instrument</a></td><td class="desc">Description of a MS instrument </td></tr>
<tr id="row_407_21_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10561.html" target="_self">InstrumentSettings</a></td><td class="desc">Description of the settings a MS <a class="el" href="a10557.html" title="Description of a MS instrument.">Instrument</a> was run with </td></tr>
<tr id="row_407_22_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10565.html" target="_self">IonDetector</a></td><td class="desc">Description of a ion detector (part of a MS <a class="el" href="a10557.html" title="Description of a MS instrument.">Instrument</a>) </td></tr>
<tr id="row_407_23_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10569.html" target="_self">IonSource</a></td><td class="desc">Description of an ion source (part of a MS <a class="el" href="a10557.html" title="Description of a MS instrument.">Instrument</a>) </td></tr>
<tr id="row_407_24_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10573.html" target="_self">MassAnalyzer</a></td><td class="desc">Description of a mass analyzer (part of a MS <a class="el" href="a10557.html" title="Description of a MS instrument.">Instrument</a>) </td></tr>
<tr id="row_407_25_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_25_" class="arrow" onclick="toggleFolder('407_25_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10581.html" target="_self">MetaInfoDescription</a></td><td class="desc">Description of the meta data arrays of <a class="el" href="a10097.html" title="The representation of a 1D spectrum.">MSSpectrum</a> </td></tr>
<tr id="row_407_25_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10413.html" target="_self">FloatDataArray</a></td><td class="desc">Float data array class </td></tr>
<tr id="row_407_25_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10417.html" target="_self">IntegerDataArray</a></td><td class="desc">Integer data array class </td></tr>
<tr id="row_407_25_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10421.html" target="_self">StringDataArray</a></td><td class="desc"><a class="el" href="a08769.html" title="A more convenient string class.">String</a> data array class </td></tr>
<tr id="row_407_26_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10617.html" target="_self">PeptideHit</a></td><td class="desc">Representation of a peptide hit </td></tr>
<tr id="row_407_27_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10645.html" target="_self">PeptideIdentification</a></td><td class="desc">Represents the peptide hits for a spectrum </td></tr>
<tr id="row_407_28_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10657.html" target="_self">ProteinHit</a></td><td class="desc">Representation of a protein hit </td></tr>
<tr id="row_407_29_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10677.html" target="_self">ProteinIdentification</a></td><td class="desc">Representation of a protein identification run </td></tr>
<tr id="row_407_30_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10685.html" target="_self">ProteinIdentification::SearchParameters</a></td><td class="desc">Search parameters of the DB search </td></tr>
<tr id="row_407_31_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_31_" class="arrow" onclick="toggleFolder('407_31_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10213.html" target="_self">RichPeak2D</a></td><td class="desc">A 2-dimensional raw data point or peak with meta information </td></tr>
<tr id="row_407_31_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_407_31_0_" class="arrow" onclick="toggleFolder('407_31_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09937.html" target="_self">BaseFeature</a></td><td class="desc">A basic LC-MS feature </td></tr>
<tr id="row_407_31_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10005.html" target="_self">ConsensusFeature</a></td><td class="desc">A consensus feature spanning multiple LC-MS/MS experiments </td></tr>
<tr id="row_407_31_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span id="arr_407_31_0_1_" class="arrow" onclick="toggleFolder('407_31_0_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10037.html" target="_self">Feature</a></td><td class="desc">An LC-MS feature </td></tr>
<tr id="row_407_31_0_1_0_" class="even" style="display:none;"><td class="entry"><span style="width:80px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10065.html" target="_self">MRMFeature</a></td><td class="desc">A multi-chromatogram MRM feature </td></tr>
<tr id="row_407_32_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10689.html" target="_self">Sample</a></td><td class="desc">Meta information about the sample </td></tr>
<tr id="row_407_33_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_33_" class="arrow" onclick="toggleFolder('407_33_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10693.html" target="_self">SampleTreatment</a></td><td class="desc">Base class for sample treatments (<a class="el" href="a10429.html" title="Meta information about digestion of a sample.">Digestion</a>, <a class="el" href="a10597.html" title="Meta information about chemical modification of a sample.">Modification</a>, <a class="el" href="a10729.html" title="Meta information about tagging of a sample e.g. ICAT labeling.">Tagging</a>, ...) </td></tr>
<tr id="row_407_33_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10429.html" target="_self">Digestion</a></td><td class="desc">Meta information about digestion of a sample </td></tr>
<tr id="row_407_33_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_407_33_1_" class="arrow" onclick="toggleFolder('407_33_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10597.html" target="_self">Modification</a></td><td class="desc">Meta information about chemical modification of a sample </td></tr>
<tr id="row_407_33_1_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10729.html" target="_self">Tagging</a></td><td class="desc">Meta information about tagging of a sample e.g. ICAT labeling </td></tr>
<tr id="row_407_34_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10697.html" target="_self">ScanWindow</a></td><td class="desc">Scan window description </td></tr>
<tr id="row_407_35_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10709.html" target="_self">SpectrumIdentification</a></td><td class="desc">Represents a object which can store the information of an analysisXML instance </td></tr>
<tr id="row_407_36_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_407_36_" class="arrow" onclick="toggleFolder('407_36_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10725.html" target="_self">SpectrumSettings</a></td><td class="desc">Representation of 1D spectrum settings </td></tr>
<tr id="row_407_36_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10097.html" target="_self">MSSpectrum</a></td><td class="desc">The representation of a 1D spectrum </td></tr>
<tr id="row_408_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10589.html" target="_self">MetaInfoInterfaceUtils</a></td><td class="desc">Utilities operating on containers inheriting from <a class="el" href="a10585.html" title="Interface for classes that can store arbitrary meta information (Type-Name-Value tuples).">MetaInfoInterface</a> </td></tr>
<tr id="row_409_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10593.html" target="_self">MetaInfoRegistry</a></td><td class="desc">Registry which assigns unique integer indices to strings </td></tr>
<tr id="row_410_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11973.html" target="_self">MetaProSIPClustering</a></td><td class="desc"></td></tr>
<tr id="row_411_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11981.html" target="_self">MetaProSIPDecomposition</a></td><td class="desc"></td></tr>
<tr id="row_412_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11969.html" target="_self">MetaProSIPInterpolation</a></td><td class="desc"></td></tr>
<tr id="row_413_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11977.html" target="_self">MetaProSIPReporting</a></td><td class="desc"></td></tr>
<tr id="row_414_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11985.html" target="_self">MetaProSIPXICExtraction</a></td><td class="desc"></td></tr>
<tr id="row_415_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11493.html" target="_self">LayerStatisticsDialog::MetaStatsValue_</a></td><td class="desc">Struct representing the statistics about one meta information </td></tr>
<tr id="row_416_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08189.html" target="_self">MinimumDistance</a></td><td class="desc">Basic data structure for distances between clusters </td></tr>
<tr id="row_417_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11193.html#a11201" target="_self">TraceFitter::ModelData</a></td><td class="desc"></td></tr>
<tr id="row_418_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11117.html" target="_self">ModelDescription&lt; D &gt;</a></td><td class="desc">Stores the name and parameters of a model </td></tr>
<tr id="row_419_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08021.html" target="_self">ModificationDefinition</a></td><td class="desc">Representation of modification definition </td></tr>
<tr id="row_420_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08025.html" target="_self">ModificationDefinitionsSet</a></td><td class="desc">Representation of a set of modification definitions </td></tr>
<tr id="row_421_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09329" target="_self">MzIdentMLDOMHandler::ModificationParam</a></td><td class="desc">Struct to hold the information from the <a class="el" href="a09309.html#a09329" title="Struct to hold the information from the ModificationParam xml tag.">ModificationParam</a> xml tag </td></tr>
<tr id="row_422_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_422_" class="arrow" onclick="toggleFolder('422_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08029.html" target="_self">ModificationsDB</a></td><td class="desc">Database which holds all residue modifications from UniMod </td></tr>
<tr id="row_422_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07877.html" target="_self">CrossLinksDB</a></td><td class="desc"></td></tr>
<tr id="row_423_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08033.html" target="_self">ModifiedNASequenceGenerator</a></td><td class="desc"></td></tr>
<tr id="row_424_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07521.html" target="_self">ModifiedPeptideGenerator</a></td><td class="desc"></td></tr>
<tr id="row_425_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10489.html" target="_self">IdentificationData::ModifyMultiIndexAddProcessingStep&lt; ElementType &gt;</a></td><td class="desc">Helper functor for adding processing steps to elements in a @t boost::multi_index_container structure </td></tr>
<tr id="row_426_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10493.html" target="_self">IdentificationData::ModifyMultiIndexAddScore&lt; ElementType &gt;</a></td><td class="desc">Helper functor for adding scores to elements in a @t boost::multi_index_container structure </td></tr>
<tr id="row_427_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10497.html" target="_self">IdentificationData::ModifyMultiIndexRemoveParentMatches&lt; ElementType &gt;</a></td><td class="desc">Helper functor for removing invalid parent matches from elements in a @t boost::multi_index_container structure </td></tr>
<tr id="row_428_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07529.html" target="_self">MorpheusScore</a></td><td class="desc">An implementation of the Morpheus PSM scoring function Inspired by a C# implementation by C. Wenger released under MIT license </td></tr>
<tr id="row_429_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07181.html" target="_self">MRMBatchFeatureSelector</a></td><td class="desc"></td></tr>
<tr id="row_430_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07197.html" target="_self">MRMFeaturePicker</a></td><td class="desc">_MRMFeaturePicker_ defines the structures containing parameters to be used in [<a class="el" href="a07253.html" title="The MRMTransitionGroupPicker finds peaks in chromatograms that belong to the same precursors.">MRMTransitionGroupPicker</a>](<a class="el" href="a07253.html">MRMTransitionGroupPicker</a>) for components and components groups </td></tr>
<tr id="row_431_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07209.html" target="_self">MRMFeatureQC</a></td><td class="desc">The <a class="el" href="a07209.html" title="The MRMFeatureQC is a class to handle the parameters and options for MRMFeatureFilter.">MRMFeatureQC</a> is a class to handle the parameters and options for <a class="el" href="a07189.html" title="The MRMFeatureFilter either flags components and/or transitions that do not pass the QC criteria or f...">MRMFeatureFilter</a> </td></tr>
<tr id="row_432_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_432_" class="arrow" onclick="toggleFolder('432_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07225.html" target="_self">MRMFeatureSelector</a></td><td class="desc"></td></tr>
<tr id="row_432_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_432_0_" class="arrow" onclick="toggleFolder('432_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07233.html" target="_self">MRMFeatureSelectorQMIP</a></td><td class="desc"></td></tr>
<tr id="row_432_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07241.html" target="_self">MRMFeatureSelector_test</a></td><td class="desc"></td></tr>
<tr id="row_432_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07237.html" target="_self">MRMFeatureSelectorScore</a></td><td class="desc"></td></tr>
<tr id="row_433_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07245.html" target="_self">MRMIonSeries</a></td><td class="desc">Generate theoretical fragment ion series for use in <a class="el" href="a07177.html" title="Generate assays from a TargetedExperiment.">MRMAssay</a> and <a class="el" href="a07185.html" title="This class generates a TargetedExperiment object with decoys based on a TargetedExperiment object.">MRMDecoy</a> </td></tr>
<tr id="row_434_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07249.html" target="_self">MRMRTNormalizer</a></td><td class="desc">The <a class="el" href="a07249.html" title="The MRMRTNormalizer will find retention time peptides in data.">MRMRTNormalizer</a> will find retention time peptides in data </td></tr>
<tr id="row_435_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06409.html" target="_self">MRMScoring</a></td><td class="desc">This class implements different scores for peaks found in SRM/MRM </td></tr>
<tr id="row_436_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10069.html" target="_self">MRMTransitionGroup&lt; ChromatogramType, TransitionType &gt;</a></td><td class="desc">The representation of a group of transitions in a targeted proteomics experiment </td></tr>
<tr id="row_437_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10069.html" target="_self">MRMTransitionGroup&lt; SpectrumT, TransitionT &gt;</a></td><td class="desc"></td></tr>
<tr id="row_438_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11369.html" target="_self">MS1FeatureMerger</a></td><td class="desc"></td></tr>
<tr id="row_439_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a11333" target="_self">MS1Signal</a></td><td class="desc"></td></tr>
<tr id="row_440_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_440_" class="arrow" onclick="toggleFolder('440_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11377.html" target="_self">MS2ConsensusSpectrum</a></td><td class="desc"></td></tr>
<tr id="row_440_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_440_0_" class="arrow" onclick="toggleFolder('440_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11325.html" target="_self">ClusteredMS2ConsensusSpectrum</a></td><td class="desc"></td></tr>
<tr id="row_440_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11381.html" target="_self">MS2Feature</a></td><td class="desc"></td></tr>
<tr id="row_441_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11385.html" target="_self">MS2Fragment</a></td><td class="desc"></td></tr>
<tr id="row_442_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11389.html" target="_self">MS2Info</a></td><td class="desc"></td></tr>
<tr id="row_443_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07473.html#a07485" target="_self">ProteinResolver::MSDGroup</a></td><td class="desc">Representation of an msd group. Contains peptides, proteins and a pointer to its ISD group </td></tr>
<tr id="row_444_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10445.html" target="_self">ExperimentalDesign::MSFileSectionEntry</a></td><td class="desc"></td></tr>
<tr id="row_445_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09489.html" target="_self">MsInspectFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MsInspect files </td></tr>
<tr id="row_446_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09493.html" target="_self">MSNumpressCoder</a></td><td class="desc">Class to encode and decode data encoded with MSNumpress </td></tr>
<tr id="row_447_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11393.html" target="_self">MSPeak</a></td><td class="desc"></td></tr>
<tr id="row_448_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09509.html" target="_self">MSPGenericFile_friend</a></td><td class="desc"></td></tr>
<tr id="row_449_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09513.html" target="_self">MSstatsFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a> files </td></tr>
<tr id="row_450_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09521.html" target="_self">MSstatsFile::MSstatsLine_</a></td><td class="desc"></td></tr>
<tr id="row_451_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09525.html" target="_self">MSstatsFile::MSstatsTMTLine_</a></td><td class="desc"></td></tr>
<tr id="row_452_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11609.html" target="_self">MultiGradient</a></td><td class="desc">A gradient of multiple colors and arbitrary distances between colors </td></tr>
<tr id="row_453_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11129.html" target="_self">MultiplexDeltaMasses</a></td><td class="desc">Data structure for mass shift pattern </td></tr>
<tr id="row_454_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11125.html" target="_self">MultiplexClustering::MultiplexDistance</a></td><td class="desc">Scaled Euclidean distance for clustering </td></tr>
<tr id="row_455_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11145.html" target="_self">MultiplexFilteredMSExperiment</a></td><td class="desc">Data structure storing all peaks (and optionally their raw data points) of an experiment corresponding to one specific peak pattern </td></tr>
<tr id="row_456_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11149.html" target="_self">MultiplexFilteredPeak</a></td><td class="desc">Data structure storing a single peak that passed all filters </td></tr>
<tr id="row_457_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11165.html" target="_self">MultiplexIsotopicPeakPattern</a></td><td class="desc">Data structure for pattern of isotopic peaks </td></tr>
<tr id="row_458_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11169.html" target="_self">MultiplexSatelliteCentroided</a></td><td class="desc">Data structure storing a single satellite peak </td></tr>
<tr id="row_459_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11173.html" target="_self">MultiplexSatelliteProfile</a></td><td class="desc">Data structure storing a single satellite data point </td></tr>
<tr id="row_460_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html" target="_self">MzIdentMLDOMHandler</a></td><td class="desc">XML DOM handler for <a class="el" href="a09533.html" title="File adapter for MzIdentML files.">MzIdentMLFile</a> </td></tr>
<tr id="row_461_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09357.html" target="_self">MzMLHandlerHelper</a></td><td class="desc">Helper for mzML file format </td></tr>
<tr id="row_462_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09365.html" target="_self">MzMLSpectrumDecoder</a></td><td class="desc">A class to decode input strings that contain an mzML chromatogram or spectrum tag </td></tr>
<tr id="row_463_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09369.html" target="_self">MzMLSqliteHandler</a></td><td class="desc">Sqlite handler for storing spectra and chromatograms in sqMass format </td></tr>
<tr id="row_464_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09373.html" target="_self">MzMLSqliteSwathHandler</a></td><td class="desc">Sqlite handler for SWATH data sets </td></tr>
<tr id="row_465_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09701.html" target="_self">MzTab</a></td><td class="desc">Data model of <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a> files. Please see the official <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a> specification at <a href="https://code.google.com/p/mztab/">https://code.google.com/p/mztab/</a> </td></tr>
<tr id="row_466_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09621" target="_self">MzTabAssayMetaData</a></td><td class="desc"></td></tr>
<tr id="row_467_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09633" target="_self">MzTabContactMetaData</a></td><td class="desc"></td></tr>
<tr id="row_468_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09625" target="_self">MzTabCVMetaData</a></td><td class="desc"></td></tr>
<tr id="row_469_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09705.html" target="_self">MzTabFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a> files </td></tr>
<tr id="row_470_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09629" target="_self">MzTabInstrumentMetaData</a></td><td class="desc"></td></tr>
<tr id="row_471_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09645.html" target="_self">MzTabMetaData</a></td><td class="desc">All meta data of a mzTab file. Please refer to specification for documentation </td></tr>
<tr id="row_472_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09617" target="_self">MzTabModificationMetaData</a></td><td class="desc"></td></tr>
<tr id="row_473_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09637" target="_self">MzTabMSRunMetaData</a></td><td class="desc"></td></tr>
<tr id="row_474_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09677" target="_self">MzTabNucleicAcidSectionRow</a></td><td class="desc">NUC - Nucleic acid section (table-based) </td></tr>
<tr id="row_475_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_475_" class="arrow" onclick="toggleFolder('475_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09545.html" target="_self">MzTabNullAbleInterface</a></td><td class="desc">Basic interface for all <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a> datatypes (can be null; are converted from and to cell string) </td></tr>
<tr id="row_475_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09597.html" target="_self">MzTabModification</a></td><td class="desc"></td></tr>
<tr id="row_475_1_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_475_1_" class="arrow" onclick="toggleFolder('475_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09553.html" target="_self">MzTabNullAbleBase</a></td><td class="desc">Base class for atomic, non-container types (Double, Int) </td></tr>
<tr id="row_475_1_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09577.html" target="_self">MzTabBoolean</a></td><td class="desc"></td></tr>
<tr id="row_475_1_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09565.html" target="_self">MzTabDoubleList</a></td><td class="desc"></td></tr>
<tr id="row_475_1_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09573.html" target="_self">MzTabIntegerList</a></td><td class="desc"></td></tr>
<tr id="row_475_1_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09601.html" target="_self">MzTabModificationList</a></td><td class="desc"></td></tr>
<tr id="row_475_2_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_475_2_" class="arrow" onclick="toggleFolder('475_2_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09549.html" target="_self">MzTabNullNaNAndInfAbleInterface</a></td><td class="desc">Interface for NaN- and Inf- able datatypes (Double and Integer in <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a>). These are as well null-able </td></tr>
<tr id="row_475_2_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_475_2_0_" class="arrow" onclick="toggleFolder('475_2_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09557.html" target="_self">MzTabNullNaNAndInfAbleBase</a></td><td class="desc">Base class for the atomic non-container like <a class="el" href="a09701.html" title="Data model of MzTab files. Please see the official MzTab specification at https://code....">MzTab</a> data types (Double, Int) </td></tr>
<tr id="row_475_2_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09561.html" target="_self">MzTabDouble</a></td><td class="desc"></td></tr>
<tr id="row_475_2_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09569.html" target="_self">MzTabInteger</a></td><td class="desc"></td></tr>
<tr id="row_475_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09585.html" target="_self">MzTabParameter</a></td><td class="desc"></td></tr>
<tr id="row_475_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09589.html" target="_self">MzTabParameterList</a></td><td class="desc"></td></tr>
<tr id="row_475_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09605.html" target="_self">MzTabSpectraRef</a></td><td class="desc"></td></tr>
<tr id="row_475_6_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09581.html" target="_self">MzTabString</a></td><td class="desc"></td></tr>
<tr id="row_475_7_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09593.html" target="_self">MzTabStringList</a></td><td class="desc"></td></tr>
<tr id="row_476_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09685" target="_self">MzTabOligonucleotideSectionRow</a></td><td class="desc">OLI - Oligonucleotide section (table-based) </td></tr>
<tr id="row_477_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09693" target="_self">MzTabOSMSectionRow</a></td><td class="desc">OSM - OSM (oligonucleotide-spectrum match) section (table-based) </td></tr>
<tr id="row_478_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09657" target="_self">MzTabPeptideSectionRow</a></td><td class="desc">PEP - Peptide section (Table based) </td></tr>
<tr id="row_479_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09649.html" target="_self">MzTabProteinSectionRow</a></td><td class="desc">PRT - Protein section (Table based) </td></tr>
<tr id="row_480_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09665" target="_self">MzTabPSMSectionRow</a></td><td class="desc">PSM - PSM section (Table based) </td></tr>
<tr id="row_481_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09609" target="_self">MzTabSampleMetaData</a></td><td class="desc"></td></tr>
<tr id="row_482_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09673" target="_self">MzTabSmallMoleculeSectionRow</a></td><td class="desc">SML Small molecule section (table based) </td></tr>
<tr id="row_483_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09613" target="_self">MzTabSoftwareMetaData</a></td><td class="desc"></td></tr>
<tr id="row_484_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a09641" target="_self">MzTabStudyVariableMetaData</a></td><td class="desc"></td></tr>
<tr id="row_485_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08833.html" target="_self">MZTrafoModel</a></td><td class="desc">Create and apply models of a mass recalibration function </td></tr>
<tr id="row_486_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08037.html" target="_self">NASequence</a></td><td class="desc">Representation of a nucleic acid sequence </td></tr>
<tr id="row_487_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08893.html" target="_self">SplineInterpolatedPeaks::Navigator</a></td><td class="desc">Iterator class for access of spline packages </td></tr>
<tr id="row_488_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08969.html" target="_self">SignalToNoiseEstimatorMedianRapid::NoiseEstimator</a></td><td class="desc">Class to compute the noise value at a given position </td></tr>
<tr id="row_489_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08673.html" target="_self">_Alloc_base&lt; _Tp, _Alloc &gt;::NoLeakAlloc</a></td><td class="desc"></td></tr>
<tr id="row_490_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10261.html" target="_self">NonNegativeLeastSquaresSolver</a></td><td class="desc">Wrapper for a non-negative least squares (<a class="el" href="a05848.html">NNLS</a>) solver </td></tr>
<tr id="row_491_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09497.html" target="_self">MSNumpressCoder::NumpressConfig</a></td><td class="desc">Configuration class for MSNumpress </td></tr>
<tr id="row_492_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06621.html" target="_self">OMS_TranslateTableAAToChar_&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_493_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06629.html" target="_self">OMS_TranslateTableByteToAA_&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_494_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06625.html" target="_self">OMS_TranslateTableCharToAA_&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_495_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09713.html" target="_self">OMSSACSVFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for OMSSACSV files </td></tr>
<tr id="row_496_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10105.html" target="_self">OnDiscMSExperiment</a></td><td class="desc">Representation of a mass spectrometry experiment on disk </td></tr>
<tr id="row_497_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a12009.html" target="_self">OpenMSOSInfo</a></td><td class="desc"></td></tr>
<tr id="row_498_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07273.html" target="_self">OpenSwath_Ind_Scores</a></td><td class="desc"></td></tr>
<tr id="row_499_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07269.html" target="_self">OpenSwath_Scores</a></td><td class="desc">A structure to hold the different scores computed by OpenSWATH </td></tr>
<tr id="row_500_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a07265" target="_self">OpenSwath_Scores_Usage</a></td><td class="desc">A structure to store which scores should be used by the OpenSWATH Algorithm </td></tr>
<tr id="row_501_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07113.html" target="_self">OpenSwathDataAccessHelper</a></td><td class="desc">Several helpers to convert <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> datastructures to structures that implement the OpenSWATH interfaces </td></tr>
<tr id="row_502_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07257.html" target="_self">OpenSwathHelper</a></td><td class="desc">A helper class that is used by several OpenSWATH tools </td></tr>
<tr id="row_503_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07261.html" target="_self">OpenSwathOSWWriter</a></td><td class="desc">Class to write out an <a class="el" href="a05820.html">OpenSwath</a> OSW SQLite output (PyProphet input) </td></tr>
<tr id="row_504_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07277.html" target="_self">OpenSwathScoring</a></td><td class="desc">A class that calls the scoring routines </td></tr>
<tr id="row_505_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07281.html" target="_self">OpenSwathTSVWriter</a></td><td class="desc">Class to write out an <a class="el" href="a05820.html">OpenSwath</a> TSV output (mProphet input) </td></tr>
<tr id="row_506_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11373.html" target="_self">MS1FeatureMerger::OPERATOR_FEATURE_TR</a></td><td class="desc"></td></tr>
<tr id="row_507_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11361.html" target="_self">LCMS::OPERATOR_FeatureCompare</a></td><td class="desc"></td></tr>
<tr id="row_508_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11357.html" target="_self">LCMS::OPERATOR_MZ</a></td><td class="desc"></td></tr>
<tr id="row_509_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11229.html" target="_self">OptimizePick</a></td><td class="desc">This class provides the non-linear optimization of the peak parameters </td></tr>
<tr id="row_510_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11237.html" target="_self">OptimizePick::OptPeakFunctor</a></td><td class="desc"></td></tr>
<tr id="row_511_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07765.html" target="_self">OPXLDataStructs</a></td><td class="desc"></td></tr>
<tr id="row_512_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07801.html" target="_self">OPXLHelper</a></td><td class="desc">Functions needed by OpenPepXL and OpenPepXLLF to reduce duplicated code </td></tr>
<tr id="row_513_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07809.html" target="_self">OPXLSpectrumProcessingAlgorithms</a></td><td class="desc"></td></tr>
<tr id="row_514_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05820.html#a06437" target="_self">OSBinaryDataArray</a></td><td class="desc">The datastructures used by the <a class="el" href="a05820.html">OpenSwath</a> interfaces </td></tr>
<tr id="row_515_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06445.html" target="_self">OSChromatogram</a></td><td class="desc">A single chromatogram </td></tr>
<tr id="row_516_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06441.html" target="_self">OSChromatogramMeta</a></td><td class="desc">Identifying information for a chromatogram </td></tr>
<tr id="row_517_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06457.html" target="_self">OSSpectrum</a></td><td class="desc">The structure that captures the generation of a peak list (including the underlying acquisitions) </td></tr>
<tr id="row_518_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06449.html" target="_self">OSSpectrumMeta</a></td><td class="desc">Identifying information for a spectrum </td></tr>
<tr id="row_519_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09729.html" target="_self">OSWFile</a></td><td class="desc">This class serves for reading in and writing OpenSWATH OSW files </td></tr>
<tr id="row_520_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05824.html#a06713" target="_self">PAM30MS</a></td><td class="desc"></td></tr>
<tr id="row_521_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08745.html" target="_self">Param</a></td><td class="desc">Management and storage of parameters / INI files </td></tr>
<tr id="row_522_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08749.html" target="_self">Param::ParamEntry</a></td><td class="desc">Parameter entry used to store the actual information inside of a <a class="el" href="a08745.html" title="Management and storage of parameters / INI files.">Param</a> entry </td></tr>
<tr id="row_523_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11925.html#a11929" target="_self">FLASHDeconv::Parameter</a></td><td class="desc"></td></tr>
<tr id="row_524_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07725.html#a07729" target="_self">FLASHDeconvHelperStructs::Parameter</a></td><td class="desc"></td></tr>
<tr id="row_525_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07833.html" target="_self">ParameterInformation</a></td><td class="desc">Struct that captures all information of a command line parameter </td></tr>
<tr id="row_526_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08757.html" target="_self">Param::ParamIterator</a></td><td class="desc">Forward const iterator for the <a class="el" href="a08745.html" title="Management and storage of parameters / INI files.">Param</a> class </td></tr>
<tr id="row_527_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08753.html" target="_self">Param::ParamNode</a></td><td class="desc">Node inside a <a class="el" href="a08745.html" title="Management and storage of parameters / INI files.">Param</a> object which is used to build the internal tree </td></tr>
<tr id="row_528_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05852.html#a10529" target="_self">ParentMoleculeGroup</a></td><td class="desc">: Group of ambiguously identified parent molecules (e.g. protein group) </td></tr>
<tr id="row_529_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06661.html" target="_self">Pattern&lt; TNeedle, FuzzyAC &gt;</a></td><td class="desc"></td></tr>
<tr id="row_530_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06657.html" target="_self">PatternAuxData&lt; TNeedle &gt;</a></td><td class="desc"></td></tr>
<tr id="row_531_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06657.html" target="_self">PatternAuxData&lt; PeptideDB &gt;</a></td><td class="desc"></td></tr>
<tr id="row_532_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07993.html" target="_self">IMSIsotopeDistribution::Peak</a></td><td class="desc">Structure that represents an isotope peak - pair of mass and abundance </td></tr>
<tr id="row_533_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_533_" class="arrow" onclick="toggleFolder('533_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10109.html" target="_self">Peak1D</a></td><td class="desc">A 1-dimensional raw data point or peak </td></tr>
<tr id="row_533_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10649.html" target="_self">Precursor</a></td><td class="desc"><a class="el" href="a10649.html" title="Precursor meta information.">Precursor</a> meta information </td></tr>
<tr id="row_534_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_534_" class="arrow" onclick="toggleFolder('534_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10125.html" target="_self">Peak2D</a></td><td class="desc">A 2-dimensional raw data point or peak </td></tr>
<tr id="row_534_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_534_0_" class="arrow" onclick="toggleFolder('534_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10041.html" target="_self">FeatureHandle</a></td><td class="desc">Representation of a <a class="el" href="a10125.html" title="A 2-dimensional raw data point or peak.">Peak2D</a>, <a class="el" href="a10213.html" title="A 2-dimensional raw data point or peak with meta information.">RichPeak2D</a> or <a class="el" href="a10037.html" title="An LC-MS feature.">Feature</a> </td></tr>
<tr id="row_534_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10049.html" target="_self">FeatureHandle::FeatureHandleMutable_</a></td><td class="desc">Helper class returned by <a class="el" href="a10041.html#a45b6e138344daa371ca90b359e8fd86f" title="Override (most of all) constness.">FeatureHandle::asMutable()</a>, which see </td></tr>
<tr id="row_534_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10213.html" target="_self">RichPeak2D</a></td><td class="desc">A 2-dimensional raw data point or peak with meta information </td></tr>
<tr id="row_535_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10621.html" target="_self">PeptideHit::PeakAnnotation</a></td><td class="desc">Contains annotations of a peak </td></tr>
<tr id="row_536_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07305.html#a07309" target="_self">PeakIntegrator::PeakArea</a></td><td class="desc"></td></tr>
<tr id="row_537_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11241.html#a11245" target="_self">PeakPickerCWT::PeakArea_</a></td><td class="desc">Class for the internal peak representation </td></tr>
<tr id="row_538_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07305.html#a07313" target="_self">PeakIntegrator::PeakBackground</a></td><td class="desc"></td></tr>
<tr id="row_539_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11249.html#a11253" target="_self">PeakPickerHiRes::PeakBoundary</a></td><td class="desc">Structure for peak boundaries </td></tr>
<tr id="row_540_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a11257" target="_self">PeakCandidate</a></td><td class="desc">A small structure to hold peak candidates </td></tr>
<tr id="row_541_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11265.html#a11269" target="_self">PeakPickerMaxima::PeakCandidate</a></td><td class="desc">The <a class="el" href="a11265.html#a11269" title="The PeakCandidate describes the output of the peak picker.">PeakCandidate</a> describes the output of the peak picker </td></tr>
<tr id="row_542_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09725.html" target="_self">PeakFileOptions</a></td><td class="desc">Options for loading files containing peak data </td></tr>
<tr id="row_543_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11941.html" target="_self">FLASHDeconv::PeakGroup</a></td><td class="desc"></td></tr>
<tr id="row_544_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07741.html" target="_self">FLASHDeconvHelperStructs::PeakGroup</a></td><td class="desc"></td></tr>
<tr id="row_545_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07749.html" target="_self">PeakGroupScoring</a></td><td class="desc"></td></tr>
<tr id="row_546_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10145.html" target="_self">PeakIndex</a></td><td class="desc">Index of a peak or feature </td></tr>
<tr id="row_547_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07381.html" target="_self">PeakIntensityPredictor</a></td><td class="desc">Predict peak heights of peptides based on Local Linear Map model </td></tr>
<tr id="row_548_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11265.html" target="_self">PeakPickerMaxima</a></td><td class="desc">This class implements a fast peak-picking algorithm best suited for high resolution MS data (FT-ICR-MS, Orbitrap). In high resolution data, the signals of ions with similar mass-to-charge ratios (m/z) exhibit little or no overlapping and therefore allow for a clear separation. Furthermore, ion signals tend to show well-defined peak shapes with narrow peak width </td></tr>
<tr id="row_549_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11277.html" target="_self">PeakShape</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> representation of a peak shape (used by the <a class="el" href="a11241.html" title="This class implements a peak picking algorithm using wavelet techniques.">PeakPickerCWT</a>) </td></tr>
<tr id="row_550_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07305.html#a07317" target="_self">PeakIntegrator::PeakShapeMetrics</a></td><td class="desc"></td></tr>
<tr id="row_551_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09737.html" target="_self">PeakTypeEstimator</a></td><td class="desc">Estimates if the data of a spectrum is raw data or peak data </td></tr>
<tr id="row_552_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11177.html" target="_self">PeakWidthEstimator</a></td><td class="desc">Rough estimation of the peak width at m/z </td></tr>
<tr id="row_553_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_553_" class="arrow" onclick="toggleFolder('553_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11225.html" target="_self">PenaltyFactors</a></td><td class="desc">Class for the penalty factors used during the optimization </td></tr>
<tr id="row_553_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11213.html" target="_self">PenaltyFactorsIntensity</a></td><td class="desc">Class for the penalty factors used during the optimization </td></tr>
<tr id="row_554_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09741.html" target="_self">PepNovoInfile</a></td><td class="desc">PepNovo input file adapter </td></tr>
<tr id="row_555_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09745.html" target="_self">PepNovoOutfile</a></td><td class="desc">Representation of a PepNovo output file </td></tr>
<tr id="row_556_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06525.html" target="_self">Peptide</a></td><td class="desc"></td></tr>
<tr id="row_557_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11041.html" target="_self">FeatureFinderIdentificationAlgorithm::PeptideCompare</a></td><td class="desc">Comparison functor for (unassigned) peptide IDs </td></tr>
<tr id="row_558_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07457.html" target="_self">PeptideAndProteinQuant::PeptideData</a></td><td class="desc">Quantitative and associated data for a peptide </td></tr>
<tr id="row_559_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08937.html" target="_self">IDFilter::PeptideDigestionFilter</a></td><td class="desc">Filter Peptide Hit by its digestion product </td></tr>
<tr id="row_560_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07473.html#a07481" target="_self">ProteinResolver::PeptideEntry</a></td><td class="desc">Peptide. First in silico. If experimental is set to true it is MS/MS derived </td></tr>
<tr id="row_561_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10613.html" target="_self">PeptideEvidence</a></td><td class="desc">Representation of a peptide evidence </td></tr>
<tr id="row_562_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09317" target="_self">MzIdentMLDOMHandler::PeptideEvidence</a></td><td class="desc">Struct to hold the <a class="el" href="a09309.html#a09317" title="Struct to hold the PeptideEvidence information.">PeptideEvidence</a> information </td></tr>
<tr id="row_563_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07805.html" target="_self">OPXLHelper::PeptideIDScoreComparator</a></td><td class="desc">A comparator for PeptideIdentifications that compares the scores in the first <a class="el" href="a10617.html" title="Representation of a peptide hit.">PeptideHit</a> </td></tr>
<tr id="row_564_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06853.html" target="_self">PeptideIndexing::PeptideProteinMatchInformation</a></td><td class="desc"></td></tr>
<tr id="row_565_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06865.html" target="_self">PeptideProteinResolution</a></td><td class="desc">Resolves shared peptides based on protein scores </td></tr>
<tr id="row_566_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10641.html" target="_self">PeptideHit::PepXMLAnalysisResult</a></td><td class="desc">Analysis Result (containing search engine / prophet results) </td></tr>
<tr id="row_567_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06869.html" target="_self">PercolatorFeatureSetHelper</a></td><td class="desc">Percolator feature set and integration helper </td></tr>
<tr id="row_568_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09761.html" target="_self">PercolatorOutfile</a></td><td class="desc">Class for reading Percolator tab-delimited output files </td></tr>
<tr id="row_569_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06557.html" target="_self">CompNovoIdentificationBase::Permut</a></td><td class="desc">Simple class to store permutations and a score </td></tr>
<tr id="row_570_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11281.html" target="_self">PeakShape::PositionLess</a></td><td class="desc">Comparison of mz_positions </td></tr>
<tr id="row_571_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08737.html" target="_self">PpmTrait</a></td><td class="desc"></td></tr>
<tr id="row_572_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07733.html" target="_self">FLASHDeconvHelperStructs::PrecalcularedAveragine</a></td><td class="desc"></td></tr>
<tr id="row_573_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11933.html" target="_self">FLASHDeconv::PrecalcularedAveragine</a></td><td class="desc"></td></tr>
<tr id="row_574_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08477.html" target="_self">PrecisionWrapper&lt; FloatingPointType &gt;</a></td><td class="desc">Wrapper class to implement output with appropriate precision. See <a class="el" href="a05822.html#af27ce8637a6a7cc24cc6fcbd3358791d" title="Wrapper function that sets the appropriate precision for output temporarily. The original precision i...">precisionWrapper()</a> </td></tr>
<tr id="row_575_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08841.html" target="_self">PrecursorCorrection</a></td><td class="desc">This class provides methods for precursor correction </td></tr>
<tr id="row_576_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11997.html" target="_self">NucleicAcidSearchEngine::PrecursorInfo</a></td><td class="desc"></td></tr>
<tr id="row_577_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06833.html" target="_self">PrecursorMassComparator</a></td><td class="desc"></td></tr>
<tr id="row_578_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06881.html" target="_self">PrecursorPurity</a></td><td class="desc"><a class="el" href="a10649.html" title="Precursor meta information.">Precursor</a> purity or noise estimation </td></tr>
<tr id="row_579_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07573.html#a07577" target="_self">SimpleSVM::Prediction</a></td><td class="desc">SVM prediction result </td></tr>
<tr id="row_580_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08445.html" target="_self">FuzzyStringComparator::PrefixInfo_</a></td><td class="desc">Wrapper for the prefix information computed for the failure report </td></tr>
<tr id="row_581_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07797.html" target="_self">OPXLDataStructs::PreprocessedPairSpectra</a></td><td class="desc">The <a class="el" href="a07797.html" title="The PreprocessedPairSpectra struct represents the result of comparing a light and a heavy labeled spe...">PreprocessedPairSpectra</a> struct represents the result of comparing a light and a heavy labeled spectra to each other </td></tr>
<tr id="row_582_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a06677" target="_self">ProbablePhosphoSites</a></td><td class="desc"></td></tr>
<tr id="row_583_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11397.html" target="_self">ProcessData</a></td><td class="desc"></td></tr>
<tr id="row_584_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05801.html#a11181" target="_self">ProductModel&lt; D &gt;</a></td><td class="desc">Class for product models i.e. models with D independent dimensions </td></tr>
<tr id="row_585_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_585_" class="arrow" onclick="toggleFolder('585_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08481.html" target="_self">ProgressLogger</a></td><td class="desc">Base class for all classes that want to report their progress </td></tr>
<tr id="row_585_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06609.html" target="_self">AccurateMassSearchEngine</a></td><td class="desc">An algorithm to search for exact mass matches from a spectrum against a database (e.g. HMDB) </td></tr>
<tr id="row_585_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08149.html" target="_self">AverageLinkage</a></td><td class="desc"><a class="el" href="a08149.html" title="AverageLinkage ClusterMethod.">AverageLinkage</a> ClusterMethod </td></tr>
<tr id="row_585_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06921.html" target="_self">BaseGroupFinder</a></td><td class="desc">The base class of all element group finding algorithms </td></tr>
<tr id="row_585_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06925.html" target="_self">BaseSuperimposer</a></td><td class="desc">The base class of all superimposer algorithms </td></tr>
<tr id="row_585_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06685.html" target="_self">BasicProteinInferenceAlgorithm</a></td><td class="desc">Algorithm class that implements simple protein inference by aggregation of peptide scores. It has multiple parameter options like the aggregation method, when to distinguish peptidoforms, and if you want to use shared peptides ("use_shared_peptides"). First, the best PSM per spectrum is used, then only the best PSM per peptidoform is aggregated. Peptidoforms can optionally be distinguished via the treat_X_separate parameters: </td></tr>
<tr id="row_585_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06689.html" target="_self">BayesianProteinInferenceAlgorithm</a></td><td class="desc">Performs a Bayesian protein inference on Protein/Peptide identifications or <a class="el" href="a10021.html" title="A container for consensus elements.">ConsensusMap</a> (experimental) </td></tr>
<tr id="row_585_6_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07089.html" target="_self">ChromatogramExtractor</a></td><td class="desc">The <a class="el" href="a07089.html" title="The ChromatogramExtractor extracts chromatograms from a spectra file.">ChromatogramExtractor</a> extracts chromatograms from a spectra file </td></tr>
<tr id="row_585_7_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07093.html" target="_self">ChromatogramExtractorAlgorithm</a></td><td class="desc">The <a class="el" href="a07093.html" title="The ChromatogramExtractorAlgorithm extracts chromatograms from a MS data.">ChromatogramExtractorAlgorithm</a> extracts chromatograms from a MS data </td></tr>
<tr id="row_585_8_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08177.html" target="_self">CompleteLinkage</a></td><td class="desc"><a class="el" href="a08177.html" title="CompleteLinkage ClusterMethod.">CompleteLinkage</a> ClusterMethod </td></tr>
<tr id="row_585_9_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07101.html" target="_self">ConfidenceScoring</a></td><td class="desc"></td></tr>
<tr id="row_585_10_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06745.html" target="_self">ConsensusMapMergerAlgorithm</a></td><td class="desc">Merges identification data in ConsensusMaps </td></tr>
<tr id="row_585_11_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09125.html" target="_self">ConsensusXMLFile</a></td><td class="desc">This class provides Input functionality for ConsensusMaps and Output functionality for alignments and quantitation </td></tr>
<tr id="row_585_12_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09237.html" target="_self">DTA2DFile</a></td><td class="desc">DTA2D <a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter </td></tr>
<tr id="row_585_13_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08861.html" target="_self">ElutionPeakDetection</a></td><td class="desc">Extracts chromatographic peaks from a mass trace </td></tr>
<tr id="row_585_14_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10965.html" target="_self">FeatureFinder</a></td><td class="desc">The main feature finder class </td></tr>
<tr id="row_585_15_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11049.html" target="_self">FeatureFinderMultiplexAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_585_16_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08877.html" target="_self">FeatureFindingMetabo</a></td><td class="desc">Method for the assembly of mass traces belonging to the same isotope pattern, i.e., that are compatible in retention times, mass-to-charge ratios, and isotope abundances </td></tr>
<tr id="row_585_17_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06957.html" target="_self">FeatureGroupingAlgorithmKD</a></td><td class="desc">A feature grouping algorithm for unlabeled data </td></tr>
<tr id="row_585_18_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09261.html" target="_self">FeatureXMLFile</a></td><td class="desc">This class provides Input/Output functionality for feature maps </td></tr>
<tr id="row_585_19_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08973.html" target="_self">GaussFilter</a></td><td class="desc">This class represents a Gaussian lowpass-filter which works on uniform as well as on non-uniform profile data </td></tr>
<tr id="row_585_20_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08193.html" target="_self">GridBasedClustering&lt; Metric &gt;</a></td><td class="desc">2D hierarchical clustering implementation optimized for large data sets containing many small clusters i.e. dimensions of clusters &lt;&lt; dimension of entire dataset </td></tr>
<tr id="row_585_21_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06805.html" target="_self">IDMergerAlgorithm</a></td><td class="desc">Creates a new Protein ID run into which other runs can be inserted. Creates union of protein hits but concatenates PSMs. Checks search engine consistency of all inserted runs. It differs from the IDMerger tool, in that it is an algorithm class and it allows inserting multiple peptide hits per peptide sequence (not only the first occurrence) </td></tr>
<tr id="row_585_22_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09437.html" target="_self">IdXMLFile</a></td><td class="desc">Used to load and store idXML files </td></tr>
<tr id="row_585_23_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_585_23_" class="arrow" onclick="toggleFolder('585_23_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09285.html" target="_self">CachedMzMLHandler</a></td><td class="desc">An class that uses on-disk caching to read and write spectra and chromatograms </td></tr>
<tr id="row_585_23_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09165.html" target="_self">MSDataCachedConsumer</a></td><td class="desc">Transforming and cached writing consumer of MS data </td></tr>
<tr id="row_585_24_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08825.html" target="_self">InternalCalibration</a></td><td class="desc">A mass recalibration method using linear/quadratic interpolation (robust/weighted) of given reference masses </td></tr>
<tr id="row_585_25_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10817.html" target="_self">IonizationSimulation</a></td><td class="desc">Simulates Protein ionization </td></tr>
<tr id="row_585_26_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09021.html" target="_self">LinearResampler</a></td><td class="desc">Linear Resampling of raw data </td></tr>
<tr id="row_585_27_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06985.html" target="_self">MapAlignmentAlgorithmIdentification</a></td><td class="desc">A map alignment algorithm based on peptide identifications from MS2 spectra </td></tr>
<tr id="row_585_28_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06993.html" target="_self">MapAlignmentAlgorithmPoseClustering</a></td><td class="desc">A map alignment algorithm based on pose clustering </td></tr>
<tr id="row_585_29_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06997.html" target="_self">MapAlignmentAlgorithmSpectrumAlignment</a></td><td class="desc">A map alignment algorithm based on spectrum similarity (dynamic programming) </td></tr>
<tr id="row_585_30_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09461.html" target="_self">MascotGenericFile</a></td><td class="desc">Read/write Mascot generic files (MGF) </td></tr>
<tr id="row_585_31_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09465.html" target="_self">MascotInfile</a></td><td class="desc">Mascot input file adapter </td></tr>
<tr id="row_585_32_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07173.html" target="_self">MasstraceCorrelator</a></td><td class="desc">Correlates individual masstraces found in mass spectrometric maps </td></tr>
<tr id="row_585_33_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08885.html" target="_self">MassTraceDetection</a></td><td class="desc">A mass trace extraction method that gathers peaks similar in m/z and moving along retention time </td></tr>
<tr id="row_585_34_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06845.html" target="_self">MetaboliteSpectralMatching</a></td><td class="desc"></td></tr>
<tr id="row_585_35_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08821.html" target="_self">MorphologicalFilter</a></td><td class="desc">This class implements baseline filtering operations using methods from mathematical morphology </td></tr>
<tr id="row_585_36_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07177.html" target="_self">MRMAssay</a></td><td class="desc">Generate assays from a <a class="el" href="a07661.html" title="A description of a targeted experiment containing precursor and production ions.">TargetedExperiment</a> </td></tr>
<tr id="row_585_37_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07185.html" target="_self">MRMDecoy</a></td><td class="desc">This class generates a <a class="el" href="a07661.html" title="A description of a targeted experiment containing precursor and production ions.">TargetedExperiment</a> object with decoys based on a <a class="el" href="a07661.html" title="A description of a targeted experiment containing precursor and production ions.">TargetedExperiment</a> object </td></tr>
<tr id="row_585_38_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07193.html" target="_self">MRMFeatureFinderScoring</a></td><td class="desc">The MRMFeatureFinder finds and scores peaks of transitions that co-elute </td></tr>
<tr id="row_585_39_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09481.html" target="_self">MRMFeatureQCFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for <a class="el" href="a07209.html" title="The MRMFeatureQC is a class to handle the parameters and options for MRMFeatureFilter.">MRMFeatureQC</a> files </td></tr>
<tr id="row_585_40_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09485.html" target="_self">MS2File</a></td><td class="desc">MS2 input file adapter </td></tr>
<tr id="row_585_41_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10845.html" target="_self">MSSim</a></td><td class="desc">Central class for simulation of mass spectrometry experiments </td></tr>
<tr id="row_585_42_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11121.html" target="_self">MultiplexClustering</a></td><td class="desc">Clusters results from multiplex filtering </td></tr>
<tr id="row_585_43_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_585_43_" class="arrow" onclick="toggleFolder('585_43_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11153.html" target="_self">MultiplexFiltering</a></td><td class="desc">Base class for filtering centroided and profile data for peak patterns </td></tr>
<tr id="row_585_43_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11157.html" target="_self">MultiplexFilteringCentroided</a></td><td class="desc">Filters centroided data for peak patterns </td></tr>
<tr id="row_585_43_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11161.html" target="_self">MultiplexFilteringProfile</a></td><td class="desc">Filters centroided and profile data for peak patterns </td></tr>
<tr id="row_585_44_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09529.html" target="_self">MzDataFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzData files </td></tr>
<tr id="row_585_45_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09533.html" target="_self">MzIdentMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzIdentML files </td></tr>
<tr id="row_585_46_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09537.html" target="_self">MzMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzML files </td></tr>
<tr id="row_585_47_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09541.html" target="_self">MzQuantMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzQuantML files </td></tr>
<tr id="row_585_48_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09709.html" target="_self">MzXMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzXML 3.1 files </td></tr>
<tr id="row_585_49_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07757.html" target="_self">OpenPepXLAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_585_50_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07761.html" target="_self">OpenPepXLLFAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_585_51_" class="even" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_585_51_" class="arrow" onclick="toggleFolder('585_51_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07289.html" target="_self">OpenSwathWorkflowBase</a></td><td class="desc"></td></tr>
<tr id="row_585_51_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07293.html" target="_self">OpenSwathCalibrationWorkflow</a></td><td class="desc">Execute all steps for retention time and m/z calibration of SWATH-MS data </td></tr>
<tr id="row_585_51_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_585_51_1_" class="arrow" onclick="toggleFolder('585_51_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07297.html" target="_self">OpenSwathWorkflow</a></td><td class="desc">Execute all steps in an <a class="el" href="a16402.html">OpenSwath</a> analysis </td></tr>
<tr id="row_585_51_1_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07301.html" target="_self">OpenSwathWorkflowSonar</a></td><td class="desc">Execute all steps in an OpenEcho analysis (<a class="el" href="a05820.html">OpenSwath</a> for SONAR data) </td></tr>
<tr id="row_585_52_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11241.html" target="_self">PeakPickerCWT</a></td><td class="desc">This class implements a peak picking algorithm using wavelet techniques </td></tr>
<tr id="row_585_53_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11249.html" target="_self">PeakPickerHiRes</a></td><td class="desc">This class implements a fast peak-picking algorithm best suited for high resolution MS data (FT-ICR-MS, Orbitrap). In high resolution data, the signals of ions with similar mass-to-charge ratios (m/z) exhibit little or no overlapping and therefore allow for a clear separation. Furthermore, ion signals tend to show well-defined peak shapes with narrow peak width </td></tr>
<tr id="row_585_54_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11261.html" target="_self">PeakPickerIterative</a></td><td class="desc">This class implements a peak-picking algorithm for high-resolution MS data (specifically designed for TOF-MS data) </td></tr>
<tr id="row_585_55_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11273.html" target="_self">PeakPickerSH</a></td><td class="desc"></td></tr>
<tr id="row_585_56_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06849.html" target="_self">PeptideIndexing</a></td><td class="desc">Refreshes the protein references for all peptide hits in a vector of PeptideIdentifications and adds target/decoy information </td></tr>
<tr id="row_585_57_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09773.html" target="_self">QcMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for QcML files used to load and store QcML files </td></tr>
<tr id="row_585_58_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10849.html" target="_self">RawMSSignalSimulation</a></td><td class="desc">Simulates MS signals for a given set of peptides </td></tr>
<tr id="row_585_59_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08985.html" target="_self">SavitzkyGolayFilter</a></td><td class="desc">Computes the Savitzky-Golay filter coefficients using QR decomposition </td></tr>
<tr id="row_585_60_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html" target="_self">SignalToNoiseEstimator&lt; Container &gt;</a></td><td class="desc">This class represents the abstract base class of a signal to noise estimator </td></tr>
<tr id="row_585_61_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06893.html" target="_self">SimpleSearchEngineAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_585_62_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08209.html" target="_self">SingleLinkage</a></td><td class="desc"><a class="el" href="a08209.html" title="SingleLinkage ClusterMethod.">SingleLinkage</a> ClusterMethod </td></tr>
<tr id="row_585_63_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09061.html" target="_self">SpectraMerger</a></td><td class="desc">Merges blocks of MS or MS2 spectra </td></tr>
<tr id="row_585_64_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07585.html" target="_self">SVMWrapper</a></td><td class="desc">Serves as a wrapper for the libsvm </td></tr>
<tr id="row_585_65_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09813.html" target="_self">SwathFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for Swath files </td></tr>
<tr id="row_585_66_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08845.html" target="_self">TOFCalibration</a></td><td class="desc">This class implements an external calibration for TOF data using external calibrant spectra </td></tr>
<tr id="row_585_67_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09821.html" target="_self">ToolDescriptionFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for ToolDescriptor files </td></tr>
<tr id="row_585_68_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09825.html" target="_self">TraMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for HUPO PSI TraML files </td></tr>
<tr id="row_585_69_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07365.html" target="_self">TransitionTSVFile</a></td><td class="desc">This class supports reading and writing of OpenSWATH transition lists </td></tr>
<tr id="row_585_70_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07813.html" target="_self">XFDRAlgorithm</a></td><td class="desc"></td></tr>
<tr id="row_585_71_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09869.html" target="_self">XMassFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for 'XMass Analysis (fid)' files </td></tr>
<tr id="row_585_72_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09877.html" target="_self">XQuestResultXMLFile</a></td><td class="desc">Used to load and store xQuest result files </td></tr>
<tr id="row_585_73_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html" target="_self">SignalToNoiseEstimator&lt; ContainerT &gt;</a></td><td class="desc"></td></tr>
<tr id="row_585_74_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08949.html" target="_self">SignalToNoiseEstimator&lt; OpenMS::MSChromatogram &gt;</a></td><td class="desc"></td></tr>
<tr id="row_586_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_586_" class="arrow" onclick="toggleFolder('586_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08485.html" target="_self">ProgressLogger::ProgressLoggerImpl</a></td><td class="desc">This class represents an actual implementation of a logger </td></tr>
<tr id="row_586_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11577.html" target="_self">GUIProgressLoggerImpl</a></td><td class="desc">Implements a GUI version of the ProgressLoggerImpl </td></tr>
<tr id="row_587_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05820.html#a06529" target="_self">Protein</a></td><td class="desc"></td></tr>
<tr id="row_588_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07461.html" target="_self">PeptideAndProteinQuant::ProteinData</a></td><td class="desc">Quantitative and associated data for a protein </td></tr>
<tr id="row_589_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07477.html" target="_self">ProteinResolver::ProteinEntry</a></td><td class="desc">Protein from FASTA file </td></tr>
<tr id="row_590_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10681.html" target="_self">ProteinIdentification::ProteinGroup</a></td><td class="desc">Bundles multiple (e.g. indistinguishable) proteins in a group </td></tr>
<tr id="row_591_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10661.html" target="_self">ProteinHit::ProteinHitAccessionHash</a></td><td class="desc">Hash of a <a class="el" href="a10657.html" title="Representation of a protein hit.">ProteinHit</a> based on its accession only! </td></tr>
<tr id="row_592_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10665.html" target="_self">ProteinHit::ProteinHitPtrAccessionHash</a></td><td class="desc"></td></tr>
<tr id="row_593_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07469.html" target="_self">ProteinInference</a></td><td class="desc">[experimental class] given a peptide quantitation, infer corresponding protein quantities </td></tr>
<tr id="row_594_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07769.html" target="_self">OPXLDataStructs::ProteinProteinCrossLink</a></td><td class="desc">The <a class="el" href="a07769.html" title="The ProteinProteinCrossLink struct represents a cross-link between two peptides in OpenPepXL.">ProteinProteinCrossLink</a> struct represents a cross-link between two peptides in OpenPepXL </td></tr>
<tr id="row_595_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11917.html" target="_self">DBsearchDeconvMass::Proteoform</a></td><td class="desc"></td></tr>
<tr id="row_596_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07537.html" target="_self">PScore</a></td><td class="desc">Implementation of the <a class="el" href="a07537.html" title="Implementation of the PScore PSM scoring algorithm.">PScore</a> PSM scoring algorithm </td></tr>
<tr id="row_597_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07657.html" target="_self">PSProteinInference</a></td><td class="desc">This class implements protein inference for the precursor ion selection strategies </td></tr>
<tr id="row_598_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11909.html" target="_self">DBsearchDeconvMass::PTM</a></td><td class="desc"></td></tr>
<tr id="row_599_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11913.html" target="_self">DBsearchDeconvMass::PTMnode</a></td><td class="desc"></td></tr>
<tr id="row_600_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07397.html" target="_self">IsobaricChannelExtractor::PuritySate_</a></td><td class="desc">Small struct to capture the current state of the purity computation </td></tr>
<tr id="row_601_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06881.html#a06885" target="_self">PrecursorPurity::PurityScores</a></td><td class="desc"></td></tr>
<tr id="row_602_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_602_" class="arrow" onclick="toggleFolder('602_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16189.html" target="_self">QApplication</a></td><td class="desc"></td></tr>
<tr id="row_602_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11445.html" target="_self">QApplicationTOPP</a></td><td class="desc">Extension to the <a class="el" href="a16189.html">QApplication</a> for running TOPPs GUI tools </td></tr>
<tr id="row_603_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_603_" class="arrow" onclick="toggleFolder('603_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10773.html" target="_self">QCBase</a></td><td class="desc">This class serves as an abstract base class for all QC classes </td></tr>
<tr id="row_603_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10733.html" target="_self">Contaminants</a></td><td class="desc">This class is a metric for the QualityControl TOPP tool </td></tr>
<tr id="row_603_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10741.html" target="_self">FragmentMassError</a></td><td class="desc"></td></tr>
<tr id="row_603_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10749.html" target="_self">MissedCleavages</a></td><td class="desc">This class is a metric for the QualityControl TOPP Tool </td></tr>
<tr id="row_603_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10753.html" target="_self">Ms2IdentificationRate</a></td><td class="desc">This class is a metric for the QualityControl-ToppTool </td></tr>
<tr id="row_603_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10761.html" target="_self">Ms2SpectrumStats</a></td><td class="desc">QC metric to determine the number of MS2 scans per MS1 scan over RT </td></tr>
<tr id="row_603_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10769.html" target="_self">MzCalibration</a></td><td class="desc">QC metric calculating (un)calibrated m/z error </td></tr>
<tr id="row_603_6_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10785.html" target="_self">RTAlignment</a></td><td class="desc">Take the original retention time before map alignment and use the alignment's trafoXML for calculation of the new alignment retention times </td></tr>
<tr id="row_603_7_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10789.html" target="_self">TIC</a></td><td class="desc"></td></tr>
<tr id="row_604_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_604_" class="arrow" onclick="toggleFolder('604_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16093.html" target="_self">QDate</a></td><td class="desc"></td></tr>
<tr id="row_604_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08577.html" target="_self">Date</a></td><td class="desc"><a class="el" href="a08577.html" title="Date Class.">Date</a> Class </td></tr>
<tr id="row_605_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_605_" class="arrow" onclick="toggleFolder('605_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16173.html" target="_self">QDateTime</a></td><td class="desc"></td></tr>
<tr id="row_605_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08581.html" target="_self">DateTime</a></td><td class="desc"><a class="el" href="a08581.html" title="DateTime Class.">DateTime</a> Class </td></tr>
<tr id="row_606_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_606_" class="arrow" onclick="toggleFolder('606_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16085.html" target="_self">QDialog</a></td><td class="desc"></td></tr>
<tr id="row_606_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11477.html" target="_self">DataFilterDialog</a></td><td class="desc">Dialog for creating and changing a DataFilter </td></tr>
<tr id="row_606_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11481.html" target="_self">FeatureEditDialog</a></td><td class="desc">Dialog for editing a feature </td></tr>
<tr id="row_606_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11485.html" target="_self">HistogramDialog</a></td><td class="desc">Dialog that show a <a class="el" href="a11581.html" title="Widget which can visualize a histogram.">HistogramWidget</a> </td></tr>
<tr id="row_606_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11505.html" target="_self">Spectrum1DPrefDialog</a></td><td class="desc">Preferences dialog for <a class="el" href="a11645.html" title="Widget for visualization of several spectra.">Spectrum1DWidget</a> </td></tr>
<tr id="row_606_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11513.html" target="_self">Spectrum2DPrefDialog</a></td><td class="desc">Preferences dialog for <a class="el" href="a11653.html" title="Widget for 2D-visualization of peak map and feature map data.">Spectrum2DWidget</a> </td></tr>
<tr id="row_606_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11517.html" target="_self">Spectrum3DPrefDialog</a></td><td class="desc">Preferences dialog for <a class="el" href="a11665.html" title="Widget for 3D-visualization of map data.">Spectrum3DWidget</a> </td></tr>
<tr id="row_606_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11561.html" target="_self">TOPPViewPrefDialog</a></td><td class="desc">Preferences dialog for TOPPView </td></tr>
<tr id="row_606_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11489.html" target="_self">LayerStatisticsDialog</a></td><td class="desc">Dialog showing statistics about the data of the current layer </td></tr>
<tr id="row_606_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11589.html" target="_self">ListEditor</a></td><td class="desc">Editor for editing int, double and string lists (including output and input file lists) </td></tr>
<tr id="row_606_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11601.html" target="_self">MetaDataBrowser</a></td><td class="desc">A meta data visualization widget </td></tr>
<tr id="row_606_10_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11497.html" target="_self">SaveImageDialog</a></td><td class="desc">Dialog for saving an image </td></tr>
<tr id="row_606_11_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11501.html" target="_self">Spectrum1DGoToDialog</a></td><td class="desc">Simple goto/set visible area dialog for exact placement of the viewing window </td></tr>
<tr id="row_606_12_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11509.html" target="_self">Spectrum2DGoToDialog</a></td><td class="desc">GoTo dialog used to zoom to a m/z and retention time range or to a feature </td></tr>
<tr id="row_606_13_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11521.html" target="_self">SpectrumAlignmentDialog</a></td><td class="desc">Lets the user select two spectra and set the parameters for the spectrum alignment </td></tr>
<tr id="row_606_14_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11525.html" target="_self">TheoreticalSpectrumGenerationDialog</a></td><td class="desc">Dialog which allows to enter an AA sequence and generates a theoretical spectrum for it </td></tr>
<tr id="row_606_15_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11529.html" target="_self">ToolsDialog</a></td><td class="desc">TOPP tool selection dialog </td></tr>
<tr id="row_606_16_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11533.html" target="_self">TOPPASInputFileDialog</a></td><td class="desc">Dialog which allows to specify an input file </td></tr>
<tr id="row_606_17_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11537.html" target="_self">TOPPASInputFilesDialog</a></td><td class="desc">Dialog which allows to specify a list of input files </td></tr>
<tr id="row_606_18_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11541.html" target="_self">TOPPASIOMappingDialog</a></td><td class="desc">Dialog which allows to configure the input/output parameter mapping of an edge </td></tr>
<tr id="row_606_19_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11545.html" target="_self">TOPPASOutputFilesDialog</a></td><td class="desc">Dialog which allows to specify the directory for the output files </td></tr>
<tr id="row_606_20_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11549.html" target="_self">TOPPASToolConfigDialog</a></td><td class="desc">TOPP tool configuration dialog </td></tr>
<tr id="row_606_21_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11553.html" target="_self">TOPPASVertexNameDialog</a></td><td class="desc">Dialog which allows to change the name of an input/output vertex </td></tr>
<tr id="row_606_22_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11557.html" target="_self">TOPPViewOpenDialog</a></td><td class="desc">Dataset opening options for TOPPView </td></tr>
<tr id="row_607_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_607_" class="arrow" onclick="toggleFolder('607_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16185.html" target="_self">QFileSystemWatcher</a></td><td class="desc"></td></tr>
<tr id="row_607_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10881.html" target="_self">FileWatcher</a></td><td class="desc">Watcher that monitors file changes </td></tr>
<tr id="row_608_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_608_" class="arrow" onclick="toggleFolder('608_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16113.html" target="_self">QGraphicsItem</a></td><td class="desc"></td></tr>
<tr id="row_608_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11677.html" target="_self">TOPPASEdge</a></td><td class="desc">An edge representing a data flow in TOPPAS </td></tr>
<tr id="row_608_1_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_608_1_" class="arrow" onclick="toggleFolder('608_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11737.html" target="_self">TOPPASVertex</a></td><td class="desc">The base class of the different vertex classes </td></tr>
<tr id="row_608_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11681.html" target="_self">TOPPASInputFileListVertex</a></td><td class="desc">A vertex representing an input file list </td></tr>
<tr id="row_608_1_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11689.html" target="_self">TOPPASMergerVertex</a></td><td class="desc">A special vertex that allows to merge several inputs </td></tr>
<tr id="row_608_1_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11693.html" target="_self">TOPPASOutputFileListVertex</a></td><td class="desc">A vertex representing an output file list </td></tr>
<tr id="row_608_1_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11717.html" target="_self">TOPPASSplitterVertex</a></td><td class="desc">A special vertex that allows to split a list of inputs </td></tr>
<tr id="row_608_1_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11725.html" target="_self">TOPPASToolVertex</a></td><td class="desc">A vertex representing a TOPP tool </td></tr>
<tr id="row_609_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_609_" class="arrow" onclick="toggleFolder('609_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16117.html" target="_self">QGraphicsScene</a></td><td class="desc"></td></tr>
<tr id="row_609_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11709.html" target="_self">TOPPASScene</a></td><td class="desc">A container for all visual items of a TOPPAS workflow </td></tr>
<tr id="row_610_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_610_" class="arrow" onclick="toggleFolder('610_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16129.html" target="_self">QGraphicsView</a></td><td class="desc"></td></tr>
<tr id="row_610_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11749.html" target="_self">TOPPASWidget</a></td><td class="desc">Widget visualizing and allowing to edit TOPP pipelines </td></tr>
<tr id="row_611_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_611_" class="arrow" onclick="toggleFolder('611_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16141.html" target="_self">QItemDelegate</a></td><td class="desc"></td></tr>
<tr id="row_611_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11597.html" target="_self">ListEditorDelegate</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> delegate class </td></tr>
<tr id="row_611_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11621.html" target="_self">ParamEditorDelegate</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> delegate class for <a class="el" href="a16121.html">QTreeWidget</a> </td></tr>
<tr id="row_612_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_612_" class="arrow" onclick="toggleFolder('612_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16101.html" target="_self">QLineEdit</a></td><td class="desc"></td></tr>
<tr id="row_612_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11617.html" target="_self">OpenMSLineEdit</a></td><td class="desc">Custom <a class="el" href="a16101.html">QLineEdit</a> which emits a signal when losing focus (such that we can commit its data) </td></tr>
<tr id="row_613_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_613_" class="arrow" onclick="toggleFolder('613_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16201.html" target="_self">QListWidget</a></td><td class="desc"></td></tr>
<tr id="row_613_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11593.html" target="_self">ListTable</a></td><td class="desc"></td></tr>
<tr id="row_614_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_614_" class="arrow" onclick="toggleFolder('614_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16073.html" target="_self">QMainWindow</a></td><td class="desc"></td></tr>
<tr id="row_614_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11441.html" target="_self">INIFileEditorWindow</a></td><td class="desc">Shows the <a class="el" href="a11629.html" title="A GUI for editing or viewing a Param object.">ParamEditor</a> widget in a <a class="el" href="a16073.html">QMainWindow</a> with a toolbar </td></tr>
<tr id="row_614_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11449.html" target="_self">TOPPASBase</a></td><td class="desc">Main window of the TOPPAS tool </td></tr>
<tr id="row_614_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11453.html" target="_self">TOPPViewBase</a></td><td class="desc">Main window of TOPPView tool </td></tr>
<tr id="row_615_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_615_" class="arrow" onclick="toggleFolder('615_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16169.html" target="_self">QMdiArea</a></td><td class="desc"></td></tr>
<tr id="row_615_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11573.html" target="_self">EnhancedWorkspace</a></td><td class="desc"></td></tr>
<tr id="row_616_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_616_" class="arrow" onclick="toggleFolder('616_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16081.html" target="_self">QObject</a></td><td class="desc"></td></tr>
<tr id="row_616_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09469.html" target="_self">MascotRemoteQuery</a></td><td class="desc">Class which handles the communication between <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> and the Mascot server </td></tr>
<tr id="row_616_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10889.html" target="_self">NetworkGetRequest</a></td><td class="desc"></td></tr>
<tr id="row_616_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11677.html" target="_self">TOPPASEdge</a></td><td class="desc">An edge representing a data flow in TOPPAS </td></tr>
<tr id="row_616_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11697.html" target="_self">TOPPASResource</a></td><td class="desc">Represents a data resource for TOPPAS workflows </td></tr>
<tr id="row_616_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11701.html" target="_self">TOPPASResources</a></td><td class="desc">A dictionary mapping string keys to lists of <a class="el" href="a11697.html" title="Represents a data resource for TOPPAS workflows.">TOPPASResource</a> objects </td></tr>
<tr id="row_616_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11737.html" target="_self">TOPPASVertex</a></td><td class="desc">The base class of the different vertex classes </td></tr>
<tr id="row_616_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11753.html" target="_self">TOPPViewIdentificationViewBehavior</a></td><td class="desc">Behavior of TOPPView in identification mode </td></tr>
<tr id="row_616_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11757.html" target="_self">TOPPViewSpectraViewBehavior</a></td><td class="desc">Behavior of TOPPView in spectra view mode </td></tr>
<tr id="row_617_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_617_" class="arrow" onclick="toggleFolder('617_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16157.html" target="_self">QOpenGLFunctions_2_0</a></td><td class="desc"></td></tr>
<tr id="row_617_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11661.html" target="_self">Spectrum3DOpenGLCanvas</a></td><td class="desc">OpenGL Canvas for 3D-visualization of map data </td></tr>
<tr id="row_618_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_618_" class="arrow" onclick="toggleFolder('618_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16153.html" target="_self">QOpenGLWidget</a></td><td class="desc"></td></tr>
<tr id="row_618_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11661.html" target="_self">Spectrum3DOpenGLCanvas</a></td><td class="desc">OpenGL Canvas for 3D-visualization of map data </td></tr>
<tr id="row_619_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_619_" class="arrow" onclick="toggleFolder('619_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16133.html" target="_self">QProcess</a></td><td class="desc"></td></tr>
<tr id="row_619_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11705.html" target="_self">FakeProcess</a></td><td class="desc">A <a class="el" href="a11705.html" title="A FakeProcess class.">FakeProcess</a> class </td></tr>
<tr id="row_620_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_620_" class="arrow" onclick="toggleFolder('620_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16109.html" target="_self">QTabBar</a></td><td class="desc"></td></tr>
<tr id="row_620_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11565.html" target="_self">EnhancedTabBar</a></td><td class="desc">Convenience tab bar implementation </td></tr>
<tr id="row_620_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11721.html" target="_self">TOPPASTabBar</a></td><td class="desc">Convenience tab bar implementation </td></tr>
<tr id="row_621_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08765.html" target="_self">QTCluster</a></td><td class="desc">A representation of a QT cluster used for feature grouping </td></tr>
<tr id="row_622_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_622_" class="arrow" onclick="toggleFolder('622_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16193.html" target="_self">QTextEdit</a></td><td class="desc"></td></tr>
<tr id="row_622_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11685.html" target="_self">TOPPASLogWindow</a></td><td class="desc"><a class="el" href="a16193.html">QTextEdit</a> implementation with a "clear" button in the context menu </td></tr>
<tr id="row_623_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_623_" class="arrow" onclick="toggleFolder('623_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16121.html" target="_self">QTreeWidget</a></td><td class="desc"></td></tr>
<tr id="row_623_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11625.html" target="_self">ParamTree</a></td><td class="desc"><a class="el" href="a16121.html">QTreeWidget</a> that emits a signal whenever a new row is selected </td></tr>
<tr id="row_623_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11733.html" target="_self">TOPPASTreeView</a></td><td class="desc">Tree view implementation for the list of TOPP tools </td></tr>
<tr id="row_624_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10353.html" target="_self">QuadraticRegression</a></td><td class="desc"></td></tr>
<tr id="row_625_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09777.html" target="_self">QcMLFile::QualityParameter</a></td><td class="desc">Representation of a quality parameter </td></tr>
<tr id="row_626_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_626_" class="arrow" onclick="toggleFolder('626_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16089.html" target="_self">QWidget</a></td><td class="desc"></td></tr>
<tr id="row_626_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11469.html" target="_self">AxisWidget</a></td><td class="desc">Widget that represents an axis of a graph </td></tr>
<tr id="row_626_1_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_626_1_" class="arrow" onclick="toggleFolder('626_1_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11773.html" target="_self">BaseVisualizerGUI</a></td><td class="desc">A base class for all visualizer classes </td></tr>
<tr id="row_626_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11761.html" target="_self">AcquisitionInfoVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10385.html" title="Description of the combination of raw data to a single spectrum.">AcquisitionInfo</a> objects </td></tr>
<tr id="row_626_1_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11765.html" target="_self">AcquisitionVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10381.html" title="Information about one raw data spectrum that was combined with several other raw data spectra.">Acquisition</a> objects </td></tr>
<tr id="row_626_1_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11777.html" target="_self">ContactPersonVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10393.html" title="Contact person information.">ContactPerson</a> objects </td></tr>
<tr id="row_626_1_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11781.html" target="_self">DataProcessingVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10425.html" title="Description of the applied preprocessing steps.">DataProcessing</a> objects </td></tr>
<tr id="row_626_1_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11785.html" target="_self">DigestionVisualizer</a></td><td class="desc">Class that displays all meta information of digestion objects </td></tr>
<tr id="row_626_1_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11789.html" target="_self">DocumentIdentifierVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10433.html" title="Manage source document information.">DocumentIdentifier</a> objects </td></tr>
<tr id="row_626_1_6_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11793.html" target="_self">ExperimentalSettingsVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10453.html" title="Description of the experimental settings.">ExperimentalSettings</a> objects </td></tr>
<tr id="row_626_1_7_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11797.html" target="_self">GradientVisualizer</a></td><td class="desc"><a class="el" href="a11797.html" title="GradientVisualizer is a visualizer class for objects of type gradient.">GradientVisualizer</a> is a visualizer class for objects of type gradient </td></tr>
<tr id="row_626_1_8_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11801.html" target="_self">HPLCVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10461.html" title="Representation of a HPLC experiment.">HPLC</a> objects </td></tr>
<tr id="row_626_1_9_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11805.html" target="_self">InstrumentSettingsVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10561.html" title="Description of the settings a MS Instrument was run with.">InstrumentSettings</a> objects </td></tr>
<tr id="row_626_1_10_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11809.html" target="_self">InstrumentVisualizer</a></td><td class="desc">Class that displays all meta information for an MS instrument </td></tr>
<tr id="row_626_1_11_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11813.html" target="_self">IonDetectorVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10565.html" title="Description of a ion detector (part of a MS Instrument)">IonDetector</a> objects </td></tr>
<tr id="row_626_1_12_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11817.html" target="_self">IonSourceVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10569.html" title="Description of an ion source (part of a MS Instrument)">IonSource</a> objects </td></tr>
<tr id="row_626_1_13_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11821.html" target="_self">MassAnalyzerVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10573.html" title="Description of a mass analyzer (part of a MS Instrument)">MassAnalyzer</a> objects </td></tr>
<tr id="row_626_1_14_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11825.html" target="_self">MetaInfoDescriptionVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10581.html" title="Description of the meta data arrays of MSSpectrum.">MetaInfoDescription</a> objects </td></tr>
<tr id="row_626_1_15_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11829.html" target="_self">MetaInfoVisualizer</a></td><td class="desc"><a class="el" href="a11829.html" title="MetaInfoVisualizer is a visualizer class for all classes that use one MetaInfo object as member.">MetaInfoVisualizer</a> is a visualizer class for all classes that use one <a class="el" href="a10577.html" title="A Type-Name-Value tuple class.">MetaInfo</a> object as member </td></tr>
<tr id="row_626_1_16_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11833.html" target="_self">ModificationVisualizer</a></td><td class="desc">Class that displays all meta information of modification objects </td></tr>
<tr id="row_626_1_17_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11837.html" target="_self">PeptideHitVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10617.html" title="Representation of a peptide hit.">PeptideHit</a> objects </td></tr>
<tr id="row_626_1_18_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11841.html" target="_self">PeptideIdentificationVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10645.html" title="Represents the peptide hits for a spectrum.">PeptideIdentification</a> objects </td></tr>
<tr id="row_626_1_19_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11845.html" target="_self">PrecursorVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10649.html" title="Precursor meta information.">Precursor</a> objects </td></tr>
<tr id="row_626_1_20_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11849.html" target="_self">ProductVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10653.html" title="Product meta information.">Product</a> objects </td></tr>
<tr id="row_626_1_21_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11853.html" target="_self">ProteinHitVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10657.html" title="Representation of a protein hit.">ProteinHit</a> objects </td></tr>
<tr id="row_626_1_22_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11857.html" target="_self">ProteinIdentificationVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10677.html" title="Representation of a protein identification run.">ProteinIdentification</a> objects </td></tr>
<tr id="row_626_1_23_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11861.html" target="_self">SampleVisualizer</a></td><td class="desc">Class that displays all meta information of sample objects </td></tr>
<tr id="row_626_1_24_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11865.html" target="_self">ScanWindowVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10697.html" title="Scan window description.">ScanWindow</a> objects </td></tr>
<tr id="row_626_1_25_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11869.html" target="_self">SoftwareVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10701.html" title="Description of the software used for processing.">Software</a> objects </td></tr>
<tr id="row_626_1_26_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11873.html" target="_self">SourceFileVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10705.html" title="Description of a file location, used to store the origin of (meta) data.">SourceFile</a> objects </td></tr>
<tr id="row_626_1_27_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11877.html" target="_self">SpectrumSettingsVisualizer</a></td><td class="desc">Class that displays all meta information for <a class="el" href="a10725.html" title="Representation of 1D spectrum settings.">SpectrumSettings</a> objects </td></tr>
<tr id="row_626_1_28_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11881.html" target="_self">TaggingVisualizer</a></td><td class="desc">Class that displays all meta information of tagging objects </td></tr>
<tr id="row_626_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11473.html" target="_self">ColorSelector</a></td><td class="desc">A widget for selecting a color </td></tr>
<tr id="row_626_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11581.html" target="_self">HistogramWidget</a></td><td class="desc">Widget which can visualize a histogram </td></tr>
<tr id="row_626_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11613.html" target="_self">MultiGradientSelector</a></td><td class="desc">A widget witch allows constructing gradients of multiple colors </td></tr>
<tr id="row_626_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11629.html" target="_self">ParamEditor</a></td><td class="desc">A GUI for editing or viewing a <a class="el" href="a08745.html" title="Management and storage of parameters / INI files.">Param</a> object </td></tr>
<tr id="row_626_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11633.html" target="_self">SpectraIdentificationViewWidget</a></td><td class="desc">Tabular visualization / selection of identified spectra </td></tr>
<tr id="row_626_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11637.html" target="_self">SpectraViewWidget</a></td><td class="desc">Hierarchical visualization and selection of spectra </td></tr>
<tr id="row_626_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11669.html" target="_self">SpectrumCanvas</a></td><td class="desc">Base class for visualization canvas classes </td></tr>
<tr id="row_626_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11673.html" target="_self">SpectrumWidget</a></td><td class="desc">Base class for spectrum widgets </td></tr>
<tr id="row_627_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10149.html" target="_self">RangeManager&lt; D &gt;</a></td><td class="desc">Handles the management of a position and intensity range </td></tr>
<tr id="row_628_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_628_" class="arrow" onclick="toggleFolder('628_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10149.html" target="_self">RangeManager&lt; 1 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_628_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10073.html" target="_self">MSChromatogram</a></td><td class="desc">The representation of a chromatogram </td></tr>
<tr id="row_628_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10097.html" target="_self">MSSpectrum</a></td><td class="desc">The representation of a 1D spectrum </td></tr>
<tr id="row_629_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_629_" class="arrow" onclick="toggleFolder('629_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10149.html" target="_self">RangeManager&lt; 2 &gt;</a></td><td class="desc"></td></tr>
<tr id="row_629_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10021.html" target="_self">ConsensusMap</a></td><td class="desc">A container for consensus elements </td></tr>
<tr id="row_629_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10057.html" target="_self">FeatureMap</a></td><td class="desc">A container for features </td></tr>
<tr id="row_629_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10081.html" target="_self">MSExperiment</a></td><td class="desc">In-Memory representation of a mass spectrometry experiment </td></tr>
<tr id="row_630_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10633.html" target="_self">PeptideHit::RankLess</a></td><td class="desc">Lesser predicate for scores of hits </td></tr>
<tr id="row_631_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10269.html" target="_self">RANSAC&lt; TModelType &gt;</a></td><td class="desc">This class provides a generic implementation of the <a class="el" href="a10269.html" title="This class provides a generic implementation of the RANSAC outlier detection algorithm....">RANSAC</a> outlier detection algorithm. Is implemented and tested after the SciPy reference: <a href="http://wiki.scipy.org/Cookbook/RANSAC">http://wiki.scipy.org/Cookbook/RANSAC</a> </td></tr>
<tr id="row_632_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10273.html" target="_self">RansacModel&lt; ModelT &gt;</a></td><td class="desc">Generic plug-in template base class using 'Curiously recurring template pattern' (CRTP) to allow for arbitrary <a class="el" href="a10269.html" title="This class provides a generic implementation of the RANSAC outlier detection algorithm....">RANSAC</a> models (e.g. linear or quadratic fits) </td></tr>
<tr id="row_633_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_633_" class="arrow" onclick="toggleFolder('633_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10273.html" target="_self">RansacModel&lt; RansacModelLinear &gt;</a></td><td class="desc"></td></tr>
<tr id="row_633_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10277.html" target="_self">RansacModelLinear</a></td><td class="desc">Implementation of a linear <a class="el" href="a10269.html" title="This class provides a generic implementation of the RANSAC outlier detection algorithm....">RANSAC</a> model fit </td></tr>
<tr id="row_634_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_634_" class="arrow" onclick="toggleFolder('634_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10273.html" target="_self">RansacModel&lt; RansacModelQuadratic &gt;</a></td><td class="desc"></td></tr>
<tr id="row_634_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10281.html" target="_self">RansacModelQuadratic</a></td><td class="desc">Implementation of a quadratic <a class="el" href="a10269.html" title="This class provides a generic implementation of the RANSAC outlier detection algorithm....">RANSAC</a> model fit </td></tr>
<tr id="row_635_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10265.html" target="_self">RANSACParam</a></td><td class="desc">A simple struct to carry all the parameters required for a <a class="el" href="a10269.html" title="This class provides a generic implementation of the RANSAC outlier detection algorithm....">RANSAC</a> run </td></tr>
<tr id="row_636_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05648.html#a11945" target="_self">RateScorePair</a></td><td class="desc"></td></tr>
<tr id="row_637_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10017.html" target="_self">ConsensusFeature::Ratio</a></td><td class="desc">Slim struct to feed the need for systematically storing of ratios ( </td></tr>
<tr id="row_638_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11401.html" target="_self">RawData</a></td><td class="desc"></td></tr>
<tr id="row_639_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_639_" class="arrow" onclick="toggleFolder('639_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>real_policies</b></td><td class="desc"></td></tr>
<tr id="row_639_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08793.html" target="_self">StringUtils::real_policies_NANfixed_&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_640_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_640_" class="arrow" onclick="toggleFolder('640_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>real_policies</b></td><td class="desc"></td></tr>
<tr id="row_640_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08785.html" target="_self">BK_PrecPolicy&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_641_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08005.html" target="_self">RealMassDecomposer</a></td><td class="desc">Handles decomposing of non-integer values/masses over a set of non-integer weights with an error allowed </td></tr>
<tr id="row_642_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09093.html#a09101" target="_self">Base64::Reinterpreter32_</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> class needed for type-punning </td></tr>
<tr id="row_643_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09093.html#a09097" target="_self">Base64::Reinterpreter64_</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> class needed for type-punning </td></tr>
<tr id="row_644_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08061.html" target="_self">Residue</a></td><td class="desc">Representation of a residue </td></tr>
<tr id="row_645_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08065.html" target="_self">ResidueDB</a></td><td class="desc"><a class="el" href="a08061.html" title="Representation of a residue.">Residue</a> data base which holds residues </td></tr>
<tr id="row_646_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08069.html" target="_self">ResidueModification</a></td><td class="desc">Representation of a modification </td></tr>
<tr id="row_647_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07493.html" target="_self">ProteinResolver::ResolverResult</a></td><td class="desc"></td></tr>
<tr id="row_648_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08621.html#a08625" target="_self">DecoyHelper::Result</a></td><td class="desc"></td></tr>
<tr id="row_649_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07529.html#a07533" target="_self">MorpheusScore::Result</a></td><td class="desc">Score and subscores </td></tr>
<tr id="row_650_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08073.html" target="_self">Ribonucleotide</a></td><td class="desc">Representation of a ribonucleotide (modified or unmodified) </td></tr>
<tr id="row_651_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08077.html" target="_self">RibonucleotideDB</a></td><td class="desc">Database of ribonucleotides (modified and unmodified) </td></tr>
<tr id="row_652_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11989.html" target="_self">RIntegration</a></td><td class="desc"></td></tr>
<tr id="row_653_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07541.html" target="_self">RNPxlFragmentAnnotationHelper</a></td><td class="desc">Convenience functions to construct appealing fragment annotation strings and store them as <a class="el" href="a10621.html" title="Contains annotations of a peak.">PeptideHit::PeakAnnotation</a> </td></tr>
<tr id="row_654_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07549.html" target="_self">RNPxlMarkerIonExtractor</a></td><td class="desc"></td></tr>
<tr id="row_655_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05822.html#a07553" target="_self">RNPxlModificationMassesResult</a></td><td class="desc"></td></tr>
<tr id="row_656_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07557.html" target="_self">RNPxlModificationsGenerator</a></td><td class="desc"></td></tr>
<tr id="row_657_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07569.html" target="_self">RNPxlReport</a></td><td class="desc">Create report </td></tr>
<tr id="row_658_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07561.html" target="_self">RNPxlReportRow</a></td><td class="desc">Struct to hold a single report line </td></tr>
<tr id="row_659_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07565.html" target="_self">RNPxlReportRowHeader</a></td><td class="desc">Create header line </td></tr>
<tr id="row_660_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10357.html" target="_self">ROCCurve</a></td><td class="desc">ROCCurves show the trade-off in sensitivity and specificity for binary classifiers using different cutoff values </td></tr>
<tr id="row_661_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09681.html" target="_self">MzTabNucleicAcidSectionRow::RowCompare</a></td><td class="desc">Comparison operator for sorting rows </td></tr>
<tr id="row_662_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09689.html" target="_self">MzTabOligonucleotideSectionRow::RowCompare</a></td><td class="desc">Comparison operator for sorting rows </td></tr>
<tr id="row_663_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09661.html" target="_self">MzTabPeptideSectionRow::RowCompare</a></td><td class="desc">Comparison operator for sorting rows </td></tr>
<tr id="row_664_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09653.html" target="_self">MzTabProteinSectionRow::RowCompare</a></td><td class="desc">Comparison operator for sorting rows </td></tr>
<tr id="row_665_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09669.html" target="_self">MzTabPSMSectionRow::RowCompare</a></td><td class="desc">Comparison operator for sorting rows </td></tr>
<tr id="row_666_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09697.html" target="_self">MzTabOSMSectionRow::RowCompare</a></td><td class="desc">Comparison operator for sorting rows </td></tr>
<tr id="row_667_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11025.html#a11029" target="_self">FeatureFinderIdentificationAlgorithm::RTRegion</a></td><td class="desc">Region in RT in which a peptide elutes: </td></tr>
<tr id="row_668_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10369.html#a10373" target="_self">AbsoluteQuantitationStandards::runConcentration</a></td><td class="desc">Structure to map runs to components to known concentrations </td></tr>
<tr id="row_669_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10893.html" target="_self">RWrapper</a></td><td class="desc">R-Wrapper Class </td></tr>
<tr id="row_670_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10449.html" target="_self">ExperimentalDesign::SampleSection</a></td><td class="desc"></td></tr>
<tr id="row_671_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10765.html" target="_self">Ms2SpectrumStats::ScanEvent</a></td><td class="desc"></td></tr>
<tr id="row_672_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10629.html" target="_self">PeptideHit::ScoreLess</a></td><td class="desc">Lesser predicate for scores of hits </td></tr>
<tr id="row_673_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10673.html" target="_self">ProteinHit::ScoreLess</a></td><td class="desc">Lesser predicate for scores of hits </td></tr>
<tr id="row_674_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10625.html" target="_self">PeptideHit::ScoreMore</a></td><td class="desc">Greater predicate for scores of hits </td></tr>
<tr id="row_675_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10669.html" target="_self">ProteinHit::ScoreMore</a></td><td class="desc">Greater predicate for scores of hits </td></tr>
<tr id="row_676_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06725.html" target="_self">ScoringMatrixData_&lt; int, AminoAcid, AdaptedIdentity &gt;</a></td><td class="desc"></td></tr>
<tr id="row_677_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06721.html" target="_self">ScoringMatrixData_&lt; int, AminoAcid, PAM30MS &gt;</a></td><td class="desc"></td></tr>
<tr id="row_678_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10997.html" target="_self">FeatureFinderAlgorithmPickedHelperStructs::Seed</a></td><td class="desc">Helper structure for seeds used in <a class="el" href="a10989.html" title="FeatureFinderAlgorithm for picked peaks.">FeatureFinderAlgorithmPicked</a> </td></tr>
<tr id="row_679_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11189.html" target="_self">SeedListGenerator</a></td><td class="desc">Generate seed lists for feature detection </td></tr>
<tr id="row_680_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07229.html" target="_self">MRMFeatureSelector::SelectorParameters</a></td><td class="desc"></td></tr>
<tr id="row_681_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10637.html" target="_self">PeptideHit::SequenceLessComparator</a></td><td class="desc">Lesser predicate for (modified) sequence of hits </td></tr>
<tr id="row_682_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09785.html" target="_self">SequestInfile</a></td><td class="desc">Sequest input file adapter </td></tr>
<tr id="row_683_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09789.html" target="_self">SequestOutfile</a></td><td class="desc">Representation of a Sequest output file </td></tr>
<tr id="row_684_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_684_" class="arrow" onclick="toggleFolder('684_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>set&lt; K &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_684_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08637.html" target="_self">IsotopeCluster::ChargedIndexSet</a></td><td class="desc">Index set with associated charge estimate </td></tr>
<tr id="row_685_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11405.html" target="_self">SHFeature</a></td><td class="desc"></td></tr>
<tr id="row_686_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08965.html" target="_self">SignalToNoiseEstimatorMedianRapid</a></td><td class="desc">Estimates the signal/noise (S/N) ratio of each data point in a scan by using the median (window based) </td></tr>
<tr id="row_687_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07133.html" target="_self">SimpleOpenMSSpectraFactory</a></td><td class="desc">A factory method that returns two ISpectrumAccess implementations </td></tr>
<tr id="row_688_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08089.html#a08093" target="_self">SimpleTSGXLMS::SimplePeak</a></td><td class="desc">A simple struct to represent peaks with mz and charge and sort them easily </td></tr>
<tr id="row_689_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08097.html" target="_self">SimpleTSGXLMS::SimplePeakComparator</a></td><td class="desc">Comparator to sort SimplePeaks by mz </td></tr>
<tr id="row_690_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10865.html" target="_self">SimProtein</a></td><td class="desc">Plain data object holding sequence and abundance information on a single protein </td></tr>
<tr id="row_691_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10869.html" target="_self">SimRandomNumberGenerator</a></td><td class="desc">Wrapper class for random number generators used by the simulation classes </td></tr>
<tr id="row_692_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10361.html" target="_self">ROCCurve::simsortdec</a></td><td class="desc">Predicate for <a class="el" href="a10357.html#a47fdc9eea42b6975cdc835bb2e08810e" title="sorts data and caches if sorted">sort()</a> </td></tr>
<tr id="row_693_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08489.html" target="_self">SingletonRegistry</a></td><td class="desc">Holds pointers to unique instance of a singleton factory </td></tr>
<tr id="row_694_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05648.html#a11949" target="_self">SIPIncorporation</a></td><td class="desc">Datastructure for reporting an incorporation event </td></tr>
<tr id="row_695_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05648.html#a11953" target="_self">SIPPeptide</a></td><td class="desc">Datastructure for reporting a peptide with one or more incorporation rates comparator for vectors of SIPPeptides based on their size. Used to sort by group size </td></tr>
<tr id="row_696_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09205.html#a09209" target="_self">SiriusMzTabWriter::SiriusAdapterHit</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> structure used in SiriusAdapter that is used for the conversion of the sirius output to an mzTab </td></tr>
<tr id="row_697_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09205.html#a09213" target="_self">SiriusMzTabWriter::SiriusAdapterIdentification</a></td><td class="desc"></td></tr>
<tr id="row_698_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09205.html#a09217" target="_self">SiriusMzTabWriter::SiriusAdapterRun</a></td><td class="desc"></td></tr>
<tr id="row_699_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09201.html" target="_self">SiriusFragmentAnnotation</a></td><td class="desc"></td></tr>
<tr id="row_700_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06909.html" target="_self">SiriusMSFile</a></td><td class="desc"></td></tr>
<tr id="row_701_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09205.html" target="_self">SiriusMzTabWriter</a></td><td class="desc"></td></tr>
<tr id="row_702_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06905.html" target="_self">SiriusAdapterAlgorithm::SiriusTemporaryFileSystemObjects</a></td><td class="desc">Struct for temporary folder structure </td></tr>
<tr id="row_703_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08713.html" target="_self">LPWrapper::SolverParam</a></td><td class="desc">Struct that holds the parameters of the LP solver </td></tr>
<tr id="row_704_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06653.html" target="_self">Spawn&lt; TNeedle &gt;</a></td><td class="desc">State of an AC spawn, operating on a trie </td></tr>
<tr id="row_705_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09793.html" target="_self">SpecArrayFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for SpecArray (.pepList) files </td></tr>
<tr id="row_706_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06797.html#a06801" target="_self">IDMapper::SpectraIdentificationState</a></td><td class="desc">Result of a partitioning by identification state with <a class="el" href="a06797.html#a9ba195cdb10338b6cd5f02f8feded441" title="Mapping of peptide identifications to spectra This helper function partitions all spectra into those ...">mapPrecursorsToIdentifications()</a> </td></tr>
<tr id="row_707_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06837.html" target="_self">SpectralMatch</a></td><td class="desc"></td></tr>
<tr id="row_708_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06841.html" target="_self">SpectralMatchScoreComparator</a></td><td class="desc"></td></tr>
<tr id="row_709_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10777.html" target="_self">QCBase::SpectraMap</a></td><td class="desc"><a class="el" href="a08717.html" title="Map class based on the STL map (containing several convenience functions)">Map</a> to find a spectrum via its NativeID </td></tr>
<tr id="row_710_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09909.html" target="_self">Spectrum</a></td><td class="desc">The structure that captures the generation of a peak list (including the underlying acquisitions) </td></tr>
<tr id="row_711_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07329.html" target="_self">SpectrumAddition</a></td><td class="desc">The <a class="el" href="a07329.html" title="The SpectrumAddition is used to add up individual spectra.">SpectrumAddition</a> is used to add up individual spectra </td></tr>
<tr id="row_712_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09345.html#a09349" target="_self">MzMLHandler::SpectrumData</a></td><td class="desc">Data necessary to generate a single spectrum </td></tr>
<tr id="row_713_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09381.html#a09385" target="_self">MzXMLHandler::SpectrumData</a></td><td class="desc">Data necessary to generate a single spectrum </td></tr>
<tr id="row_714_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07753.html" target="_self">SpectrumDeconvolution</a></td><td class="desc"></td></tr>
<tr id="row_715_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09325" target="_self">MzIdentMLDOMHandler::SpectrumIdentification</a></td><td class="desc">Struct to hold the information from the <a class="el" href="a09309.html#a09325" title="Struct to hold the information from the SpectrumIdentification xml tag.">SpectrumIdentification</a> xml tag </td></tr>
<tr id="row_716_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09309.html#a09333" target="_self">MzIdentMLDOMHandler::SpectrumIdentificationProtocol</a></td><td class="desc">Struct to hold the information from the <a class="el" href="a09309.html#a09333" title="Struct to hold the information from the SpectrumIdentificationProtocol xml tag.">SpectrumIdentificationProtocol</a> xml tag </td></tr>
<tr id="row_717_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_717_" class="arrow" onclick="toggleFolder('717_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10713.html" target="_self">SpectrumLookup</a></td><td class="desc">Helper class for looking up spectra based on different attributes </td></tr>
<tr id="row_717_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10717.html" target="_self">SpectrumMetaDataLookup</a></td><td class="desc">Helper class for looking up spectrum meta data </td></tr>
<tr id="row_718_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09905.html" target="_self">SpectrumMeta</a></td><td class="desc">Identifying information for a spectrum </td></tr>
<tr id="row_719_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10721.html" target="_self">SpectrumMetaDataLookup::SpectrumMetaData</a></td><td class="desc">Meta data of a spectrum </td></tr>
<tr id="row_720_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08889.html" target="_self">SplineInterpolatedPeaks</a></td><td class="desc">Data structure for spline interpolation of MS1 spectra and chromatograms </td></tr>
<tr id="row_721_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08897.html" target="_self">SplinePackage</a></td><td class="desc">Fundamental data structure for <a class="el" href="a08889.html" title="Data structure for spline interpolation of MS1 spectra and chromatograms.">SplineInterpolatedPeaks</a> </td></tr>
<tr id="row_722_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09797.html" target="_self">SqliteConnector</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for Sqlite files </td></tr>
<tr id="row_723_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09805.html" target="_self">SqMassFile::SqMassConfig</a></td><td class="desc">Configuration class for <a class="el" href="a09801.html" title="An class that uses on-disk SQLite database to read and write spectra and chromatograms.">SqMassFile</a> </td></tr>
<tr id="row_724_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09801.html" target="_self">SqMassFile</a></td><td class="desc">An class that uses on-disk SQLite database to read and write spectra and chromatograms </td></tr>
<tr id="row_725_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08661.html" target="_self">squared_difference&lt; _Tp, _Dist &gt;</a></td><td class="desc"></td></tr>
<tr id="row_726_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08661.html" target="_self">squared_difference&lt; typename _Bracket_accessor&lt; KDTreeFeatureNode &gt; ::result_type, typename _Bracket_accessor&lt; KDTreeFeatureNode &gt; ::result_type &gt;</a></td><td class="desc"></td></tr>
<tr id="row_727_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08665.html" target="_self">squared_difference_counted&lt; _Tp, _Dist &gt;</a></td><td class="desc"></td></tr>
<tr id="row_728_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_728_" class="arrow" onclick="toggleFolder('728_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>static_visitor</b></td><td class="desc"></td></tr>
<tr id="row_728_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06781.html" target="_self">IDBoostGraph::GetPosteriorVisitor</a></td><td class="desc"></td></tr>
<tr id="row_728_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06769.html" target="_self">IDBoostGraph::LabelVisitor</a></td><td class="desc">Visits nodes in the boost graph (ptrs to an ID Object) and depending on their type creates a label </td></tr>
<tr id="row_728_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06773.html" target="_self">IDBoostGraph::PrintAddressVisitor&lt; CharT &gt;</a></td><td class="desc"></td></tr>
<tr id="row_728_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06777.html" target="_self">IDBoostGraph::SetPosteriorVisitor</a></td><td class="desc"></td></tr>
<tr id="row_729_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07465.html" target="_self">PeptideAndProteinQuant::Statistics</a></td><td class="desc"><a class="el" href="a07465.html" title="Statistics for processing summary.">Statistics</a> for processing summary </td></tr>
<tr id="row_730_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10781.html" target="_self">QCBase::Status</a></td><td class="desc">Storing a status of available/needed inputs (i.e. a set of Requires) as UInt64 </td></tr>
<tr id="row_731_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10897.html" target="_self">StopWatch</a></td><td class="desc"><a class="el" href="a10897.html" title="StopWatch Class.">StopWatch</a> Class </td></tr>
<tr id="row_732_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_732_" class="arrow" onclick="toggleFolder('732_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>streambuf</b></td><td class="desc"></td></tr>
<tr id="row_732_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08457.html" target="_self">LogStreamBuf</a></td><td class="desc">Stream buffer used by <a class="el" href="a08473.html" title="Log Stream Class.">LogStream</a> </td></tr>
<tr id="row_733_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08441.html" target="_self">FuzzyStringComparator::StreamElement_</a></td><td class="desc">Stores information about characters, numbers, and white spaces loaded from the InputStream </td></tr>
<tr id="row_734_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08493.html" target="_self">StreamHandler</a></td><td class="desc">Provides a central class to register globally used output streams. Currently supported streams are </td></tr>
<tr id="row_735_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08461.html" target="_self">LogStreamBuf::StreamStruct</a></td><td class="desc">Holds a stream that is connected to the <a class="el" href="a08473.html" title="Log Stream Class.">LogStream</a>. It also includes the minimum and maximum level at which the <a class="el" href="a08473.html" title="Log Stream Class.">LogStream</a> redirects messages to this stream </td></tr>
<tr id="row_736_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08769.html" target="_self">String&lt; AAcid, Alloc&lt; void &gt; &gt;</a></td><td class="desc"></td></tr>
<tr id="row_737_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08769.html" target="_self">String&lt; TSize &gt;</a></td><td class="desc"></td></tr>
<tr id="row_738_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08781.html" target="_self">StringListUtils</a></td><td class="desc">Utilities operating on lists of Strings </td></tr>
<tr id="row_739_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09413.html" target="_self">StringManager</a></td><td class="desc"></td></tr>
<tr id="row_740_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08789.html" target="_self">StringUtils</a></td><td class="desc"></td></tr>
<tr id="row_741_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08773.html" target="_self">StringView</a></td><td class="desc"><a class="el" href="a08773.html" title="StringView provides a non-owning view on an existing string.">StringView</a> provides a non-owning view on an existing string </td></tr>
<tr id="row_742_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10969.html" target="_self">Summary</a></td><td class="desc"><a class="el" href="a10969.html" title="Summary of fitting results.">Summary</a> of fitting results </td></tr>
<tr id="row_743_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10365.html" target="_self">SummaryStatistics&lt; T &gt;</a></td><td class="desc">Helper class to gather (and dump) some statistics from a e.g. vector&lt;double&gt; </td></tr>
<tr id="row_744_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11409.html" target="_self">SuperHirnParameters</a></td><td class="desc">SuperHirn parameters singleton class containing all static configuration variables </td></tr>
<tr id="row_745_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11413.html" target="_self">SuperHirnUtil</a></td><td class="desc"></td></tr>
<tr id="row_746_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07581.html" target="_self">SVMData</a></td><td class="desc">Data structure used in <a class="el" href="a07585.html" title="Serves as a wrapper for the libsvm.">SVMWrapper</a> </td></tr>
<tr id="row_747_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08109.html#a08121" target="_self">SvmTheoreticalSpectrumGenerator::SvmModelParameterSet</a></td><td class="desc">Simple container storing the model parameters required for simulation </td></tr>
<tr id="row_748_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08125.html" target="_self">SvmTheoreticalSpectrumGeneratorSet</a></td><td class="desc">Loads <a class="el" href="a08109.html" title="Simulates MS2 spectra with support vector machines.">SvmTheoreticalSpectrumGenerator</a> instances for different charges </td></tr>
<tr id="row_749_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06497.html" target="_self">SwathMap</a></td><td class="desc">Data structure to hold one SWATH map with information about upper / lower isolation window and whether the map is MS1 or MS2 </td></tr>
<tr id="row_750_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07337.html" target="_self">SwathQC</a></td><td class="desc">Quality Control function for <a class="el" href="a05820.html">OpenSwath</a> </td></tr>
<tr id="row_751_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07341.html" target="_self">SwathWindowLoader</a></td><td class="desc">Class to read a file describing the Swath Windows </td></tr>
<tr id="row_752_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10901.html" target="_self">SysInfo</a></td><td class="desc">Some functions to get system information </td></tr>
<tr id="row_753_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07661.html" target="_self">TargetedExperiment</a></td><td class="desc">A description of a targeted experiment containing precursor and production ions </td></tr>
<tr id="row_754_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05820.html#a06533" target="_self">TargetedExperiment</a></td><td class="desc"></td></tr>
<tr id="row_755_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10877.html" target="_self">File::TemporaryFiles_</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> helper class, which holds temporary filenames and deletes these files at program exit </td></tr>
<tr id="row_756_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_756_" class="arrow" onclick="toggleFolder('756_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09817.html" target="_self">TextFile</a></td><td class="desc">This class provides some basic file handling methods for text files </td></tr>
<tr id="row_756_0_" style="display:none;"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_756_0_" class="arrow" onclick="toggleFolder('756_0_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09137.html" target="_self">CsvFile</a></td><td class="desc">This class handles csv files. Currently only loading is implemented </td></tr>
<tr id="row_756_0_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09085.html" target="_self">AbsoluteQuantitationMethodFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for <a class="el" href="a07389.html" title="AbsoluteQuantitationMethod is a class to hold information about the quantitation method and for apply...">AbsoluteQuantitationMethod</a> files </td></tr>
<tr id="row_756_0_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09477.html" target="_self">MRMFeaturePickerFile</a></td><td class="desc">_MRMFeaturePickerFile_ loads components and components groups parameters from a .csv file </td></tr>
<tr id="row_756_0_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09481.html" target="_self">MRMFeatureQCFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for <a class="el" href="a07209.html" title="The MRMFeatureQC is a class to handle the parameters and options for MRMFeatureFilter.">MRMFeatureQC</a> files </td></tr>
<tr id="row_757_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11009.html" target="_self">FeatureFinderAlgorithmPickedHelperStructs::TheoreticalIsotopePattern</a></td><td class="desc">Helper structure for a theoretical isotope pattern used in <a class="el" href="a10989.html" title="FeatureFinderAlgorithm for picked peaks.">FeatureFinderAlgorithmPicked</a> </td></tr>
<tr id="row_758_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_758_" class="arrow" onclick="toggleFolder('758_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08805.html" target="_self">ToolDescriptionInternal</a></td><td class="desc"><a class="el" href="a08813.html">ToolDescription</a> Class </td></tr>
<tr id="row_758_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08813.html" target="_self">ToolDescription</a></td><td class="desc"></td></tr>
<tr id="row_759_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a05823.html#a08809" target="_self">ToolExternalDetails</a></td><td class="desc"></td></tr>
<tr id="row_760_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07837.html" target="_self">ToolHandler</a></td><td class="desc"></td></tr>
<tr id="row_761_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11741.html" target="_self">TOPPASVertex::TOPPASFilenames</a></td><td class="desc"></td></tr>
<tr id="row_762_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_762_" class="arrow" onclick="toggleFolder('762_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07857.html" target="_self">TOPPBase</a></td><td class="desc">Base class for TOPP applications </td></tr>
<tr id="row_762_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11905.html" target="_self">DBsearchDeconvMass</a></td><td class="desc"></td></tr>
<tr id="row_762_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11905.html" target="_self">DBsearchDeconvMass</a></td><td class="desc"></td></tr>
<tr id="row_762_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11921.html" target="_self">Epifany</a></td><td class="desc"></td></tr>
<tr id="row_762_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11925.html" target="_self">FLASHDeconv</a></td><td class="desc"></td></tr>
<tr id="row_762_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11993.html" target="_self">NucleicAcidSearchEngine</a></td><td class="desc"></td></tr>
<tr id="row_762_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07829.html" target="_self">TOPPOpenSwathBase</a></td><td class="desc"></td></tr>
<tr id="row_762_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a12017.html" target="_self">TDfeatureDeconvolution</a></td><td class="desc"></td></tr>
<tr id="row_762_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11885.html" target="_self">TOPPFLASHDeconv</a></td><td class="desc"></td></tr>
<tr id="row_762_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11889.html" target="_self">TOPPGNPSExport</a></td><td class="desc"></td></tr>
<tr id="row_762_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11893.html" target="_self">TOPPOpenPepXL</a></td><td class="desc"></td></tr>
<tr id="row_762_10_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11897.html" target="_self">TOPPOpenPepXLLF</a></td><td class="desc"></td></tr>
<tr id="row_762_11_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11901.html" target="_self">TOPPXFDR</a></td><td class="desc"></td></tr>
<tr id="row_762_12_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a12013.html" target="_self">UTILProteomicsLFQ</a></td><td class="desc"></td></tr>
<tr id="row_763_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11713.html" target="_self">TOPPASScene::TOPPProcess</a></td><td class="desc">Stores the information for a TOPP process </td></tr>
<tr id="row_764_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08761.html" target="_self">Param::ParamIterator::TraceInfo</a></td><td class="desc">Struct that captures information on entered / left nodes for <a class="el" href="a08757.html" title="Forward const iterator for the Param class.">ParamIterator</a> </td></tr>
<tr id="row_765_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06793.html" target="_self">IDDecoyProbability::Transformation_</a></td><td class="desc">Struct to be used to store a transformation (used for fitting) </td></tr>
<tr id="row_766_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07041.html" target="_self">TransformationDescription</a></td><td class="desc">Generic description of a coordinate transformation </td></tr>
<tr id="row_767_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_767_" class="arrow" onclick="toggleFolder('767_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07049.html" target="_self">TransformationModel</a></td><td class="desc">Base class for transformation models </td></tr>
<tr id="row_767_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07057.html" target="_self">TransformationModelBSpline</a></td><td class="desc">B-spline (non-linear) model for transformations </td></tr>
<tr id="row_767_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07061.html" target="_self">TransformationModelInterpolated</a></td><td class="desc">Interpolation model for transformations </td></tr>
<tr id="row_767_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07069.html" target="_self">TransformationModelLinear</a></td><td class="desc">Linear model for transformations </td></tr>
<tr id="row_767_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07073.html" target="_self">TransformationModelLowess</a></td><td class="desc">Lowess (non-linear) model for transformations </td></tr>
<tr id="row_768_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07045.html" target="_self">TransformationDescription::TransformationStatistics</a></td><td class="desc"><a class="el" href="a10969.html" title="Summary of fitting results.">Summary</a> statistics before/after applying the transformation. For deviations before/after transformation, the percentiles 100, 99, 95, 90, 75, 50, 25 are returned </td></tr>
<tr id="row_769_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06521.html" target="_self">TransitionHelper</a></td><td class="desc"></td></tr>
<tr id="row_770_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11101.html" target="_self">IsotopeWaveletTransform&lt; PeakType &gt;::TransSpectrum</a></td><td class="desc">Internally (only by GPUs) used data structure . It allows efficient data exchange between CPU and GPU and avoids unnecessary memory moves. The class is tailored on the isotope wavelet transform and is in general not applicable on similar - but different - situations </td></tr>
<tr id="row_771_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07369.html" target="_self">TransitionTSVFile::TSVTransition</a></td><td class="desc"><a class="el" href="a05823.html" title="Namespace used to hide implementation details from users.">Internal</a> structure to represent a transition </td></tr>
<tr id="row_772_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11293.html" target="_self">TwoDOptimization::TwoDOptFunctor</a></td><td class="desc"></td></tr>
<tr id="row_773_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_773_" class="arrow" onclick="toggleFolder('773_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>unary_function</b></td><td class="desc"></td></tr>
<tr id="row_773_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10181.html" target="_self">HasActivationMethod&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum was generated using any activation method given in the constructor list </td></tr>
<tr id="row_773_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10153.html" target="_self">HasMetaValue&lt; MetaContainer &gt;</a></td><td class="desc">Predicate that determines if a class has a certain metavalue </td></tr>
<tr id="row_773_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10189.html" target="_self">HasPrecursorCharge&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum has a certain precursor charge as given in the constructor list </td></tr>
<tr id="row_773_3_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10165.html" target="_self">HasScanMode&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum has a certain scan mode </td></tr>
<tr id="row_773_4_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10169.html" target="_self">HasScanPolarity&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum has a certain scan polarity </td></tr>
<tr id="row_773_5_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10197.html" target="_self">InIntensityRange&lt; PeakType &gt;</a></td><td class="desc">Predicate that determines if a peak lies inside/outside a specific intensity range </td></tr>
<tr id="row_773_6_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10161.html" target="_self">InMSLevelRange&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum lies inside/outside a specific MS level set </td></tr>
<tr id="row_773_7_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10193.html" target="_self">InMzRange&lt; PeakType &gt;</a></td><td class="desc">Predicate that determines if a peak lies inside/outside a specific m/z range </td></tr>
<tr id="row_773_8_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10185.html" target="_self">InPrecursorMZRange&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum's precursor is within a certain m/z range </td></tr>
<tr id="row_773_9_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10157.html" target="_self">InRTRange&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum lies inside/outside a specific retention time range </td></tr>
<tr id="row_773_10_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10173.html" target="_self">IsEmptySpectrum&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum is empty </td></tr>
<tr id="row_773_11_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10201.html" target="_self">IsInCollisionEnergyRange&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if an MSn spectrum was generated with a collision energy in the given range </td></tr>
<tr id="row_773_12_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10209.html" target="_self">IsInIsolationWindow&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if the isolation window covers ANY of the given m/z values </td></tr>
<tr id="row_773_13_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10205.html" target="_self">IsInIsolationWindowSizeRange&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if the width of the isolation window of an MSn spectrum is in the given range </td></tr>
<tr id="row_773_14_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10177.html" target="_self">IsZoomSpectrum&lt; SpectrumType &gt;</a></td><td class="desc">Predicate that determines if a spectrum is a zoom (enhanced resolution) spectrum </td></tr>
<tr id="row_773_15_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06417.html" target="_self">mySqrt</a></td><td class="desc"></td></tr>
<tr id="row_774_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08497.html" target="_self">UniqueIdGenerator</a></td><td class="desc">A generator for unique ids </td></tr>
<tr id="row_775_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08501.html" target="_self">UniqueIdIndexer&lt; RandomAccessContainer &gt;</a></td><td class="desc">A base class for random access containers for classes derived from <a class="el" href="a08505.html" title="A base class defining a common interface for all classes having a unique id.">UniqueIdInterface</a> that adds functionality to convert a unique id into an index into the container </td></tr>
<tr id="row_776_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_776_" class="arrow" onclick="toggleFolder('776_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08501.html" target="_self">UniqueIdIndexer&lt; ConsensusMap &gt;</a></td><td class="desc"></td></tr>
<tr id="row_776_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10021.html" target="_self">ConsensusMap</a></td><td class="desc">A container for consensus elements </td></tr>
<tr id="row_777_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_777_" class="arrow" onclick="toggleFolder('777_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08501.html" target="_self">UniqueIdIndexer&lt; FeatureMap &gt;</a></td><td class="desc"></td></tr>
<tr id="row_777_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10057.html" target="_self">FeatureMap</a></td><td class="desc">A container for features </td></tr>
<tr id="row_778_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_778_" class="arrow" onclick="toggleFolder('778_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08505.html" target="_self">UniqueIdInterface</a></td><td class="desc">A base class defining a common interface for all classes having a unique id </td></tr>
<tr id="row_778_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10021.html" target="_self">ConsensusMap</a></td><td class="desc">A container for consensus elements </td></tr>
<tr id="row_778_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10041.html" target="_self">FeatureHandle</a></td><td class="desc">Representation of a <a class="el" href="a10125.html" title="A 2-dimensional raw data point or peak.">Peak2D</a>, <a class="el" href="a10213.html" title="A 2-dimensional raw data point or peak with meta information.">RichPeak2D</a> or <a class="el" href="a10037.html" title="An LC-MS feature.">Feature</a> </td></tr>
<tr id="row_778_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10057.html" target="_self">FeatureMap</a></td><td class="desc">A container for features </td></tr>
<tr id="row_778_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10213.html" target="_self">RichPeak2D</a></td><td class="desc">A 2-dimensional raw data point or peak with meta information </td></tr>
<tr id="row_779_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10401.html" target="_self">CVTerm::Unit</a></td><td class="desc"></td></tr>
<tr id="row_780_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10909.html" target="_self">UpdateCheck</a></td><td class="desc">Helper Functions to perform an update query to the <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a> REST server </td></tr>
<tr id="row_781_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06637.html" target="_self">ValueSize&lt; AAcid &gt;</a></td><td class="desc"></td></tr>
<tr id="row_782_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08733.html" target="_self">ValueTrait</a></td><td class="desc">Trait for <a class="el" href="a08729.html" title="For each element in the reference container the closest peak in the target will be searched....">MatchedIterator</a> to find pairs with a certain distance, which is computed directly on the value_type of the container </td></tr>
<tr id="row_783_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08701.html" target="_self">VecLowPrecision&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_784_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_784_" class="arrow" onclick="toggleFolder('784_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><b>vector&lt; T &gt;</b></td><td class="desc">STL class </td></tr>
<tr id="row_784_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07417.html" target="_self">Matrix&lt; double &gt;</a></td><td class="desc"></td></tr>
<tr id="row_784_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07417.html" target="_self">Matrix&lt; UInt &gt;</a></td><td class="desc"></td></tr>
<tr id="row_784_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10385.html" target="_self">AcquisitionInfo</a></td><td class="desc">Description of the combination of raw data to a single spectrum </td></tr>
<tr id="row_784_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10021.html" target="_self">ConsensusMap</a></td><td class="desc">A container for consensus elements </td></tr>
<tr id="row_784_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10413.html" target="_self">FloatDataArray</a></td><td class="desc">Float data array class </td></tr>
<tr id="row_784_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10417.html" target="_self">IntegerDataArray</a></td><td class="desc">Integer data array class </td></tr>
<tr id="row_784_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10421.html" target="_self">StringDataArray</a></td><td class="desc"><a class="el" href="a08769.html" title="A more convenient string class.">String</a> data array class </td></tr>
<tr id="row_784_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11005.html" target="_self">FeatureFinderAlgorithmPickedHelperStructs::MassTraces</a></td><td class="desc">Helper struct for a collection of mass traces used in <a class="el" href="a10989.html" title="FeatureFinderAlgorithm for picked peaks.">FeatureFinderAlgorithmPicked</a> </td></tr>
<tr id="row_784_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10057.html" target="_self">FeatureMap</a></td><td class="desc">A container for features </td></tr>
<tr id="row_784_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07417.html" target="_self">Matrix&lt; Value &gt;</a></td><td class="desc">A two-dimensional matrix. Similar to std::vector, but uses a binary operator(,) for element access </td></tr>
<tr id="row_784_10_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10073.html" target="_self">MSChromatogram</a></td><td class="desc">The representation of a chromatogram </td></tr>
<tr id="row_784_11_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10097.html" target="_self">MSSpectrum</a></td><td class="desc">The representation of a 1D spectrum </td></tr>
<tr id="row_784_12_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06813.html" target="_self">ScoreToTgtDecLabelPairs</a></td><td class="desc"></td></tr>
<tr id="row_785_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08513.html" target="_self">VersionInfo::VersionDetails</a></td><td class="desc"></td></tr>
<tr id="row_786_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08509.html" target="_self">VersionInfo</a></td><td class="desc">Version information class </td></tr>
<tr id="row_787_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a11737.html#a11745" target="_self">TOPPASVertex::VertexRoundPackage</a></td><td class="desc">Info for one edge and round, to be passed to next node </td></tr>
<tr id="row_788_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08009.html" target="_self">Weights</a></td><td class="desc">Represents a set of weights (double values and scaled with a certain precision their integer counterparts) with a quick access </td></tr>
<tr id="row_789_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a08145.html" target="_self">WeightWrapper</a></td><td class="desc">Encapsulated weight queries to simplify mono vs average weight computation </td></tr>
<tr id="row_790_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07601.html" target="_self">InclusionExclusionList::WindowDistance_</a></td><td class="desc">Determine distance between two spectra </td></tr>
<tr id="row_791_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a06413.html" target="_self">XCorrArrayType</a></td><td class="desc"></td></tr>
<tr id="row_792_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07765.html#a07781" target="_self">OPXLDataStructs::XLPrecursor</a></td><td class="desc">The <a class="el" href="a07765.html#a07781" title="The XLPrecursor struct represents a cross-link candidate in the process of filtering candidates by pr...">XLPrecursor</a> struct represents a cross-link candidate in the process of filtering candidates by precursor masses in OpenPepXL </td></tr>
<tr id="row_793_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07785.html" target="_self">OPXLDataStructs::XLPrecursorComparator</a></td><td class="desc">The <a class="el" href="a07785.html" title="The XLPrecursorComparator is a comparator for XLPrecursors, that allows direct comparison of the XLPr...">XLPrecursorComparator</a> is a comparator for XLPrecursors, that allows direct comparison of the <a class="el" href="a07765.html#a07781" title="The XLPrecursor struct represents a cross-link candidate in the process of filtering candidates by pr...">XLPrecursor</a> precursor mass with double numbers </td></tr>
<tr id="row_794_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_794_" class="arrow" onclick="toggleFolder('794_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09873.html" target="_self">XMLFile</a></td><td class="desc">Base class for loading/storing XML files that have a handler derived from <a class="el" href="a09417.html" title="Base class for XML handlers.">XMLHandler</a> </td></tr>
<tr id="row_794_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09125.html" target="_self">ConsensusXMLFile</a></td><td class="desc">This class provides Input functionality for ConsensusMaps and Output functionality for alignments and quantitation </td></tr>
<tr id="row_794_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09141.html" target="_self">CVMappingFile</a></td><td class="desc">Used to load CvMapping files </td></tr>
<tr id="row_794_2_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09261.html" target="_self">FeatureXMLFile</a></td><td class="desc">This class provides Input/Output functionality for feature maps </td></tr>
<tr id="row_794_3_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09437.html" target="_self">IdXMLFile</a></td><td class="desc">Used to load and store idXML files </td></tr>
<tr id="row_794_4_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09853.html" target="_self">SemanticValidator</a></td><td class="desc">Semantically validates XML files using <a class="el" href="a08557.html" title="Representation of controlled vocabulary mapping rules (for PSI formats)">CVMappings</a> and a <a class="el" href="a09129.html" title="Representation of a controlled vocabulary.">ControlledVocabulary</a> </td></tr>
<tr id="row_794_5_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09473.html" target="_self">MascotXMLFile</a></td><td class="desc">Used to load Mascot XML files </td></tr>
<tr id="row_794_6_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09529.html" target="_self">MzDataFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzData files </td></tr>
<tr id="row_794_7_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09533.html" target="_self">MzIdentMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzIdentML files </td></tr>
<tr id="row_794_8_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09537.html" target="_self">MzMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzML files </td></tr>
<tr id="row_794_9_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09541.html" target="_self">MzQuantMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzQuantML files </td></tr>
<tr id="row_794_10_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09709.html" target="_self">MzXMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for MzXML 3.1 files </td></tr>
<tr id="row_794_11_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09717.html" target="_self">OMSSAXMLFile</a></td><td class="desc">Used to load OMSSAXML files </td></tr>
<tr id="row_794_12_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09733.html" target="_self">ParamXMLFile</a></td><td class="desc">The file pendant of the <a class="el" href="a08745.html" title="Management and storage of parameters / INI files.">Param</a> class used to load and store the param datastructure as paramXML </td></tr>
<tr id="row_794_13_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09749.html" target="_self">PepXMLFile</a></td><td class="desc">Used to load and store PepXML files </td></tr>
<tr id="row_794_14_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09757.html" target="_self">PepXMLFileMascot</a></td><td class="desc">Used to load Mascot PepXML files </td></tr>
<tr id="row_794_15_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09765.html" target="_self">ProtXMLFile</a></td><td class="desc">Used to load (storing not supported, yet) ProtXML files </td></tr>
<tr id="row_794_16_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09769.html" target="_self">PTMXMLFile</a></td><td class="desc">Used to load and store PTMXML files </td></tr>
<tr id="row_794_17_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09773.html" target="_self">QcMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for QcML files used to load and store QcML files </td></tr>
<tr id="row_794_18_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09821.html" target="_self">ToolDescriptionFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for ToolDescriptor files </td></tr>
<tr id="row_794_19_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09825.html" target="_self">TraMLFile</a></td><td class="desc"><a class="el" href="a10873.html" title="Basic file handling operations.">File</a> adapter for HUPO PSI TraML files </td></tr>
<tr id="row_794_20_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09829.html" target="_self">TransformationXMLFile</a></td><td class="desc">Used to load and store TransformationXML files </td></tr>
<tr id="row_794_21_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09833.html" target="_self">UnimodXMLFile</a></td><td class="desc">Used to load XML files from unimod.org files </td></tr>
<tr id="row_794_22_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09877.html" target="_self">XQuestResultXMLFile</a></td><td class="desc">Used to load and store xQuest result files </td></tr>
<tr id="row_794_23_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09881.html" target="_self">XTandemInfile</a></td><td class="desc">XTandem input file </td></tr>
<tr id="row_794_24_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09885.html" target="_self">XTandemXMLFile</a></td><td class="desc">Used to load XTandemXML files </td></tr>
<tr id="row_795_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a07817.html" target="_self">XQuestScores</a></td><td class="desc">An implementation of the scores for cross-link identification from the xQuest algorithm (O. Rinner et al., 2008, "Identification of cross-linked peptides from large sequence databases") </td></tr>
<tr id="row_796_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a09889.html" target="_self">ZlibCompression</a></td><td class="desc">Compresses and uncompresses data using zlib </td></tr>
<tr id="row_797_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a00001.html#a12109" target="_self">double</a></td><td class="desc"></td></tr>
<tr id="row_798_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a00001.html#a12189" target="_self">float</a></td><td class="desc"></td></tr>
<tr id="row_799_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a00001.html#a12329" target="_self">int</a></td><td class="desc"></td></tr>
<tr id="row_800_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_800_" class="arrow" onclick="toggleFolder('800_')">&#9658;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a16149.html" target="_self">Iterator</a></td><td class="desc"></td></tr>
<tr id="row_800_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a10513.html" target="_self">IteratorWrapper&lt; Iterator &gt;</a></td><td class="desc">Wrapper that adds <code>operator&lt;</code> to iterators, so they can be used as (part of) keys in maps/sets or <code>multi_index_containers</code> </td></tr>
<tr id="row_801_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a00001.html#a12149" target="_self">pair&lt; double, bool &gt;</a></td><td class="desc"></td></tr>
<tr id="row_802_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="a00001.html#a12321" target="_self">Value</a></td><td class="desc"></td></tr>
</table>
</div><!-- directory -->
</div><!-- contents -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:07:14 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
