<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.16"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>OpenMS: Related Pages</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="icon" href="favicon.ico" type="image/x-icon" />
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="style_ini.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="OpenMS_75x55_transparent.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">OpenMS
   &#160;<span id="projectnumber">2.4.0</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.16 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">Related Pages</div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock">Here is a list of all related documentation pages:</div><div class="directory">
<div class="levels">[detail level <span onclick="javascript:toggleLevel(1);">1</span><span onclick="javascript:toggleLevel(2);">2</span><span onclick="javascript:toggleLevel(3);">3</span><span onclick="javascript:toggleLevel(4);">4</span>]</div><table class="directory">
<tr id="row_0_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16206.html" target="_self">OpenMS Developer Quickstart Guide</a></td><td class="desc"></td></tr>
<tr id="row_1_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16233.html" target="_self">ChangeLog</a></td><td class="desc"></td></tr>
<tr id="row_2_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16235.html" target="_self">FAQ</a></td><td class="desc"></td></tr>
<tr id="row_3_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16236.html" target="_self">Known Issues</a></td><td class="desc"></td></tr>
<tr id="row_4_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16237.html" target="_self">Developer C++ Quickstart Guide</a></td><td class="desc"></td></tr>
<tr id="row_5_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16238.html" target="_self">pyOpenMS (Python bindings)</a></td><td class="desc"></td></tr>
<tr id="row_6_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_6_" class="arrow" onclick="toggleFolder('6_')">&#9660;</span><a class="el" href="a16239.html" target="_self">User Quickstart Guide</a></td><td class="desc"></td></tr>
<tr id="row_6_0_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_6_0_" class="arrow" onclick="toggleFolder('6_0_')">&#9658;</span><a class="el" href="a16241.html" target="_self">TOPP tutorial</a></td><td class="desc"></td></tr>
<tr id="row_6_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16208.html" target="_self">TOPP and OpenMS</a></td><td class="desc"></td></tr>
<tr id="row_6_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16227.html" target="_self">Introduction</a></td><td class="desc"></td></tr>
<tr id="row_6_0_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16228.html" target="_self">The views</a></td><td class="desc"></td></tr>
<tr id="row_6_0_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16229.html" target="_self">Display modes and view options</a></td><td class="desc"></td></tr>
<tr id="row_6_0_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16230.html" target="_self">Data analysis</a></td><td class="desc"></td></tr>
<tr id="row_6_0_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16231.html" target="_self">Data editing</a></td><td class="desc"></td></tr>
<tr id="row_6_0_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16232.html" target="_self">Hotkeys table</a></td><td class="desc"></td></tr>
<tr id="row_6_0_7_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16224.html" target="_self">Smoothing raw data</a></td><td class="desc"></td></tr>
<tr id="row_6_0_8_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16223.html" target="_self">Subtracting a baseline from a spectrum</a></td><td class="desc"></td></tr>
<tr id="row_6_0_9_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16225.html" target="_self">Picking peaks</a></td><td class="desc"></td></tr>
<tr id="row_6_0_10_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16226.html" target="_self">Feature detection on centroided data</a></td><td class="desc"></td></tr>
<tr id="row_6_0_11_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_0_12_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_0_12_" class="arrow" onclick="toggleFolder('6_0_12_')">&#9658;</span><a class="el" href="a16210.html" target="_self">General introduction</a></td><td class="desc"></td></tr>
<tr id="row_6_0_12_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16255.html" target="_self">INIFileEditor</a></td><td class="desc">Can be used to visually edit INI files of TOPP tools </td></tr>
<tr id="row_6_0_12_1_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16381.html" target="_self">INIUpdater</a></td><td class="desc">Update INI and TOPPAS files from previous versions of OpenMS/TOPP </td></tr>
<tr id="row_6_0_13_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16211.html" target="_self">File Handling</a></td><td class="desc"></td></tr>
<tr id="row_6_0_14_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16212.html" target="_self">Profile data processing</a></td><td class="desc"></td></tr>
<tr id="row_6_0_15_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16217.html" target="_self">Calibration</a></td><td class="desc"></td></tr>
<tr id="row_6_0_16_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16214.html" target="_self">Map alignment</a></td><td class="desc"></td></tr>
<tr id="row_6_0_17_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16215.html" target="_self">Feature detection</a></td><td class="desc"></td></tr>
<tr id="row_6_0_18_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16216.html" target="_self">Feature grouping</a></td><td class="desc"></td></tr>
<tr id="row_6_0_19_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16213.html" target="_self">Consensus peptide identification</a></td><td class="desc"></td></tr>
<tr id="row_6_0_20_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16218.html" target="_self">Peptide property prediction</a></td><td class="desc"></td></tr>
<tr id="row_6_0_21_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16209.html" target="_self">Quality Control</a></td><td class="desc"></td></tr>
<tr id="row_6_0_22_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16219.html" target="_self">Conversion between OpenMS XML formats and text formats</a></td><td class="desc"></td></tr>
<tr id="row_6_1_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_6_1_" class="arrow" onclick="toggleFolder('6_1_')">&#9658;</span><a class="el" href="a16240.html" target="_self">TOPP documentation</a></td><td class="desc"></td></tr>
<tr id="row_6_1_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16257.html" target="_self">TOPPView</a></td><td class="desc"></td></tr>
<tr id="row_6_1_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16256.html" target="_self">TOPPAS</a></td><td class="desc">An assistant for GUI-driven TOPP workflow design </td></tr>
<tr id="row_6_1_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16255.html" target="_self">INIFileEditor</a></td><td class="desc">Can be used to visually edit INI files of TOPP tools </td></tr>
<tr id="row_6_1_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16266.html" target="_self">DTAExtractor</a></td><td class="desc">Extracts scans of an mzML file to several files in DTA format </td></tr>
<tr id="row_6_1_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16283.html" target="_self">FileConverter</a></td><td class="desc">Converts between different MS file formats </td></tr>
<tr id="row_6_1_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16284.html" target="_self">FileFilter</a></td><td class="desc">Extracts portions of the data from an mzML, featureXML or consensusXML file </td></tr>
<tr id="row_6_1_6_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16285.html" target="_self">FileInfo</a></td><td class="desc">Shows basic information about the data in an OpenMS readable file </td></tr>
<tr id="row_6_1_7_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16286.html" target="_self">FileMerger</a></td><td class="desc">Merges several files. Multiple output format supported, depending on input format </td></tr>
<tr id="row_6_1_8_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16294.html" target="_self">IDMerger</a></td><td class="desc">Merges several idXML files into one idXML file </td></tr>
<tr id="row_6_1_9_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16296.html" target="_self">IDRipper</a></td><td class="desc">IDRipper splits the protein/peptide identifications of an idXML file into several idXML files according their annotated file origin </td></tr>
<tr id="row_6_1_10_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16291.html" target="_self">IDFileConverter</a></td><td class="desc">Converts peptide/protein identification engine file formats </td></tr>
<tr id="row_6_1_11_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16308.html" target="_self">MapStatistics</a></td><td class="desc">Extract extended statistics on the features of a map for quality control </td></tr>
<tr id="row_6_1_12_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16425.html" target="_self">TargetedFileConverter</a></td><td class="desc">Converts different spectral libraries / transition files for targeted proteomics and metabolomics analysis </td></tr>
<tr id="row_6_1_13_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16356.html" target="_self">TextExporter</a></td><td class="desc">This application converts several OpenMS XML formats (featureXML, consensusXML, and idXML) to text files </td></tr>
<tr id="row_6_1_14_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16316.html" target="_self">MzTabExporter</a></td><td class="desc">This application converts several OpenMS XML formats (featureXML, consensusXML, and idXML) to mzTab </td></tr>
<tr id="row_6_1_15_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16258.html" target="_self">BaselineFilter</a></td><td class="desc">Executes the top-hat filter to remove the baseline of an MS experiment </td></tr>
<tr id="row_6_1_16_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16300.html" target="_self">InternalCalibration</a></td><td class="desc">Performs an internal mass recalibration on an MS experiment </td></tr>
<tr id="row_6_1_17_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16269.html" target="_self">ExternalCalibration</a></td><td class="desc">Performs an mass recalibration on an MS experiment using an external calibration function </td></tr>
<tr id="row_6_1_18_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16306.html" target="_self">MapNormalizer</a></td><td class="desc">Normalizes peak intensities to the percentage of the maximum intensity in the HPLC-MS map </td></tr>
<tr id="row_6_1_19_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16312.html" target="_self">MassTraceExtractor</a></td><td class="desc">MassTraceExtractor extracts mass traces from a MSExperiment map and stores them into a FeatureXMLFile </td></tr>
<tr id="row_6_1_20_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16317.html" target="_self">NoiseFilterGaussian</a></td><td class="desc">Executes a Gaussian filter to reduce the noise in an MS experiment </td></tr>
<tr id="row_6_1_21_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16318.html" target="_self">NoiseFilterSGolay</a></td><td class="desc">Executes a Savitzky Golay filter to reduce the noise in an MS experiment </td></tr>
<tr id="row_6_1_22_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16327.html" target="_self">PeakPickerHiRes</a></td><td class="desc">A tool for peak detection in profile data. Executes the peak picking with <a class="el" href="a11249.html">high_res</a> algorithm </td></tr>
<tr id="row_6_1_23_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16328.html" target="_self">PeakPickerWavelet</a></td><td class="desc">A tool for peak detection in profile data. Executes the peak picking with the algorithm described in described in Lange et al. (2006) Proc. PSB-06 </td></tr>
<tr id="row_6_1_24_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16334.html" target="_self">PrecursorMassCorrector</a></td><td class="desc">Corrects the precursor entries of MS/MS spectra, by using MS1 information </td></tr>
<tr id="row_6_1_25_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16290.html" target="_self">HighResPrecursorMassCorrector</a></td><td class="desc">Corrects the precursor mz of high resolution data </td></tr>
<tr id="row_6_1_26_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16341.html" target="_self">Resampler</a></td><td class="desc">Resampler can be used to transform an LC/MS map into a resampled map </td></tr>
<tr id="row_6_1_27_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16346.html" target="_self">SpectraFilterBernNorm</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_28_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16347.html" target="_self">SpectraFilterMarkerMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_29_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16348.html" target="_self">SpectraFilterNLargest</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_30_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16349.html" target="_self">SpectraFilterNormalizer</a></td><td class="desc">Normalizes intensity of peak spectra </td></tr>
<tr id="row_6_1_31_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16350.html" target="_self">SpectraFilterParentPeakMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_32_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16351.html" target="_self">SpectraFilterScaler</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_33_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16352.html" target="_self">SpectraFilterSqrtMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_34_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16353.html" target="_self">SpectraFilterThresholdMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_35_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16354.html" target="_self">SpectraFilterWindowMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_1_36_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16355.html" target="_self">SpectraMerger</a></td><td class="desc">Allows to add up several spectra </td></tr>
<tr id="row_6_1_37_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16357.html" target="_self">TOFCalibration</a></td><td class="desc">Performs an external calibration for tof spectra </td></tr>
<tr id="row_6_1_38_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16265.html" target="_self">Decharger</a></td><td class="desc">Decharges a feature map by clustering charge variants of a peptide to zero-charge entities </td></tr>
<tr id="row_6_1_39_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16267.html" target="_self">EICExtractor</a></td><td class="desc">Extracts EICs from an MS experiment, in order to quantify analytes at a given position </td></tr>
<tr id="row_6_1_40_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16271.html" target="_self">FeatureFinderCentroided</a></td><td class="desc">The feature detection application for quantitation (centroided) </td></tr>
<tr id="row_6_1_41_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16272.html" target="_self">FeatureFinderIdentification</a></td><td class="desc">Detects features in MS1 data based on peptide identifications </td></tr>
<tr id="row_6_1_42_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16273.html" target="_self">FeatureFinderIsotopeWavelet</a></td><td class="desc">The feature detection application for quantitation </td></tr>
<tr id="row_6_1_43_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16274.html" target="_self">FeatureFinderMetabo</a></td><td class="desc">FeatureFinderMetabo assembles metabolite features from singleton mass traces </td></tr>
<tr id="row_6_1_44_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16275.html" target="_self">FeatureFinderMRM</a></td><td class="desc">The feature detection application for quantitation </td></tr>
<tr id="row_6_1_45_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16276.html" target="_self">FeatureFinderMultiplex</a></td><td class="desc">Detects peptide pairs in LC-MS data and determines their relative abundance </td></tr>
<tr id="row_6_1_46_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16301.html" target="_self">IsobaricAnalyzer</a></td><td class="desc">Extracts and normalizes isobaric labeling information from an LC-MS/MS experiment </td></tr>
<tr id="row_6_1_47_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16336.html" target="_self">ProteinQuantifier</a></td><td class="desc">Compute peptide and protein abundances from annotated feature/consensus maps or from identification results </td></tr>
<tr id="row_6_1_48_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16337.html" target="_self">ProteinResolver</a></td><td class="desc">A peptide-centric algorithm for protein inference </td></tr>
<tr id="row_6_1_49_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16344.html" target="_self">SeedListGenerator</a></td><td class="desc">Application to generate seed lists for feature detection </td></tr>
<tr id="row_6_1_50_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16263.html" target="_self">ConsensusMapNormalizer</a></td><td class="desc">Normalization of intensities in a set of maps using robust regression </td></tr>
<tr id="row_6_1_51_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16303.html" target="_self">MapAlignerIdentification</a></td><td class="desc">Corrects retention time distortions between maps, using information from peptides identified in different maps </td></tr>
<tr id="row_6_1_52_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16304.html" target="_self">MapAlignerPoseClustering</a></td><td class="desc">Corrects retention time distortions between maps, using a pose clustering approach </td></tr>
<tr id="row_6_1_53_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16305.html" target="_self">MapAlignerSpectrum</a></td><td class="desc">Corrects retention time distortions between maps by aligning spectra </td></tr>
<tr id="row_6_1_54_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16307.html" target="_self">MapRTTransformer</a></td><td class="desc">Applies retention time transformations to maps </td></tr>
<tr id="row_6_1_55_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16278.html" target="_self">FeatureLinkerLabeled</a></td><td class="desc">Groups corresponding isotope-labeled features in a feature map </td></tr>
<tr id="row_6_1_56_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_1_56_" class="arrow" onclick="toggleFolder('6_1_56_')">&#9658;</span><a class="el" href="a16279.html" target="_self">FeatureLinkerUnlabeled</a></td><td class="desc">Groups corresponding features from multiple maps </td></tr>
<tr id="row_6_1_56_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16283.html" target="_self">FileConverter</a></td><td class="desc">Converts between different MS file formats </td></tr>
<tr id="row_6_1_57_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16281.html" target="_self">FeatureLinkerUnlabeledQT</a></td><td class="desc">Groups corresponding features from multiple maps using a QT clustering approach </td></tr>
<tr id="row_6_1_58_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16280.html" target="_self">FeatureLinkerUnlabeledKD</a></td><td class="desc">Group corresponding features across labelfree experiments </td></tr>
<tr id="row_6_1_59_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16259.html" target="_self">CometAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Comet </td></tr>
<tr id="row_6_1_60_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16260.html" target="_self">CompNovo</a></td><td class="desc">Performs a peptide/protein identification with the CompNovo engine </td></tr>
<tr id="row_6_1_61_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16261.html" target="_self">CompNovoCID</a></td><td class="desc">Performs a peptide/protein identification with the CompNovo engine </td></tr>
<tr id="row_6_1_62_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16264.html" target="_self">CruxAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Crux and tide-search </td></tr>
<tr id="row_6_1_63_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16299.html" target="_self">InspectAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Inspect </td></tr>
<tr id="row_6_1_64_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16310.html" target="_self">MascotAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Mascot </td></tr>
<tr id="row_6_1_65_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16311.html" target="_self">MascotAdapterOnline</a></td><td class="desc">Identifies peptides in MS/MS spectra via Mascot </td></tr>
<tr id="row_6_1_66_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_1_66_" class="arrow" onclick="toggleFolder('6_1_66_')">&#9658;</span><a class="el" href="a16314.html" target="_self">MSGFPlusAdapter</a></td><td class="desc">Adapter for the MS-GF+ protein identification (database search) engine </td></tr>
<tr id="row_6_1_66_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_1_67_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16315.html" target="_self">MyriMatchAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via MyriMatch </td></tr>
<tr id="row_6_1_68_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_1_68_" class="arrow" onclick="toggleFolder('6_1_68_')">&#9658;</span><a class="el" href="a16319.html" target="_self">OMSSAAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via OMSSA (Open Mass Spectrometry Search Algorithm) </td></tr>
<tr id="row_6_1_68_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_1_69_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16329.html" target="_self">PepNovoAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via PepNovo </td></tr>
<tr id="row_6_1_70_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16345.html" target="_self">SpecLibSearcher</a></td><td class="desc">Identifies peptide MS/MS spectra by spectral matching with a searchable spectral library </td></tr>
<tr id="row_6_1_71_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16420.html" target="_self">SimpleSearchEngine</a></td><td class="desc">Identifies peptides in MS/MS spectra </td></tr>
<tr id="row_6_1_72_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_1_72_" class="arrow" onclick="toggleFolder('6_1_72_')">&#9658;</span><a class="el" href="a16358.html" target="_self">XTandemAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via the search engine X! Tandem </td></tr>
<tr id="row_6_1_72_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_1_73_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16262.html" target="_self">ConsensusID</a></td><td class="desc">Computes a consensus from results of multiple peptide identification engines </td></tr>
<tr id="row_6_1_74_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16270.html" target="_self">FalseDiscoveryRate</a></td><td class="desc">Tool to estimate the false discovery rate on peptide and protein level </td></tr>
<tr id="row_6_1_75_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16282.html" target="_self">FidoAdapter</a></td><td class="desc">Runs the protein inference engine Fido </td></tr>
<tr id="row_6_1_76_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16250.html" target="_self">IDConflictResolver</a></td><td class="desc">Resolves ambiguous annotations of features with peptide identifications </td></tr>
<tr id="row_6_1_77_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16292.html" target="_self">IDFilter</a></td><td class="desc">Filters peptide/protein identification results by different criteria </td></tr>
<tr id="row_6_1_78_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16293.html" target="_self">IDMapper</a></td><td class="desc">Assigns protein/peptide identifications to features or consensus features </td></tr>
<tr id="row_6_1_79_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16295.html" target="_self">IDPosteriorErrorProbability</a></td><td class="desc">Tool to estimate the probability of peptide hits to be incorrectly assigned </td></tr>
<tr id="row_6_1_80_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16297.html" target="_self">IDRTCalibration</a></td><td class="desc">Can be used to calibrate the RTs of peptide hits linearly to standards </td></tr>
<tr id="row_6_1_81_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16302.html" target="_self">LuciphorAdapter</a></td><td class="desc">Adapter for the LuciPHOr2: a site localisation tool of generic post-translational modifications from tandem mass spectrometry data </td></tr>
<tr id="row_6_1_82_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_1_82_" class="arrow" onclick="toggleFolder('6_1_82_')">&#9658;</span><a class="el" href="a16330.html" target="_self">PeptideIndexer</a></td><td class="desc">Refreshes the protein references for all peptide hits from an idXML file and adds target/decoy information </td></tr>
<tr id="row_6_1_82_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_1_83_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16332.html" target="_self">PhosphoScoring</a></td><td class="desc">Tool to score phosphorylation sites of peptides </td></tr>
<tr id="row_6_1_84_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16335.html" target="_self">ProteinInference</a></td><td class="desc">Computes a protein identification score based on an aggregation of scores of identified peptides </td></tr>
<tr id="row_6_1_85_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16331.html" target="_self">PercolatorAdapter</a></td><td class="desc">PercolatorAdapter facilitates the input to, the call of and output integration of Percolator. Percolator (<a href="http://per-colator.com/">http://per-colator.com/</a>) is a tool to apply semi-supervised learning for peptide identification from shotgun proteomics datasets </td></tr>
<tr id="row_6_1_86_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16402.html" target="_self">OpenSwathWorkflow</a></td><td class="desc">Complete workflow to run OpenSWATH </td></tr>
<tr id="row_6_1_87_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16360.html" target="_self">AssayGeneratorMetabo</a></td><td class="desc">Generates an assay library using DDA data (Metabolomics) </td></tr>
<tr id="row_6_1_88_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16298.html" target="_self">InclusionExclusionListCreator</a></td><td class="desc">A tool for creating inclusion and/or exclusion lists for LC-MS/MS </td></tr>
<tr id="row_6_1_89_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16313.html" target="_self">MRMMapper</a></td><td class="desc">MRMMapper maps measured chromatograms (mzML) and the transitions used (TraML) </td></tr>
<tr id="row_6_1_90_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16390.html" target="_self">MRMTransitionGroupPicker</a></td><td class="desc">Picks peaks in SRM/MRM chromatograms that belong to the same precursors </td></tr>
<tr id="row_6_1_91_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16333.html" target="_self">PrecursorIonSelector</a></td><td class="desc">A tool for precursor ion selection based on identification results </td></tr>
<tr id="row_6_1_92_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16320.html" target="_self">OpenSwathAnalyzer</a></td><td class="desc">Executes a peak-picking and scoring algorithm on MRM/SRM data </td></tr>
<tr id="row_6_1_93_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16321.html" target="_self">OpenSwathAssayGenerator</a></td><td class="desc">Generates filtered and optimized assays using TraML files </td></tr>
<tr id="row_6_1_94_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16322.html" target="_self">OpenSwathChromatogramExtractor</a></td><td class="desc">Extracts chromatograms (XICs) from a file containing spectra </td></tr>
<tr id="row_6_1_95_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16323.html" target="_self">OpenSwathConfidenceScoring</a></td><td class="desc">Computes confidence scores for <a class="el" href="a05820.html">OpenSwath</a> results </td></tr>
<tr id="row_6_1_96_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16324.html" target="_self">OpenSwathDecoyGenerator</a></td><td class="desc">Generates decoys according to different models for a specific TraML </td></tr>
<tr id="row_6_1_97_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16325.html" target="_self">OpenSwathFeatureXMLToTSV</a></td><td class="desc">Converts a featureXML to a mProphet tsv </td></tr>
<tr id="row_6_1_98_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16326.html" target="_self">OpenSwathRTNormalizer</a></td><td class="desc">The OpenSwathRTNormalizer will find retention time peptides in data </td></tr>
<tr id="row_6_1_99_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16338.html" target="_self">PTModel</a></td><td class="desc">Used to train a model for the prediction of proteotypic peptides </td></tr>
<tr id="row_6_1_100_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16339.html" target="_self">PTPredict</a></td><td class="desc">This application is used to predict the likelihood of peptides to be proteotypic </td></tr>
<tr id="row_6_1_101_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16342.html" target="_self">RTModel</a></td><td class="desc">Used to train a model for peptide retention time prediction or peptide separation prediction </td></tr>
<tr id="row_6_1_102_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16343.html" target="_self">RTPredict</a></td><td class="desc">This application is used to predict retention times for peptides or peptide separation </td></tr>
<tr id="row_6_1_103_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16340.html" target="_self">QualityControl</a></td><td class="desc">Generates an mzTab file from various sources of a pipeline (mainly a ConsensusXML) which can be used for QC plots (e.g. via the R package 'PTX-QC') </td></tr>
<tr id="row_6_1_104_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16288.html" target="_self">GenericWrapper</a></td><td class="desc">Allows generically the wrapping of external tools </td></tr>
<tr id="row_6_1_105_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16268.html" target="_self">ExecutePipeline</a></td><td class="desc">Executes workflows created by TOPPAS </td></tr>
<tr id="row_6_1_106_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16289.html" target="_self">GNPSExport</a></td><td class="desc">Export MS/MS data in .MGF format for GNPS (<a href="http://gnps.ucsd.edu">http://gnps.ucsd.edu</a>) </td></tr>
<tr id="row_6_1_107_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16309.html" target="_self">MaRaClusterAdapter</a></td><td class="desc">MaRaClusterAdapter facilitates the input to, the call of and output integration of MaRaCluster. MaRaCluster (<a href="https://github.com/statisticalbiotechnology/maracluster">https://github.com/statisticalbiotechnology/maracluster</a>) is a tool to apply unsupervised clustering of ms2 spectra from shotgun proteomics datasets </td></tr>
<tr id="row_6_1_108_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16287.html" target="_self">TOPP_FLASHDeconv</a></td><td class="desc"></td></tr>
<tr id="row_6_2_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_6_2_" class="arrow" onclick="toggleFolder('6_2_')">&#9658;</span><a class="el" href="a16243.html" target="_self">UTILS documentation</a></td><td class="desc"></td></tr>
<tr id="row_6_2_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16381.html" target="_self">INIUpdater</a></td><td class="desc">Update INI and TOPPAS files from previous versions of OpenMS/TOPP </td></tr>
<tr id="row_6_2_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16383.html" target="_self">LowMemPeakPickerHiRes</a></td><td class="desc">A tool for peak detection on streamed profile data (low-memory requirements). Executes the peak picking with <a class="el" href="a11249.html">high_res</a> algorithm </td></tr>
<tr id="row_6_2_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16384.html" target="_self">LowMemPeakPickerHiResRandomAccess</a></td><td class="desc">A tool for peak detection in profile data. Executes the peak picking with <a class="el" href="a11249.html">high_res</a> algorithm </td></tr>
<tr id="row_6_2_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16403.html" target="_self">PeakPickerIterative</a></td><td class="desc">A tool for peak detection in profile data. Executes the peak picking with <a class="el" href="a11261.html">high_res</a> algorithm </td></tr>
<tr id="row_6_2_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16363.html" target="_self">CVInspector</a></td><td class="desc">A tool for visualization and validation of PSI mapping and CV files </td></tr>
<tr id="row_6_2_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16374.html" target="_self">FuzzyDiff</a></td><td class="desc">Compares two files, tolerating numeric differences </td></tr>
<tr id="row_6_2_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16379.html" target="_self">IDSplitter</a></td><td class="desc">Splits protein/peptide identifications off of annotated data files </td></tr>
<tr id="row_6_2_7_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16395.html" target="_self">MzMLSplitter</a></td><td class="desc">Splits an mzML file into multiple parts </td></tr>
<tr id="row_6_2_8_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16418.html" target="_self">SemanticValidator</a></td><td class="desc">SemanticValidator for XML files which can be semantically validated </td></tr>
<tr id="row_6_2_9_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16428.html" target="_self">XMLValidator</a></td><td class="desc">Validates XML files against an XSD schema </td></tr>
<tr id="row_6_2_10_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16373.html" target="_self">FFEval</a></td><td class="desc">Evaluation tool for feature detection algorithms </td></tr>
<tr id="row_6_2_11_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16382.html" target="_self">LabeledEval</a></td><td class="desc">Evaluation tool for isotope-labeled quantitation experiments </td></tr>
<tr id="row_6_2_12_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16417.html" target="_self">RTEvaluation</a></td><td class="desc">Application that evaluates TPs (true positives), TNs, FPs, and FNs for an idXML file with predicted RTs </td></tr>
<tr id="row_6_2_13_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16427.html" target="_self">TransformationEvaluation</a></td><td class="desc">Applies a transformation to a range of values and records the results </td></tr>
<tr id="row_6_2_14_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16364.html" target="_self">DatabaseFilter</a></td><td class="desc">The DatabaseFilter tool filters a protein database in fasta format according to one or multiple filtering criteria </td></tr>
<tr id="row_6_2_15_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16365.html" target="_self">DecoyDatabase</a></td><td class="desc">Create a decoy peptide database from standard FASTA databases </td></tr>
<tr id="row_6_2_16_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16367.html" target="_self">Digestor</a></td><td class="desc">Digests a protein database in-silico </td></tr>
<tr id="row_6_2_17_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16368.html" target="_self">DigestorMotif</a></td><td class="desc">This application is used to digest a protein database to get all peptides given a cleavage enzyme. It will also produce peptide statistics given the mass accuracy of the instrument. You can extract peptides with specific motifs,e.g. onyl cysteine containing peptides for ICAT experiments. At the moment only trypsin is supported </td></tr>
<tr id="row_6_2_18_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16369.html" target="_self">Epifany</a></td><td class="desc">EPIFANY - Efficient protein inference for any peptide-protein network is a Bayesian protein inference engine. It uses PSM (posterior) probabilities from Percolator, <a class="el" href="a05822.html" title="Main OpenMS namespace.">OpenMS</a>' IDPosteriorErrorProbability or similar tools to calculate posterior probabilities for proteins and protein groups </td></tr>
<tr id="row_6_2_19_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16376.html" target="_self">IDExtractor</a></td><td class="desc">Extracts 'n' peptides randomly or best 'n' from idXML files </td></tr>
<tr id="row_6_2_20_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16377.html" target="_self">IDMassAccuracy</a></td><td class="desc">Calculates a distribution of the mass error from given mass spectra and IDs </td></tr>
<tr id="row_6_2_21_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16378.html" target="_self">IDScoreSwitcher</a></td><td class="desc">Switches between different scores of peptide hits (PSMs) or protein hits in identification data </td></tr>
<tr id="row_6_2_22_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16391.html" target="_self">MSFraggerAdapter</a></td><td class="desc">Peptide Identification with MSFragger </td></tr>
<tr id="row_6_2_23_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16396.html" target="_self">NovorAdapter</a></td><td class="desc">Novoradapter for de novo sequencing from tandem mass spectrometry data </td></tr>
<tr id="row_6_2_24_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16405.html" target="_self">PSMFeatureExtractor</a></td><td class="desc">PSMFeatureExtractor computes extra features for each input PSM </td></tr>
<tr id="row_6_2_25_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16419.html" target="_self">SequenceCoverageCalculator</a></td><td class="desc">Prints information about idXML files </td></tr>
<tr id="row_6_2_26_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16422.html" target="_self">SpecLibCreator</a></td><td class="desc">Creates with given data a .MSP format spectral library </td></tr>
<tr id="row_6_2_27_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16423.html" target="_self">SpectraSTSearchAdapter</a></td><td class="desc">This util provides an interface to the 'SEARCH' mode of the SpectraST program. All non-advanced parameters of the executable of SpectraST were translated into parameters of this util </td></tr>
<tr id="row_6_2_28_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16415.html" target="_self">RNPxlSearch</a></td><td class="desc">Annotate RNA to peptide crosslinks in MS/MS spectra </td></tr>
<tr id="row_6_2_29_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16416.html" target="_self">RNPxlXICFilter</a></td><td class="desc">Filters MS2 spectra based on XIC intensities in control and treatment. Used in RNPxl experiments to reduce candidate spectra </td></tr>
<tr id="row_6_2_30_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16370.html" target="_self">ERPairFinder</a></td><td class="desc">Util which can be used to evaluate pair ratios on enhanced resolution (zoom) scans </td></tr>
<tr id="row_6_2_31_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16371.html" target="_self">FeatureFinderMetaboIdent</a></td><td class="desc">Detects features in MS1 data corresponding to small molecule identifications </td></tr>
<tr id="row_6_2_32_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16372.html" target="_self">FeatureFinderSuperHirn</a></td><td class="desc"></td></tr>
<tr id="row_6_2_33_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16386.html" target="_self">UTILS_MetaboliteAdductDecharger</a></td><td class="desc">Decharges a feature map by clustering charge variants of metabolites to zero-charge entities </td></tr>
<tr id="row_6_2_34_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16394.html" target="_self">MultiplexResolver</a></td><td class="desc">Completes peptide multiplets and resolves conflicts within them </td></tr>
<tr id="row_6_2_35_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16359.html" target="_self">AccurateMassSearch</a></td><td class="desc">An algorithm to search for exact mass matches from a spectrum against a database (e.g. HMDB) </td></tr>
<tr id="row_6_2_36_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16387.html" target="_self">MetaboliteSpectralMatcher</a></td><td class="desc">MetaboliteSpectralMatcher identify small molecules from tandem MS spectra </td></tr>
<tr id="row_6_2_37_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16421.html" target="_self">SiriusAdapter</a></td><td class="desc">De novo metabolite identification </td></tr>
<tr id="row_6_2_38_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16362.html" target="_self">CorrelateMassTraces</a></td><td class="desc">Identifies precursor mass traces and tries to correlate them with fragment ion mass traces in SWATH maps </td></tr>
<tr id="row_6_2_39_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16389.html" target="_self">MRMPairFinder</a></td><td class="desc">Util which can be used to evaluate pairs of MRM experiments </td></tr>
<tr id="row_6_2_40_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16398.html" target="_self">OpenSwathDIAPreScoring</a></td><td class="desc">.. </td></tr>
<tr id="row_6_2_41_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16399.html" target="_self">OpenSwathFileSplitter</a></td><td class="desc">A tool for splitting a single SWATH / DIA file into a set of files, each containing one SWATH window (plus one file for the MS1 survey scans) </td></tr>
<tr id="row_6_2_42_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16400.html" target="_self">OpenSwathMzMLFileCacher</a></td><td class="desc">Serialize a spectra and/or chromatogram mzML file </td></tr>
<tr id="row_6_2_43_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16401.html" target="_self">OpenSwathRewriteToFeatureXML</a></td><td class="desc">Combines featureXML and mProphet tsv to FDR filtered featureXML </td></tr>
<tr id="row_6_2_44_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16397.html" target="_self">NucleicAcidSearchEngine</a></td><td class="desc">Matches tandem mass spectra to nucleic acid sequences </td></tr>
<tr id="row_6_2_45_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16413.html" target="_self">RNADigestor</a></td><td class="desc">Digests an RNA sequence database in-silico </td></tr>
<tr id="row_6_2_46_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16414.html" target="_self">RNAMassCalculator</a></td><td class="desc">Calculates masses and mass-to-charge ratios of RNA sequences </td></tr>
<tr id="row_6_2_47_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16406.html" target="_self">QCCalculator</a></td><td class="desc">Calculates basic quality parameters from MS experiments and compiles data for subsequent QC into a qcML file </td></tr>
<tr id="row_6_2_48_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16407.html" target="_self">QCEmbedder</a></td><td class="desc">This application is used to embed tables or plots generated externally as attachments to existing quality parameters in qcML files </td></tr>
<tr id="row_6_2_49_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16408.html" target="_self">QCExporter</a></td><td class="desc">Will extract several quality parameter from several run/sets from a qcML file into a tabular (text) format - counterpart to QCImporter </td></tr>
<tr id="row_6_2_50_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16409.html" target="_self">QCExtractor</a></td><td class="desc">Extracts a table attachment of a given quality parameter from a qcML file as tabular (text) format </td></tr>
<tr id="row_6_2_51_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16410.html" target="_self">QCImporter</a></td><td class="desc">Will import several quality parameter from a tabular (text) format into a qcML file - counterpart to QCExporter </td></tr>
<tr id="row_6_2_52_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16411.html" target="_self">QCMerger</a></td><td class="desc">Merges two qcml files together </td></tr>
<tr id="row_6_2_53_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16412.html" target="_self">QCShrinker</a></td><td class="desc">This application is used to remove extra verbose table attachments from a qcML file that are not needed anymore, e.g. for a final report </td></tr>
<tr id="row_6_2_54_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16361.html" target="_self">ClusterMassTraces</a></td><td class="desc">Cluster mass traces occurring in the same map together </td></tr>
<tr id="row_6_2_55_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16366.html" target="_self">DeMeanderize</a></td><td class="desc">Repairs MALDI experiments which were spotted line by line </td></tr>
<tr id="row_6_2_56_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16380.html" target="_self">ImageCreator</a></td><td class="desc">Transforms an LC-MS map into a png image </td></tr>
<tr id="row_6_2_57_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16385.html" target="_self">MassCalculator</a></td><td class="desc">Calculates masses and mass-to-charge ratios of peptide sequences </td></tr>
<tr id="row_6_2_58_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16388.html" target="_self">MetaProSIP</a></td><td class="desc">Performs proteinSIP on peptide features for elemental flux analysis </td></tr>
<tr id="row_6_2_59_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16392.html" target="_self">MSSimulator</a></td><td class="desc">A highly configurable simulator for mass spectrometry experiments </td></tr>
<tr id="row_6_2_60_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16424.html" target="_self">SvmTheoreticalSpectrumGeneratorTrainer</a></td><td class="desc">Trainer for SVM model as input for SvmTheoreticalSpectrumGenerator </td></tr>
<tr id="row_6_2_61_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16426.html" target="_self">TICCalculator</a></td><td class="desc">Calculates the TIC of a raw mass spectrometric file </td></tr>
<tr id="row_6_2_62_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16393.html" target="_self">MSstatsConverter</a></td><td class="desc">Converter to input for MSstats </td></tr>
<tr id="row_6_2_63_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16375.html" target="_self">IDDecoyProbability</a></td><td class="desc">Util to estimate probability of peptide hits </td></tr>
<tr id="row_6_3_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_6_3_" class="arrow" onclick="toggleFolder('6_3_')">&#9658;</span><a class="el" href="a16241.html" target="_self">TOPP tutorial</a></td><td class="desc"></td></tr>
<tr id="row_6_3_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16208.html" target="_self">TOPP and OpenMS</a></td><td class="desc"></td></tr>
<tr id="row_6_3_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16227.html" target="_self">Introduction</a></td><td class="desc"></td></tr>
<tr id="row_6_3_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16228.html" target="_self">The views</a></td><td class="desc"></td></tr>
<tr id="row_6_3_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16229.html" target="_self">Display modes and view options</a></td><td class="desc"></td></tr>
<tr id="row_6_3_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16230.html" target="_self">Data analysis</a></td><td class="desc"></td></tr>
<tr id="row_6_3_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16231.html" target="_self">Data editing</a></td><td class="desc"></td></tr>
<tr id="row_6_3_6_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16232.html" target="_self">Hotkeys table</a></td><td class="desc"></td></tr>
<tr id="row_6_3_7_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16224.html" target="_self">Smoothing raw data</a></td><td class="desc"></td></tr>
<tr id="row_6_3_8_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16223.html" target="_self">Subtracting a baseline from a spectrum</a></td><td class="desc"></td></tr>
<tr id="row_6_3_9_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16225.html" target="_self">Picking peaks</a></td><td class="desc"></td></tr>
<tr id="row_6_3_10_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16226.html" target="_self">Feature detection on centroided data</a></td><td class="desc"></td></tr>
<tr id="row_6_3_11_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_3_12_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_3_12_" class="arrow" onclick="toggleFolder('6_3_12_')">&#9658;</span><a class="el" href="a16210.html" target="_self">General introduction</a></td><td class="desc"></td></tr>
<tr id="row_6_3_12_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16255.html" target="_self">INIFileEditor</a></td><td class="desc">Can be used to visually edit INI files of TOPP tools </td></tr>
<tr id="row_6_3_12_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16381.html" target="_self">INIUpdater</a></td><td class="desc">Update INI and TOPPAS files from previous versions of OpenMS/TOPP </td></tr>
<tr id="row_6_3_13_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16211.html" target="_self">File Handling</a></td><td class="desc"></td></tr>
<tr id="row_6_3_14_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16212.html" target="_self">Profile data processing</a></td><td class="desc"></td></tr>
<tr id="row_6_3_15_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16217.html" target="_self">Calibration</a></td><td class="desc"></td></tr>
<tr id="row_6_3_16_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16214.html" target="_self">Map alignment</a></td><td class="desc"></td></tr>
<tr id="row_6_3_17_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16215.html" target="_self">Feature detection</a></td><td class="desc"></td></tr>
<tr id="row_6_3_18_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16216.html" target="_self">Feature grouping</a></td><td class="desc"></td></tr>
<tr id="row_6_3_19_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16213.html" target="_self">Consensus peptide identification</a></td><td class="desc"></td></tr>
<tr id="row_6_3_20_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16218.html" target="_self">Peptide property prediction</a></td><td class="desc"></td></tr>
<tr id="row_6_3_21_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16209.html" target="_self">Quality Control</a></td><td class="desc"></td></tr>
<tr id="row_6_3_22_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16219.html" target="_self">Conversion between OpenMS XML formats and text formats</a></td><td class="desc"></td></tr>
<tr id="row_6_4_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_6_4_" class="arrow" onclick="toggleFolder('6_4_')">&#9658;</span><a class="el" href="a16240.html" target="_self">TOPP documentation</a></td><td class="desc"></td></tr>
<tr id="row_6_4_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16257.html" target="_self">TOPPView</a></td><td class="desc"></td></tr>
<tr id="row_6_4_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16256.html" target="_self">TOPPAS</a></td><td class="desc">An assistant for GUI-driven TOPP workflow design </td></tr>
<tr id="row_6_4_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16255.html" target="_self">INIFileEditor</a></td><td class="desc">Can be used to visually edit INI files of TOPP tools </td></tr>
<tr id="row_6_4_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16266.html" target="_self">DTAExtractor</a></td><td class="desc">Extracts scans of an mzML file to several files in DTA format </td></tr>
<tr id="row_6_4_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16283.html" target="_self">FileConverter</a></td><td class="desc">Converts between different MS file formats </td></tr>
<tr id="row_6_4_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16284.html" target="_self">FileFilter</a></td><td class="desc">Extracts portions of the data from an mzML, featureXML or consensusXML file </td></tr>
<tr id="row_6_4_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16285.html" target="_self">FileInfo</a></td><td class="desc">Shows basic information about the data in an OpenMS readable file </td></tr>
<tr id="row_6_4_7_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16286.html" target="_self">FileMerger</a></td><td class="desc">Merges several files. Multiple output format supported, depending on input format </td></tr>
<tr id="row_6_4_8_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16294.html" target="_self">IDMerger</a></td><td class="desc">Merges several idXML files into one idXML file </td></tr>
<tr id="row_6_4_9_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16296.html" target="_self">IDRipper</a></td><td class="desc">IDRipper splits the protein/peptide identifications of an idXML file into several idXML files according their annotated file origin </td></tr>
<tr id="row_6_4_10_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16291.html" target="_self">IDFileConverter</a></td><td class="desc">Converts peptide/protein identification engine file formats </td></tr>
<tr id="row_6_4_11_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16308.html" target="_self">MapStatistics</a></td><td class="desc">Extract extended statistics on the features of a map for quality control </td></tr>
<tr id="row_6_4_12_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16425.html" target="_self">TargetedFileConverter</a></td><td class="desc">Converts different spectral libraries / transition files for targeted proteomics and metabolomics analysis </td></tr>
<tr id="row_6_4_13_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16356.html" target="_self">TextExporter</a></td><td class="desc">This application converts several OpenMS XML formats (featureXML, consensusXML, and idXML) to text files </td></tr>
<tr id="row_6_4_14_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16316.html" target="_self">MzTabExporter</a></td><td class="desc">This application converts several OpenMS XML formats (featureXML, consensusXML, and idXML) to mzTab </td></tr>
<tr id="row_6_4_15_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16258.html" target="_self">BaselineFilter</a></td><td class="desc">Executes the top-hat filter to remove the baseline of an MS experiment </td></tr>
<tr id="row_6_4_16_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16300.html" target="_self">InternalCalibration</a></td><td class="desc">Performs an internal mass recalibration on an MS experiment </td></tr>
<tr id="row_6_4_17_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16269.html" target="_self">ExternalCalibration</a></td><td class="desc">Performs an mass recalibration on an MS experiment using an external calibration function </td></tr>
<tr id="row_6_4_18_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16306.html" target="_self">MapNormalizer</a></td><td class="desc">Normalizes peak intensities to the percentage of the maximum intensity in the HPLC-MS map </td></tr>
<tr id="row_6_4_19_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16312.html" target="_self">MassTraceExtractor</a></td><td class="desc">MassTraceExtractor extracts mass traces from a MSExperiment map and stores them into a FeatureXMLFile </td></tr>
<tr id="row_6_4_20_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16317.html" target="_self">NoiseFilterGaussian</a></td><td class="desc">Executes a Gaussian filter to reduce the noise in an MS experiment </td></tr>
<tr id="row_6_4_21_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16318.html" target="_self">NoiseFilterSGolay</a></td><td class="desc">Executes a Savitzky Golay filter to reduce the noise in an MS experiment </td></tr>
<tr id="row_6_4_22_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16327.html" target="_self">PeakPickerHiRes</a></td><td class="desc">A tool for peak detection in profile data. Executes the peak picking with <a class="el" href="a11249.html">high_res</a> algorithm </td></tr>
<tr id="row_6_4_23_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16328.html" target="_self">PeakPickerWavelet</a></td><td class="desc">A tool for peak detection in profile data. Executes the peak picking with the algorithm described in described in Lange et al. (2006) Proc. PSB-06 </td></tr>
<tr id="row_6_4_24_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16334.html" target="_self">PrecursorMassCorrector</a></td><td class="desc">Corrects the precursor entries of MS/MS spectra, by using MS1 information </td></tr>
<tr id="row_6_4_25_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16290.html" target="_self">HighResPrecursorMassCorrector</a></td><td class="desc">Corrects the precursor mz of high resolution data </td></tr>
<tr id="row_6_4_26_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16341.html" target="_self">Resampler</a></td><td class="desc">Resampler can be used to transform an LC/MS map into a resampled map </td></tr>
<tr id="row_6_4_27_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16346.html" target="_self">SpectraFilterBernNorm</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_28_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16347.html" target="_self">SpectraFilterMarkerMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_29_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16348.html" target="_self">SpectraFilterNLargest</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_30_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16349.html" target="_self">SpectraFilterNormalizer</a></td><td class="desc">Normalizes intensity of peak spectra </td></tr>
<tr id="row_6_4_31_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16350.html" target="_self">SpectraFilterParentPeakMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_32_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16351.html" target="_self">SpectraFilterScaler</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_33_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16352.html" target="_self">SpectraFilterSqrtMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_34_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16353.html" target="_self">SpectraFilterThresholdMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_35_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16354.html" target="_self">SpectraFilterWindowMower</a></td><td class="desc">Filters the top Peaks in the given spectra according to a given schema/thresholdset </td></tr>
<tr id="row_6_4_36_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16355.html" target="_self">SpectraMerger</a></td><td class="desc">Allows to add up several spectra </td></tr>
<tr id="row_6_4_37_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16357.html" target="_self">TOFCalibration</a></td><td class="desc">Performs an external calibration for tof spectra </td></tr>
<tr id="row_6_4_38_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16265.html" target="_self">Decharger</a></td><td class="desc">Decharges a feature map by clustering charge variants of a peptide to zero-charge entities </td></tr>
<tr id="row_6_4_39_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16267.html" target="_self">EICExtractor</a></td><td class="desc">Extracts EICs from an MS experiment, in order to quantify analytes at a given position </td></tr>
<tr id="row_6_4_40_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16271.html" target="_self">FeatureFinderCentroided</a></td><td class="desc">The feature detection application for quantitation (centroided) </td></tr>
<tr id="row_6_4_41_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16272.html" target="_self">FeatureFinderIdentification</a></td><td class="desc">Detects features in MS1 data based on peptide identifications </td></tr>
<tr id="row_6_4_42_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16273.html" target="_self">FeatureFinderIsotopeWavelet</a></td><td class="desc">The feature detection application for quantitation </td></tr>
<tr id="row_6_4_43_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16274.html" target="_self">FeatureFinderMetabo</a></td><td class="desc">FeatureFinderMetabo assembles metabolite features from singleton mass traces </td></tr>
<tr id="row_6_4_44_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16275.html" target="_self">FeatureFinderMRM</a></td><td class="desc">The feature detection application for quantitation </td></tr>
<tr id="row_6_4_45_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16276.html" target="_self">FeatureFinderMultiplex</a></td><td class="desc">Detects peptide pairs in LC-MS data and determines their relative abundance </td></tr>
<tr id="row_6_4_46_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16301.html" target="_self">IsobaricAnalyzer</a></td><td class="desc">Extracts and normalizes isobaric labeling information from an LC-MS/MS experiment </td></tr>
<tr id="row_6_4_47_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16336.html" target="_self">ProteinQuantifier</a></td><td class="desc">Compute peptide and protein abundances from annotated feature/consensus maps or from identification results </td></tr>
<tr id="row_6_4_48_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16337.html" target="_self">ProteinResolver</a></td><td class="desc">A peptide-centric algorithm for protein inference </td></tr>
<tr id="row_6_4_49_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16344.html" target="_self">SeedListGenerator</a></td><td class="desc">Application to generate seed lists for feature detection </td></tr>
<tr id="row_6_4_50_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16263.html" target="_self">ConsensusMapNormalizer</a></td><td class="desc">Normalization of intensities in a set of maps using robust regression </td></tr>
<tr id="row_6_4_51_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16303.html" target="_self">MapAlignerIdentification</a></td><td class="desc">Corrects retention time distortions between maps, using information from peptides identified in different maps </td></tr>
<tr id="row_6_4_52_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16304.html" target="_self">MapAlignerPoseClustering</a></td><td class="desc">Corrects retention time distortions between maps, using a pose clustering approach </td></tr>
<tr id="row_6_4_53_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16305.html" target="_self">MapAlignerSpectrum</a></td><td class="desc">Corrects retention time distortions between maps by aligning spectra </td></tr>
<tr id="row_6_4_54_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16307.html" target="_self">MapRTTransformer</a></td><td class="desc">Applies retention time transformations to maps </td></tr>
<tr id="row_6_4_55_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16278.html" target="_self">FeatureLinkerLabeled</a></td><td class="desc">Groups corresponding isotope-labeled features in a feature map </td></tr>
<tr id="row_6_4_56_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_4_56_" class="arrow" onclick="toggleFolder('6_4_56_')">&#9658;</span><a class="el" href="a16279.html" target="_self">FeatureLinkerUnlabeled</a></td><td class="desc">Groups corresponding features from multiple maps </td></tr>
<tr id="row_6_4_56_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16283.html" target="_self">FileConverter</a></td><td class="desc">Converts between different MS file formats </td></tr>
<tr id="row_6_4_57_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16281.html" target="_self">FeatureLinkerUnlabeledQT</a></td><td class="desc">Groups corresponding features from multiple maps using a QT clustering approach </td></tr>
<tr id="row_6_4_58_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16280.html" target="_self">FeatureLinkerUnlabeledKD</a></td><td class="desc">Group corresponding features across labelfree experiments </td></tr>
<tr id="row_6_4_59_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16259.html" target="_self">CometAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Comet </td></tr>
<tr id="row_6_4_60_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16260.html" target="_self">CompNovo</a></td><td class="desc">Performs a peptide/protein identification with the CompNovo engine </td></tr>
<tr id="row_6_4_61_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16261.html" target="_self">CompNovoCID</a></td><td class="desc">Performs a peptide/protein identification with the CompNovo engine </td></tr>
<tr id="row_6_4_62_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16264.html" target="_self">CruxAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Crux and tide-search </td></tr>
<tr id="row_6_4_63_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16299.html" target="_self">InspectAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Inspect </td></tr>
<tr id="row_6_4_64_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16310.html" target="_self">MascotAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via Mascot </td></tr>
<tr id="row_6_4_65_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16311.html" target="_self">MascotAdapterOnline</a></td><td class="desc">Identifies peptides in MS/MS spectra via Mascot </td></tr>
<tr id="row_6_4_66_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_4_66_" class="arrow" onclick="toggleFolder('6_4_66_')">&#9658;</span><a class="el" href="a16314.html" target="_self">MSGFPlusAdapter</a></td><td class="desc">Adapter for the MS-GF+ protein identification (database search) engine </td></tr>
<tr id="row_6_4_66_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_4_67_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16315.html" target="_self">MyriMatchAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via MyriMatch </td></tr>
<tr id="row_6_4_68_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_4_68_" class="arrow" onclick="toggleFolder('6_4_68_')">&#9658;</span><a class="el" href="a16319.html" target="_self">OMSSAAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via OMSSA (Open Mass Spectrometry Search Algorithm) </td></tr>
<tr id="row_6_4_68_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_4_69_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16329.html" target="_self">PepNovoAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via PepNovo </td></tr>
<tr id="row_6_4_70_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16345.html" target="_self">SpecLibSearcher</a></td><td class="desc">Identifies peptide MS/MS spectra by spectral matching with a searchable spectral library </td></tr>
<tr id="row_6_4_71_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16420.html" target="_self">SimpleSearchEngine</a></td><td class="desc">Identifies peptides in MS/MS spectra </td></tr>
<tr id="row_6_4_72_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_4_72_" class="arrow" onclick="toggleFolder('6_4_72_')">&#9658;</span><a class="el" href="a16358.html" target="_self">XTandemAdapter</a></td><td class="desc">Identifies peptides in MS/MS spectra via the search engine X! Tandem </td></tr>
<tr id="row_6_4_72_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_4_73_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16262.html" target="_self">ConsensusID</a></td><td class="desc">Computes a consensus from results of multiple peptide identification engines </td></tr>
<tr id="row_6_4_74_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16270.html" target="_self">FalseDiscoveryRate</a></td><td class="desc">Tool to estimate the false discovery rate on peptide and protein level </td></tr>
<tr id="row_6_4_75_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16282.html" target="_self">FidoAdapter</a></td><td class="desc">Runs the protein inference engine Fido </td></tr>
<tr id="row_6_4_76_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16250.html" target="_self">IDConflictResolver</a></td><td class="desc">Resolves ambiguous annotations of features with peptide identifications </td></tr>
<tr id="row_6_4_77_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16292.html" target="_self">IDFilter</a></td><td class="desc">Filters peptide/protein identification results by different criteria </td></tr>
<tr id="row_6_4_78_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16293.html" target="_self">IDMapper</a></td><td class="desc">Assigns protein/peptide identifications to features or consensus features </td></tr>
<tr id="row_6_4_79_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16295.html" target="_self">IDPosteriorErrorProbability</a></td><td class="desc">Tool to estimate the probability of peptide hits to be incorrectly assigned </td></tr>
<tr id="row_6_4_80_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16297.html" target="_self">IDRTCalibration</a></td><td class="desc">Can be used to calibrate the RTs of peptide hits linearly to standards </td></tr>
<tr id="row_6_4_81_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16302.html" target="_self">LuciphorAdapter</a></td><td class="desc">Adapter for the LuciPHOr2: a site localisation tool of generic post-translational modifications from tandem mass spectrometry data </td></tr>
<tr id="row_6_4_82_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_6_4_82_" class="arrow" onclick="toggleFolder('6_4_82_')">&#9658;</span><a class="el" href="a16330.html" target="_self">PeptideIndexer</a></td><td class="desc">Refreshes the protein references for all peptide hits from an idXML file and adds target/decoy information </td></tr>
<tr id="row_6_4_82_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><a class="el" href="a16207.html" target="_self">TOPP for Advanced Users</a></td><td class="desc"></td></tr>
<tr id="row_6_4_83_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16332.html" target="_self">PhosphoScoring</a></td><td class="desc">Tool to score phosphorylation sites of peptides </td></tr>
<tr id="row_6_4_84_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16335.html" target="_self">ProteinInference</a></td><td class="desc">Computes a protein identification score based on an aggregation of scores of identified peptides </td></tr>
<tr id="row_6_4_85_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16331.html" target="_self">PercolatorAdapter</a></td><td class="desc">PercolatorAdapter facilitates the input to, the call of and output integration of Percolator. Percolator (<a href="http://per-colator.com/">http://per-colator.com/</a>) is a tool to apply semi-supervised learning for peptide identification from shotgun proteomics datasets </td></tr>
<tr id="row_6_4_86_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16402.html" target="_self">OpenSwathWorkflow</a></td><td class="desc">Complete workflow to run OpenSWATH </td></tr>
<tr id="row_6_4_87_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16360.html" target="_self">AssayGeneratorMetabo</a></td><td class="desc">Generates an assay library using DDA data (Metabolomics) </td></tr>
<tr id="row_6_4_88_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16298.html" target="_self">InclusionExclusionListCreator</a></td><td class="desc">A tool for creating inclusion and/or exclusion lists for LC-MS/MS </td></tr>
<tr id="row_6_4_89_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16313.html" target="_self">MRMMapper</a></td><td class="desc">MRMMapper maps measured chromatograms (mzML) and the transitions used (TraML) </td></tr>
<tr id="row_6_4_90_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16390.html" target="_self">MRMTransitionGroupPicker</a></td><td class="desc">Picks peaks in SRM/MRM chromatograms that belong to the same precursors </td></tr>
<tr id="row_6_4_91_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16333.html" target="_self">PrecursorIonSelector</a></td><td class="desc">A tool for precursor ion selection based on identification results </td></tr>
<tr id="row_6_4_92_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16320.html" target="_self">OpenSwathAnalyzer</a></td><td class="desc">Executes a peak-picking and scoring algorithm on MRM/SRM data </td></tr>
<tr id="row_6_4_93_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16321.html" target="_self">OpenSwathAssayGenerator</a></td><td class="desc">Generates filtered and optimized assays using TraML files </td></tr>
<tr id="row_6_4_94_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16322.html" target="_self">OpenSwathChromatogramExtractor</a></td><td class="desc">Extracts chromatograms (XICs) from a file containing spectra </td></tr>
<tr id="row_6_4_95_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16323.html" target="_self">OpenSwathConfidenceScoring</a></td><td class="desc">Computes confidence scores for <a class="el" href="a05820.html">OpenSwath</a> results </td></tr>
<tr id="row_6_4_96_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16324.html" target="_self">OpenSwathDecoyGenerator</a></td><td class="desc">Generates decoys according to different models for a specific TraML </td></tr>
<tr id="row_6_4_97_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16325.html" target="_self">OpenSwathFeatureXMLToTSV</a></td><td class="desc">Converts a featureXML to a mProphet tsv </td></tr>
<tr id="row_6_4_98_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16326.html" target="_self">OpenSwathRTNormalizer</a></td><td class="desc">The OpenSwathRTNormalizer will find retention time peptides in data </td></tr>
<tr id="row_6_4_99_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16338.html" target="_self">PTModel</a></td><td class="desc">Used to train a model for the prediction of proteotypic peptides </td></tr>
<tr id="row_6_4_100_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16339.html" target="_self">PTPredict</a></td><td class="desc">This application is used to predict the likelihood of peptides to be proteotypic </td></tr>
<tr id="row_6_4_101_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16342.html" target="_self">RTModel</a></td><td class="desc">Used to train a model for peptide retention time prediction or peptide separation prediction </td></tr>
<tr id="row_6_4_102_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16343.html" target="_self">RTPredict</a></td><td class="desc">This application is used to predict retention times for peptides or peptide separation </td></tr>
<tr id="row_6_4_103_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16340.html" target="_self">QualityControl</a></td><td class="desc">Generates an mzTab file from various sources of a pipeline (mainly a ConsensusXML) which can be used for QC plots (e.g. via the R package 'PTX-QC') </td></tr>
<tr id="row_6_4_104_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16288.html" target="_self">GenericWrapper</a></td><td class="desc">Allows generically the wrapping of external tools </td></tr>
<tr id="row_6_4_105_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16268.html" target="_self">ExecutePipeline</a></td><td class="desc">Executes workflows created by TOPPAS </td></tr>
<tr id="row_6_4_106_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16289.html" target="_self">GNPSExport</a></td><td class="desc">Export MS/MS data in .MGF format for GNPS (<a href="http://gnps.ucsd.edu">http://gnps.ucsd.edu</a>) </td></tr>
<tr id="row_6_4_107_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16309.html" target="_self">MaRaClusterAdapter</a></td><td class="desc">MaRaClusterAdapter facilitates the input to, the call of and output integration of MaRaCluster. MaRaCluster (<a href="https://github.com/statisticalbiotechnology/maracluster">https://github.com/statisticalbiotechnology/maracluster</a>) is a tool to apply unsupervised clustering of ms2 spectra from shotgun proteomics datasets </td></tr>
<tr id="row_6_4_108_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16287.html" target="_self">TOPP_FLASHDeconv</a></td><td class="desc"></td></tr>
<tr id="row_6_5_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_6_5_" class="arrow" onclick="toggleFolder('6_5_')">&#9658;</span><a class="el" href="a16242.html" target="_self">TOPPAS tutorial</a></td><td class="desc"></td></tr>
<tr id="row_6_5_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16220.html" target="_self">General introduction</a></td><td class="desc"></td></tr>
<tr id="row_6_5_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16221.html" target="_self">User interface</a></td><td class="desc"></td></tr>
<tr id="row_6_5_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><a class="el" href="a16222.html" target="_self">Examples</a></td><td class="desc"></td></tr>
<tr id="row_6_6_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><a class="el" href="a16255.html" target="_self">INIFileEditor</a></td><td class="desc">Can be used to visually edit INI files of TOPP tools </td></tr>
<tr id="row_7_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_7_" class="arrow" onclick="toggleFolder('7_')">&#9660;</span><a class="el" href="a16244.html" target="_self">Installation on GNU/Linux</a></td><td class="desc"></td></tr>
<tr id="row_7_0_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><a class="el" href="a16245.html" target="_self">Building OpenMS on GNU/Linux</a></td><td class="desc"></td></tr>
<tr id="row_8_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16246.html" target="_self">Installation on MacOS</a></td><td class="desc"></td></tr>
<tr id="row_9_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16247.html" target="_self">Building OpenMS on Mac OS X</a></td><td class="desc"></td></tr>
<tr id="row_10_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16248.html" target="_self">Installation on Windows</a></td><td class="desc"></td></tr>
<tr id="row_11_"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_11_" class="arrow" onclick="toggleFolder('11_')">&#9660;</span><a class="el" href="a16249.html" target="_self">Building OpenMS on Windows</a></td><td class="desc"></td></tr>
<tr id="row_11_0_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><a class="el" href="a16234.html" target="_self">External Code using OpenMS</a></td><td class="desc"></td></tr>
<tr id="row_12_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16251.html" target="_self">OpenPepXL</a></td><td class="desc">Search for peptide pairs linked with a labeled cross-linker </td></tr>
<tr id="row_13_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16252.html" target="_self">OpenPepXLLF</a></td><td class="desc">Search for cross-linked peptide pairs in tandem MS spectra </td></tr>
<tr id="row_14_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16253.html" target="_self">XFDR</a></td><td class="desc">Calculates false discovery rate estimates on crosslink identifications </td></tr>
<tr id="row_15_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16254.html" target="_self">MapAlignerBase</a></td><td class="desc">Base class for different MapAligner TOPP tools </td></tr>
<tr id="row_16_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16277.html" target="_self">FeatureLinkerBase</a></td><td class="desc">Base class for different FeatureLinker tools </td></tr>
<tr id="row_17_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a16404.html" target="_self">UTILS_ProteomicsLFQ</a></td><td class="desc"></td></tr>
<tr id="row_18_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a05779.html" target="_self">Experimental List</a></td><td class="desc"></td></tr>
<tr id="row_19_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a05778.html" target="_self">Improvements List</a></td><td class="desc"></td></tr>
<tr id="row_20_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a05777.html" target="_self">Todo List</a></td><td class="desc"></td></tr>
<tr id="row_21_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a05780.html" target="_self">Deprecated List</a></td><td class="desc"></td></tr>
<tr id="row_22_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="a05781.html" target="_self">Bug List</a></td><td class="desc"></td></tr>
</table>
</div><!-- directory -->
</div><!-- contents -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Mon Dec 9 2019 15:07:12 for OpenMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.16
</small></address>
</body>
</html>
